{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# N starvation \n",
    "\n",
    "## Which N sources allow MED4 to grow"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "check out Escher visualization https://escher.readthedocs.io/en/latest/getting_started.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-12-15T09:30:48.353507Z",
     "start_time": "2020-12-15T09:30:43.130260Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import cobra\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import itertools\n",
    "from matplotlib.colors import LogNorm, Normalize\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import time\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.flux_analysis import flux_variability_analysis\n",
    "from cobra.flux_analysis import production_envelope\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.flux_analysis import (\n",
    "    single_gene_deletion, single_reaction_deletion, double_gene_deletion,\n",
    "    double_reaction_deletion)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['MED4_DE_DESEQ_results.csv',\n",
       " 'MED4_DE_genes_DESEQ_wKEGG.csv',\n",
       " 'MED4_counts_combined.txt',\n",
       " 'MED4_PMM2locus.csv']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RNA_dpath =os.path.join('data', 'RNASEQ') \n",
    "os.listdir(RNA_dpath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "pmm_df = pd.read_csv(os.path.join(RNA_dpath, 'MED4_PMM2locus.csv'), names=['gene_id', 'pmm'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "deseq_df = pd.read_csv(os.path.join(RNA_dpath, 'MED4_DE_genes_DESEQ_wKEGG.csv'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-12-15T09:30:48.389369Z",
     "start_time": "2020-12-15T09:30:48.353507Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function seaborn.palettes.hls_palette(n_colors=6, h=0.01, l=0.6, s=0.65, as_cmap=False)>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sns.hls_palette"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# values from Ofaim at el.\n",
    "\n",
    "PARAMETER_VALUES = [#\"Name\",     \"Reaction ID\",          \"Lower bound\", \"UpperBound\"\n",
    "                   [\"HCO3\",      \"HCO3EXcar\",            -8,            0],\n",
    "                   #[\"Nitrogen\",  \"AmmoniaEX\",            -0.56,         0],\n",
    "                   #[\"Phosphate\", \"FAKEOrthophosphateEX\", -0.1,          0],\n",
    "                   [\"Light\",     \"LightEX\",              -150,          0]]\n",
    "CO2MAX = 0.82\n",
    "\n",
    "# Block fake reactions\n",
    "FAKE_TRANSPORT = [\"AminosugarsOUT\", \"FAKEAAOUT\", \"FAKEABPOUT\", \"FAKEacpTRANS\", \"FAKEApoacpTRANS\", \"FAKEThioredoxinTRANS\", \n",
    "                  \"FreefattyacidsOUT\", \"7NMeth7carbOUT\", \"ArtificialproteinOUT\", \"FADOUT\", \"LipoylproteinTRANS\", \"MenaquinoneOUT\", \n",
    "                  \"NicotinateOUT\", \"THFpolyglutOUT\", \"Thiamin_dpOUT\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_elements(rid):\n",
    "    r = model.reactions.get_by_id(rid)\n",
    "    m = r.reactants[0]    \n",
    "    elements = m.elements.copy()\n",
    "    \n",
    "    res = {\n",
    "        'reaction' : rid,\n",
    "        'metabolite' : m.name,\n",
    "        'N' : 'N' in elements,\n",
    "        'C' : 'C' in elements,\n",
    "        \n",
    "    }\n",
    "    return res"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import model and manipulate based on Ofaim at el"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "model_dpath = os.path.join('..', 'Model_files')\n",
    "model_fname = 'iSO595v7.xml'\n",
    "model_fpath = os.path.join(model_dpath, model_fname)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-12-15T09:30:49.550482Z",
     "start_time": "2020-12-15T09:30:48.441726Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Set parameter Username\n",
      "Academic license - for non-commercial use only - expires 2023-07-13\n"
     ]
    }
   ],
   "source": [
    "model = cobra.io.read_sbml_model(model_fpath)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "# fails with error! \n",
    "model.solver = 'gurobi'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# manipulations copied from Ofaim at el.\n",
    "\n",
    "remove_blocked = False\n",
    "\n",
    "# Block H2S\n",
    "model.reactions.H2SEX.lower_bound = 0\n",
    "\n",
    "# Block fake transports\n",
    "for rid in FAKE_TRANSPORT:\n",
    "    model.reactions.get_by_id(rid).bounds = (0,0)\n",
    "\n",
    "# Remove blocked reactions\n",
    "if remove_blocked:\n",
    "    blocked = cobra.flux_analysis.find_blocked_reactions(model, open_exchanges = True)\n",
    "    print('blocked', len(blocked), blocked)\n",
    "    model.remove_reactions([model.reactions.get_by_id(r_id) for r_id in blocked])\n",
    "\n",
    "# Block maximum CO2 production\n",
    "model.reactions.CO2EX.bounds = (0, CO2MAX)\n",
    "\n",
    "for i, row in enumerate(PARAMETER_VALUES):\n",
    "    # Row: Name, Reaction ID, lower bound, upper bound\n",
    "    key = row[0]\n",
    "    reaction_id = row[1]\n",
    "    lower_bound = row[2]\n",
    "    upper_bound = row[3]\n",
    "    r = model.reactions.get_by_id(reaction_id)\n",
    "    # Fix flux\n",
    "    r.bounds = (lower_bound, upper_bound)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "exchange_ids = [r.id for r in model.exchanges] # + [\"R00024\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reactions 994\n",
      "metabolites 802\n",
      "genes 595\n"
     ]
    }
   ],
   "source": [
    "print('reactions', len(model.reactions))\n",
    "print('metabolites', len(model.metabolites))\n",
    "print('genes', len(model.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>COBRAModel</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f911edecf10</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>802</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>994</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>55</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*BIOMASS - 1.0*BIOMASS_reverse_69053</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytoplasm, Extracellular, unknownCompartment2, Periplasm, unknownCompartment5</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model COBRAModel at 0x7f911edecf10>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>BIOMASS</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Biomass formation</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x7f911ea4bb20</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>104.0 ATP[c] + 0.0324116276714271 BioPool[c] + 0.0268408791654006 Bmineral[c] + 0.0559910868023904 Cell_wall[c] + 0.00111921401802897 Free_nucleic_acids[c] + 104.0 H2O[c] + 0.649245416793275...</p>\n",
       "                    <p style='text-align:right'>104.0 ATP + 0.0324116276714271 BioPool + 0.0268408791654006 Trace metals and minerals + 0.0559910868023904 Cell wall Biomass precursor + 0.00111921401802897 Free nucleic acids + 104.0 H2O +...</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction BIOMASS at 0x7f911ea4bb20>"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions.get_by_id('BIOMASS')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "fpath = os.path.join('data','PRO_N_sources.xlsx')\n",
    "df_growing = pd.read_excel(fpath, sheet_name='uptake')\n",
    "n_exchange_df = pd.read_excel(fpath, sheet_name='secretion')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Knockout genes - to find genes required for uptake/secretion"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Partition genes and reactions to essential and non-essential"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Objective\n",
      "=========\n",
      "1.0 BIOMASS = 1.318332141786151\n",
      "\n",
      "Uptake\n",
      "------\n",
      "         Metabolite             Reaction      Flux  C-Number C-Flux\n",
      "         Cadmium[e]            CadmiumEX 6.363E-05         0  0.00%\n",
      "  Calcium_cation[e]            CalciumEX   0.03529         0  0.00%\n",
      "    Chloride_ion[e]           ChlorideEX    0.3136         0  0.00%\n",
      "      Cobalt_ion[e]             CobaltEX  0.002046         0  0.00%\n",
      "          Copper[e]             CopperEX 0.0001151         0  0.00%\n",
      "  Orthophosphate[e] FAKEOrthophosphateEX     0.479         0  0.00%\n",
      "     Glutathione[e]        GlutathioneEX    0.6745         0  0.00%\n",
      "       Guanosine[e]          GuanosineEX     22.81         0  0.00%\n",
      "            HCO3[e]            HCO3EXcar         8         1  5.49%\n",
      "             Fe2[e]               IronEX  0.004261         0  0.00%\n",
      "       L_Alanine[e]          L_AlanineEX    0.5777         3  1.19%\n",
      "      L_Arginine[e]         L_ArginineEX    0.5092         6  2.09%\n",
      "     L_Glutamine[e]        L_GlutamineEX    0.9174         5  3.15%\n",
      "       L_Leucine[e]          L_LeucineEX    0.5581         6  2.30%\n",
      "        L_Serine[e]           L_SerineEX     40.93         3 84.19%\n",
      "        L_Valine[e]           L_ValineEX    0.4673         5  1.60%\n",
      "          Photon[e]              LightEX    0.2396         0  0.00%\n",
      "Magnesium_cation[e]          MagnesiumEX    0.2095         0  0.00%\n",
      "      Molybdenum[e]         MolybdenumEX  9.09E-06         0  0.00%\n",
      "          Oxygen[e]                 O2EX     23.34         0  0.00%\n",
      "               K[e]          PotassiumEX    0.1255         0  0.00%\n",
      "        Selenate[e]           SelenateEX    0.1007         0  0.00%\n",
      "   Sodium_cation[e]             SodiumEX    0.5881         0  0.00%\n",
      "Strontium_cation[e]          StrontiumEX  0.001515         0  0.00%\n",
      "         Sulfate[e]            SulfateEX    0.3284         0  0.00%\n",
      "             Zn2[e]               ZincEX 0.0002424         0  0.00%\n",
      "\n",
      "Secretion\n",
      "---------\n",
      "                               Metabolite                            Reaction       Flux  C-Number C-Flux\n",
      "                               biomass[e]                           BiomassEX     -1.318         0  0.00%\n",
      "                                   CO2[e]                               CO2EX      -0.82         1  1.08%\n",
      "                                    CO[e]                                COEX -1.122E-05         1  0.00%\n",
      "                               Citrate[e]                           CitrateEX     -5.034         0  0.00%\n",
      "                               Formate[e]                           FormateEX     -53.16         0  0.00%\n",
      "                               Glycine[e]                           GlycineEX     -37.27         2 98.20%\n",
      "                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.007952         2  0.02%\n",
      "                               Guanine[e]                           GuanineEX     -22.73         0  0.00%\n",
      "                     Hydrogen_peroxide[e]                              H2O2EX    -0.1467         0  0.00%\n",
      "                                   H2O[e]                               H2OEX     -36.97         0  0.00%\n",
      "                              Methanol[e]                          MethanolEX  -0.006139         1  0.01%\n",
      "                              Pyruvate[e]                          PyruvateEX     -5.994         0  0.00%\n",
      "S_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX  -0.006139        15  0.12%\n",
      "              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.06004         6  0.47%\n",
      "                              S_Malate[e]                          S_MalateEX     -5.814         0  0.00%\n",
      "                                  Urea[e]                              UreaEX   -0.06004         1  0.08%\n",
      "                                  tRNA[e]                              tRNAEX -0.0006875        15  0.01%\n",
      "\n",
      "done genes\n"
     ]
    }
   ],
   "source": [
    "# open all potential mediums and run gene knockout to see what is blocking growth when all N sources are available \n",
    "with model:\n",
    "    medium = model.medium\n",
    "    medium[\"AmmoniaEX\"] = 0.0\n",
    "    for i in df_growing.uptake:\n",
    "        medium[i] = 1000.0\n",
    "    model.medium = medium\n",
    "    model.reactions.BIOMASS.lower_bound = 1e-5\n",
    "    solution = model.optimize()\n",
    "    print(model.summary())\n",
    "    essential_gene_df = single_gene_deletion(model, processes=10)\n",
    "    print('done genes')\n",
    "    essential_reaction_df = single_reaction_deletion(model, processes=10)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ids</th>\n",
       "      <th>growth</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>{PMM1337}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>{PMM0304}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>{PMM0508}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>{PMM0243}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>{PMM1142}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>{PMM0433}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>{PMM1592}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>{PMM0888}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>591</th>\n",
       "      <td>{PMM0045}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592</th>\n",
       "      <td>{PMM0825}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>286 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           ids  growth      status\n",
       "0    {PMM1337}     NaN  infeasible\n",
       "2    {PMM0304}     NaN  infeasible\n",
       "3    {PMM0508}     NaN  infeasible\n",
       "8    {PMM0243}     NaN  infeasible\n",
       "11   {PMM1142}     NaN  infeasible\n",
       "..         ...     ...         ...\n",
       "585  {PMM0433}     NaN  infeasible\n",
       "586  {PMM1592}     NaN  infeasible\n",
       "588  {PMM0888}     NaN  infeasible\n",
       "591  {PMM0045}     NaN  infeasible\n",
       "592  {PMM0825}     NaN  infeasible\n",
       "\n",
       "[286 rows x 3 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "essential_gene_df.loc[~essential_gene_df.status.isin(['optimal'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ids</th>\n",
       "      <th>growth</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>{R04519}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>{R00416}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>{R03508}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>{R03471}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>{R00084}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>{R01150}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>{R05637}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>{R04550}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>{R04606}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>{R00480}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>335 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          ids  growth      status\n",
       "0    {R04519}     NaN  infeasible\n",
       "1    {R00416}     NaN  infeasible\n",
       "2    {R03508}     NaN  infeasible\n",
       "3    {R03471}     NaN  infeasible\n",
       "4    {R00084}     NaN  infeasible\n",
       "..        ...     ...         ...\n",
       "979  {R01150}     NaN  infeasible\n",
       "980  {R05637}     NaN  infeasible\n",
       "984  {R04550}     NaN  infeasible\n",
       "987  {R04606}     NaN  infeasible\n",
       "989  {R00480}     NaN  infeasible\n",
       "\n",
       "[335 rows x 3 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "essential_reaction_df.loc[~essential_reaction_df.status.isin(['optimal'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "non_essential_genes = set().union(*essential_gene_df.loc[essential_gene_df.status.isin(['optimal']), 'ids'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "non_essential_reactions = set().union(*essential_reaction_df.loc[essential_reaction_df.status.isin(['optimal']), 'ids'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(309, 659)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_essential_genes), len(non_essential_reactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "essential_gene_df.to_csv(os.path.join('data', 'MED4_essential_genes.csv'))\n",
    "essential_reaction_df.to_csv(os.path.join('data', 'MED4_essential_reactions.csv'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "essential_gene_df = pd.read_csv(os.path.join('data', 'MED4_essential_genes.csv'))\n",
    "essential_reaction_df = pd.read_csv(os.path.join('data', 'MED4_essential_reactions.csv'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "essential_gene_df['gid']  = essential_gene_df.ids.str.extract(r\"([^\\{\\}\\']+)\")\n",
    "essential_reaction_df['rid']  = essential_reaction_df.ids.str.extract(r\"([^\\{\\}\\']+)\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Double knockdowns\n",
    "\n",
    "Some of the genes removed by double knockout are clearly related to uptake. So decided to avoid doing double knockdowns"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "# open all potential mediums and run gene knockout to see what is blocking growth when all N sources are available \n",
    "with model:\n",
    "    medium = model.medium\n",
    "    medium[\"AmmoniaEX\"] = 0.0\n",
    "    for i in df_growing.uptake:\n",
    "        medium[i] = 1000.0\n",
    "    model.medium = medium\n",
    "    model.reactions.BIOMASS.lower_bound = 1e-5\n",
    "    solution = model.optimize()\n",
    "    print(model.summary())\n",
    "    essential_2gene_df = double_gene_deletion(model, processes=10, \n",
    "                                             gene_list1=list(non_essential_genes),\n",
    "                                             gene_list2=list(non_essential_genes), \n",
    "                                            )\n",
    "    print('done genes')\n",
    "    #essential_reaction_df = single_reaction_deletion(model, processes=10)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1      {'PMM1084'}\n",
       "4      {'PMM0468'}\n",
       "5      {'PMM1450'}\n",
       "6      {'PMM1627'}\n",
       "7      {'PMM1525'}\n",
       "          ...     \n",
       "587    {'PMM0618'}\n",
       "589    {'PMM1216'}\n",
       "590    {'PMM0841'}\n",
       "593    {'PMM0297'}\n",
       "594    {'PMM1152'}\n",
       "Name: ids, Length: 309, dtype: object"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "essential_gene_df.loc[essential_gene_df.growth > 1e-5].ids"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "essential_2gene_df.loc[essential_2gene_df.growth > 1e-5].ids"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "essential_2gene_df.to_csv('data/MED4_double_gene_knockout.csv.gz')"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "essential_2genes_list = essential_2gene_df.loc[essential_2gene_df.status.isin(['infeasible']) | \n",
    "                      (essential_2gene_df.growth<1e-5)]['ids']"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "essential_2genes_list"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "essential_2genes_set = set().union(*essential_2genes_list)\n",
    "len(essential_2genes_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "essential_genes_list = essential_gene_df.loc[essential_gene_df.status.isin(['infeasible']) | \n",
    "                      (essential_gene_df.growth<=1e-5)]['gid']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      PMM1337\n",
       "2      PMM0304\n",
       "3      PMM0508\n",
       "8      PMM0243\n",
       "11     PMM1142\n",
       "        ...   \n",
       "585    PMM0433\n",
       "586    PMM1592\n",
       "588    PMM0888\n",
       "591    PMM0045\n",
       "592    PMM0825\n",
       "Name: gid, Length: 286, dtype: object"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "essential_genes_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#non_essential_genes = [g.id for g in model.genes if g.id not in essential_combined_genes_set]\n",
    "non_essential_genes = [g.id for g in model.genes if g.id not in essential_genes_list]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7              R05221\n",
       "8              R01055\n",
       "12      Trans_Formate\n",
       "13             R01078\n",
       "16             R00963\n",
       "            ...      \n",
       "985            R01618\n",
       "988            R01708\n",
       "990            R02063\n",
       "992            R10086\n",
       "993    Trans_S_Malate\n",
       "Name: rid, Length: 579, dtype: object"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_essential_reactions = essential_reaction_df.loc[\n",
    "    essential_reaction_df.status.isin(['optimal']) & \n",
    "    (essential_reaction_df.growth > 1e-5) &\n",
    "    (~essential_reaction_df.rid.isin(exchange_ids))\n",
    "    , 'rid']\n",
    "non_essential_reactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(579, 595)"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_essential_reactions), len(non_essential_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(os.path.join('data', 'MED4_del_gene_list.txt'), 'w') as fh:\n",
    "    fh.writelines(f'{i}\\n' for i in non_essential_genes)\n",
    "with open(os.path.join('data', 'MED4_del_reaction_list.txt'), 'w') as fh:\n",
    "    fh.writelines(f'{i}\\n' for i in non_essential_reactions)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Try to delete genes and reactions and see what is breaking the model per uptake option"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(os.path.join('data', 'MED4_del_gene_list.txt')) as fh:\n",
    "    non_essential_genes = [i.rstrip() for i in fh.readlines()]\n",
    "    \n",
    "with open(os.path.join('data', 'MED4_del_reaction_list.txt')) as fh:\n",
    "    non_essential_reactions = [i.rstrip() for i in fh.readlines()]\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _uptake_deletion_feeding_with_timeout(dpath, uptake1, deltype, timeout=10):\n",
    "    try:\n",
    "        print (uptake1)\n",
    "        list_file = os.path.join('data', f'MED4_del_{deltype}_list.txt')\n",
    "            \n",
    "        result = subprocess.run([\n",
    "            './prochlorococcus_feed_knockout.py', \n",
    "            '--uptake', uptake1, \n",
    "            '--out_dpath', dpath,\n",
    "            '--deltype', deltype,\n",
    "            '--list_file', list_file,\n",
    "            ],\n",
    "            capture_output=True, text=True, check=True, timeout=timeout,\n",
    "        )\n",
    "        #print(\"stdout:\", result.stdout)\n",
    "        #print(\"stderr:\", result.stderr)\n",
    "        return 'success'\n",
    "        \n",
    "    except subprocess.CalledProcessError as err:\n",
    "        print('CalledProcessError', err.returncode)\n",
    "        print(\"stdout:\", err.stdout)\n",
    "        print(\"stderr:\", err.stderr)\n",
    "        return 'error'\n",
    "                \n",
    "    except subprocess.TimeoutExpired as err:\n",
    "        print('TimeoutExpired', err.timeout)\n",
    "        print(\"stdout:\", err.stdout)\n",
    "        print(\"stderr:\", err.stderr)\n",
    "        return 'timeout'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GuanosineEX\n",
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGuanosineEX None Objective\\n=========\\n1.0 BIOMASS = 1.23944740203336\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 5.982E-05         0   0.00%\\n  Calcium_cation[e]            CalciumEX   0.03317         0   0.00%\\n    Chloride_ion[e]           ChlorideEX    0.2949         0   0.00%\\n      Cobalt_ion[e]             CobaltEX  0.001923         0   0.00%\\n          Copper[e]             CopperEX 0.0001083         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX    0.4503         0   0.00%\\n       Guanosine[e]          GuanosineEX     48.05         0   0.00%\\n            HCO3[e]            HCO3EXcar         8         1 100.00%\\n             Fe2[e]               IronEX  0.004006         0   0.00%\\n          Photon[e]              LightEX    0.2252         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX    0.1969         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 8.546E-06         0   0.00%\\n          Oxygen[e]                 O2EX     13.78         0   0.00%\\n               K[e]          PotassiumEX     0.118         0   0.00%\\n        Selenate[e]           SelenateEX   0.09464         0   0.00%\\n   Sodium_cation[e]             SodiumEX    0.5529         0   0.00%\\nStrontium_cation[e]          StrontiumEX  0.001424         0   0.00%\\n         Sulfate[e]            SulfateEX    0.9429         0   0.00%\\n             Zn2[e]               ZincEX 0.0002279         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -1.239         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 64.28%\\n                                    CO[e]                                COEX -1.055E-05         1  0.00%\\n                               Formate[e]                           FormateEX     -29.11         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.007477         2  1.17%\\n                               Guanine[e]                           GuanineEX      -37.6         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX    -0.1379         0  0.00%\\n                                   H2O[e]                               H2OEX     -36.49         0  0.00%\\n                              Methanol[e]                          MethanolEX  -0.005772         1  0.45%\\n                              Pyruvate[e]                          PyruvateEX     -34.61         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX  -0.005772        15  6.79%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.05645         6 26.55%\\n                            Xanthosine[e]                        XanthosineEX     -10.37         0  0.00%\\n                                  tRNA[e]                              tRNAEX -0.0006464        15  0.76%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX\n",
      "GlutathioneEX\n",
      "L_GlutamineEX\n",
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamineEX None Objective\\n=========\\n1.0 BIOMASS = 0.12192265750448544\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 5.885E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.003263         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02901         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001892         0  0.00%\\n          Copper[e]             CopperEX 1.065E-05         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.04429         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.495         1 58.16%\\n             Fe2[e]               IronEX  0.000394         0  0.00%\\n     L_Glutamine[e]        L_GlutamineEX    0.7904         5 41.84%\\n          Photon[e]              LightEX     78.68         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01937         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 8.407E-07         0  0.00%\\n               K[e]          PotassiumEX    0.0116         0  0.00%\\n        Selenate[e]           SelenateEX  0.009309         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.05439         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001401         0  0.00%\\n         Sulfate[e]            SulfateEX   0.09275         0  0.00%\\n             Zn2[e]               ZincEX 2.242E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.1219         0  0.00%\\n                                   CO2[e]                               CO2EX     -0.637         1 19.35%\\n                                    CO[e]                                COEX -1.037E-06         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0007355         2  0.04%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01356         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.218         0  0.00%\\n                                     H[e]                                 HEX     -14.12         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX     -0.522         5 79.29%\\n                              Methanol[e]                          MethanolEX -0.0005678         1  0.02%\\n                                Oxygen[e]                                O2EX      -6.04         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0005678        15  0.26%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.005553         6  1.01%\\n                                  tRNA[e]                              tRNAEX -6.358E-05        15  0.03%\\n\\n'\n",
      "stderr: None\n",
      "L_SerineEX\n",
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_SerineEX None Objective\\n=========\\n1.0 BIOMASS = 0.1189721893957353\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 5.742E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.003184         0  0.00%\\n    Chloride_ion[e]           ChlorideEX    0.0283         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001846         0  0.00%\\n          Copper[e]             CopperEX 1.039E-05         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.04322         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.475         1  8.04%\\n               H[e]                  HEX     4.669         0  0.00%\\n             Fe2[e]               IronEX 0.0003845         0  0.00%\\n        L_Serine[e]           L_SerineEX     20.88         3 91.96%\\n          Photon[e]              LightEX   0.02162         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX    0.0189         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 8.203E-07         0  0.00%\\n          Oxygen[e]                 O2EX     3.996         0  0.00%\\n               K[e]          PotassiumEX   0.01132         0  0.00%\\n        Selenate[e]           SelenateEX  0.009084         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.05307         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001367         0  0.00%\\n         Sulfate[e]            SulfateEX   0.09051         0  0.00%\\n             Zn2[e]               ZincEX 2.188E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.119         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  2.02%\\n                                    CO[e]                                COEX -1.012E-06         1  0.00%\\n                               Formate[e]                           FormateEX     -19.79         0  0.00%\\n                               Glycine[e]                           GlycineEX     -19.83         2 97.86%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0007177         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01324         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.166         0  0.00%\\n                              Methanol[e]                          MethanolEX  -0.000554         1  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.5902         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX  -0.000554        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.005418         6  0.08%\\n                                  Urea[e]                              UreaEX  -0.005418         1  0.01%\\n                                  tRNA[e]                              tRNAEX -6.205E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_CystineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_CystineEX None Objective\\n=========\\n1.0 BIOMASS = 0.11645172899918435\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 5.621E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.003117         0  0.00%\\n    Chloride_ion[e]           ChlorideEX    0.0277         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001807         0  0.00%\\n          Copper[e]             CopperEX 1.017E-05         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.04231         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.459         1 64.04%\\n             Fe2[e]               IronEX 0.0003764         0  0.00%\\n       L_Cystine[e]          L_CystineEX     0.511         6 35.96%\\n          Photon[e]              LightEX     77.06         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX    0.0185         0  0.00%\\n      Molybdenum[e]         MolybdenumEX  8.03E-07         0  0.00%\\n               K[e]          PotassiumEX   0.01108         0  0.00%\\n        Selenate[e]           SelenateEX  0.008892         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.05195         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001338         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02901         0  0.00%\\n             Zn2[e]               ZincEX 2.141E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.1165         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 94.46%\\n                                    CO[e]                                COEX -9.908E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0007025         2  0.16%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01296         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.556         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX    -0.9624         0  0.00%\\n                                     H[e]                                 HEX     -13.93         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0005423         1  0.06%\\n                                Oxygen[e]                                O2EX     -5.986         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.5926         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0005423        15  0.94%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.005303         6  3.67%\\n                                  Urea[e]                              UreaEX  -0.005303         1  0.61%\\n                                  tRNA[e]                              tRNAEX -6.073E-05        15  0.10%\\n\\n'\n",
      "stderr: None\n",
      "L_AlanineEX\n",
      "L_LeucineEX\n",
      "GlycineEX\n",
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlycineEX None Objective\\n=========\\n1.0 BIOMASS = 0.101625655084699\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.905E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX   0.00272         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02418         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001577         0  0.00%\\n          Copper[e]             CopperEX 8.876E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03692         0  0.00%\\n         Glycine[e]            GlycineEX    0.8826         2 24.77%\\n            HCO3[e]            HCO3EXcar     5.362         1 75.23%\\n             Fe2[e]               IronEX 0.0003284         0  0.00%\\n          Photon[e]              LightEX      72.8         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01615         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 7.007E-07         0  0.00%\\n               K[e]          PotassiumEX  0.009671         0  0.00%\\n        Selenate[e]           SelenateEX   0.00776         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.04533         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001168         0  0.00%\\n         Sulfate[e]            SulfateEX   0.07731         0  0.00%\\n             Zn2[e]               ZincEX 1.869E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.1016         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 95.64%\\n                                    CO[e]                                COEX -8.646E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000613         2  0.14%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01131         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.409         0  0.00%\\n                                     H[e]                                 HEX     -6.104         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0004733         1  0.06%\\n                                Oxygen[e]                                O2EX     -5.668         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3799         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004733        15  0.83%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004628         6  3.24%\\n                                  tRNA[e]                              tRNAEX   -5.3E-05        15  0.09%\\n\\n'\n",
      "stderr: None\n",
      "UreaEX\n",
      "AmmoniaEX\n",
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX None Objective\\n=========\\n1.0 BIOMASS = 0.09846248800237388\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.8551         0   0.00%\\n         Cadmium[e]            CadmiumEX 4.752E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.002635         0   0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02343         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001528         0   0.00%\\n          Copper[e]             CopperEX 8.599E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03577         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.342         1 100.00%\\n             Fe2[e]               IronEX 0.0003182         0   0.00%\\n          Photon[e]              LightEX     73.12         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01565         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 6.789E-07         0   0.00%\\n               K[e]          PotassiumEX   0.00937         0   0.00%\\n        Selenate[e]           SelenateEX  0.007518         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.04392         0   0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001131         0   0.00%\\n         Sulfate[e]            SulfateEX   0.07491         0   0.00%\\n             Zn2[e]               ZincEX  1.81E-05         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.09846         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.3348         1 90.24%\\n                                    CO[e]                                COEX -8.377E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0005939         2  0.32%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01095         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.299         0  0.00%\\n                                     H[e]                                 HEX     -12.94         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0004585         1  0.12%\\n                                Oxygen[e]                                O2EX     -5.854         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004585        15  1.85%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004484         6  7.25%\\n                                  tRNA[e]                              tRNAEX -5.135E-05        15  0.21%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX None Objective\\n=========\\n1.0 BIOMASS = 0.09726135236073205\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.694E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.002603         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02314         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001509         0  0.00%\\n          Copper[e]             CopperEX 8.495E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03534         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.334         1 41.94%\\n             Fe2[e]               IronEX 0.0003143         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX     1.477         5 58.06%\\n          Photon[e]              LightEX     82.63         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01545         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 6.706E-07         0  0.00%\\n               K[e]          PotassiumEX  0.009256         0  0.00%\\n        Selenate[e]           SelenateEX  0.007426         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.04339         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001118         0  0.00%\\n         Sulfate[e]            SulfateEX   0.07399         0  0.00%\\n             Zn2[e]               ZincEX 1.788E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.09726         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 95.82%\\n                                    CO[e]                                COEX -8.275E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0005867         2  0.14%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01082         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.416         0  0.00%\\n                                     H[e]                                 HEX     -15.33         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0004529         1  0.05%\\n                                Oxygen[e]                                O2EX     -6.484         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004529        15  0.79%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004429         6  3.11%\\n                                  tRNA[e]                              tRNAEX -5.072E-05        15  0.09%\\n\\n'\n",
      "stderr: None\n",
      "L_ArginineEX\n",
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX None Objective\\n=========\\n1.0 BIOMASS = 0.09479172165768338\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.575E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.002537         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02255         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001471         0  0.00%\\n          Copper[e]             CopperEX 8.279E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03444         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.281         1 70.12%\\n             Fe2[e]               IronEX 0.0003064         0  0.00%\\n      L_Arginine[e]         L_ArginineEX     0.375         6 29.88%\\n          Photon[e]              LightEX     72.37         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01506         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 6.536E-07         0  0.00%\\n               K[e]          PotassiumEX  0.009021         0  0.00%\\n        Selenate[e]           SelenateEX  0.007238         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.04228         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001089         0  0.00%\\n         Sulfate[e]            SulfateEX   0.07211         0  0.00%\\n             Zn2[e]               ZincEX 1.743E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.09479         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 95.92%\\n                                    CO[e]                                COEX -8.065E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0005718         2  0.13%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01055         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.279         0  0.00%\\n                                     H[e]                                 HEX     -12.78         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0004414         1  0.05%\\n                                Oxygen[e]                                O2EX     -5.882         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.3384         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1791         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004414        15  0.77%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004317         6  3.03%\\n                                  tRNA[e]                              tRNAEX -4.943E-05        15  0.09%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX\n",
      "L_ValineEX\n",
      "TimeoutExpired 20\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX None Objective\\n=========\\n1.0 BIOMASS = 0.07493417708047853\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 3.617E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.002006         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01783         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001163         0  0.00%\\n          Copper[e]             CopperEX 6.545E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02722         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.188         1 61.46%\\n             Fe2[e]               IronEX 0.0002422         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.6508         5 38.54%\\n          Photon[e]              LightEX     63.81         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01191         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 5.167E-07         0  0.00%\\n               K[e]          PotassiumEX  0.007131         0  0.00%\\n        Selenate[e]           SelenateEX  0.005722         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.03343         0  0.00%\\nStrontium_cation[e]          StrontiumEX 8.611E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.05701         0  0.00%\\n             Zn2[e]               ZincEX 1.378E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.5925         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.07493         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 96.75%\\n                                    CO[e]                                COEX -6.375E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000452         2  0.11%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.008337         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.864         0  0.00%\\n                                     H[e]                                 HEX     -4.732         0  0.00%\\n                              Methanol[e]                          MethanolEX  -0.000349         1  0.04%\\n                                Oxygen[e]                                O2EX     -4.912         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX   -0.08559         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX  -0.000349        15  0.62%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.003413         6  2.42%\\n                                  tRNA[e]                              tRNAEX -3.908E-05        15  0.07%\\n\\n'\n",
      "stderr: None\n",
      "GuanineEX\n"
     ]
    }
   ],
   "source": [
    "deltype = 'gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_uptake_29082022')\n",
    "knockout_success = [\n",
    "    _uptake_deletion_feeding_with_timeout(dpath, u, deltype, timeout=20)\n",
    "    for u in df_growing.uptake\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['timeout', 'success', 'success', 'success', 'success', 'success']"
      ]
     },
     "execution_count": 380,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knockout_success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "timeout    9\n",
       "success    7\n",
       "Name: gknockout_success1, dtype: int64"
      ]
     },
     "execution_count": 375,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_growing['gknockout_success1'] = knockout_success\n",
    "df_growing['gknockout_success1'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 382,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_GlutamineEX\n",
      "L_SerineEX\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_1858647/2110985753.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mdeltype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'gene'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mdpath\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'data'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34mf'{deltype}_knockout_uptake_29082022'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m knockout_success = [\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0m_uptake_deletion_feeding_with_timeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdpath\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mu\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdeltype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mu\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdf_growing\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf_growing\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgknockout_success2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'timeout'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'uptake'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/tmp/ipykernel_1858647/2110985753.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mdpath\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'data'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34mf'{deltype}_knockout_uptake_29082022'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m knockout_success = [\n\u001b[0;32m----> 4\u001b[0;31m     \u001b[0m_uptake_deletion_feeding_with_timeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdpath\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mu\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdeltype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mu\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdf_growing\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf_growing\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgknockout_success2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'timeout'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'uptake'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m ]\n",
      "\u001b[0;32m/tmp/ipykernel_1858647/3663326180.py\u001b[0m in \u001b[0;36m_uptake_deletion_feeding_with_timeout\u001b[0;34m(dpath, uptake1, deltype, timeout)\u001b[0m\n\u001b[1;32m      4\u001b[0m         \u001b[0mlist_file\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'data'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34mf'MED4_del_{deltype}_list.txt'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m         result = subprocess.run([\n\u001b[0m\u001b[1;32m      7\u001b[0m             \u001b[0;34m'./prochlorococcus_feed_knockout.py'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m             \u001b[0;34m'--uptake'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0muptake1\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/work/tools/anaconda3/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36mrun\u001b[0;34m(input, capture_output, timeout, check, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    505\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mPopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mpopenargs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mprocess\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    506\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 507\u001b[0;31m             \u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcommunicate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    508\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mTimeoutExpired\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    509\u001b[0m             \u001b[0mprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkill\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/work/tools/anaconda3/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36mcommunicate\u001b[0;34m(self, input, timeout)\u001b[0m\n\u001b[1;32m   1132\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1133\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1134\u001b[0;31m                 \u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_communicate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mendtime\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1135\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyboardInterrupt\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1136\u001b[0m                 \u001b[0;31m# https://bugs.python.org/issue25942\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/work/tools/anaconda3/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36m_communicate\u001b[0;34m(self, input, endtime, orig_timeout)\u001b[0m\n\u001b[1;32m   1977\u001b[0m                             'failed to raise TimeoutExpired.')\n\u001b[1;32m   1978\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1979\u001b[0;31m                     \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mselector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mselect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1980\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_timeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mendtime\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morig_timeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1981\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/work/tools/anaconda3/lib/python3.9/selectors.py\u001b[0m in \u001b[0;36mselect\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    414\u001b[0m         \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    415\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 416\u001b[0;31m             \u001b[0mfd_event_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_selector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    417\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mInterruptedError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    418\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mready\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "deltype = 'gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_uptake_29082022')\n",
    "knockout_success = [\n",
    "    _uptake_deletion_feeding_with_timeout(dpath, u, deltype, timeout=1000)\n",
    "    for u in df_growing.loc[df_growing.gknockout_success2.isin(['timeout']), 'uptake']\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "success    5\n",
       "timeout    1\n",
       "Name: gknockout_success3, dtype: int64"
      ]
     },
     "execution_count": 381,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_growing.loc[df_growing.gknockout_success2.isin(['timeout']), 'gknockout_success3'] = knockout_success\n",
    "df_growing['gknockout_success3'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GuanosineEX\n",
      "L_AspartateEX\n",
      "GlutathioneEX\n",
      "L_GlutamineEX\n",
      "L_SerineEX\n",
      "TimeoutExpired 26\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_SerineEX None Objective\\n=========\\n1.0 BIOMASS = 0.1189721893957353\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 5.742E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.003184         0  0.00%\\n    Chloride_ion[e]           ChlorideEX    0.0283         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001846         0  0.00%\\n          Copper[e]             CopperEX 1.039E-05         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.04322         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.475         1  8.04%\\n               H[e]                  HEX     4.669         0  0.00%\\n             Fe2[e]               IronEX 0.0003845         0  0.00%\\n        L_Serine[e]           L_SerineEX     20.88         3 91.96%\\n          Photon[e]              LightEX   0.02162         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX    0.0189         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 8.203E-07         0  0.00%\\n          Oxygen[e]                 O2EX     3.996         0  0.00%\\n               K[e]          PotassiumEX   0.01132         0  0.00%\\n        Selenate[e]           SelenateEX  0.009084         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.05307         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001367         0  0.00%\\n         Sulfate[e]            SulfateEX   0.09051         0  0.00%\\n             Zn2[e]               ZincEX 2.188E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.119         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  2.02%\\n                                    CO[e]                                COEX -1.012E-06         1  0.00%\\n                               Formate[e]                           FormateEX     -19.79         0  0.00%\\n                               Glycine[e]                           GlycineEX     -19.83         2 97.86%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0007177         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01324         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.166         0  0.00%\\n                              Methanol[e]                          MethanolEX  -0.000554         1  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.5902         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX  -0.000554        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.005418         6  0.08%\\n                                  Urea[e]                              UreaEX  -0.005418         1  0.01%\\n                                  tRNA[e]                              tRNAEX -6.205E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_CystineEX\n",
      "L_AlanineEX\n",
      "L_LeucineEX\n",
      "GlycineEX\n",
      "TimeoutExpired 26\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlycineEX None Objective\\n=========\\n1.0 BIOMASS = 0.101625655084699\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.905E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX   0.00272         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02418         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001577         0  0.00%\\n          Copper[e]             CopperEX 8.876E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03692         0  0.00%\\n         Glycine[e]            GlycineEX    0.8826         2 24.77%\\n            HCO3[e]            HCO3EXcar     5.362         1 75.23%\\n             Fe2[e]               IronEX 0.0003284         0  0.00%\\n          Photon[e]              LightEX      72.8         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01615         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 7.007E-07         0  0.00%\\n               K[e]          PotassiumEX  0.009671         0  0.00%\\n        Selenate[e]           SelenateEX   0.00776         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.04533         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001168         0  0.00%\\n         Sulfate[e]            SulfateEX   0.07731         0  0.00%\\n             Zn2[e]               ZincEX 1.869E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.1016         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 95.64%\\n                                    CO[e]                                COEX -8.646E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000613         2  0.14%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01131         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.409         0  0.00%\\n                                     H[e]                                 HEX     -6.104         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0004733         1  0.06%\\n                                Oxygen[e]                                O2EX     -5.668         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3799         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004733        15  0.83%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004628         6  3.24%\\n                                  tRNA[e]                              tRNAEX   -5.3E-05        15  0.09%\\n\\n'\n",
      "stderr: None\n",
      "UreaEX\n",
      "AmmoniaEX\n",
      "L_GlutamateEX\n",
      "TimeoutExpired 26\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX None Objective\\n=========\\n1.0 BIOMASS = 0.09726135236073205\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.694E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.002603         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02314         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001509         0  0.00%\\n          Copper[e]             CopperEX 8.495E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03534         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.334         1 41.94%\\n             Fe2[e]               IronEX 0.0003143         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX     1.477         5 58.06%\\n          Photon[e]              LightEX     82.63         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01545         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 6.706E-07         0  0.00%\\n               K[e]          PotassiumEX  0.009256         0  0.00%\\n        Selenate[e]           SelenateEX  0.007426         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.04339         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001118         0  0.00%\\n         Sulfate[e]            SulfateEX   0.07399         0  0.00%\\n             Zn2[e]               ZincEX 1.788E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.09726         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 95.82%\\n                                    CO[e]                                COEX -8.275E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0005867         2  0.14%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01082         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.416         0  0.00%\\n                                     H[e]                                 HEX     -15.33         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0004529         1  0.05%\\n                                Oxygen[e]                                O2EX     -6.484         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004529        15  0.79%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004429         6  3.11%\\n                                  tRNA[e]                              tRNAEX -5.072E-05        15  0.09%\\n\\n'\n",
      "stderr: None\n",
      "L_ArginineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CyanateEX\n",
      "TimeoutExpired 26\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX None Objective\\n=========\\n1.0 BIOMASS = 0.08817277250778842\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.256E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX   0.00236         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02098         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001368         0  0.00%\\n          Copper[e]             CopperEX 7.701E-06         0  0.00%\\n         Cyanate[e]            CyanateEX    0.7658         1 12.68%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03203         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.275         1 87.32%\\n             Fe2[e]               IronEX  0.000285         0  0.00%\\n          Photon[e]              LightEX     71.18         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01401         0  0.00%\\n      Molybdenum[e]         MolybdenumEX  6.08E-07         0  0.00%\\n               K[e]          PotassiumEX  0.008391         0  0.00%\\n        Selenate[e]           SelenateEX  0.006732         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.03933         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001013         0  0.00%\\n         Sulfate[e]            SulfateEX   0.06708         0  0.00%\\n             Zn2[e]               ZincEX 1.621E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.08817         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 96.20%\\n                                    CO[e]                                COEX -7.502E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0005319         2  0.12%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.009809         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.084         0  0.00%\\n                                     H[e]                                 HEX     -11.76         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0004106         1  0.05%\\n                                Oxygen[e]                                O2EX     -5.856         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.2456         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004106        15  0.72%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004016         6  2.83%\\n                                  tRNA[e]                              tRNAEX -4.598E-05        15  0.08%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX\n",
      "GuanineEX\n"
     ]
    }
   ],
   "source": [
    "deltype = 'reaction'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_uptake_29082022')\n",
    "knockout_success = [\n",
    "    _uptake_deletion_feeding_with_timeout(dpath, u, deltype, timeout=26)\n",
    "    for u in df_growing.uptake\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['timeout', 'success', 'success', 'success']"
      ]
     },
     "execution_count": 388,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knockout_success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "success    12\n",
       "timeout     4\n",
       "Name: rknockout_success1, dtype: int64"
      ]
     },
     "execution_count": 385,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_growing['rknockout_success1'] = knockout_success\n",
    "df_growing['rknockout_success1'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_SerineEX\n"
     ]
    }
   ],
   "source": [
    "deltype = 'reaction'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_uptake_29082022')\n",
    "knockout_success = [\n",
    "    _uptake_deletion_feeding_with_timeout(dpath, u, deltype, timeout=60)\n",
    "    for u in df_growing.loc[df_growing.rknockout_success2.isin(['timeout']), 'uptake']\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "success    3\n",
       "timeout    1\n",
       "Name: rknockout_success2, dtype: int64"
      ]
     },
     "execution_count": 389,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_growing.loc[df_growing.rknockout_success1.isin(['timeout']), 'rknockout_success2'] = knockout_success\n",
    "df_growing['rknockout_success2'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 391,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['success']"
      ]
     },
     "execution_count": 391,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knockout_success"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Knockouts for secretion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 799,
   "metadata": {},
   "outputs": [],
   "source": [
    "non_essential_genes_secretion = [g for g in non_essential_genes if g not in gene_uptake_knockout_df.gid.values]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 790,
   "metadata": {},
   "outputs": [],
   "source": [
    "non_essential_reactions_secretion = [r for r in non_essential_reactions if r not in reaction_uptake_knockout_df.rid.values]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 791,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(595, 579, 171, 504)"
      ]
     },
     "execution_count": 791,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_essential_genes), len(non_essential_reactions), len(non_essential_genes_secretion), len(non_essential_reactions_secretion)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 794,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(os.path.join('data', 'MED4_del_gene_secretion_list.txt'), 'w') as fh:\n",
    "    fh.writelines(f'{i}\\n' for i in non_essential_genes_secretion)\n",
    "with open(os.path.join('data', 'MED4_del_reaction_secretion_list.txt'), 'w') as fh:\n",
    "    fh.writelines(f'{i}\\n' for i in non_essential_reactions_secretion)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 798,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _secretion_deletion_feeding_with_timeout(dpath, uptake1, sec1, bound, deltype, timeout=10):\n",
    "    try:\n",
    "        print (uptake1, sec1)\n",
    "        list_file = os.path.join('data', f'MED4_del_{deltype}_secretion_list.txt')\n",
    "            \n",
    "        result = subprocess.run([\n",
    "            './prochlorococcus_feed_knockout.py', \n",
    "            '--uptake', uptake1, \n",
    "            '--secretion', sec1,\n",
    "            '--bound', str(bound),\n",
    "            '--out_dpath', dpath,\n",
    "            '--deltype', deltype,\n",
    "            '--list_file', list_file,\n",
    "            ],\n",
    "            capture_output=True, text=True, check=True, timeout=timeout,\n",
    "        )\n",
    "        #print(\"stdout:\", result.stdout)\n",
    "        #print(\"stderr:\", result.stderr)\n",
    "        return 'success'\n",
    "        \n",
    "    except subprocess.CalledProcessError as err:\n",
    "        print('CalledProcessError', err.returncode)\n",
    "        print(\"stdout:\", err.stdout)\n",
    "        print(\"stderr:\", err.stderr)\n",
    "        return 'error'\n",
    "                \n",
    "    except subprocess.TimeoutExpired as err:\n",
    "        print('TimeoutExpired', err.timeout)\n",
    "        print(\"stdout:\", err.stdout)\n",
    "        print(\"stderr:\", err.stderr)\n",
    "        return 'timeout'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 800,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GuanosineEX B12EX\n",
      "GuanosineEX BiotinEX\n",
      "GuanosineEX MethanolEX\n",
      "GuanosineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "GuanosineEX SMethyl5thioDriboseEX\n",
      "GuanosineEX GlycolateEX\n",
      "GuanosineEX AmmoniaEX\n",
      "GuanosineEX GlycogenEX\n",
      "GuanosineEX H2SEX\n",
      "GuanosineEX SpermineEX\n",
      "GuanosineEX ThiosulfateEX\n",
      "GuanosineEX UreaEX\n",
      "GuanosineEX L_AlanineEX\n",
      "GuanosineEX L_ArginineEX\n",
      "GuanosineEX L_AsparagineEX\n",
      "GuanosineEX L_AspartateEX\n",
      "GuanosineEX L_CystineEX\n",
      "GuanosineEX L_GlutamateEX\n",
      "GuanosineEX L_GlutamineEX\n",
      "GuanosineEX GlycineEX\n",
      "GuanosineEX L_HistidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GuanosineEX L_IsoleucineEX\n",
      "GuanosineEX L_LeucineEX\n",
      "GuanosineEX L_LysineEX\n",
      "GuanosineEX L_MethionineEX\n",
      "GuanosineEX L_PhenylalanineEX\n",
      "GuanosineEX L_ProlineEX\n",
      "GuanosineEX L_SerineEX\n",
      "GuanosineEX L_ThreonineEX\n",
      "GuanosineEX L_TryptophanEX\n",
      "GuanosineEX L_TyrosineEX\n",
      "GuanosineEX L_ValineEX\n",
      "GuanosineEX PhospholipidEX\n",
      "GuanosineEX LPSEX\n",
      "GuanosineEX AcetateEX\n",
      "GuanosineEX CitrateEX\n",
      "GuanosineEX FormateEX\n",
      "GuanosineEX PyruvateEX\n",
      "GuanosineEX FumarateEX\n",
      "GuanosineEX GlutathioneEX\n",
      "GuanosineEX 4_Methyl_2_oxopentanoateEX\n",
      "GuanosineEX PantothenateEX\n",
      "GuanosineEX SuccinateEX\n",
      "GuanosineEX ThymidineEX\n",
      "GuanosineEX XanthosineEX\n",
      "GuanosineEX 4_AminobenzoateEX\n",
      "GuanosineEX 5_MethylthioadenosineEX\n",
      "GuanosineEX AdenineEX\n",
      "GuanosineEX DethiobiotinEX\n",
      "GuanosineEX GuanineEX\n",
      "GuanosineEX PutrescineEX\n",
      "GuanosineEX SpermidineEX\n",
      "GuanosineEX EthanolEX\n",
      "GuanosineEX S_MalateEX\n",
      "GuanosineEX H2O2EX\n",
      "L_AspartateEX B12EX\n",
      "L_AspartateEX BiotinEX\n",
      "L_AspartateEX MethanolEX\n",
      "L_AspartateEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_AspartateEX SMethyl5thioDriboseEX\n",
      "L_AspartateEX GlycolateEX\n",
      "L_AspartateEX AmmoniaEX\n",
      "L_AspartateEX GlycogenEX\n",
      "L_AspartateEX H2SEX\n",
      "L_AspartateEX SpermineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_AspartateEX ThiosulfateEX\n",
      "L_AspartateEX UreaEX\n",
      "L_AspartateEX L_AlanineEX\n",
      "L_AspartateEX L_ArginineEX\n",
      "L_AspartateEX L_AsparagineEX\n",
      "L_AspartateEX L_CystineEX\n",
      "L_AspartateEX L_GlutamateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_AspartateEX L_GlutamineEX\n",
      "L_AspartateEX GlycineEX\n",
      "L_AspartateEX L_HistidineEX\n",
      "L_AspartateEX L_IsoleucineEX\n",
      "L_AspartateEX L_LeucineEX\n",
      "L_AspartateEX L_LysineEX\n",
      "L_AspartateEX L_MethionineEX\n",
      "L_AspartateEX L_PhenylalanineEX\n",
      "L_AspartateEX L_ProlineEX\n",
      "L_AspartateEX L_SerineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX L_SerineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000023080871796\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX     36.56         0  0.00%\\n            HCO3[e]            HCO3EXcar   0.05866         1  0.07%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     22.42         4 99.93%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n          Oxygen[e]                 O2EX     4.205         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -14.48         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  3.36%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -22.39         0  0.00%\\n                                     H[e]                                 HEX     -37.44         0  0.00%\\n                              L_Serine[e]                          L_SerineEX     -7.856         3 96.62%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -21.63         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.01%\\n                                  Urea[e]                              UreaEX -0.0004099         1  0.00%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX L_ThreonineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_AspartateEX L_TryptophanEX\n",
      "L_AspartateEX L_TyrosineEX\n",
      "L_AspartateEX L_ValineEX\n",
      "L_AspartateEX PhospholipidEX\n",
      "L_AspartateEX LPSEX\n",
      "L_AspartateEX AcetateEX\n",
      "L_AspartateEX CitrateEX\n",
      "L_AspartateEX FormateEX\n",
      "L_AspartateEX PyruvateEX\n",
      "L_AspartateEX FumarateEX\n",
      "L_AspartateEX GlutathioneEX\n",
      "L_AspartateEX 4_Methyl_2_oxopentanoateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_AspartateEX PantothenateEX\n",
      "L_AspartateEX SuccinateEX\n",
      "L_AspartateEX ThymidineEX\n",
      "L_AspartateEX XanthosineEX\n",
      "L_AspartateEX 4_AminobenzoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX 4_AminobenzoateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000023080869621\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX     25.71         0  0.00%\\n            HCO3[e]            HCO3EXcar   0.05866         1  0.08%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     19.16         4 99.92%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n          Oxygen[e]                 O2EX     1.292         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                       4_Aminobenzoate[e]                   4_AminobenzoateEX     -2.031         0  0.00%\\n                               Ammonia[e]                           AmmoniaEX     -17.05         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.55%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -19.14         0  0.00%\\n                                     H[e]                                 HEX     -37.44         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                              Pyruvate[e]                          PyruvateEX      -20.4         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                                  Urea[e]                              UreaEX -0.0004099         1  0.05%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX 5_MethylthioadenosineEX\n",
      "L_AspartateEX AdenineEX\n",
      "L_AspartateEX DethiobiotinEX\n",
      "L_AspartateEX GuanineEX\n",
      "L_AspartateEX GuanosineEX\n",
      "L_AspartateEX PutrescineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX PutrescineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000000000002813\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX     20.33         0  0.00%\\n            HCO3[e]            HCO3EXcar     2.619         1  3.76%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     16.77         4 96.24%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -6.452         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 24.23%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -14.19         0  0.00%\\n                                     H[e]                                 HEX     -37.44         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -3.564         0  0.00%\\n                            Putrescine[e]                        PutrescineEX      -2.56         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -18.54         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  Urea[e]                              UreaEX     -2.561         1 75.67%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX SpermidineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_AspartateEX EthanolEX\n",
      "L_AspartateEX S_MalateEX\n",
      "L_AspartateEX H2O2EX\n",
      "GlutathioneEX B12EX\n",
      "GlutathioneEX BiotinEX\n",
      "GlutathioneEX MethanolEX\n",
      "GlutathioneEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "GlutathioneEX SMethyl5thioDriboseEX\n",
      "GlutathioneEX GlycolateEX\n",
      "GlutathioneEX AmmoniaEX\n",
      "GlutathioneEX GlycogenEX\n",
      "GlutathioneEX H2SEX\n",
      "GlutathioneEX SpermineEX\n",
      "GlutathioneEX ThiosulfateEX\n",
      "GlutathioneEX UreaEX\n",
      "GlutathioneEX L_AlanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX L_AlanineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000008956397838\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0   0.00%\\n          Copper[e]             CopperEX  7.86E-07         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0   0.00%\\n     Glutathione[e]        GlutathioneEX     489.9         0   0.00%\\n             H2O[e]                H2OEX      1000         0   0.00%\\n            HCO3[e]            HCO3EXcar    0.3715         1 100.00%\\n             Fe2[e]               IronEX 2.909E-05         0   0.00%\\n          Photon[e]              LightEX       150         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0   0.00%\\n               K[e]          PotassiumEX 0.0008565         0   0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0   0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0   0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0   0.00%\\n         Sulfate[e]            SulfateEX     16.53         0   0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  0.02%\\n                               Formate[e]                           FormateEX    -0.3089         0  0.00%\\n                               Glycine[e]                           GlycineEX       -489         2 19.97%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                               Guanine[e]                           GuanineEX    -0.1805         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -172.6         0  0.00%\\n                                     H[e]                                 HEX     -52.99         0  0.00%\\n                             L_Alanine[e]                         L_AlanineEX       -189         3 11.58%\\n                             L_Cystine[e]                         L_CystineEX     -150.4         6 18.42%\\n                           L_Glutamate[e]                       L_GlutamateEX     -489.9         5 50.01%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -18.74         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.00%\\n                           Thiosulfate[e]                       ThiosulfateEX     -16.53         0  0.00%\\n                                  Urea[e]                              UreaEX -0.0004099         1  0.00%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_ArginineEX\n",
      "GlutathioneEX L_AsparagineEX\n",
      "GlutathioneEX L_AspartateEX\n",
      "GlutathioneEX L_CystineEX\n",
      "GlutathioneEX L_GlutamateEX\n",
      "GlutathioneEX L_GlutamineEX\n",
      "GlutathioneEX GlycineEX\n",
      "GlutathioneEX L_HistidineEX\n",
      "GlutathioneEX L_IsoleucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX L_IsoleucineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000054113781927\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0   0.00%\\n          Copper[e]             CopperEX  7.86E-07         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0   0.00%\\n     Glutathione[e]        GlutathioneEX     3.632         0   0.00%\\n             H2O[e]                H2OEX     8.414         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 100.00%\\n             Fe2[e]               IronEX 2.909E-05         0   0.00%\\n          Photon[e]              LightEX     91.79         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0   0.00%\\n               K[e]          PotassiumEX 0.0008565         0   0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0   0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0   0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0   0.00%\\n         Sulfate[e]            SulfateEX  0.002242         0   0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -1.442         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  3.00%\\n                               Glycine[e]                           GlycineEX      -5.45         2 39.92%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -3.627         0  0.00%\\n                                     H[e]                                 HEX     -16.18         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX    -0.6997         5 12.81%\\n                          L_Isoleucine[e]                      L_IsoleucineEX     -2.014         6 44.25%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -10.16         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.01%\\n                                  Urea[e]                              UreaEX -0.0004099         1  0.00%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_LeucineEX\n",
      "GlutathioneEX L_LysineEX\n",
      "GlutathioneEX L_MethionineEX\n",
      "GlutathioneEX L_PhenylalanineEX\n",
      "GlutathioneEX L_ProlineEX\n",
      "GlutathioneEX L_SerineEX\n",
      "GlutathioneEX L_ThreonineEX\n",
      "GlutathioneEX L_TryptophanEX\n",
      "GlutathioneEX L_TyrosineEX\n",
      "GlutathioneEX L_ValineEX\n",
      "GlutathioneEX PhospholipidEX\n",
      "GlutathioneEX LPSEX\n",
      "GlutathioneEX AcetateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GlutathioneEX CitrateEX\n",
      "GlutathioneEX FormateEX\n",
      "GlutathioneEX PyruvateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "GlutathioneEX FumarateEX\n",
      "GlutathioneEX 4_Methyl_2_oxopentanoateEX\n",
      "GlutathioneEX PantothenateEX\n",
      "GlutathioneEX SuccinateEX\n",
      "GlutathioneEX ThymidineEX\n",
      "GlutathioneEX XanthosineEX\n",
      "GlutathioneEX 4_AminobenzoateEX\n",
      "GlutathioneEX 5_MethylthioadenosineEX\n",
      "GlutathioneEX AdenineEX\n",
      "GlutathioneEX DethiobiotinEX\n",
      "GlutathioneEX GuanineEX\n",
      "GlutathioneEX GuanosineEX\n",
      "GlutathioneEX PutrescineEX\n",
      "GlutathioneEX SpermidineEX\n",
      "GlutathioneEX EthanolEX\n",
      "GlutathioneEX S_MalateEX\n",
      "GlutathioneEX H2O2EX\n",
      "L_GlutamineEX B12EX\n",
      "L_GlutamineEX BiotinEX\n",
      "L_GlutamineEX MethanolEX\n",
      "L_GlutamineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_GlutamineEX SMethyl5thioDriboseEX\n",
      "L_GlutamineEX GlycolateEX\n",
      "L_GlutamineEX AmmoniaEX\n",
      "L_GlutamineEX GlycogenEX\n",
      "L_GlutamineEX H2SEX\n",
      "L_GlutamineEX SpermineEX\n",
      "L_GlutamineEX ThiosulfateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamineEX UreaEX\n",
      "L_GlutamineEX L_AlanineEX\n",
      "L_GlutamineEX L_ArginineEX\n",
      "L_GlutamineEX L_AsparagineEX\n",
      "L_GlutamineEX L_AspartateEX\n",
      "L_GlutamineEX L_CystineEX\n",
      "L_GlutamineEX L_GlutamateEX\n",
      "L_GlutamineEX GlycineEX\n",
      "L_GlutamineEX L_HistidineEX\n",
      "L_GlutamineEX L_IsoleucineEX\n",
      "L_GlutamineEX L_LeucineEX\n",
      "L_GlutamineEX L_LysineEX\n",
      "L_GlutamineEX L_MethionineEX\n",
      "L_GlutamineEX L_PhenylalanineEX\n",
      "L_GlutamineEX L_ProlineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamineEX L_SerineEX\n",
      "L_GlutamineEX L_ThreonineEX\n",
      "L_GlutamineEX L_TryptophanEX\n",
      "L_GlutamineEX L_TyrosineEX\n",
      "L_GlutamineEX L_ValineEX\n",
      "L_GlutamineEX PhospholipidEX\n",
      "L_GlutamineEX LPSEX\n",
      "L_GlutamineEX AcetateEX\n",
      "L_GlutamineEX CitrateEX\n",
      "L_GlutamineEX FormateEX\n",
      "L_GlutamineEX PyruvateEX\n",
      "L_GlutamineEX FumarateEX\n",
      "L_GlutamineEX GlutathioneEX\n",
      "L_GlutamineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_GlutamineEX PantothenateEX\n",
      "L_GlutamineEX SuccinateEX\n",
      "L_GlutamineEX ThymidineEX\n",
      "L_GlutamineEX XanthosineEX\n",
      "L_GlutamineEX 4_AminobenzoateEX\n",
      "L_GlutamineEX 5_MethylthioadenosineEX\n",
      "L_GlutamineEX AdenineEX\n",
      "L_GlutamineEX DethiobiotinEX\n",
      "L_GlutamineEX GuanineEX\n",
      "L_GlutamineEX GuanosineEX\n",
      "L_GlutamineEX PutrescineEX\n",
      "L_GlutamineEX SpermidineEX\n",
      "L_GlutamineEX EthanolEX\n",
      "L_GlutamineEX S_MalateEX\n",
      "L_GlutamineEX H2O2EX\n",
      "L_SerineEX B12EX\n",
      "L_SerineEX BiotinEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX MethanolEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX SMethyl5thioDriboseEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX GlycolateEX\n",
      "L_SerineEX AmmoniaEX\n",
      "L_SerineEX GlycogenEX\n",
      "L_SerineEX H2SEX\n",
      "L_SerineEX SpermineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX ThiosulfateEX\n",
      "L_SerineEX UreaEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX L_AlanineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_SerineEX L_ArginineEX\n",
      "L_SerineEX L_AsparagineEX\n",
      "L_SerineEX L_AspartateEX\n",
      "L_SerineEX L_CystineEX\n",
      "L_SerineEX L_GlutamateEX\n",
      "L_SerineEX L_GlutamineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX GlycineEX\n",
      "L_SerineEX L_HistidineEX\n",
      "L_SerineEX L_IsoleucineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX L_LeucineEX\n",
      "L_SerineEX L_LysineEX\n",
      "L_SerineEX L_MethionineEX\n",
      "L_SerineEX L_PhenylalanineEX\n",
      "L_SerineEX L_ProlineEX\n",
      "L_SerineEX L_ThreonineEX\n",
      "L_SerineEX L_TryptophanEX\n",
      "L_SerineEX L_TyrosineEX\n",
      "L_SerineEX L_ValineEX\n",
      "L_SerineEX PhospholipidEX\n",
      "L_SerineEX LPSEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX AcetateEX\n",
      "L_SerineEX CitrateEX\n",
      "L_SerineEX FormateEX\n",
      "L_SerineEX PyruvateEX\n",
      "L_SerineEX FumarateEX\n",
      "L_SerineEX GlutathioneEX\n",
      "L_SerineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_SerineEX PantothenateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX SuccinateEX\n",
      "L_SerineEX ThymidineEX\n",
      "L_SerineEX XanthosineEX\n",
      "L_SerineEX 4_AminobenzoateEX\n",
      "L_SerineEX 5_MethylthioadenosineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX AdenineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_SerineEX DethiobiotinEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX GuanineEX\n",
      "L_SerineEX GuanosineEX\n",
      "L_SerineEX PutrescineEX\n",
      "L_SerineEX SpermidineEX\n",
      "L_SerineEX EthanolEX\n",
      "L_SerineEX S_MalateEX\n",
      "L_SerineEX H2O2EX\n",
      "L_CystineEX B12EX\n",
      "L_CystineEX BiotinEX\n",
      "L_CystineEX MethanolEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_CystineEX SAdenosyl4methylthio2oxobutanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000028186071726\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.754         1 51.29%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n       L_Cystine[e]          L_CystineEX    0.9108         6 48.71%\\n          Photon[e]              LightEX     60.63         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.002242         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 17.06%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.2489         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.017         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -1.568         0  0.00%\\n                                     H[e]                                 HEX     -8.666         0  0.00%\\n                              Methanol[e]                          MethanolEX     -0.249         1  5.18%\\n                                Oxygen[e]                                O2EX     -5.666         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -1.156         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX     -0.249        15 77.70%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.05%\\n                                  Urea[e]                              UreaEX -0.0004099         1  0.01%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_CystineEX SMethyl5thioDriboseEX\n",
      "L_CystineEX GlycolateEX\n",
      "L_CystineEX AmmoniaEX\n",
      "L_CystineEX GlycogenEX\n",
      "L_CystineEX H2SEX\n",
      "L_CystineEX SpermineEX\n",
      "L_CystineEX ThiosulfateEX\n",
      "L_CystineEX UreaEX\n",
      "L_CystineEX L_AlanineEX\n",
      "L_CystineEX L_ArginineEX\n",
      "L_CystineEX L_AsparagineEX\n",
      "L_CystineEX L_AspartateEX\n",
      "L_CystineEX L_GlutamateEX\n",
      "L_CystineEX L_GlutamineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX GlycineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_CystineEX L_HistidineEX\n",
      "L_CystineEX L_IsoleucineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX L_LeucineEX\n",
      "L_CystineEX L_LysineEX\n",
      "L_CystineEX L_MethionineEX\n",
      "L_CystineEX L_PhenylalanineEX\n",
      "L_CystineEX L_ProlineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX L_SerineEX\n",
      "L_CystineEX L_ThreonineEX\n",
      "L_CystineEX L_TryptophanEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX L_TyrosineEX\n",
      "L_CystineEX L_ValineEX\n",
      "L_CystineEX PhospholipidEX\n",
      "L_CystineEX LPSEX\n",
      "L_CystineEX AcetateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX CitrateEX\n",
      "L_CystineEX FormateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX PyruvateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_CystineEX FumarateEX\n",
      "L_CystineEX GlutathioneEX\n",
      "L_CystineEX 4_Methyl_2_oxopentanoateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX PantothenateEX\n",
      "L_CystineEX SuccinateEX\n",
      "L_CystineEX ThymidineEX\n",
      "L_CystineEX XanthosineEX\n",
      "L_CystineEX 4_AminobenzoateEX\n",
      "L_CystineEX 5_MethylthioadenosineEX\n",
      "L_CystineEX AdenineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX DethiobiotinEX\n",
      "L_CystineEX GuanineEX\n",
      "L_CystineEX GuanosineEX\n",
      "L_CystineEX PutrescineEX\n",
      "L_CystineEX SpermidineEX\n",
      "L_CystineEX EthanolEX\n",
      "L_CystineEX S_MalateEX\n",
      "L_CystineEX H2O2EX\n",
      "L_AlanineEX B12EX\n",
      "L_AlanineEX BiotinEX\n",
      "L_AlanineEX MethanolEX\n",
      "L_AlanineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_AlanineEX SMethyl5thioDriboseEX\n",
      "L_AlanineEX GlycolateEX\n",
      "L_AlanineEX AmmoniaEX\n",
      "L_AlanineEX GlycogenEX\n",
      "L_AlanineEX H2SEX\n",
      "L_AlanineEX SpermineEX\n",
      "L_AlanineEX ThiosulfateEX\n",
      "L_AlanineEX UreaEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_AlanineEX L_ArginineEX\n",
      "L_AlanineEX L_AsparagineEX\n",
      "L_AlanineEX L_AspartateEX\n",
      "L_AlanineEX L_CystineEX\n",
      "L_AlanineEX L_GlutamateEX\n",
      "L_AlanineEX L_GlutamineEX\n",
      "L_AlanineEX GlycineEX\n",
      "L_AlanineEX L_HistidineEX\n",
      "L_AlanineEX L_IsoleucineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_AlanineEX L_LeucineEX\n",
      "L_AlanineEX L_LysineEX\n",
      "L_AlanineEX L_MethionineEX\n",
      "L_AlanineEX L_PhenylalanineEX\n",
      "L_AlanineEX L_ProlineEX\n",
      "L_AlanineEX L_SerineEX\n",
      "L_AlanineEX L_ThreonineEX\n",
      "L_AlanineEX L_TryptophanEX\n",
      "L_AlanineEX L_TyrosineEX\n",
      "L_AlanineEX L_ValineEX\n",
      "L_AlanineEX PhospholipidEX\n",
      "L_AlanineEX LPSEX\n",
      "L_AlanineEX AcetateEX\n",
      "L_AlanineEX CitrateEX\n",
      "L_AlanineEX FormateEX\n",
      "L_AlanineEX PyruvateEX\n",
      "L_AlanineEX FumarateEX\n",
      "L_AlanineEX GlutathioneEX\n",
      "L_AlanineEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_AlanineEX PantothenateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_AlanineEX SuccinateEX\n",
      "L_AlanineEX ThymidineEX\n",
      "L_AlanineEX XanthosineEX\n",
      "L_AlanineEX 4_AminobenzoateEX\n",
      "L_AlanineEX 5_MethylthioadenosineEX\n",
      "L_AlanineEX AdenineEX\n",
      "L_AlanineEX DethiobiotinEX\n",
      "L_AlanineEX GuanineEX\n",
      "L_AlanineEX GuanosineEX\n",
      "L_AlanineEX PutrescineEX\n",
      "L_AlanineEX SpermidineEX\n",
      "L_AlanineEX EthanolEX\n",
      "L_AlanineEX S_MalateEX\n",
      "L_AlanineEX H2O2EX\n",
      "L_LeucineEX B12EX\n",
      "L_LeucineEX BiotinEX\n",
      "L_LeucineEX MethanolEX\n",
      "L_LeucineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_LeucineEX SMethyl5thioDriboseEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX GlycolateEX\n",
      "L_LeucineEX AmmoniaEX\n",
      "L_LeucineEX GlycogenEX\n",
      "L_LeucineEX H2SEX\n",
      "L_LeucineEX SpermineEX\n",
      "L_LeucineEX ThiosulfateEX\n",
      "L_LeucineEX UreaEX\n",
      "L_LeucineEX L_AlanineEX\n",
      "L_LeucineEX L_ArginineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX L_AsparagineEX\n",
      "L_LeucineEX L_AspartateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX L_CystineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX L_GlutamateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_LeucineEX L_GlutamineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX GlycineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX L_HistidineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX L_IsoleucineEX\n",
      "L_LeucineEX L_LysineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX L_MethionineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_LeucineEX L_PhenylalanineEX\n",
      "L_LeucineEX L_ProlineEX\n",
      "L_LeucineEX L_SerineEX\n",
      "L_LeucineEX L_ThreonineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX L_TryptophanEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_LeucineEX L_TyrosineEX\n",
      "L_LeucineEX L_ValineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX PhospholipidEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX LPSEX\n",
      "L_LeucineEX AcetateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX CitrateEX\n",
      "L_LeucineEX FormateEX\n",
      "L_LeucineEX PyruvateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX FumarateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX GlutathioneEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_LeucineEX PantothenateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX SuccinateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX ThymidineEX\n",
      "L_LeucineEX XanthosineEX\n",
      "L_LeucineEX 4_AminobenzoateEX\n",
      "L_LeucineEX 5_MethylthioadenosineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX AdenineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX DethiobiotinEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_LeucineEX GuanineEX\n",
      "L_LeucineEX GuanosineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX PutrescineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX SpermidineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_LeucineEX EthanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_LeucineEX S_MalateEX\n",
      "L_LeucineEX H2O2EX\n",
      "GlycineEX B12EX\n",
      "GlycineEX BiotinEX\n",
      "GlycineEX MethanolEX\n",
      "GlycineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "GlycineEX SMethyl5thioDriboseEX\n",
      "GlycineEX GlycolateEX\n",
      "GlycineEX AmmoniaEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "GlycineEX GlycogenEX\n",
      "GlycineEX H2SEX\n",
      "GlycineEX SpermineEX\n",
      "GlycineEX ThiosulfateEX\n",
      "GlycineEX UreaEX\n",
      "GlycineEX L_AlanineEX\n",
      "GlycineEX L_ArginineEX\n",
      "GlycineEX L_AsparagineEX\n",
      "GlycineEX L_ValineEX\n",
      "GlycineEX PhospholipidEX\n",
      "GlycineEX LPSEX\n",
      "GlycineEX AcetateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "GlycineEX CitrateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GlycineEX FormateEX\n",
      "GlycineEX PyruvateEX\n",
      "GlycineEX FumarateEX\n",
      "GlycineEX GlutathioneEX\n",
      "GlycineEX 4_Methyl_2_oxopentanoateEX\n",
      "GlycineEX PantothenateEX\n",
      "GlycineEX SuccinateEX\n",
      "GlycineEX ThymidineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "GlycineEX XanthosineEX\n",
      "GlycineEX 4_AminobenzoateEX\n",
      "GlycineEX 5_MethylthioadenosineEX\n",
      "GlycineEX AdenineEX\n",
      "GlycineEX DethiobiotinEX\n",
      "GlycineEX GuanineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "GlycineEX GuanosineEX\n",
      "GlycineEX PutrescineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlycineEX SpermidineEX\n",
      "GlycineEX EthanolEX\n",
      "GlycineEX S_MalateEX\n",
      "GlycineEX H2O2EX\n",
      "UreaEX B12EX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "UreaEX BiotinEX\n",
      "UreaEX MethanolEX\n",
      "UreaEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "UreaEX SMethyl5thioDriboseEX\n",
      "UreaEX GlycolateEX\n",
      "UreaEX AmmoniaEX\n",
      "UreaEX GlycogenEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "UreaEX H2SEX\n",
      "UreaEX SpermineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "UreaEX ThiosulfateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UreaEX L_AlanineEX\n",
      "UreaEX L_ArginineEX\n",
      "UreaEX L_AsparagineEX\n",
      "UreaEX L_AspartateEX\n",
      "UreaEX L_CystineEX\n",
      "UreaEX L_GlutamateEX\n",
      "UreaEX L_GlutamineEX\n",
      "UreaEX GlycineEX\n",
      "UreaEX L_HistidineEX\n",
      "UreaEX L_IsoleucineEX\n",
      "UreaEX L_LeucineEX\n",
      "UreaEX L_LysineEX\n",
      "UreaEX L_MethionineEX\n",
      "UreaEX L_PhenylalanineEX\n",
      "UreaEX L_ProlineEX\n",
      "UreaEX L_SerineEX\n",
      "UreaEX L_ThreonineEX\n",
      "UreaEX L_TryptophanEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "UreaEX L_TyrosineEX\n",
      "UreaEX L_ValineEX\n",
      "UreaEX PhospholipidEX\n",
      "UreaEX LPSEX\n",
      "UreaEX AcetateEX\n",
      "UreaEX CitrateEX\n",
      "UreaEX FormateEX\n",
      "UreaEX PyruvateEX\n",
      "UreaEX FumarateEX\n",
      "UreaEX GlutathioneEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "UreaEX 4_Methyl_2_oxopentanoateEX\n",
      "UreaEX PantothenateEX\n",
      "UreaEX SuccinateEX\n",
      "UreaEX ThymidineEX\n",
      "UreaEX XanthosineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "UreaEX 4_AminobenzoateEX\n",
      "UreaEX 5_MethylthioadenosineEX\n",
      "UreaEX AdenineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "UreaEX DethiobiotinEX\n",
      "UreaEX GuanineEX\n",
      "UreaEX GuanosineEX\n",
      "UreaEX PutrescineEX\n",
      "UreaEX SpermidineEX\n",
      "UreaEX EthanolEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "UreaEX S_MalateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UreaEX H2O2EX\n",
      "AmmoniaEX B12EX\n",
      "AmmoniaEX BiotinEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX MethanolEX\n",
      "AmmoniaEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "AmmoniaEX SMethyl5thioDriboseEX\n",
      "AmmoniaEX GlycolateEX\n",
      "AmmoniaEX GlycogenEX\n",
      "AmmoniaEX H2SEX\n",
      "AmmoniaEX SpermineEX\n",
      "AmmoniaEX ThiosulfateEX\n",
      "AmmoniaEX UreaEX\n",
      "AmmoniaEX L_AlanineEX\n",
      "AmmoniaEX L_ArginineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX L_AsparagineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX L_AspartateEX\n",
      "AmmoniaEX L_CystineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX L_GlutamateEX\n",
      "AmmoniaEX L_GlutamineEX\n",
      "AmmoniaEX GlycineEX\n",
      "AmmoniaEX L_HistidineEX\n",
      "AmmoniaEX L_IsoleucineEX\n",
      "AmmoniaEX L_LeucineEX\n",
      "AmmoniaEX L_LysineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX L_MethionineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX L_MethionineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000032270979771\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.146         0   0.00%\\n         Cadmium[e]            CadmiumEX 4.344E-07         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0   0.00%\\n          Copper[e]             CopperEX  7.86E-07         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.799         1 100.00%\\n             Fe2[e]               IronEX 2.909E-05         0   0.00%\\n          Photon[e]              LightEX     72.84         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0   0.00%\\n               K[e]          PotassiumEX 0.0008565         0   0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0   0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0   0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0   0.00%\\n         Sulfate[e]            SulfateEX     1.075         0   0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.499         0  0.00%\\n                                     H[e]                                 HEX     -10.27         0  0.00%\\n                          L_Methionine[e]                      L_MethionineEX     -1.068         5 99.94%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -8.547         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.05%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_PhenylalanineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX L_ProlineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX L_SerineEX\n",
      "AmmoniaEX L_ThreonineEX\n",
      "AmmoniaEX L_TryptophanEX\n",
      "AmmoniaEX L_TyrosineEX\n",
      "AmmoniaEX L_ValineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX PhospholipidEX\n",
      "AmmoniaEX LPSEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX AcetateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AmmoniaEX CitrateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX CitrateEX Objective\\n=========\\n1.0 BIOMASS = 0.00900003208316052\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX   0.07817         0   0.00%\\n         Cadmium[e]            CadmiumEX 4.344E-07         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0   0.00%\\n          Copper[e]             CopperEX  7.86E-07         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.582         1 100.00%\\n             Fe2[e]               IronEX 2.909E-05         0   0.00%\\n          Photon[e]              LightEX     47.41         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0   0.00%\\n               K[e]          PotassiumEX 0.0008565         0   0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0   0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0   0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0   0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0   0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                               Citrate[e]                           CitrateEX    -0.8541         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX      -2.07         0  0.00%\\n                                     H[e]                                 HEX      -6.27         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  1.27%\\n                                Oxygen[e]                                O2EX     -4.378         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6 74.32%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX FormateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX PyruvateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX FumarateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX GlutathioneEX\n",
      "AmmoniaEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX PantothenateEX\n",
      "AmmoniaEX SuccinateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AmmoniaEX ThymidineEX\n",
      "AmmoniaEX XanthosineEX\n",
      "AmmoniaEX 4_AminobenzoateEX\n",
      "AmmoniaEX 5_MethylthioadenosineEX\n",
      "AmmoniaEX AdenineEX\n",
      "AmmoniaEX DethiobiotinEX\n",
      "AmmoniaEX GuanineEX\n",
      "AmmoniaEX GuanosineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX PutrescineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX SpermidineEX\n",
      "AmmoniaEX EthanolEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX S_MalateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "AmmoniaEX H2O2EX\n",
      "L_GlutamateEX B12EX\n",
      "L_GlutamateEX BiotinEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX MethanolEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX SMethyl5thioDriboseEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX GlycolateEX\n",
      "L_GlutamateEX AmmoniaEX\n",
      "L_GlutamateEX GlycogenEX\n",
      "L_GlutamateEX H2SEX\n",
      "L_GlutamateEX SpermineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX ThiosulfateEX\n",
      "L_GlutamateEX UreaEX\n",
      "L_GlutamateEX L_AlanineEX\n",
      "L_GlutamateEX L_ArginineEX\n",
      "L_GlutamateEX L_AsparagineEX\n",
      "L_GlutamateEX L_AspartateEX\n",
      "L_GlutamateEX L_CystineEX\n",
      "L_GlutamateEX L_GlutamineEX\n",
      "L_GlutamateEX GlycineEX\n",
      "L_GlutamateEX L_HistidineEX\n",
      "L_GlutamateEX L_IsoleucineEX\n",
      "L_GlutamateEX L_LeucineEX\n",
      "L_GlutamateEX L_LysineEX\n",
      "L_GlutamateEX L_MethionineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX L_PhenylalanineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_GlutamateEX L_ProlineEX\n",
      "L_GlutamateEX L_SerineEX\n",
      "L_GlutamateEX L_ThreonineEX\n",
      "L_GlutamateEX L_TryptophanEX\n",
      "L_GlutamateEX L_TyrosineEX\n",
      "L_GlutamateEX L_ValineEX\n",
      "L_GlutamateEX PhospholipidEX\n",
      "L_GlutamateEX LPSEX\n",
      "L_GlutamateEX AcetateEX\n",
      "L_GlutamateEX CitrateEX\n",
      "L_GlutamateEX FormateEX\n",
      "L_GlutamateEX PyruvateEX\n",
      "L_GlutamateEX FumarateEX\n",
      "L_GlutamateEX GlutathioneEX\n",
      "L_GlutamateEX 4_Methyl_2_oxopentanoateEX\n",
      "L_GlutamateEX PantothenateEX\n",
      "L_GlutamateEX SuccinateEX\n",
      "L_GlutamateEX ThymidineEX\n",
      "L_GlutamateEX XanthosineEX\n",
      "L_GlutamateEX 4_AminobenzoateEX\n",
      "L_GlutamateEX 5_MethylthioadenosineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX AdenineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX DethiobiotinEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_GlutamateEX GuanineEX\n",
      "L_GlutamateEX GuanosineEX\n",
      "L_GlutamateEX PutrescineEX\n",
      "L_GlutamateEX SpermidineEX\n",
      "L_GlutamateEX EthanolEX\n",
      "L_GlutamateEX S_MalateEX\n",
      "L_GlutamateEX H2O2EX\n",
      "L_ArginineEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX BiotinEX\n",
      "L_ArginineEX MethanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_ArginineEX SMethyl5thioDriboseEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX GlycolateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX AmmoniaEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX AmmoniaEX Objective\\n=========\\n1.0 BIOMASS = 0.009000068692351489\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX     2.558         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.755         1 35.03%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n      L_Arginine[e]         L_ArginineEX      1.47         6 64.97%\\n          Photon[e]              LightEX     56.23         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -2.869         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.60%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                     H[e]                                 HEX     -9.299         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -5.876         0  0.00%\\n                            Putrescine[e]                        PutrescineEX     -1.467         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -2.144         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_ArginineEX GlycogenEX\n",
      "L_ArginineEX H2SEX\n",
      "L_ArginineEX SpermineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX ThiosulfateEX\n",
      "L_ArginineEX UreaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX UreaEX Objective\\n=========\\n1.0 BIOMASS = 0.009000034346045483\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX     2.558         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.755         1 21.44%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n      L_Arginine[e]         L_ArginineEX     2.905         6 78.56%\\n          Photon[e]              LightEX     56.23         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 22.21%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                     H[e]                                 HEX     -9.299         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.876         0  0.00%\\n                            Putrescine[e]                        PutrescineEX     -2.901         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -2.144         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  Urea[e]                              UreaEX     -2.869         1 77.70%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_AlanineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_AsparagineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_AspartateEX\n",
      "L_ArginineEX L_CystineEX\n",
      "L_ArginineEX L_GlutamateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_GlutamineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX GlycineEX\n",
      "L_ArginineEX L_HistidineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_IsoleucineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_LeucineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_LysineEX\n",
      "L_ArginineEX L_MethionineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_PhenylalanineEX\n",
      "L_ArginineEX L_ProlineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_SerineEX\n",
      "L_ArginineEX L_ThreonineEX\n",
      "L_ArginineEX L_TryptophanEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX L_TyrosineEX\n",
      "L_ArginineEX L_ValineEX\n",
      "L_ArginineEX PhospholipidEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX LPSEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_ArginineEX AcetateEX\n",
      "L_ArginineEX CitrateEX\n",
      "L_ArginineEX FormateEX\n",
      "L_ArginineEX PyruvateEX\n",
      "L_ArginineEX FumarateEX\n",
      "L_ArginineEX GlutathioneEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX 4_Methyl_2_oxopentanoateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX PantothenateEX\n",
      "L_ArginineEX SuccinateEX\n",
      "L_ArginineEX ThymidineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX XanthosineEX\n",
      "L_ArginineEX 4_AminobenzoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX 5_MethylthioadenosineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX AdenineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX DethiobiotinEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX GuanineEX\n",
      "L_ArginineEX GuanosineEX\n",
      "L_ArginineEX PutrescineEX\n",
      "L_ArginineEX SpermidineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ArginineEX EthanolEX\n",
      "L_ArginineEX S_MalateEX\n",
      "L_ArginineEX H2O2EX\n",
      "CyanateEX B12EX\n",
      "CyanateEX BiotinEX\n",
      "CyanateEX MethanolEX\n",
      "CyanateEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "CyanateEX SMethyl5thioDriboseEX\n",
      "CyanateEX GlycolateEX\n",
      "CyanateEX AmmoniaEX\n",
      "CyanateEX GlycogenEX\n",
      "CyanateEX H2SEX\n",
      "CyanateEX SpermineEX\n",
      "CyanateEX ThiosulfateEX\n",
      "CyanateEX UreaEX\n",
      "CyanateEX L_AlanineEX\n",
      "CyanateEX L_ArginineEX\n",
      "CyanateEX L_AsparagineEX\n",
      "CyanateEX L_AspartateEX\n",
      "CyanateEX L_CystineEX\n",
      "CyanateEX L_GlutamateEX\n",
      "CyanateEX L_GlutamineEX\n",
      "CyanateEX GlycineEX\n",
      "CyanateEX L_HistidineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "CyanateEX L_IsoleucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX L_IsoleucineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000032270980507\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX    0.9327         1 16.39%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX    0.5423         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 83.61%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     61.75         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1063         1  2.03%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                     H[e]                                 HEX     -9.746         0  0.00%\\n                          L_Isoleucine[e]                      L_IsoleucineEX    -0.8546         6 97.91%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.944         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.05%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX L_LeucineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX L_LeucineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000021298495227\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX    0.7898         1 14.23%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 85.77%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     56.23         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 16.10%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.3148         0  0.00%\\n                                     H[e]                                 HEX      -8.51         0  0.00%\\n                             L_Leucine[e]                         L_LeucineEX    -0.7116         6 83.83%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.873         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX -0.0003746         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.05%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status     uptake    secretion\\n141  {PMM0225}     NaN  infeasible  CyanateEX  L_LeucineEX\\n142  {PMM1075}     NaN  infeasible  CyanateEX  L_LeucineEX\\n157  {PMM0770}     NaN  infeasible  CyanateEX  L_LeucineEX\\n158  {PMM0642}     NaN  infeasible  CyanateEX  L_LeucineEX\\n159  {PMM0912}     NaN  infeasible  CyanateEX  L_LeucineEX\\n'\n",
      "stderr: None\n",
      "CyanateEX L_LysineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX L_LysineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000063895787856\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX     1.705         1 26.37%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX    0.4985         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 73.63%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     59.37         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 14.38%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                     H[e]                                 HEX      -8.38         0  0.00%\\n                              L_Lysine[e]                          L_LysineEX    -0.8133         6 85.56%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.483         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -0.102         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.04%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status     uptake   secretion\\n149  {PMM0770}     NaN  infeasible  CyanateEX  L_LysineEX\\n169  {PMM0225}     NaN  infeasible  CyanateEX  L_LysineEX\\n'\n",
      "stderr: None\n",
      "CyanateEX L_MethionineEX\n",
      "CyanateEX L_PhenylalanineEX\n",
      "CyanateEX L_ProlineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX L_ProlineEX Objective\\n=========\\n1.0 BIOMASS = 0.008999999999999937\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX    0.8914         1 15.78%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 84.22%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX      53.1         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 16.77%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.128         0  0.00%\\n                                     H[e]                                 HEX     -7.624         0  0.00%\\n                             L_Proline[e]                         L_ProlineEX    -0.8133         5 83.16%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.263         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -0.102         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.05%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status     uptake    secretion\\n156  {PMM0770}     NaN  infeasible  CyanateEX  L_ProlineEX\\n166  {PMM0225}     NaN  infeasible  CyanateEX  L_ProlineEX\\n'\n",
      "stderr: None\n",
      "CyanateEX L_SerineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CyanateEX L_ThreonineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX L_ThreonineEX Objective\\n=========\\n1.0 BIOMASS = 0.00900000000000048\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX     1.502         1 23.99%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX     1.109         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 76.01%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     59.96         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1088         1  1.87%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                     H[e]                                 HEX     -6.209         0  0.00%\\n                           L_Threonine[e]                       L_ThreonineEX     -1.423         4 98.07%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.229         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.04%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\nEmpty DataFrame\\nColumns: [ids, growth, status, uptake, secretion]\\nIndex: []\\n'\n",
      "stderr: None\n",
      "CyanateEX L_TryptophanEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "CyanateEX L_TyrosineEX\n",
      "CyanateEX L_ValineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "CyanateEX PhospholipidEX\n",
      "CyanateEX LPSEX\n",
      "CyanateEX AcetateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "CyanateEX CitrateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX CitrateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000032083162081\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX    0.8235         1 14.75%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 85.25%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     47.41         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX    -0.7453         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                               Citrate[e]                           CitrateEX    -0.8541         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.4235         0  0.00%\\n                                     H[e]                                 HEX      -6.27         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  1.27%\\n                                Oxygen[e]                                O2EX     -4.378         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6 74.32%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  2.13%\\n\\n           ids  growth      status     uptake  secretion\\n8    {PMM0225}     NaN  infeasible  CyanateEX  CitrateEX\\n146  {PMM0770}     NaN  infeasible  CyanateEX  CitrateEX\\n'\n",
      "stderr: None\n",
      "CyanateEX FormateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CyanateEX PyruvateEX\n",
      "CyanateEX FumarateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX FumarateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000032083160699\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX     1.393         1 22.64%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 77.36%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     48.84         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -1.315         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                              Fumarate[e]                          FumarateEX     -1.423         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.4235         0  0.00%\\n                                     H[e]                                 HEX      -6.06         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  1.27%\\n                                Oxygen[e]                                O2EX     -4.805         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6 74.32%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  2.13%\\n\\n           ids  growth      status     uptake   secretion\\n165  {PMM1023}     NaN  infeasible  CyanateEX  FumarateEX\\n'\n",
      "stderr: None\n",
      "CyanateEX GlutathioneEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX GlutathioneEX Objective\\n=========\\n1.0 BIOMASS = 0.009000021538091652\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX       1.3         1 21.45%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 78.55%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     53.66         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX     0.414         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.60%\\n                               Formate[e]                           FormateEX    -0.4043         0  0.00%\\n                           Glutathione[e]                       GlutathioneEX    -0.4071         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.535         0  0.00%\\n                                     H[e]                                 HEX     -6.139         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -5.266         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1016         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n           ids  growth      status     uptake      secretion\\n128  {PMM1702}     NaN  infeasible  CyanateEX  GlutathioneEX\\n147  {PMM0415}     NaN  infeasible  CyanateEX  GlutathioneEX\\n149  {PMM0225}     NaN  infeasible  CyanateEX  GlutathioneEX\\n157  {PMM0770}     NaN  infeasible  CyanateEX  GlutathioneEX\\n165  {PMM1354}     NaN  infeasible  CyanateEX  GlutathioneEX\\n'\n",
      "stderr: None\n",
      "CyanateEX 4_Methyl_2_oxopentanoateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "CyanateEX PantothenateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CyanateEX SuccinateEX\n",
      "CyanateEX ThymidineEX\n",
      "CyanateEX XanthosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX XanthosineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000021298493662\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX     1.309         1 20.53%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.066         1 79.47%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX      51.6         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.60%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.315         0  0.00%\\n                                     H[e]                                 HEX     -5.908         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -4.373         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.6735         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                            Xanthosine[e]                        XanthosineEX    -0.3077         0  0.00%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n           ids  growth      status     uptake     secretion\\n160  {PMM1354}     NaN  infeasible  CyanateEX  XanthosineEX\\n'\n",
      "stderr: None\n",
      "CyanateEX 4_AminobenzoateEX\n",
      "CyanateEX 5_MethylthioadenosineEX\n",
      "CyanateEX AdenineEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "CyanateEX DethiobiotinEX\n",
      "CyanateEX GuanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX GuanineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000021298495912\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX     1.915         1 27.19%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.126         1 72.81%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     53.06         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.60%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                               Guanine[e]                           GuanineEX    -0.3673         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.335         0  0.00%\\n                                     H[e]                                 HEX      -5.49         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -4.542         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -1.309         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n           ids  growth      status     uptake  secretion\\n133  {PMM1354}     NaN  infeasible  CyanateEX  GuanineEX\\n148  {PMM0225}     NaN  infeasible  CyanateEX  GuanineEX\\n154  {PMM0770}     NaN  infeasible  CyanateEX  GuanineEX\\n'\n",
      "stderr: None\n",
      "CyanateEX GuanosineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CyanateEX PutrescineEX\n",
      "CyanateEX SpermidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX SpermidineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000021548262899\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX    0.7907         1 13.66%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.996         1 86.34%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX     60.03         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.2444         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 36.47%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.025         0  0.00%\\n                                     H[e]                                 HEX     -8.984         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.995         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.4738         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.03%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.2379         6 63.49%\\n                            Spermidine[e]                        SpermidineEX    -0.2375         0  0.00%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX EthanolEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "CyanateEX S_MalateEX\n",
      "CyanateEX H2O2EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX H2O2EX Objective\\n=========\\n1.0 BIOMASS = 0.009000061240633357\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n         Cyanate[e]            CyanateEX   0.07817         1 12.11%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n             H2O[e]                H2OEX     36.03         0  0.00%\\n            HCO3[e]            HCO3EXcar    0.5671         1 87.89%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n          Oxygen[e]                 O2EX     17.67         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1876         1 98.27%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.06%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -36.42         0  0.00%\\n                                     H[e]                                 HEX     -36.93         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.02%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.33%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  1.29%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.04%\\n\\nEmpty DataFrame\\nColumns: [ids, growth, status, uptake, secretion]\\nIndex: []\\n'\n",
      "stderr: None\n",
      "L_ValineEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX B12EX Objective\\n=========\\n1.0 BIOMASS = 0.009000061542511859\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX   0.05665         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX    0.0599         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.872         1 49.75%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.9843         5 50.25%\\n          Photon[e]              LightEX     50.84         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.9773         0  0.00%\\n                     Cobamide_coenzyme[e]                               B12EX   -0.05663        72 83.20%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 16.73%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.183         0  0.00%\\n                                     H[e]                                 HEX     -3.248         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -3.997         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.2799         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.05%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake secretion\\n150  {PMM0770}     NaN  infeasible  L_ValineEX     B12EX\\n154  {PMM0225}     NaN  infeasible  L_ValineEX     B12EX\\n'\n",
      "stderr: None\n",
      "L_ValineEX BiotinEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX BiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079643411719\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.192         1 55.39%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8364         5 44.61%\\n          Photon[e]              LightEX     53.53         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.2235         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8294         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                Biotin[e]                            BiotinEX    -0.1083        10 29.76%\\n                                   CO2[e]                               CO2EX      -0.82         1 22.53%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.118         0  0.00%\\n                                     H[e]                                 HEX     -4.036         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1084         1  2.98%\\n                                Oxygen[e]                                O2EX     -4.771         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1012         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1084        15 44.66%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX MethanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX MethanolEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079869813984\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.192         1 55.39%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8364         5 44.61%\\n          Photon[e]              LightEX     52.12         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.1152         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8294         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 32.08%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1083         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.118         0  0.00%\\n                                     H[e]                                 HEX     -7.298         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1084         1  4.24%\\n                                Oxygen[e]                                O2EX     -4.554         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1012         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1084        15 63.58%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.10%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX SAdenosyl4methylthio2oxobutanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079869814895\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.192         1 55.39%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8364         5 44.61%\\n          Photon[e]              LightEX     52.12         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.1152         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8294         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 32.08%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1083         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.118         0  0.00%\\n                                     H[e]                                 HEX     -7.298         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1084         1  4.24%\\n                                Oxygen[e]                                O2EX     -4.554         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1012         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1084        15 63.58%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.10%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX SMethyl5thioDriboseEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX SMethyl5thioDriboseEX Objective\\n=========\\n1.0 BIOMASS = 0.009000081387017012\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.889         1 62.06%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.5978         5 37.94%\\n          Photon[e]              LightEX      55.6         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.2667         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.5908         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 22.27%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX      -2.15         0  0.00%\\n                                     H[e]                                 HEX     -8.622         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX      -5.21         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX   -0.06571         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.2602         6 42.41%\\n                              Spermine[e]                          SpermineEX    -0.1299        10 35.29%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake              secretion\\n124  {PMM0301}     NaN  infeasible  L_ValineEX  SMethyl5thioDriboseEX\\n145  {PMM0225}     NaN  infeasible  L_ValineEX  SMethyl5thioDriboseEX\\n160  {PMM0770}     NaN  infeasible  L_ValineEX  SMethyl5thioDriboseEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX GlycolateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX AmmoniaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX AmmoniaEX Objective\\n=========\\n1.0 BIOMASS = 0.009000048871927017\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 39.25%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.473         5 60.75%\\n          Photon[e]              LightEX     44.89         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX     -1.466         0  0.00%\\n                               Ammonia[e]                           AmmoniaEX     -1.395         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.60%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.859         0  0.00%\\n                                     H[e]                                 HEX     -6.464         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.667         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.6834         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n           ids  growth      status      uptake  secretion\\n9    {PMM0770}     NaN  infeasible  L_ValineEX  AmmoniaEX\\n161  {PMM0225}     NaN  infeasible  L_ValineEX  AmmoniaEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX GlycogenEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX H2SEX\n",
      "L_ValineEX SpermineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX SpermineEX Objective\\n=========\\n1.0 BIOMASS = 0.00900008043577019\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.889         1 62.06%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.5978         5 37.94%\\n          Photon[e]              LightEX      55.6         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.2667         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.5908         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 22.27%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX      -2.15         0  0.00%\\n                                     H[e]                                 HEX     -5.154         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX      -5.21         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX   -0.06571         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.2602         6 42.41%\\n                              Spermine[e]                          SpermineEX    -0.1299        10 35.29%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake   secretion\\n128  {PMM0770}     NaN  infeasible  L_ValineEX  SpermineEX\\n153  {PMM0301}     NaN  infeasible  L_ValineEX  SpermineEX\\n164  {PMM0225}     NaN  infeasible  L_ValineEX  SpermineEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX ThiosulfateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ValineEX UreaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX UreaEX Objective\\n=========\\n1.0 BIOMASS = 0.009000018267812389\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.001         1 48.82%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.049         5 51.18%\\n          Photon[e]              LightEX     48.52         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX     -1.042         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 76.91%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.405         0  0.00%\\n                                     H[e]                                 HEX     -3.496         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.122         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.2424         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.5014         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.06%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.23%\\n                                  Urea[e]                              UreaEX    -0.2428         1 22.78%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n           ids  growth      status      uptake secretion\\n144  {PMM0225}     NaN  infeasible  L_ValineEX    UreaEX\\n164  {PMM0770}     NaN  infeasible  L_ValineEX    UreaEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_AlanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_AlanineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000055747260347\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 46.00%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.117         5 54.00%\\n          Photon[e]              LightEX      50.3         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX      -1.11         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1088         1  3.37%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.503         0  0.00%\\n                                     H[e]                                 HEX     -7.818         0  0.00%\\n                             L_Alanine[e]                         L_AlanineEX     -1.039         3 96.53%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX      -4.72         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.08%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake    secretion\\n139  {PMM0770}     NaN  infeasible  L_ValineEX  L_AlanineEX\\n157  {PMM0225}     NaN  infeasible  L_ValineEX  L_AlanineEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_ArginineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_ArginineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000048871925685\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.024         1 46.83%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.141         5 53.17%\\n          Photon[e]              LightEX     47.84         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX     -1.134         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 33.92%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.855         0  0.00%\\n                                     H[e]                                 HEX     -7.203         0  0.00%\\n                            L_Arginine[e]                        L_ArginineEX    -0.2656         6 65.94%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -3.966         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3513         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.03%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.10%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_AsparagineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_AsparagineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000086155152662\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 42.92%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.266         5 57.08%\\n          Photon[e]              LightEX     44.99         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX     -1.259         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.7025         1 22.80%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.839         0  0.00%\\n                                     H[e]                                 HEX     -2.319         0  0.00%\\n                          L_Asparagine[e]                      L_AsparagineEX    -0.5937         4 77.09%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -3.533         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.08%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake       secretion\\n32   {PMM0770}     NaN  infeasible  L_ValineEX  L_AsparagineEX\\n170  {PMM0225}     NaN  infeasible  L_ValineEX  L_AsparagineEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_AspartateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_CystineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_GlutamateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_GlutamateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000086155151058\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 58.62%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.6719         5 41.38%\\n          Photon[e]              LightEX     44.93         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.6649         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.7025         1 19.12%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.245         0  0.00%\\n                                     H[e]                                 HEX     -2.912         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX    -0.5937         5 80.79%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX      -3.83         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake      secretion\\n145  {PMM0770}     NaN  infeasible  L_ValineEX  L_GlutamateEX\\n163  {PMM0225}     NaN  infeasible  L_ValineEX  L_GlutamateEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_GlutamineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_GlutamineEX Objective\\n=========\\n1.0 BIOMASS = 0.00900007964341047\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 50.41%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.9364         5 49.59%\\n          Photon[e]              LightEX     45.17         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.9294         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 27.62%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX      -2.61         0  0.00%\\n                                     H[e]                                 HEX     -6.533         0  0.00%\\n                           L_Glutamine[e]                       L_GlutamineEX    -0.4291         5 72.27%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -3.721         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -0.147         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.08%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX GlycineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX GlycineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000018273407654\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 52.07%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8761         5 47.93%\\n          Photon[e]              LightEX     40.21         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8691         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 33.90%\\n                               Formate[e]                           FormateEX    -0.7951         0  0.00%\\n                               Glycine[e]                           GlycineEX     -0.798         2 65.97%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.059         0  0.00%\\n                                     H[e]                                 HEX     -4.499         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX      -3.17         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX   -0.08529         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.03%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.10%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_HistidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_HistidineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079643410165\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 50.41%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.9364         5 49.59%\\n          Photon[e]              LightEX     46.53         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.9294         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 32.29%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.182         0  0.00%\\n                                     H[e]                                 HEX     -6.588         0  0.00%\\n                           L_Histidine[e]                       L_HistidineEX    -0.2861         6 67.58%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -3.578         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX      -0.29         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.10%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_IsoleucineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_IsoleucineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000086553015482\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 58.60%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.6722         5 41.40%\\n          Photon[e]              LightEX      54.9         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.6652         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1063         1  2.89%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.056         0  0.00%\\n                                     H[e]                                 HEX     -8.968         0  0.00%\\n                          L_Isoleucine[e]                      L_IsoleucineEX    -0.5941         6 97.02%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.614         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake       secretion\\n141  {PMM0225}     NaN  infeasible  L_ValineEX  L_IsoleucineEX\\n167  {PMM0770}     NaN  infeasible  L_ValineEX  L_IsoleucineEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_LeucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_LeucineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079643411266\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 39.25%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.473         5 60.75%\\n          Photon[e]              LightEX     48.48         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX   -0.07117         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  8.92%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.859         0  0.00%\\n                                     H[e]                                 HEX     -7.361         0  0.00%\\n                             L_Leucine[e]                         L_LeucineEX     -1.395         6 91.04%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.365         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.6834         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.03%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake    secretion\\n164  {PMM0770}     NaN  infeasible  L_ValineEX  L_LeucineEX\\n169  {PMM0225}     NaN  infeasible  L_ValineEX  L_LeucineEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_LysineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_LysineEX Objective\\n=========\\n1.0 BIOMASS = 0.00900008615515114\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 48.72%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.002         5 51.28%\\n          Photon[e]              LightEX      50.3         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.9947         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.5706         1 17.06%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.849         0  0.00%\\n                                     H[e]                                 HEX     -7.818         0  0.00%\\n                              L_Lysine[e]                          L_LysineEX    -0.4618         6 82.84%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX      -4.72         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_MethionineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_MethionineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000086553015804\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 55.24%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.7713         5 44.76%\\n          Photon[e]              LightEX     61.08         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.6999         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.7643         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1063         1  2.97%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.848         0  0.00%\\n                                     H[e]                                 HEX     -6.778         0  0.00%\\n                          L_Methionine[e]                      L_MethionineEX    -0.6931         5 96.93%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.456         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake       secretion\\n160  {PMM0770}     NaN  infeasible  L_ValineEX  L_MethionineEX\\n163  {PMM0225}     NaN  infeasible  L_ValineEX  L_MethionineEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_PhenylalanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_PhenylalanineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000025339577361\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 69.15%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.4245         5 30.85%\\n          Photon[e]              LightEX     48.06         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.4175         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.8014         1 20.44%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.888         0  0.00%\\n                                     H[e]                                 HEX     -7.257         0  0.00%\\n                       L_Phenylalanine[e]                   L_PhenylalanineEX    -0.3463         9 79.48%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.374         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.06%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_ProlineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_ProlineEX Objective\\n=========\\n1.0 BIOMASS = 0.00900008615515076\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 58.62%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.6719         5 41.38%\\n          Photon[e]              LightEX     48.78         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.6649         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.7025         1 19.12%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.245         0  0.00%\\n                                     H[e]                                 HEX     -7.436         0  0.00%\\n                             L_Proline[e]                         L_ProlineEX    -0.5937         5 80.79%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.423         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n           ids  growth      status      uptake    secretion\\n145  {PMM0225}     NaN  infeasible  L_ValineEX  L_ProlineEX\\n165  {PMM0770}     NaN  infeasible  L_ValineEX  L_ProlineEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_SerineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_SerineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000018100651234\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 52.10%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8751         5 47.90%\\n          Photon[e]              LightEX     43.31         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8681         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 25.51%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.058         0  0.00%\\n                                     H[e]                                 HEX      -6.07         0  0.00%\\n                              L_Serine[e]                          L_SerineEX    -0.7969         3 74.38%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -3.568         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.0857         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.08%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_ThreonineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX L_ThreonineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000025339579842\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 51.14%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.9094         5 48.86%\\n          Photon[e]              LightEX     51.41         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.9024         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1088         1  3.16%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.295         0  0.00%\\n                                     H[e]                                 HEX     -4.693         0  0.00%\\n                           L_Threonine[e]                       L_ThreonineEX    -0.8312         4 96.74%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX      -4.72         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_TryptophanEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_TyrosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX PhospholipidEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX LPSEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX AcetateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX CitrateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX CitrateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000025339577264\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.481         1 93.34%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX   0.07817         5  6.66%\\n          Photon[e]              LightEX     47.06         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX   -0.07117         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                               Citrate[e]                           CitrateEX    -0.8312         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.018         0  0.00%\\n                                     H[e]                                 HEX     -6.284         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  1.27%\\n                                Oxygen[e]                                O2EX     -4.305         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6 74.32%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX FormateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX PyruvateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX FumarateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX GlutathioneEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX GlutathioneEX Objective\\n=========\\n1.0 BIOMASS = 0.009000086844788395\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 52.58%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8582         5 47.42%\\n          Photon[e]              LightEX        48         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.2669         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8512         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.6274         1 99.48%\\n                               Formate[e]                           FormateEX    -0.2571         0  0.00%\\n                           Glutathione[e]                       GlutathioneEX      -0.26         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.803         0  0.00%\\n                                     H[e]                                 HEX     -6.464         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -4.203         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.10%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.39%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX 4_Methyl_2_oxopentanoateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ValineEX PantothenateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX PantothenateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000025339578752\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 48.72%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.002         5 51.28%\\n          Photon[e]              LightEX     48.55         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX   -0.07117         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.5706         1 99.42%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.849         0  0.00%\\n                                     H[e]                                 HEX     -7.378         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -4.258         0  0.00%\\n                          Pantothenate[e]                      PantothenateEX    -0.9236         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.11%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.43%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX SuccinateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX SuccinateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000055747263896\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     6.035         1 93.92%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX   0.07817         5  6.08%\\n          Photon[e]              LightEX     52.02         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX   -0.07117         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.741         0  0.00%\\n                                     H[e]                                 HEX      -6.97         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.413         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6 74.32%\\n                             Succinate[e]                         SuccinateEX     -1.385         0  0.00%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX ThymidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX XanthosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX 4_AminobenzoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX 4_AminobenzoateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079643411344\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 66.63%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.4766         5 33.37%\\n          Photon[e]              LightEX     46.43         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                       4_Aminobenzoate[e]                   4_AminobenzoateEX    -0.3985         0  0.00%\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.4696         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.60%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.253         0  0.00%\\n                                     H[e]                                 HEX     -6.848         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.966         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.0857         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX 5_MethylthioadenosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX AdenineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX AdenineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079643411544\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.977         1 45.93%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.172         5 54.07%\\n          Photon[e]              LightEX     45.59         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX     -1.165         0  0.00%\\n                               Adenine[e]                           AdenineEX    -0.2188         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.60%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.855         0  0.00%\\n                                     H[e]                                 HEX     -1.466         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.435         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -0.492         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n           ids  growth      status      uptake  secretion\\n134  {PMM0225}     NaN  infeasible  L_ValineEX  AdenineEX\\n151  {PMM1354}     NaN  infeasible  L_ValineEX  AdenineEX\\n161  {PMM0770}     NaN  infeasible  L_ValineEX  AdenineEX\\n'\n",
      "stderr: None\n",
      "L_ValineEX DethiobiotinEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX DethiobiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079643410179\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.192         1 55.39%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8364         5 44.61%\\n          Photon[e]              LightEX     52.12         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.1152         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8294         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 32.08%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1083         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.118         0  0.00%\\n                                     H[e]                                 HEX     -7.298         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1084         1  4.24%\\n                                Oxygen[e]                                O2EX     -4.554         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1012         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1084        15 63.58%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.10%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX GuanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX GuanosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX PutrescineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX SpermidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ValineEX EthanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX EthanolEX Objective\\n=========\\n1.0 BIOMASS = 0.009000018100648825\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.928         1 51.63%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.9234         5 48.37%\\n          Photon[e]              LightEX     50.79         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.9164         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 24.41%\\n                               Ethanol[e]                           EthanolEX     -0.169         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.169         0  0.00%\\n                                     H[e]                                 HEX     -7.771         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.027         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -0.134         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\\n                                  tRNA[e]                              tRNAEX    -0.1691        15 75.49%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX S_MalateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX S_MalateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000025339579441\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     6.035         1 93.92%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX   0.07817         5  6.08%\\n          Photon[e]              LightEX     48.46         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX   -0.07117         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.741         0  0.00%\\n                                     H[e]                                 HEX      -6.08         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  1.27%\\n                                Oxygen[e]                                O2EX      -4.72         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6 74.32%\\n                              S_Malate[e]                          S_MalateEX     -1.385         0  0.00%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX H2O2EX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "GuanineEX B12EX\n",
      "CalledProcessError -9\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "GuanineEX B12EX Objective\n",
      "=========\n",
      "1.0 BIOMASS = 0.009000017150792973\n",
      "\n",
      "Uptake\n",
      "------\n",
      "         Metabolite             Reaction      Flux  C-Number  C-Flux\n",
      "         Cadmium[e]            CadmiumEX 4.344E-07         0   0.00%\n",
      "  Calcium_cation[e]            CalciumEX 0.0002409         0   0.00%\n",
      "    Chloride_ion[e]           ChlorideEX  0.002141         0   0.00%\n",
      "      Cobalt_ion[e]             CobaltEX   0.04753         0   0.00%\n",
      "          Copper[e]             CopperEX  7.86E-07         0   0.00%\n",
      "  Orthophosphate[e] FAKEOrthophosphateEX   0.05079         0   0.00%\n",
      "         Guanine[e]            GuanineEX    0.6461         0   0.00%\n",
      "            HCO3[e]            HCO3EXcar     4.806         1 100.00%\n",
      "             Fe2[e]               IronEX 2.909E-05         0   0.00%\n",
      "          Photon[e]              LightEX     54.63         0   0.00%\n",
      "Magnesium_cation[e]          MagnesiumEX   0.00143         0   0.00%\n",
      "               K[e]          PotassiumEX 0.0008565         0   0.00%\n",
      "        Selenate[e]           SelenateEX 0.0006872         0   0.00%\n",
      "   Sodium_cation[e]             SodiumEX  0.004015         0   0.00%\n",
      "Strontium_cation[e]          StrontiumEX 1.034E-05         0   0.00%\n",
      "         Sulfate[e]            SulfateEX  0.006847         0   0.00%\n",
      "             Zn2[e]               ZincEX 1.655E-06         0   0.00%\n",
      "\n",
      "Secretion\n",
      "---------\n",
      "                               Metabolite                            Reaction       Flux  C-Number C-Flux\n",
      "                     Cobamide_coenzyme[e]                               B12EX   -0.04752        72 97.68%\n",
      "                               biomass[e]                           BiomassEX     -0.009         0  0.00%\n",
      "                                   CO2[e]                               CO2EX   -0.07783         1  2.22%\n",
      "                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\n",
      "                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\n",
      "                                   H2O[e]                               H2OEX     -1.409         0  0.00%\n",
      "                                     H[e]                                 HEX     -8.904         0  0.00%\n",
      "                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\n",
      "                                Oxygen[e]                                O2EX     -4.961         0  0.00%\n",
      "S_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.02%\n",
      "              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.07%\n",
      "                            Xanthosine[e]                        XanthosineEX     -0.598         0  0.00%\n",
      "                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\n",
      "\n",
      "Empty DataFrame\n",
      "Columns: [ids, growth, status, uptake, secretion]\n",
      "Index: []\n",
      "\n",
      "stderr: \n",
      "GuanineEX BiotinEX\n",
      "GuanineEX MethanolEX\n",
      "GuanineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "GuanineEX SMethyl5thioDriboseEX\n",
      "GuanineEX GlycolateEX\n",
      "GuanineEX AmmoniaEX\n",
      "GuanineEX GlycogenEX\n",
      "GuanineEX H2SEX\n",
      "GuanineEX SpermineEX\n",
      "GuanineEX ThiosulfateEX\n",
      "GuanineEX UreaEX\n",
      "GuanineEX L_AlanineEX\n",
      "GuanineEX L_ArginineEX\n",
      "GuanineEX L_AsparagineEX\n",
      "GuanineEX L_AspartateEX\n",
      "GuanineEX L_CystineEX\n",
      "GuanineEX L_GlutamateEX\n",
      "GuanineEX L_GlutamineEX\n",
      "GuanineEX GlycineEX\n",
      "GuanineEX L_HistidineEX\n",
      "GuanineEX L_IsoleucineEX\n",
      "GuanineEX L_LeucineEX\n",
      "GuanineEX L_LysineEX\n",
      "GuanineEX L_MethionineEX\n",
      "GuanineEX L_PhenylalanineEX\n",
      "GuanineEX L_ProlineEX\n",
      "GuanineEX L_SerineEX\n",
      "GuanineEX L_ThreonineEX\n",
      "GuanineEX L_TryptophanEX\n",
      "GuanineEX L_TyrosineEX\n",
      "GuanineEX L_ValineEX\n",
      "GuanineEX PhospholipidEX\n",
      "GuanineEX LPSEX\n",
      "GuanineEX AcetateEX\n",
      "GuanineEX CitrateEX\n",
      "GuanineEX FormateEX\n",
      "GuanineEX PyruvateEX\n",
      "GuanineEX FumarateEX\n",
      "GuanineEX GlutathioneEX\n",
      "GuanineEX 4_Methyl_2_oxopentanoateEX\n",
      "GuanineEX PantothenateEX\n",
      "GuanineEX SuccinateEX\n",
      "GuanineEX ThymidineEX\n",
      "GuanineEX XanthosineEX\n",
      "GuanineEX 4_AminobenzoateEX\n",
      "GuanineEX 5_MethylthioadenosineEX\n",
      "GuanineEX AdenineEX\n",
      "GuanineEX DethiobiotinEX\n",
      "GuanineEX GuanosineEX\n",
      "GuanineEX PutrescineEX\n",
      "GuanineEX SpermidineEX\n",
      "GuanineEX EthanolEX\n",
      "GuanineEX S_MalateEX\n",
      "GuanineEX H2O2EX\n"
     ]
    }
   ],
   "source": [
    "tdf = n_exchange_df\n",
    "deltype = 'gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_secretion_30082022')\n",
    "knockout_success = [\n",
    "    _secretion_deletion_feeding_with_timeout(dpath, u, s, m, deltype, timeout=10)\n",
    "    for (u, s, m) in zip(tdf['uptake'], tdf['secretion'], tdf['maximum'])\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 801,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'error',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success']"
      ]
     },
     "execution_count": 801,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knockout_success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 802,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "success    671\n",
       "error      126\n",
       "timeout     84\n",
       "Name: gknockout_success1, dtype: int64"
      ]
     },
     "execution_count": 802,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_exchange_df['gknockout_success1'] = knockout_success\n",
    "n_exchange_df['gknockout_success1'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 804,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GuanosineEX L_HistidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_AspartateEX L_SerineEX\n",
      "L_AspartateEX 4_AminobenzoateEX\n",
      "L_AspartateEX PutrescineEX\n",
      "GlutathioneEX L_AlanineEX\n",
      "GlutathioneEX L_IsoleucineEX\n",
      "L_CystineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_AlanineEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_LeucineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_LeucineEX L_MethionineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_LeucineEX EthanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlycineEX PutrescineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_MethionineEX\n",
      "AmmoniaEX CitrateEX\n",
      "AmmoniaEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX MethanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX AmmoniaEX\n",
      "L_ArginineEX UreaEX\n",
      "L_ArginineEX 4_AminobenzoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "CyanateEX L_IsoleucineEX\n",
      "CyanateEX L_LeucineEX\n",
      "CyanateEX L_LysineEX\n",
      "CyanateEX L_ProlineEX\n",
      "CyanateEX L_ThreonineEX\n",
      "CyanateEX AcetateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "CyanateEX CitrateEX\n",
      "CyanateEX FumarateEX\n",
      "CyanateEX GlutathioneEX\n",
      "CyanateEX XanthosineEX\n",
      "CyanateEX GuanineEX\n",
      "CyanateEX SpermidineEX\n",
      "CyanateEX H2O2EX\n",
      "L_ValineEX B12EX\n",
      "L_ValineEX BiotinEX\n",
      "L_ValineEX MethanolEX\n",
      "L_ValineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX SAdenosyl4methylthio2oxobutanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.009000079869814895\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.192         1 55.39%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8364         5 44.61%\\n          Photon[e]              LightEX     52.12         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.1152         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8294         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 32.08%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1083         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.118         0  0.00%\\n                                     H[e]                                 HEX     -7.298         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1084         1  4.24%\\n                                Oxygen[e]                                O2EX     -4.554         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1012         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1084        15 63.58%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.10%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX SMethyl5thioDriboseEX\n",
      "L_ValineEX GlycolateEX\n",
      "L_ValineEX AmmoniaEX\n",
      "L_ValineEX GlycogenEX\n",
      "L_ValineEX SpermineEX\n",
      "L_ValineEX UreaEX\n",
      "L_ValineEX L_AlanineEX\n",
      "L_ValineEX L_ArginineEX\n",
      "L_ValineEX L_AsparagineEX\n",
      "L_ValineEX L_AspartateEX\n",
      "L_ValineEX L_CystineEX\n",
      "L_ValineEX L_GlutamateEX\n",
      "L_ValineEX L_GlutamineEX\n",
      "L_ValineEX GlycineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX GlycineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000018273407654\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.759         1 52.07%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.8761         5 47.93%\\n          Photon[e]              LightEX     40.21         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.8691         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 33.90%\\n                               Formate[e]                           FormateEX    -0.7951         0  0.00%\\n                               Glycine[e]                           GlycineEX     -0.798         2 65.97%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.059         0  0.00%\\n                                     H[e]                                 HEX     -4.499         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.00%\\n                                Oxygen[e]                                O2EX      -3.17         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX   -0.08529         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.03%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.10%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX L_HistidineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_ValineEX L_IsoleucineEX\n",
      "L_ValineEX L_LeucineEX\n",
      "L_ValineEX L_LysineEX\n",
      "L_ValineEX L_MethionineEX\n",
      "L_ValineEX L_PhenylalanineEX\n",
      "L_ValineEX L_ProlineEX\n",
      "L_ValineEX L_SerineEX\n",
      "L_ValineEX L_ThreonineEX\n",
      "L_ValineEX L_TryptophanEX\n",
      "L_ValineEX L_TyrosineEX\n",
      "L_ValineEX PhospholipidEX\n",
      "L_ValineEX LPSEX\n",
      "L_ValineEX AcetateEX\n",
      "L_ValineEX CitrateEX\n",
      "L_ValineEX FormateEX\n",
      "L_ValineEX PyruvateEX\n",
      "L_ValineEX FumarateEX\n",
      "L_ValineEX GlutathioneEX\n",
      "L_ValineEX PantothenateEX\n",
      "L_ValineEX SuccinateEX\n",
      "L_ValineEX ThymidineEX\n",
      "L_ValineEX XanthosineEX\n",
      "L_ValineEX 4_AminobenzoateEX\n",
      "L_ValineEX 5_MethylthioadenosineEX\n",
      "L_ValineEX AdenineEX\n",
      "L_ValineEX DethiobiotinEX\n",
      "L_ValineEX GuanineEX\n",
      "L_ValineEX GuanosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX GuanosineEX Objective\\n=========\\n1.0 BIOMASS = 0.009000061542511754\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.344E-07         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0002409         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.002141         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 1.397E-05         0  0.00%\\n          Copper[e]             CopperEX  7.86E-07         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00327         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.942         1 49.89%\\n             Fe2[e]               IronEX 2.909E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.9927         5 50.11%\\n          Photon[e]              LightEX     46.32         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00143         0  0.00%\\n               K[e]          PotassiumEX 0.0008565         0  0.00%\\n        Selenate[e]           SelenateEX 0.0006872         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.004015         0  0.00%\\nStrontium_cation[e]          StrontiumEX 1.034E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.006847         0  0.00%\\n             Zn2[e]               ZincEX 1.655E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.9857         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.009         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.60%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -5.429E-05         2  0.01%\\n                             Guanosine[e]                         GuanosineEX    -0.1829         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001001         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.482         0  0.00%\\n                                     H[e]                                 HEX     -1.696         0  0.00%\\n                              Methanol[e]                          MethanolEX -4.191E-05         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.496         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.2947         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -4.191E-05        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0004099         6  0.30%\\n                                  tRNA[e]                              tRNAEX -4.694E-06        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX PutrescineEX\n",
      "L_ValineEX SpermidineEX\n",
      "L_ValineEX EthanolEX\n",
      "L_ValineEX S_MalateEX\n"
     ]
    }
   ],
   "source": [
    "tdf = n_exchange_df.loc[n_exchange_df['gknockout_success1'].isin(['timeout'])]\n",
    "deltype = 'gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_secretion_30082022')\n",
    "knockout_success = [\n",
    "    _secretion_deletion_feeding_with_timeout(dpath, u, s, m, deltype, timeout=10)\n",
    "    for (u, s, m) in zip(tdf['uptake'], tdf['secretion'], tdf['maximum'])\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 814,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success']"
      ]
     },
     "execution_count": 814,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knockout_success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 815,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "success    123\n",
       "timeout      3\n",
       "Name: gknockout_success4, dtype: int64"
      ]
     },
     "execution_count": 815,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_exchange_df.loc[n_exchange_df.gknockout_success1.isin(['error']), 'gknockout_success4'] = knockout_success\n",
    "n_exchange_df['gknockout_success4'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 816,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_AspartateEX SpermineEX\n",
      "AmmoniaEX FumarateEX\n",
      "CyanateEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 30\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX 4_Methyl_2_oxopentanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047467\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n         Cyanate[e]            CyanateEX    0.2336         1  4.57%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 95.43%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX     58.15         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.5694         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX     -0.325         1 97.05%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.10%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX      -1.51         0  0.00%\\n                                     H[e]                                 HEX     -9.429         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.04%\\n                                Oxygen[e]                                O2EX     -5.584         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.56%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  2.19%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.06%\\n\\n'\n",
      "stderr: None\n"
     ]
    }
   ],
   "source": [
    "tdf = n_exchange_df.loc[n_exchange_df['gknockout_success4'].isin(['timeout'])]\n",
    "deltype = 'gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_secretion_30082022')\n",
    "knockout_success = [\n",
    "    _secretion_deletion_feeding_with_timeout(dpath, u, s, m, deltype, timeout=30)\n",
    "    for (u, s, m) in zip(tdf['uptake'], tdf['secretion'], tdf['maximum']*0.8)\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 813,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_AspartateEX SpermineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX SpermineEX Objective\\n=========\\n1.0 BIOMASS = 0.055262585003112714\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.667E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001479         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01315         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 8.575E-05         0  0.00%\\n          Copper[e]             CopperEX 4.826E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02008         0  0.00%\\n             H2O[e]                H2OEX     18.23         0  0.00%\\n            HCO3[e]            HCO3EXcar    0.9079         1  1.60%\\n             Fe2[e]               IronEX 0.0001786         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     13.95         4 98.40%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.008781         0  0.00%\\n      Molybdenum[e]         MolybdenumEX  3.81E-07         0  0.00%\\n               K[e]          PotassiumEX  0.005259         0  0.00%\\n        Selenate[e]           SelenateEX   0.00422         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02465         0  0.00%\\nStrontium_cation[e]          StrontiumEX 6.351E-05         0  0.00%\\n         Sulfate[e]            SulfateEX     1.138         0  0.00%\\n             Zn2[e]               ZincEX 1.016E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -9.115         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.05526         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  5.27%\\n                                    CO[e]                                COEX -4.702E-07         1  0.00%\\n                               Glycine[e]                           GlycineEX     -1.065         2 13.68%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003334         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -12.16         0  0.00%\\n                                     H[e]                                 HEX     -34.95         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002574         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.618         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -12.78         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002574        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX     -1.098         6 42.31%\\n                              Spermine[e]                          SpermineEX    -0.5477        10 35.18%\\n                                  Urea[e]                              UreaEX    -0.5502         1  3.53%\\n                                  tRNA[e]                              tRNAEX -2.882E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX L_GlutamateEX\n",
      "L_AspartateEX L_ThreonineEX\n",
      "L_AspartateEX 4_Methyl_2_oxopentanoateEX\n",
      "GlutathioneEX PyruvateEX\n",
      "L_GlutamineEX ThiosulfateEX\n",
      "L_GlutamineEX L_ProlineEX\n",
      "L_SerineEX BiotinEX\n",
      "L_SerineEX MethanolEX\n",
      "L_SerineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_SerineEX SMethyl5thioDriboseEX\n",
      "L_SerineEX SpermineEX\n",
      "L_SerineEX UreaEX\n",
      "L_SerineEX L_GlutamineEX\n",
      "L_SerineEX L_IsoleucineEX\n",
      "L_SerineEX LPSEX\n",
      "L_SerineEX PantothenateEX\n",
      "L_SerineEX 5_MethylthioadenosineEX\n",
      "L_SerineEX DethiobiotinEX\n",
      "L_CystineEX MethanolEX\n",
      "L_CystineEX L_GlutamineEX\n",
      "L_CystineEX L_IsoleucineEX\n",
      "L_CystineEX L_ProlineEX\n",
      "L_CystineEX L_TryptophanEX\n",
      "L_CystineEX AcetateEX\n",
      "L_CystineEX FormateEX\n",
      "L_CystineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_CystineEX AdenineEX\n",
      "L_AlanineEX UreaEX\n",
      "L_AlanineEX L_IsoleucineEX\n",
      "L_AlanineEX PantothenateEX\n",
      "L_LeucineEX SMethyl5thioDriboseEX\n",
      "L_LeucineEX L_ArginineEX\n",
      "L_LeucineEX L_AspartateEX\n",
      "L_LeucineEX L_CystineEX\n",
      "L_LeucineEX L_GlutamineEX\n",
      "L_LeucineEX GlycineEX\n",
      "L_LeucineEX L_HistidineEX\n",
      "L_LeucineEX L_LysineEX\n",
      "L_LeucineEX L_ThreonineEX\n",
      "L_LeucineEX L_ValineEX\n",
      "L_LeucineEX PhospholipidEX\n",
      "L_LeucineEX AcetateEX\n",
      "L_LeucineEX PyruvateEX\n",
      "L_LeucineEX FumarateEX\n",
      "L_LeucineEX GlutathioneEX\n",
      "L_LeucineEX PantothenateEX\n",
      "L_LeucineEX SuccinateEX\n",
      "L_LeucineEX 5_MethylthioadenosineEX\n",
      "L_LeucineEX AdenineEX\n",
      "L_LeucineEX GuanosineEX\n",
      "L_LeucineEX PutrescineEX\n",
      "L_LeucineEX SpermidineEX\n",
      "GlycineEX AmmoniaEX\n",
      "GlycineEX L_CystineEX\n",
      "GlycineEX L_GlutamateEX\n",
      "GlycineEX L_GlutamineEX\n",
      "GlycineEX L_ProlineEX\n",
      "GlycineEX L_TryptophanEX\n",
      "GlycineEX AcetateEX\n",
      "GlycineEX ThymidineEX\n",
      "GlycineEX GuanineEX\n",
      "UreaEX B12EX\n",
      "UreaEX GlycogenEX\n",
      "UreaEX SpermineEX\n",
      "UreaEX L_TryptophanEX\n",
      "UreaEX GlutathioneEX\n",
      "UreaEX XanthosineEX\n",
      "UreaEX AdenineEX\n",
      "UreaEX EthanolEX\n",
      "AmmoniaEX BiotinEX\n",
      "AmmoniaEX L_ArginineEX\n",
      "AmmoniaEX L_AsparagineEX\n",
      "AmmoniaEX L_CystineEX\n",
      "AmmoniaEX L_LysineEX\n",
      "AmmoniaEX L_PhenylalanineEX\n",
      "AmmoniaEX L_ProlineEX\n",
      "AmmoniaEX L_ValineEX\n",
      "AmmoniaEX LPSEX\n",
      "AmmoniaEX FormateEX\n",
      "AmmoniaEX PyruvateEX\n",
      "AmmoniaEX FumarateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX FumarateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600474182\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.923         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX      53.7         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                              Fumarate[e]                          FumarateEX     -1.139         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX      -3.36         0  0.00%\\n                                     H[e]                                 HEX     -7.504         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.015         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX GuanosineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AmmoniaEX PutrescineEX\n",
      "AmmoniaEX EthanolEX\n",
      "AmmoniaEX S_MalateEX\n",
      "L_GlutamateEX BiotinEX\n",
      "L_GlutamateEX MethanolEX\n",
      "L_GlutamateEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_GlutamateEX SMethyl5thioDriboseEX\n",
      "L_GlutamateEX SpermineEX\n",
      "L_GlutamateEX L_MethionineEX\n",
      "L_GlutamateEX 5_MethylthioadenosineEX\n",
      "L_GlutamateEX AdenineEX\n",
      "L_GlutamateEX DethiobiotinEX\n",
      "L_ArginineEX SMethyl5thioDriboseEX\n",
      "L_ArginineEX GlycolateEX\n",
      "L_ArginineEX SpermineEX\n",
      "L_ArginineEX L_AlanineEX\n",
      "L_ArginineEX L_AsparagineEX\n",
      "L_ArginineEX L_GlutamateEX\n",
      "L_ArginineEX L_GlutamineEX\n",
      "L_ArginineEX L_HistidineEX\n",
      "L_ArginineEX L_IsoleucineEX\n",
      "L_ArginineEX L_LeucineEX\n",
      "L_ArginineEX L_MethionineEX\n",
      "L_ArginineEX L_ProlineEX\n",
      "L_ArginineEX L_TryptophanEX\n",
      "L_ArginineEX PhospholipidEX\n",
      "L_ArginineEX GlutathioneEX\n",
      "L_ArginineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_ArginineEX ThymidineEX\n",
      "L_ArginineEX 5_MethylthioadenosineEX\n",
      "L_ArginineEX AdenineEX\n",
      "L_ArginineEX DethiobiotinEX\n",
      "L_ArginineEX SpermidineEX\n",
      "CyanateEX L_HistidineEX\n",
      "CyanateEX L_TryptophanEX\n",
      "CyanateEX L_ValineEX\n",
      "CyanateEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX 4_Methyl_2_oxopentanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047467\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n         Cyanate[e]            CyanateEX    0.2336         1  4.57%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 95.43%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX     58.15         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.5694         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX     -0.325         1 97.05%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.10%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX      -1.51         0  0.00%\\n                                     H[e]                                 HEX     -9.429         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.04%\\n                                Oxygen[e]                                O2EX     -5.584         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.56%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  2.19%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.06%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX AdenineEX\n",
      "CyanateEX EthanolEX\n",
      "L_ValineEX ThiosulfateEX\n",
      "L_ValineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_ValineEX H2O2EX\n",
      "GuanineEX B12EX\n"
     ]
    }
   ],
   "source": [
    "tdf = n_exchange_df.loc[n_exchange_df['gknockout_success1'].isin(['error'])]\n",
    "deltype = 'gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_secretion_30082022')\n",
    "knockout_success = [\n",
    "    _secretion_deletion_feeding_with_timeout(dpath, u, s, m, deltype, timeout=10)\n",
    "    for (u, s, m) in zip(tdf['uptake'], tdf['secretion'], tdf['maximum']*0.8)\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 817,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GuanosineEX B12EX\n",
      "GuanosineEX BiotinEX\n",
      "GuanosineEX MethanolEX\n",
      "GuanosineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "GuanosineEX SMethyl5thioDriboseEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGuanosineEX SMethyl5thioDriboseEX Objective\\n=========\\n1.0 BIOMASS = 0.25596710573264236\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 1.235E-05         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.006851         0   0.00%\\n    Chloride_ion[e]           ChlorideEX    0.0609         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0003972         0   0.00%\\n          Copper[e]             CopperEX 2.236E-05         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.09299         0   0.00%\\n       Guanosine[e]          GuanosineEX     107.4         0   0.00%\\n            HCO3[e]            HCO3EXcar         8         1 100.00%\\n             Fe2[e]               IronEX 0.0008272         0   0.00%\\n          Photon[e]              LightEX   0.04651         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.04067         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.765E-06         0   0.00%\\n          Oxygen[e]                 O2EX    0.1535         0   0.00%\\n               K[e]          PotassiumEX   0.02436         0   0.00%\\n        Selenate[e]           SelenateEX   0.01954         0   0.00%\\n   Sodium_cation[e]             SodiumEX    0.1142         0   0.00%\\nStrontium_cation[e]          StrontiumEX 0.0002941         0   0.00%\\n         Sulfate[e]            SulfateEX     12.89         0   0.00%\\n             Zn2[e]               ZincEX 4.706E-05         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX     -0.256         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  0.58%\\n                                    CO[e]                                COEX -2.178E-06         1  0.00%\\n                               Citrate[e]                           CitrateEX      -42.1         0  0.00%\\n                               Formate[e]                           FormateEX  -0.005152         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.001544         2  0.00%\\n                               Guanine[e]                           GuanineEX      -79.9         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.02848         0  0.00%\\n                                   H2O[e]                               H2OEX     -32.54         0  0.00%\\n                              Methanol[e]                          MethanolEX  -0.001192         1  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX  -0.001192        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX     -12.71         6 54.24%\\n                              S_Malate[e]                          S_MalateEX    -0.3892         0  0.00%\\n                              Spermine[e]                          SpermineEX     -6.348        10 45.16%\\n                            Xanthosine[e]                        XanthosineEX     -27.53         0  0.00%\\n                                  tRNA[e]                              tRNAEX -0.0001335        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "GuanosineEX GlycolateEX\n",
      "GuanosineEX AmmoniaEX\n",
      "GuanosineEX GlycogenEX\n",
      "GuanosineEX H2SEX\n",
      "GuanosineEX SpermineEX\n",
      "GuanosineEX ThiosulfateEX\n",
      "GuanosineEX UreaEX\n",
      "GuanosineEX L_AlanineEX\n",
      "GuanosineEX L_ArginineEX\n",
      "GuanosineEX L_AsparagineEX\n",
      "GuanosineEX L_AspartateEX\n",
      "GuanosineEX L_CystineEX\n",
      "GuanosineEX L_GlutamateEX\n",
      "GuanosineEX L_GlutamineEX\n",
      "GuanosineEX GlycineEX\n",
      "GuanosineEX L_HistidineEX\n",
      "GuanosineEX L_IsoleucineEX\n",
      "GuanosineEX L_LeucineEX\n",
      "GuanosineEX L_LysineEX\n",
      "GuanosineEX L_MethionineEX\n",
      "GuanosineEX L_PhenylalanineEX\n",
      "GuanosineEX L_ProlineEX\n",
      "GuanosineEX L_SerineEX\n",
      "GuanosineEX L_ThreonineEX\n",
      "GuanosineEX L_TryptophanEX\n",
      "GuanosineEX L_TyrosineEX\n",
      "GuanosineEX L_ValineEX\n",
      "GuanosineEX PhospholipidEX\n",
      "GuanosineEX LPSEX\n",
      "GuanosineEX AcetateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGuanosineEX AcetateEX Objective\\n=========\\n1.0 BIOMASS = 1.2394474020333732\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 5.982E-05         0   0.00%\\n  Calcium_cation[e]            CalciumEX   0.03317         0   0.00%\\n    Chloride_ion[e]           ChlorideEX    0.2949         0   0.00%\\n      Cobalt_ion[e]             CobaltEX  0.001923         0   0.00%\\n          Copper[e]             CopperEX 0.0001083         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX    0.4503         0   0.00%\\n       Guanosine[e]          GuanosineEX      1000         0   0.00%\\n            HCO3[e]            HCO3EXcar         8         1 100.00%\\n             Fe2[e]               IronEX  0.004006         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX    0.1969         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 8.546E-06         0   0.00%\\n          Oxygen[e]                 O2EX     1.258         0   0.00%\\n               K[e]          PotassiumEX     0.118         0   0.00%\\n        Selenate[e]           SelenateEX   0.09464         0   0.00%\\n   Sodium_cation[e]             SodiumEX    0.5529         0   0.00%\\nStrontium_cation[e]          StrontiumEX  0.001424         0   0.00%\\n         Sulfate[e]            SulfateEX     51.91         0   0.00%\\n             Zn2[e]               ZincEX 0.0002279         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Acetate[e]                           AcetateEX       -800         0  0.00%\\n                               biomass[e]                           BiomassEX     -1.239         0  0.00%\\n                                    CO[e]                                COEX -1.055E-05         1  0.00%\\n                               Formate[e]                           FormateEX       -154         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.007477         2  0.00%\\n                               Guanine[e]                           GuanineEX     -807.7         0  0.00%\\n                                   H2O[e]                               H2OEX      -21.3         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -37.22         0  0.00%\\n                                     H[e]                                 HEX     -58.05         0  0.00%\\n                          L_Methionine[e]                      L_MethionineEX     -13.75         5  2.65%\\n                              Methanol[e]                          MethanolEX  -0.005772         1  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -608.4         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX  -0.005772        15  0.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.05645         6  0.01%\\n                            Xanthosine[e]                        XanthosineEX     -24.12         0  0.00%\\n                                  tRNA[e]                              tRNAEX     -168.1        15 97.33%\\n\\n'\n",
      "stderr: None\n",
      "GuanosineEX CitrateEX\n",
      "GuanosineEX FormateEX\n",
      "GuanosineEX PyruvateEX\n",
      "GuanosineEX FumarateEX\n",
      "GuanosineEX GlutathioneEX\n",
      "GuanosineEX 4_Methyl_2_oxopentanoateEX\n",
      "GuanosineEX PantothenateEX\n",
      "GuanosineEX SuccinateEX\n",
      "GuanosineEX ThymidineEX\n",
      "GuanosineEX XanthosineEX\n",
      "GuanosineEX 4_AminobenzoateEX\n",
      "GuanosineEX 5_MethylthioadenosineEX\n",
      "GuanosineEX AdenineEX\n",
      "GuanosineEX DethiobiotinEX\n",
      "GuanosineEX GuanineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GuanosineEX PutrescineEX\n",
      "GuanosineEX SpermidineEX\n",
      "GuanosineEX EthanolEX\n",
      "GuanosineEX S_MalateEX\n",
      "GuanosineEX H2O2EX\n",
      "L_AspartateEX B12EX\n",
      "L_AspartateEX BiotinEX\n",
      "L_AspartateEX MethanolEX\n",
      "L_AspartateEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_AspartateEX SMethyl5thioDriboseEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX SMethyl5thioDriboseEX Objective\\n=========\\n1.0 BIOMASS = 0.055632476403817396\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.685E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001489         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01324         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 8.633E-05         0  0.00%\\n          Copper[e]             CopperEX 4.859E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02021         0  0.00%\\n             H2O[e]                H2OEX     18.22         0  0.00%\\n            HCO3[e]            HCO3EXcar    0.9092         1  1.60%\\n             Fe2[e]               IronEX 0.0001798         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     13.95         4 98.40%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00884         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.836E-07         0  0.00%\\n               K[e]          PotassiumEX  0.005294         0  0.00%\\n        Selenate[e]           SelenateEX  0.004248         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02482         0  0.00%\\nStrontium_cation[e]          StrontiumEX 6.393E-05         0  0.00%\\n         Sulfate[e]            SulfateEX     1.136         0  0.00%\\n             Zn2[e]               ZincEX 1.023E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -9.115         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.05563         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  5.28%\\n                                    CO[e]                                COEX -4.733E-07         1  0.00%\\n                               Glycine[e]                           GlycineEX     -1.062         2 13.67%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003356         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -12.16         0  0.00%\\n                                     H[e]                                 HEX     -34.95         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002591         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.613         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -12.78         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002591        15  0.03%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX     -1.096         6 42.31%\\n                              Spermine[e]                          SpermineEX    -0.5466        10 35.18%\\n                                  Urea[e]                              UreaEX    -0.5492         1  3.53%\\n                                  tRNA[e]                              tRNAEX -2.901E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX GlycolateEX\n",
      "L_AspartateEX AmmoniaEX\n",
      "L_AspartateEX GlycogenEX\n",
      "L_AspartateEX H2SEX\n",
      "L_AspartateEX SpermineEX\n",
      "L_AspartateEX ThiosulfateEX\n",
      "L_AspartateEX UreaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX UreaEX Objective\\n=========\\n1.0 BIOMASS = 0.05511207878108888\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX  2.66E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001475         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01311         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 8.552E-05         0  0.00%\\n          Copper[e]             CopperEX 4.813E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02002         0  0.00%\\n             H2O[e]                H2OEX     18.58         0  0.00%\\n            HCO3[e]            HCO3EXcar     2.405         1  3.69%\\n             Fe2[e]               IronEX 0.0001781         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     15.68         4 96.31%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.008757         0  0.00%\\n      Molybdenum[e]         MolybdenumEX   3.8E-07         0  0.00%\\n               K[e]          PotassiumEX  0.005245         0  0.00%\\n        Selenate[e]           SelenateEX  0.004208         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02458         0  0.00%\\nStrontium_cation[e]          StrontiumEX 6.333E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.04193         0  0.00%\\n             Zn2[e]               ZincEX 1.013E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -7.015         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.05511         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 28.38%\\n                                    CO[e]                                COEX -4.689E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003324         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -13.49         0  0.00%\\n                                     H[e]                                 HEX     -37.14         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002567         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.474         0  0.00%\\n                            Putrescine[e]                        PutrescineEX     -2.046         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -17.09         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002567        15  0.13%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00251         6  0.52%\\n                                  Urea[e]                              UreaEX     -2.049         1 70.91%\\n                                  tRNA[e]                              tRNAEX -2.874E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX L_AlanineEX\n",
      "L_AspartateEX L_ArginineEX\n",
      "L_AspartateEX L_AsparagineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX L_AsparagineEX Objective\\n=========\\n1.0 BIOMASS = 0.07420852187817394\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 3.582E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001986         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01765         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001152         0  0.00%\\n          Copper[e]             CopperEX 6.481E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02696         0  0.00%\\n            HCO3[e]            HCO3EXcar    0.4837         1  0.27%\\n             Fe2[e]               IronEX 0.0002398         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     45.27         4 99.73%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01179         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 5.117E-07         0  0.00%\\n          Oxygen[e]                 O2EX     9.594         0  0.00%\\n               K[e]          PotassiumEX  0.007062         0  0.00%\\n        Selenate[e]           SelenateEX  0.005666         0  0.00%\\n   Sodium_cation[e]             SodiumEX    0.0331         0  0.00%\\nStrontium_cation[e]          StrontiumEX 8.528E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.05645         0  0.00%\\n             Zn2[e]               ZincEX 1.364E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.07421         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  0.91%\\n                                    CO[e]                                COEX -6.314E-07         1  0.00%\\n                              Fumarate[e]                          FumarateEX     -20.16         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0004476         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -22.77         0  0.00%\\n                                     H[e]                                 HEX     -25.93         0  0.00%\\n                          L_Asparagine[e]                      L_AsparagineEX     -22.31         4 99.06%\\n                              Methanol[e]                          MethanolEX -0.0003456         1  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -1.829         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0003456        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00338         6  0.02%\\n                              S_Malate[e]                          S_MalateEX    -0.3976         0  0.00%\\n                                  Urea[e]                              UreaEX   -0.00338         1  0.00%\\n                                  tRNA[e]                              tRNAEX  -3.87E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX L_CystineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_AspartateEX L_GlutamateEX\n",
      "L_AspartateEX L_GlutamineEX\n",
      "L_AspartateEX GlycineEX\n",
      "L_AspartateEX L_HistidineEX\n",
      "L_AspartateEX L_IsoleucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX L_IsoleucineEX Objective\\n=========\\n1.0 BIOMASS = 0.06704803016205906\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 3.236E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001795         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01595         0  0.00%\\n      Cobalt_ion[e]             CobaltEX  0.000104         0  0.00%\\n          Copper[e]             CopperEX 5.856E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02436         0  0.00%\\n             H2O[e]                H2OEX     15.79         0  0.00%\\n            HCO3[e]            HCO3EXcar     1.057         1  1.77%\\n             Fe2[e]               IronEX 0.0002167         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     14.67         4 98.23%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01065         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 4.623E-07         0  0.00%\\n               K[e]          PotassiumEX  0.006381         0  0.00%\\n        Selenate[e]           SelenateEX  0.005119         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02991         0  0.00%\\nStrontium_cation[e]          StrontiumEX 7.705E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.05101         0  0.00%\\n             Zn2[e]               ZincEX 1.233E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                 5_Methylthioadenosine[e]             5_MethylthioadenosineEX  -0.003053         0  0.00%\\n                               Ammonia[e]                           AmmoniaEX     -3.117         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.06705         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  2.45%\\n                                    CO[e]                                COEX -5.704E-07         1  0.00%\\n                               Formate[e]                           FormateEX     -2.975         0  0.00%\\n                               Glycine[e]                           GlycineEX     -3.616         2 21.57%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0004044         2  0.00%\\n                               Guanine[e]                           GuanineEX    -0.6169         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX      -9.62         0  0.00%\\n                                     H[e]                                 HEX     -32.23         0  0.00%\\n                          L_Isoleucine[e]                      L_IsoleucineEX     -4.244         6 75.95%\\n                              Methanol[e]                          MethanolEX -0.0003122         1  0.00%\\n                                Oxygen[e]                                O2EX     -9.096         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -5.571         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0003122        15  0.01%\\n                                  Urea[e]                              UreaEX  -0.003053         1  0.01%\\n                                  tRNA[e]                              tRNAEX -3.497E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX L_LeucineEX\n",
      "L_AspartateEX L_LysineEX\n",
      "L_AspartateEX L_MethionineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX L_MethionineEX Objective\\n=========\\n1.0 BIOMASS = 0.10134289290530496\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 4.891E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.002712         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02411         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001573         0  0.00%\\n          Copper[e]             CopperEX 8.851E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03682         0  0.00%\\n             H2O[e]                H2OEX     3.882         0  0.00%\\n            HCO3[e]            HCO3EXcar    0.6605         1  1.94%\\n             Fe2[e]               IronEX 0.0003275         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     8.354         4 98.06%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX    0.0161         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 6.988E-07         0  0.00%\\n               K[e]          PotassiumEX  0.009644         0  0.00%\\n        Selenate[e]           SelenateEX  0.007738         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.04521         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001165         0  0.00%\\n         Sulfate[e]            SulfateEX     3.958         0  0.00%\\n             Zn2[e]               ZincEX 1.863E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.1013         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  2.99%\\n                                    CO[e]                                COEX -8.622E-07         1  0.00%\\n                              Fumarate[e]                          FumarateEX    -0.3812         0  0.00%\\n                               Glycine[e]                           GlycineEX     -3.583         2 26.12%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0006113         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -4.203         0  0.00%\\n                                     H[e]                                 HEX     -29.03         0  0.00%\\n                          L_Methionine[e]                      L_MethionineEX     -3.881         5 70.74%\\n                              Methanol[e]                          MethanolEX -0.0004719         1  0.00%\\n                                Oxygen[e]                                O2EX      -14.5         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004719        15  0.03%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004615         6  0.10%\\n                                  Urea[e]                              UreaEX  -0.004615         1  0.02%\\n                                  tRNA[e]                              tRNAEX -5.285E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX L_PhenylalanineEX\n",
      "L_AspartateEX L_ProlineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX L_ProlineEX Objective\\n=========\\n1.0 BIOMASS = 0.06144439479840995\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.966E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001645         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01462         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 9.534E-05         0  0.00%\\n          Copper[e]             CopperEX 5.366E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02232         0  0.00%\\n             H2O[e]                H2OEX     24.06         0  0.00%\\n            HCO3[e]            HCO3EXcar    0.4005         1  0.61%\\n             Fe2[e]               IronEX 0.0001986         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     16.39         4 99.39%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.009763         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 4.237E-07         0  0.00%\\n               K[e]          PotassiumEX  0.005847         0  0.00%\\n        Selenate[e]           SelenateEX  0.004692         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02741         0  0.00%\\nStrontium_cation[e]          StrontiumEX 7.061E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.04674         0  0.00%\\n             Zn2[e]               ZincEX  1.13E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -12.92         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.06144         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  5.28%\\n                                    CO[e]                                COEX -5.228E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003706         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -16.23         0  0.00%\\n                                     H[e]                                 HEX      -37.1         0  0.00%\\n                             L_Proline[e]                         L_ProlineEX     -2.938         5 94.56%\\n                              Methanol[e]                          MethanolEX -0.0002861         1  0.00%\\n                                Oxygen[e]                                O2EX     -1.967         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -15.78         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002861        15  0.03%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.002798         6  0.11%\\n                                  Urea[e]                              UreaEX  -0.002798         1  0.02%\\n                                  tRNA[e]                              tRNAEX -3.204E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX L_SerineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_AspartateEX L_ThreonineEX\n",
      "L_AspartateEX L_TryptophanEX\n",
      "L_AspartateEX L_TyrosineEX\n",
      "L_AspartateEX L_ValineEX\n",
      "L_AspartateEX PhospholipidEX\n",
      "L_AspartateEX LPSEX\n",
      "L_AspartateEX AcetateEX\n",
      "L_AspartateEX CitrateEX\n",
      "L_AspartateEX FormateEX\n",
      "L_AspartateEX PyruvateEX\n",
      "L_AspartateEX FumarateEX\n",
      "L_AspartateEX GlutathioneEX\n",
      "L_AspartateEX 4_Methyl_2_oxopentanoateEX\n",
      "L_AspartateEX PantothenateEX\n",
      "L_AspartateEX SuccinateEX\n",
      "L_AspartateEX ThymidineEX\n",
      "L_AspartateEX XanthosineEX\n",
      "L_AspartateEX 4_AminobenzoateEX\n",
      "L_AspartateEX 5_MethylthioadenosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX 5_MethylthioadenosineEX Objective\\n=========\\n1.0 BIOMASS = 0.055276610259259604\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.668E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001479         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01315         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 8.577E-05         0  0.00%\\n          Copper[e]             CopperEX 4.828E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02008         0  0.00%\\n             H2O[e]                H2OEX     12.21         0  0.00%\\n            HCO3[e]            HCO3EXcar     1.608         1  2.60%\\n             Fe2[e]               IronEX 0.0001786         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     15.08         4 97.40%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.008783         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.811E-07         0  0.00%\\n               K[e]          PotassiumEX   0.00526         0  0.00%\\n        Selenate[e]           SelenateEX  0.004221         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02466         0  0.00%\\nStrontium_cation[e]          StrontiumEX 6.352E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.8724         0  0.00%\\n             Zn2[e]               ZincEX 1.016E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                 5_Methylthioadenosine[e]             5_MethylthioadenosineEX    -0.8328         0  0.00%\\n                               Ammonia[e]                           AmmoniaEX     -7.943         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.05528         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 15.20%\\n                                    CO[e]                                COEX -4.703E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003334         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -12.03         0  0.00%\\n                                     H[e]                                 HEX     -32.98         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002574         1  0.00%\\n                                Oxygen[e]                                O2EX     -3.934         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -14.87         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002574        15  0.07%\\n                              Spermine[e]                          SpermineEX    -0.4152        10 76.96%\\n                                  Urea[e]                              UreaEX    -0.4177         1  7.74%\\n                                  tRNA[e]                              tRNAEX -2.883E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX AdenineEX\n",
      "L_AspartateEX DethiobiotinEX\n",
      "L_AspartateEX GuanineEX\n",
      "L_AspartateEX GuanosineEX\n",
      "L_AspartateEX PutrescineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX PutrescineEX Objective\\n=========\\n1.0 BIOMASS = 0.05491643121164597\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.651E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX   0.00147         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01307         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 8.521E-05         0  0.00%\\n          Copper[e]             CopperEX 4.796E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01995         0  0.00%\\n             H2O[e]                H2OEX     18.58         0  0.00%\\n            HCO3[e]            HCO3EXcar     2.406         1  3.69%\\n             Fe2[e]               IronEX 0.0001775         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     15.69         4 96.31%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.008726         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.787E-07         0  0.00%\\n               K[e]          PotassiumEX  0.005226         0  0.00%\\n        Selenate[e]           SelenateEX  0.004193         0  0.00%\\n   Sodium_cation[e]             SodiumEX    0.0245         0  0.00%\\nStrontium_cation[e]          StrontiumEX 6.311E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.04178         0  0.00%\\n             Zn2[e]               ZincEX  1.01E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -7.013         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.05492         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 28.36%\\n                                    CO[e]                                COEX -4.672E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003313         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -13.49         0  0.00%\\n                                     H[e]                                 HEX     -37.14         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002557         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.474         0  0.00%\\n                            Putrescine[e]                        PutrescineEX     -2.048         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX      -17.1         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002557        15  0.13%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.002501         6  0.52%\\n                                  Urea[e]                              UreaEX     -2.051         1 70.94%\\n                                  tRNA[e]                              tRNAEX -2.864E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX SpermidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX SpermidineEX Objective\\n=========\\n1.0 BIOMASS = 0.0549940348085391\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.654E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001472         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01308         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 8.534E-05         0  0.00%\\n          Copper[e]             CopperEX 4.803E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01998         0  0.00%\\n             H2O[e]                H2OEX     18.31         0  0.00%\\n            HCO3[e]            HCO3EXcar     1.224         1  2.09%\\n             Fe2[e]               IronEX 0.0001777         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     14.32         4 97.91%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.008738         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.792E-07         0  0.00%\\n               K[e]          PotassiumEX  0.005233         0  0.00%\\n        Selenate[e]           SelenateEX  0.004199         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02453         0  0.00%\\nStrontium_cation[e]          StrontiumEX  6.32E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.9072         0  0.00%\\n             Zn2[e]               ZincEX 1.011E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -8.675         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.05499         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  9.57%\\n                                    CO[e]                                COEX -4.679E-07         1  0.00%\\n                               Glycine[e]                           GlycineEX    -0.8348         2 19.48%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003317         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -12.44         0  0.00%\\n                                     H[e]                                 HEX     -35.41         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002561         1  0.00%\\n                                Oxygen[e]                                O2EX      -5.17         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX      -13.7         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002561        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.8679         6 60.76%\\n                            Spermidine[e]                        SpermidineEX    -0.8653         0  0.00%\\n                                  Urea[e]                              UreaEX    -0.8679         1 10.13%\\n                                  tRNA[e]                              tRNAEX -2.868E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX EthanolEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX EthanolEX Objective\\n=========\\n1.0 BIOMASS = 0.05491643121164532\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.651E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX   0.00147         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01307         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 8.521E-05         0  0.00%\\n          Copper[e]             CopperEX 4.796E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01995         0  0.00%\\n             H2O[e]                H2OEX      14.6         0  0.00%\\n            HCO3[e]            HCO3EXcar      1.39         1  2.30%\\n             Fe2[e]               IronEX 0.0001775         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     14.77         4 97.70%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.008726         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.787E-07         0  0.00%\\n               K[e]          PotassiumEX  0.005226         0  0.00%\\n        Selenate[e]           SelenateEX  0.004193         0  0.00%\\n   Sodium_cation[e]             SodiumEX    0.0245         0  0.00%\\nStrontium_cation[e]          StrontiumEX 6.311E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.04178         0  0.00%\\n             Zn2[e]               ZincEX  1.01E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -9.127         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.05492         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  5.02%\\n                                    CO[e]                                COEX -4.672E-07         1  0.00%\\n                               Ethanol[e]                           EthanolEX     -1.032         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003313         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -13.59         0  0.00%\\n                                     H[e]                                 HEX     -36.11         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002557         1  0.00%\\n                                Oxygen[e]                                O2EX     -1.434         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -14.14         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002557        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.002501         6  0.09%\\n                                  Urea[e]                              UreaEX  -0.002501         1  0.02%\\n                                  tRNA[e]                              tRNAEX     -1.032        15 94.84%\\n\\n'\n",
      "stderr: None\n",
      "L_AspartateEX S_MalateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_AspartateEX H2O2EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AspartateEX H2O2EX Objective\\n=========\\n1.0 BIOMASS = 0.1073720038455029\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 5.182E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.002874         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02554         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001666         0  0.00%\\n          Copper[e]             CopperEX 9.378E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03901         0  0.00%\\n             H2O[e]                H2OEX     27.35         0  0.00%\\n            HCO3[e]            HCO3EXcar    0.6998         1  2.29%\\n             Fe2[e]               IronEX  0.000347         0  0.00%\\n     L_Aspartate[e]        L_AspartateEX     7.469         4 97.71%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01706         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 7.403E-07         0  0.00%\\n          Oxygen[e]                 O2EX     12.42         0  0.00%\\n               K[e]          PotassiumEX   0.01022         0  0.00%\\n        Selenate[e]           SelenateEX  0.008198         0  0.00%\\n   Sodium_cation[e]             SodiumEX    0.0479         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001234         0  0.00%\\n         Sulfate[e]            SulfateEX   0.08168         0  0.00%\\n             Zn2[e]               ZincEX 1.974E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.1074         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  5.89%\\n                                    CO[e]                                COEX -9.135E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0006477         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -31.12         0  0.00%\\n                                     H[e]                                 HEX     -36.75         0  0.00%\\n                          L_Asparagine[e]                      L_AsparagineEX     -3.263         4 93.79%\\n                              Methanol[e]                          MethanolEX    -0.0005         1  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -3.746         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.0005        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00489         6  0.21%\\n                                  Urea[e]                              UreaEX   -0.00489         1  0.04%\\n                                  tRNA[e]                              tRNAEX   -5.6E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX B12EX Objective\\n=========\\n1.0 BIOMASS = 0.034647716952061126\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 1.672E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0009273         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008243         0   0.00%\\n      Cobalt_ion[e]             CobaltEX    0.1482         0   0.00%\\n          Copper[e]             CopperEX 3.026E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX    0.1607         0   0.00%\\n     Glutathione[e]        GlutathioneEX     1.859         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.222         1 100.00%\\n             Fe2[e]               IronEX  0.000112         0   0.00%\\n          Photon[e]              LightEX     69.09         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005505         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 2.389E-07         0   0.00%\\n               K[e]          PotassiumEX  0.003297         0   0.00%\\n        Selenate[e]           SelenateEX  0.002645         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01546         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.982E-05         0   0.00%\\n         Sulfate[e]            SulfateEX  0.008632         0   0.00%\\n             Zn2[e]               ZincEX 6.371E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                     Cobamide_coenzyme[e]                               B12EX    -0.1481        72 46.95%\\n                               biomass[e]                           BiomassEX   -0.03465         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  3.61%\\n                                    CO[e]                                COEX -2.948E-07         1  0.00%\\n                               Glycine[e]                           GlycineEX     -1.099         2  9.68%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000209         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003855         0  0.00%\\n                                   H2O[e]                               H2OEX     -0.187         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -1.841         0  0.00%\\n                                     H[e]                                 HEX     -5.479         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX     -1.804         5 39.70%\\n                              Methanol[e]                          MethanolEX -0.0001614         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.158         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -1.757         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001614        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001578         6  0.04%\\n                                  Urea[e]                              UreaEX  -0.001578         1  0.01%\\n                                  tRNA[e]                              tRNAEX -1.807E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX BiotinEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX MethanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX SMethyl5thioDriboseEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX GlycolateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX GlycolateEX Objective\\n=========\\n1.0 BIOMASS = 0.058138057435281325\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 2.806E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.001556         0   0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01383         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 9.021E-05         0   0.00%\\n          Copper[e]             CopperEX 5.078E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02112         0   0.00%\\n     Glutathione[e]        GlutathioneEX    0.6088         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.079         1 100.00%\\n             Fe2[e]               IronEX 0.0001879         0   0.00%\\n          Photon[e]              LightEX     71.04         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.009238         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 4.009E-07         0   0.00%\\n               K[e]          PotassiumEX  0.005533         0   0.00%\\n        Selenate[e]           SelenateEX  0.004439         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.02593         0   0.00%\\nStrontium_cation[e]          StrontiumEX 6.681E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.01448         0   0.00%\\n             Zn2[e]               ZincEX 1.069E-05         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX    -0.2235         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.05814         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  9.96%\\n                                    CO[e]                                COEX -4.946E-07         1  0.00%\\n                               Glycine[e]                           GlycineEX    -0.5765         2 14.01%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003507         2  0.01%\\n                             Glycolate[e]                         GlycolateEX     -1.827         2 44.39%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.006468         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.5129         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX    -0.5791         0  0.00%\\n                                     H[e]                                 HEX      -7.01         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX    -0.5162         5 31.35%\\n                              Methanol[e]                          MethanolEX -0.0002707         1  0.00%\\n                                Oxygen[e]                                O2EX      -4.15         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002707        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.002648         6  0.19%\\n                                  Urea[e]                              UreaEX  -0.002648         1  0.03%\\n                                  tRNA[e]                              tRNAEX -3.032E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX AmmoniaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX AmmoniaEX Objective\\n=========\\n1.0 BIOMASS = 0.13723858476030898\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 6.624E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.003673         0   0.00%\\n    Chloride_ion[e]           ChlorideEX   0.03265         0   0.00%\\n      Cobalt_ion[e]             CobaltEX  0.000213         0   0.00%\\n          Copper[e]             CopperEX 1.199E-05         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.04986         0   0.00%\\n     Glutathione[e]        GlutathioneEX     283.4         0   0.00%\\n             H2O[e]                H2OEX     844.6         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.594         1 100.00%\\n             Fe2[e]               IronEX 0.0004435         0   0.00%\\n          Photon[e]              LightEX     81.99         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.02181         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 9.463E-07         0   0.00%\\n               K[e]          PotassiumEX   0.01306         0   0.00%\\n        Selenate[e]           SelenateEX   0.01048         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.06122         0   0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001577         0   0.00%\\n         Sulfate[e]            SulfateEX   0.03419         0   0.00%\\n             Zn2[e]               ZincEX 2.523E-05         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -282.5         0  0.00%\\n                               biomass[e]                           BiomassEX    -0.1372         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  0.04%\\n                                    CO[e]                                COEX -1.168E-06         1  0.00%\\n                               Glycine[e]                           GlycineEX     -283.3         2 28.57%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0008278         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01527         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -283.3         0  0.00%\\n                                     H[e]                                 HEX     -15.05         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX     -283.2         5 71.39%\\n                              Methanol[e]                          MethanolEX -0.0006391         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.116         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -283.1         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0006391        15  0.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00625         6  0.00%\\n                                  Urea[e]                              UreaEX   -0.00625         1  0.00%\\n                                  tRNA[e]                              tRNAEX -7.157E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX GlycogenEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX H2SEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX SpermineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX ThiosulfateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX UreaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_AlanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX L_AlanineEX Objective\\n=========\\n1.0 BIOMASS = 0.13723858476030637\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 6.624E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.003673         0   0.00%\\n    Chloride_ion[e]           ChlorideEX   0.03265         0   0.00%\\n      Cobalt_ion[e]             CobaltEX  0.000213         0   0.00%\\n          Copper[e]             CopperEX 1.199E-05         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.04986         0   0.00%\\n     Glutathione[e]        GlutathioneEX     404.4         0   0.00%\\n             H2O[e]                H2OEX     829.3         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.594         1 100.00%\\n             Fe2[e]               IronEX 0.0004435         0   0.00%\\n          Photon[e]              LightEX       150         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.02181         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 9.463E-07         0   0.00%\\n               K[e]          PotassiumEX   0.01306         0   0.00%\\n        Selenate[e]           SelenateEX   0.01048         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.06122         0   0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001577         0   0.00%\\n         Sulfate[e]            SulfateEX   0.03419         0   0.00%\\n             Zn2[e]               ZincEX 2.523E-05         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.1372         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  0.02%\\n                                    CO[e]                                COEX -1.168E-06         1  0.00%\\n                               Glycine[e]                           GlycineEX     -404.3         2 20.01%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0008278         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01527         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -152.1         0  0.00%\\n                                     H[e]                                 HEX     -31.99         0  0.00%\\n                             L_Alanine[e]                         L_AlanineEX     -151.2         3 11.23%\\n                             L_Cystine[e]                         L_CystineEX     -126.1         6 18.73%\\n                           L_Glutamate[e]                       L_GlutamateEX     -404.2         5 50.01%\\n                              Methanol[e]                          MethanolEX -0.0006391         1  0.00%\\n                                Oxygen[e]                                O2EX     -18.66         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -0.588         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0006391        15  0.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00625         6  0.00%\\n                                  Urea[e]                              UreaEX   -0.00625         1  0.00%\\n                                  tRNA[e]                              tRNAEX -7.157E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_ArginineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_AsparagineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_AspartateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX L_AspartateEX Objective\\n=========\\n1.0 BIOMASS = 0.07117022482529485\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 3.435E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.001905         0   0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01693         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001104         0   0.00%\\n          Copper[e]             CopperEX 6.216E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02586         0   0.00%\\n     Glutathione[e]        GlutathioneEX    0.9782         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.164         1 100.00%\\n             Fe2[e]               IronEX   0.00023         0   0.00%\\n          Photon[e]              LightEX     56.42         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01131         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 4.907E-07         0   0.00%\\n               K[e]          PotassiumEX  0.006773         0   0.00%\\n        Selenate[e]           SelenateEX  0.005434         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.03175         0   0.00%\\nStrontium_cation[e]          StrontiumEX 8.179E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.01773         0   0.00%\\n             Zn2[e]               ZincEX 1.309E-05         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX   -0.04656         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.07117         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  7.81%\\n                                    CO[e]                                COEX -6.055E-07         1  0.00%\\n                               Formate[e]                           FormateEX    -0.8521         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0004293         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.007918         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.483         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX    -0.7179         0  0.00%\\n                                     H[e]                                 HEX     -8.164         0  0.00%\\n                           L_Aspartate[e]                       L_AspartateEX     -1.218         4 46.40%\\n                             L_Cystine[e]                         L_CystineEX     -0.112         6  6.40%\\n                           L_Glutamate[e]                       L_GlutamateEX    -0.8215         5 39.12%\\n                              Methanol[e]                          MethanolEX -0.0003314         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.124         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0003314        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.003241         6  0.19%\\n                                  Urea[e]                              UreaEX  -0.003241         1  0.03%\\n                                  tRNA[e]                              tRNAEX -3.712E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_CystineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_GlutamateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_GlutamineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX GlycineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_HistidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX L_HistidineEX Objective\\n=========\\n1.0 BIOMASS = 0.03464771695201939\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 1.672E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0009273         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008243         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 5.376E-05         0   0.00%\\n          Copper[e]             CopperEX 3.026E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01259         0   0.00%\\n     Glutathione[e]        GlutathioneEX     1.584         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.926         1 100.00%\\n             Fe2[e]               IronEX  0.000112         0   0.00%\\n          Photon[e]              LightEX     57.31         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005505         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 2.389E-07         0   0.00%\\n               K[e]          PotassiumEX  0.003297         0   0.00%\\n        Selenate[e]           SelenateEX  0.002645         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01546         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.982E-05         0   0.00%\\n         Sulfate[e]            SulfateEX  0.008632         0   0.00%\\n             Zn2[e]               ZincEX 6.371E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.03465         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  5.73%\\n                                    CO[e]                                COEX -2.948E-07         1  0.00%\\n                               Glycine[e]                           GlycineEX    -0.8874         2 12.40%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000209         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003855         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.7054         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -1.566         0  0.00%\\n                                     H[e]                                 HEX     -2.624         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX     -1.528         5 53.39%\\n                           L_Histidine[e]                       L_HistidineEX    -0.6771         6 28.38%\\n                              Methanol[e]                          MethanolEX -0.0001614         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.074         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -1.567         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001614        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001578         6  0.07%\\n                                  Urea[e]                              UreaEX  -0.001578         1  0.01%\\n                                  tRNA[e]                              tRNAEX -1.807E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX L_IsoleucineEX\n",
      "GlutathioneEX L_LeucineEX\n",
      "GlutathioneEX L_LysineEX\n",
      "GlutathioneEX L_MethionineEX\n",
      "GlutathioneEX L_PhenylalanineEX\n",
      "GlutathioneEX L_ProlineEX\n",
      "GlutathioneEX L_SerineEX\n",
      "GlutathioneEX L_ThreonineEX\n",
      "GlutathioneEX L_TryptophanEX\n",
      "GlutathioneEX L_TyrosineEX\n",
      "GlutathioneEX L_ValineEX\n",
      "GlutathioneEX PhospholipidEX\n",
      "GlutathioneEX ThymidineEX\n",
      "GlutathioneEX XanthosineEX\n",
      "GlutathioneEX 4_AminobenzoateEX\n",
      "GlutathioneEX 5_MethylthioadenosineEX\n",
      "GlutathioneEX AdenineEX\n",
      "GlutathioneEX DethiobiotinEX\n",
      "GlutathioneEX GuanineEX\n",
      "GlutathioneEX GuanosineEX\n",
      "GlutathioneEX PutrescineEX\n",
      "GlutathioneEX SpermidineEX\n",
      "GlutathioneEX EthanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGlutathioneEX EthanolEX Objective\\n=========\\n1.0 BIOMASS = 0.04231795197029573\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 2.042E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.001133         0   0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01007         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 6.567E-05         0   0.00%\\n          Copper[e]             CopperEX 3.696E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01537         0   0.00%\\n     Glutathione[e]        GlutathioneEX     1.377         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.341         1 100.00%\\n             Fe2[e]               IronEX 0.0001368         0   0.00%\\n          Photon[e]              LightEX      68.5         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.006724         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 2.918E-07         0   0.00%\\n               K[e]          PotassiumEX  0.004027         0   0.00%\\n        Selenate[e]           SelenateEX  0.003231         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01888         0   0.00%\\nStrontium_cation[e]          StrontiumEX 4.863E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.01054         0   0.00%\\n             Zn2[e]               ZincEX 7.781E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.04232         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  5.81%\\n                                    CO[e]                                COEX   -3.6E-07         1  0.00%\\n                               Ethanol[e]                           EthanolEX    -0.3654         0  0.00%\\n                               Glycine[e]                           GlycineEX    -0.6223         2  8.82%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002553         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.004708         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.227         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -1.355         0  0.00%\\n                                     H[e]                                 HEX     -11.83         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX     -1.309         5 46.40%\\n                              Methanol[e]                          MethanolEX -0.0001971         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.219         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -1.076         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001971        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001927         6  0.08%\\n                                  Urea[e]                              UreaEX  -0.001927         1  0.01%\\n                                  tRNA[e]                              tRNAEX    -0.3654        15 38.85%\\n\\n'\n",
      "stderr: None\n",
      "GlutathioneEX S_MalateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GlutathioneEX H2O2EX\n",
      "L_GlutamineEX B12EX\n",
      "L_GlutamineEX BiotinEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamineEX BiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.03158453150089693\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.524E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0008454         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.007514         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.901E-05         0  0.00%\\n          Copper[e]             CopperEX 2.759E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01147         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.585         1 47.72%\\n             Fe2[e]               IronEX 0.0001021         0  0.00%\\n     L_Glutamine[e]        L_GlutamineEX     1.224         5 52.28%\\n          Photon[e]              LightEX     64.43         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005019         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.178E-07         0  0.00%\\n               K[e]          PotassiumEX  0.003006         0  0.00%\\n        Selenate[e]           SelenateEX  0.002412         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01409         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.63E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.3638         0  0.00%\\n             Zn2[e]               ZincEX 5.807E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.03158         0  0.00%\\n                                Biotin[e]                            BiotinEX    -0.1699        10 16.80%\\n                                   CO2[e]                               CO2EX    -0.7596         1  7.51%\\n                                    CO[e]                                COEX -2.687E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001905         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003514         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.216         0  0.00%\\n                                     H[e]                                 HEX     -5.554         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX    -0.9846         5 48.69%\\n                              Methanol[e]                          MethanolEX      -0.17         1  1.68%\\n                                Oxygen[e]                                O2EX     -5.812         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX      -0.17        15 25.22%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001438         6  0.09%\\n                                  tRNA[e]                              tRNAEX -1.647E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamineEX MethanolEX\n",
      "L_GlutamineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_GlutamineEX SMethyl5thioDriboseEX\n",
      "L_GlutamineEX GlycolateEX\n",
      "L_GlutamineEX AmmoniaEX\n",
      "L_GlutamineEX GlycogenEX\n",
      "L_GlutamineEX H2SEX\n",
      "L_GlutamineEX SpermineEX\n",
      "L_GlutamineEX ThiosulfateEX\n",
      "L_GlutamineEX UreaEX\n",
      "L_GlutamineEX L_AlanineEX\n",
      "L_GlutamineEX L_ArginineEX\n",
      "L_GlutamineEX L_AsparagineEX\n",
      "L_GlutamineEX L_AspartateEX\n",
      "L_GlutamineEX L_CystineEX\n",
      "L_GlutamineEX L_GlutamateEX\n",
      "L_GlutamineEX GlycineEX\n",
      "L_GlutamineEX L_HistidineEX\n",
      "L_GlutamineEX L_IsoleucineEX\n",
      "L_GlutamineEX L_LeucineEX\n",
      "L_GlutamineEX L_LysineEX\n",
      "L_GlutamineEX L_MethionineEX\n",
      "L_GlutamineEX L_PhenylalanineEX\n",
      "L_GlutamineEX L_ProlineEX\n",
      "L_GlutamineEX L_SerineEX\n",
      "L_GlutamineEX L_ThreonineEX\n",
      "L_GlutamineEX L_TryptophanEX\n",
      "L_GlutamineEX L_TyrosineEX\n",
      "L_GlutamineEX L_ValineEX\n",
      "L_GlutamineEX PhospholipidEX\n",
      "L_GlutamineEX LPSEX\n",
      "L_GlutamineEX AcetateEX\n",
      "L_GlutamineEX CitrateEX\n",
      "L_GlutamineEX FormateEX\n",
      "L_GlutamineEX PyruvateEX\n",
      "L_GlutamineEX FumarateEX\n",
      "L_GlutamineEX GlutathioneEX\n",
      "L_GlutamineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_GlutamineEX PantothenateEX\n",
      "L_GlutamineEX SuccinateEX\n",
      "L_GlutamineEX ThymidineEX\n",
      "L_GlutamineEX XanthosineEX\n",
      "L_GlutamineEX 4_AminobenzoateEX\n",
      "L_GlutamineEX 5_MethylthioadenosineEX\n",
      "L_GlutamineEX AdenineEX\n",
      "L_GlutamineEX DethiobiotinEX\n",
      "L_GlutamineEX GuanineEX\n",
      "L_GlutamineEX GuanosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamineEX GuanosineEX Objective\\n=========\\n1.0 BIOMASS = 0.03158453150089729\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.524E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0008454         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.007514         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.901E-05         0  0.00%\\n          Copper[e]             CopperEX 2.759E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01147         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.272         1 34.10%\\n             Fe2[e]               IronEX 0.0001021         0  0.00%\\n     L_Glutamine[e]        L_GlutamineEX     2.038         5 65.90%\\n          Photon[e]              LightEX     55.27         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005019         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.178E-07         0  0.00%\\n               K[e]          PotassiumEX  0.003006         0  0.00%\\n        Selenate[e]           SelenateEX  0.002412         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01409         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.63E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02403         0  0.00%\\n             Zn2[e]               ZincEX 5.807E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.03158         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.3483         1  3.41%\\n                                    CO[e]                                COEX -2.687E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001905         2  0.00%\\n                             Guanosine[e]                         GuanosineEX    -0.3666         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003514         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.475         0  0.00%\\n                                     H[e]                                 HEX     -7.797         0  0.00%\\n                           L_Glutamate[e]                       L_GlutamateEX     -1.968         5 96.47%\\n                              Methanol[e]                          MethanolEX -0.0001471         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.131         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001471        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001438         6  0.08%\\n                                  tRNA[e]                              tRNAEX -1.647E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamineEX PutrescineEX\n",
      "L_GlutamineEX SpermidineEX\n",
      "L_GlutamineEX EthanolEX\n",
      "L_GlutamineEX S_MalateEX\n",
      "L_GlutamineEX H2O2EX\n",
      "L_SerineEX B12EX\n",
      "L_SerineEX BiotinEX\n",
      "L_SerineEX MethanolEX\n",
      "L_SerineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_SerineEX SMethyl5thioDriboseEX\n",
      "L_SerineEX GlycolateEX\n",
      "L_SerineEX AmmoniaEX\n",
      "L_SerineEX GlycogenEX\n",
      "L_SerineEX H2SEX\n",
      "L_SerineEX SpermineEX\n",
      "L_SerineEX ThiosulfateEX\n",
      "L_SerineEX UreaEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_SerineEX L_AlanineEX\n",
      "L_SerineEX L_ArginineEX\n",
      "L_SerineEX L_AsparagineEX\n",
      "L_SerineEX L_AspartateEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_SerineEX L_CystineEX\n",
      "L_SerineEX L_GlutamateEX\n",
      "L_SerineEX L_GlutamineEX\n",
      "L_SerineEX GlycineEX\n",
      "L_SerineEX L_HistidineEX\n",
      "L_SerineEX L_IsoleucineEX\n",
      "L_SerineEX L_LeucineEX\n",
      "L_SerineEX L_LysineEX\n",
      "L_SerineEX L_MethionineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_SerineEX L_MethionineEX Objective\\n=========\\n1.0 BIOMASS = 0.036801883565595946\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.776E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.000985         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008756         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 5.711E-05         0  0.00%\\n          Copper[e]             CopperEX 3.214E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01337         0  0.00%\\n            HCO3[e]            HCO3EXcar      4.94         1  5.57%\\n               H[e]                  HEX     2.438         0  0.00%\\n             Fe2[e]               IronEX 0.0001189         0  0.00%\\n        L_Serine[e]           L_SerineEX     27.91         3 94.43%\\n          Photon[e]              LightEX  0.006688         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005848         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.538E-07         0  0.00%\\n          Oxygen[e]                 O2EX      1.09         0  0.00%\\n               K[e]          PotassiumEX  0.003502         0  0.00%\\n        Selenate[e]           SelenateEX   0.00281         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01642         0  0.00%\\nStrontium_cation[e]          StrontiumEX 4.229E-05         0  0.00%\\n         Sulfate[e]            SulfateEX     2.102         0  0.00%\\n             Zn2[e]               ZincEX 6.767E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.0368         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  1.32%\\n                                    CO[e]                                COEX -3.131E-07         1  0.00%\\n                               Formate[e]                           FormateEX     -23.43         0  0.00%\\n                               Glycine[e]                           GlycineEX     -25.52         2 82.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000222         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.004094         0  0.00%\\n                                   H2O[e]                               H2OEX      -4.44         0  0.00%\\n                          L_Methionine[e]                      L_MethionineEX     -2.074         5 16.66%\\n                              Methanol[e]                          MethanolEX -0.0001714         1  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3846         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001714        15  0.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001676         6  0.02%\\n                                  Urea[e]                              UreaEX  -0.001676         1  0.00%\\n                                  tRNA[e]                              tRNAEX -1.919E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_SerineEX L_PhenylalanineEX\n",
      "L_SerineEX L_ProlineEX\n",
      "L_SerineEX L_ThreonineEX\n",
      "L_SerineEX L_TryptophanEX\n",
      "L_SerineEX L_TyrosineEX\n",
      "L_SerineEX L_ValineEX\n",
      "L_SerineEX PhospholipidEX\n",
      "L_SerineEX LPSEX\n",
      "L_SerineEX AcetateEX\n",
      "L_SerineEX CitrateEX\n",
      "L_SerineEX FormateEX\n",
      "L_SerineEX PyruvateEX\n",
      "L_SerineEX FumarateEX\n",
      "L_SerineEX GlutathioneEX\n",
      "L_SerineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_SerineEX PantothenateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_SerineEX PantothenateEX Objective\\n=========\\n1.0 BIOMASS = 0.03099443787914727\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.496E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0008296         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.007374         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.809E-05         0  0.00%\\n          Copper[e]             CopperEX 2.707E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01126         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.902         1  8.03%\\n               H[e]                  HEX     4.064         0  0.00%\\n             Fe2[e]               IronEX 0.0001002         0  0.00%\\n        L_Serine[e]           L_SerineEX     18.72         3 91.97%\\n          Photon[e]              LightEX  0.005632         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004925         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.137E-07         0  0.00%\\n          Oxygen[e]                 O2EX     1.941         0  0.00%\\n               K[e]          PotassiumEX   0.00295         0  0.00%\\n        Selenate[e]           SelenateEX  0.002367         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01383         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.562E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02358         0  0.00%\\n             Zn2[e]               ZincEX 5.699E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX    -0.4526         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.03099         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  2.35%\\n                                    CO[e]                                COEX -2.637E-07         1  0.00%\\n                               Formate[e]                           FormateEX     -16.11         0  0.00%\\n                               Glycine[e]                           GlycineEX     -17.06         2 97.62%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000187         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003448         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.284         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001443         1  0.00%\\n                          Pantothenate[e]                      PantothenateEX    -0.9377         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001443        15  0.01%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001412         6  0.02%\\n                                  Urea[e]                              UreaEX  -0.001412         1  0.00%\\n                                  tRNA[e]                              tRNAEX -1.616E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_SerineEX SuccinateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_SerineEX ThymidineEX\n",
      "L_SerineEX XanthosineEX\n",
      "L_SerineEX 4_AminobenzoateEX\n",
      "L_SerineEX 5_MethylthioadenosineEX\n",
      "L_SerineEX AdenineEX\n",
      "L_SerineEX DethiobiotinEX\n",
      "L_SerineEX GuanineEX\n",
      "L_SerineEX GuanosineEX\n",
      "L_SerineEX PutrescineEX\n",
      "L_SerineEX SpermidineEX\n",
      "L_SerineEX EthanolEX\n",
      "L_SerineEX S_MalateEX\n",
      "L_SerineEX H2O2EX\n",
      "L_CystineEX B12EX\n",
      "L_CystineEX BiotinEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_CystineEX MethanolEX\n",
      "L_CystineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_CystineEX SMethyl5thioDriboseEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_CystineEX SMethyl5thioDriboseEX Objective\\n=========\\n1.0 BIOMASS = 0.034251640333123966\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.653E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0009167         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008149         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 5.315E-05         0  0.00%\\n          Copper[e]             CopperEX 2.991E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01244         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.118         1 48.09%\\n             Fe2[e]               IronEX 0.0001107         0  0.00%\\n       L_Cystine[e]          L_CystineEX    0.9207         6 51.91%\\n          Photon[e]              LightEX     64.55         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005442         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.362E-07         0  0.00%\\n               K[e]          PotassiumEX   0.00326         0  0.00%\\n        Selenate[e]           SelenateEX  0.002615         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01528         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.936E-05         0  0.00%\\n         Sulfate[e]            SulfateEX  0.008533         0  0.00%\\n             Zn2[e]               ZincEX 6.298E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.03425         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 13.53%\\n                                    CO[e]                                COEX -2.914E-07         1  0.00%\\n                               Glycine[e]                           GlycineEX    -0.3709         2 12.24%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002066         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003811         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.6751         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -1.434         0  0.00%\\n                                     H[e]                                 HEX     -11.25         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001595         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.959         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.9509         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001595        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.3915         6 38.76%\\n                              Spermine[e]                          SpermineEX     -0.195        10 32.17%\\n                                  Urea[e]                              UreaEX    -0.1965         1  3.24%\\n                                  tRNA[e]                              tRNAEX -1.786E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_CystineEX GlycolateEX\n",
      "L_CystineEX AmmoniaEX\n",
      "L_CystineEX GlycogenEX\n",
      "L_CystineEX H2SEX\n",
      "L_CystineEX SpermineEX\n",
      "L_CystineEX ThiosulfateEX\n",
      "L_CystineEX UreaEX\n",
      "L_CystineEX L_AlanineEX\n",
      "L_CystineEX L_ArginineEX\n",
      "L_CystineEX L_AsparagineEX\n",
      "L_CystineEX L_AspartateEX\n",
      "L_CystineEX L_GlutamateEX\n",
      "L_CystineEX L_GlutamineEX\n",
      "L_CystineEX GlycineEX\n",
      "L_CystineEX L_HistidineEX\n",
      "L_CystineEX L_IsoleucineEX\n",
      "L_CystineEX L_LeucineEX\n",
      "L_CystineEX L_LysineEX\n",
      "L_CystineEX L_MethionineEX\n",
      "L_CystineEX L_PhenylalanineEX\n",
      "L_CystineEX L_ProlineEX\n",
      "L_CystineEX L_SerineEX\n",
      "L_CystineEX L_ThreonineEX\n",
      "L_CystineEX L_TryptophanEX\n",
      "L_CystineEX L_TyrosineEX\n",
      "L_CystineEX L_ValineEX\n",
      "L_CystineEX PhospholipidEX\n",
      "L_CystineEX LPSEX\n",
      "L_CystineEX AcetateEX\n",
      "L_CystineEX CitrateEX\n",
      "L_CystineEX FormateEX\n",
      "L_CystineEX PyruvateEX\n",
      "L_CystineEX FumarateEX\n",
      "L_CystineEX GlutathioneEX\n",
      "L_CystineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_CystineEX PantothenateEX\n",
      "L_CystineEX SuccinateEX\n",
      "L_CystineEX ThymidineEX\n",
      "L_CystineEX XanthosineEX\n",
      "L_CystineEX 4_AminobenzoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_CystineEX 4_AminobenzoateEX Objective\\n=========\\n1.0 BIOMASS = 0.04966951326277404\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.397E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001329         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01182         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 7.707E-05         0  0.00%\\n          Copper[e]             CopperEX 4.338E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01804         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.024         1 67.78%\\n             Fe2[e]               IronEX 0.0001605         0  0.00%\\n       L_Cystine[e]          L_CystineEX     0.398         6 32.22%\\n          Photon[e]              LightEX     58.97         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.007892         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.425E-07         0  0.00%\\n               K[e]          PotassiumEX  0.004727         0  0.00%\\n        Selenate[e]           SelenateEX  0.003792         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02216         0  0.00%\\nStrontium_cation[e]          StrontiumEX 5.708E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.01237         0  0.00%\\n             Zn2[e]               ZincEX 9.133E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                       4_Aminobenzoate[e]                   4_AminobenzoateEX    -0.3601         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.04967         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 97.56%\\n                                    CO[e]                                COEX -4.226E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002996         2  0.07%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.005526         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.149         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX    -0.7706         0  0.00%\\n                                     H[e]                                 HEX     -9.771         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002313         1  0.03%\\n                                Oxygen[e]                                O2EX     -4.918         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.5144         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002313        15  0.41%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.002262         6  1.61%\\n                                  Urea[e]                              UreaEX  -0.002262         1  0.27%\\n                                  tRNA[e]                              tRNAEX  -2.59E-05        15  0.05%\\n\\n'\n",
      "stderr: None\n",
      "L_CystineEX 5_MethylthioadenosineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_CystineEX AdenineEX\n",
      "L_CystineEX DethiobiotinEX\n",
      "L_CystineEX GuanineEX\n",
      "L_CystineEX GuanosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_CystineEX GuanosineEX Objective\\n=========\\n1.0 BIOMASS = 0.04335433183240598\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.093E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX   0.00116         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01031         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 6.727E-05         0  0.00%\\n          Copper[e]             CopperEX 3.786E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01575         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.362         1 43.99%\\n             Fe2[e]               IronEX 0.0001401         0  0.00%\\n       L_Cystine[e]          L_CystineEX     1.138         6 56.01%\\n          Photon[e]              LightEX     60.65         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.006889         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.989E-07         0  0.00%\\n               K[e]          PotassiumEX  0.004126         0  0.00%\\n        Selenate[e]           SelenateEX   0.00331         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01934         0  0.00%\\nStrontium_cation[e]          StrontiumEX 4.982E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.0108         0  0.00%\\n             Zn2[e]               ZincEX 7.971E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.04335         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 97.86%\\n                                    CO[e]                                COEX -3.689E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002615         2  0.06%\\n                             Guanosine[e]                         GuanosineEX     -0.379         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.004823         0  0.00%\\n                                   H2O[e]                               H2OEX      -2.84         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -2.253         0  0.00%\\n                                     H[e]                                 HEX      -9.09         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002019         1  0.02%\\n                                Oxygen[e]                                O2EX     -4.542         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX      -1.79         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002019        15  0.36%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001974         6  1.41%\\n                                  Urea[e]                              UreaEX  -0.001974         1  0.24%\\n                                  tRNA[e]                              tRNAEX -2.261E-05        15  0.04%\\n\\n'\n",
      "stderr: None\n",
      "L_CystineEX PutrescineEX\n",
      "L_CystineEX SpermidineEX\n",
      "L_CystineEX EthanolEX\n",
      "L_CystineEX S_MalateEX\n",
      "L_CystineEX H2O2EX\n",
      "L_AlanineEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AlanineEX B12EX Objective\\n=========\\n1.0 BIOMASS = 0.04401734677269456\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.125E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001178         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01047         0  0.00%\\n      Cobalt_ion[e]             CobaltEX    0.1144         0  0.00%\\n          Copper[e]             CopperEX 3.844E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX    0.1303         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.216         1 14.68%\\n             Fe2[e]               IronEX 0.0001423         0  0.00%\\n       L_Alanine[e]          L_AlanineEX      10.1         3 85.32%\\n          Photon[e]              LightEX     41.27         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.006994         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.035E-07         0  0.00%\\n               K[e]          PotassiumEX  0.004189         0  0.00%\\n        Selenate[e]           SelenateEX  0.003361         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01964         0  0.00%\\nStrontium_cation[e]          StrontiumEX 5.058E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.03349         0  0.00%\\n             Zn2[e]               ZincEX 8.093E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -7.887         0  0.00%\\n                     Cobamide_coenzyme[e]                               B12EX    -0.1143        72 90.76%\\n                               biomass[e]                           BiomassEX   -0.04402         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1  9.04%\\n                                    CO[e]                                COEX -3.745E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002655         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.004897         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.582         0  0.00%\\n                                     H[e]                                 HEX     -5.113         0  0.00%\\n                              Methanol[e]                          MethanolEX  -0.000205         1  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -9.601         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX  -0.000205        15  0.03%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.002005         6  0.13%\\n                                  Urea[e]                              UreaEX  -0.002005         1  0.02%\\n                                  tRNA[e]                              tRNAEX -2.296E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_AlanineEX BiotinEX\n",
      "L_AlanineEX MethanolEX\n",
      "L_AlanineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_AlanineEX SMethyl5thioDriboseEX\n",
      "L_AlanineEX GlycolateEX\n",
      "L_AlanineEX AmmoniaEX\n",
      "L_AlanineEX GlycogenEX\n",
      "L_AlanineEX H2SEX\n",
      "L_AlanineEX SpermineEX\n",
      "L_AlanineEX ThiosulfateEX\n",
      "L_AlanineEX UreaEX\n",
      "L_AlanineEX L_ArginineEX\n",
      "L_AlanineEX L_AsparagineEX\n",
      "L_AlanineEX L_AspartateEX\n",
      "L_AlanineEX L_CystineEX\n",
      "L_AlanineEX L_GlutamateEX\n",
      "L_AlanineEX L_GlutamineEX\n",
      "L_AlanineEX GlycineEX\n",
      "L_AlanineEX L_HistidineEX\n",
      "L_AlanineEX L_IsoleucineEX\n",
      "L_AlanineEX L_LeucineEX\n",
      "L_AlanineEX L_LysineEX\n",
      "L_AlanineEX L_MethionineEX\n",
      "L_AlanineEX L_PhenylalanineEX\n",
      "L_AlanineEX L_ProlineEX\n",
      "L_AlanineEX L_SerineEX\n",
      "L_AlanineEX L_ThreonineEX\n",
      "L_AlanineEX L_TryptophanEX\n",
      "L_AlanineEX L_TyrosineEX\n",
      "L_AlanineEX L_ValineEX\n",
      "L_AlanineEX PhospholipidEX\n",
      "L_AlanineEX LPSEX\n",
      "L_AlanineEX AcetateEX\n",
      "L_AlanineEX CitrateEX\n",
      "L_AlanineEX FormateEX\n",
      "L_AlanineEX PyruvateEX\n",
      "L_AlanineEX FumarateEX\n",
      "L_AlanineEX GlutathioneEX\n",
      "L_AlanineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_AlanineEX PantothenateEX\n",
      "L_AlanineEX SuccinateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AlanineEX SuccinateEX Objective\\n=========\\n1.0 BIOMASS = 0.04966951326270743\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.397E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001329         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01182         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 7.707E-05         0  0.00%\\n          Copper[e]             CopperEX 4.338E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01804         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.024         1 13.64%\\n             Fe2[e]               IronEX 0.0001605         0  0.00%\\n       L_Alanine[e]          L_AlanineEX      10.6         3 86.36%\\n          Photon[e]              LightEX     32.11         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.007892         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.425E-07         0  0.00%\\n               K[e]          PotassiumEX  0.004727         0  0.00%\\n        Selenate[e]           SelenateEX  0.003792         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02216         0  0.00%\\nStrontium_cation[e]          StrontiumEX 5.708E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.03779         0  0.00%\\n             Zn2[e]               ZincEX 9.133E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.3757         0  0.00%\\n                               Ammonia[e]                           AmmoniaEX     -10.17         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.04967         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 97.56%\\n                                    CO[e]                                COEX -4.226E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002996         2  0.07%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.005526         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.607         0  0.00%\\n                                     H[e]                                 HEX     -3.005         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002313         1  0.03%\\n                              Pyruvate[e]                          PyruvateEX     -8.808         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002313        15  0.41%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.002262         6  1.61%\\n                             Succinate[e]                         SuccinateEX       -1.2         0  0.00%\\n                                  Urea[e]                              UreaEX  -0.002262         1  0.27%\\n                                  tRNA[e]                              tRNAEX  -2.59E-05        15  0.05%\\n\\n'\n",
      "stderr: None\n",
      "L_AlanineEX ThymidineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_AlanineEX XanthosineEX\n",
      "L_AlanineEX 4_AminobenzoateEX\n",
      "L_AlanineEX 5_MethylthioadenosineEX\n",
      "L_AlanineEX AdenineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AlanineEX AdenineEX Objective\\n=========\\n1.0 BIOMASS = 0.03561915752904322\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.719E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0009533         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008474         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 5.527E-05         0  0.00%\\n          Copper[e]             CopperEX 3.111E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01294         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.732         1 22.10%\\n             Fe2[e]               IronEX 0.0001151         0  0.00%\\n       L_Alanine[e]          L_AlanineEX     6.734         3 77.90%\\n          Photon[e]              LightEX     38.02         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00566         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.456E-07         0  0.00%\\n               K[e]          PotassiumEX   0.00339         0  0.00%\\n        Selenate[e]           SelenateEX   0.00272         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01589         0  0.00%\\nStrontium_cation[e]          StrontiumEX 4.093E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.0271         0  0.00%\\n             Zn2[e]               ZincEX 6.549E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Adenine[e]                           AdenineEX    -0.8001         0  0.00%\\n                               Ammonia[e]                           AmmoniaEX      -2.42         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.03562         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.24%\\n                                    CO[e]                                COEX  -3.03E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002149         2  0.05%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003963         0  0.00%\\n                                   H2O[e]                               H2OEX     -9.958         0  0.00%\\n                                     H[e]                                 HEX     -2.172         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001659         1  0.02%\\n                              Pyruvate[e]                          PyruvateEX     -6.433         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001659        15  0.30%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001622         6  1.17%\\n                                  Urea[e]                              UreaEX  -0.001622         1  0.19%\\n                                  tRNA[e]                              tRNAEX -1.858E-05        15  0.03%\\n\\n'\n",
      "stderr: None\n",
      "L_AlanineEX DethiobiotinEX\n",
      "L_AlanineEX GuanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_AlanineEX GuanineEX Objective\\n=========\\n1.0 BIOMASS = 0.03561915752904217\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.719E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0009533         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008474         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 5.527E-05         0  0.00%\\n          Copper[e]             CopperEX 3.111E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01294         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.732         1 24.36%\\n             Fe2[e]               IronEX 0.0001151         0  0.00%\\n       L_Alanine[e]          L_AlanineEX     5.934         3 75.64%\\n          Photon[e]              LightEX     39.08         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00566         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.456E-07         0  0.00%\\n               K[e]          PotassiumEX   0.00339         0  0.00%\\n        Selenate[e]           SelenateEX   0.00272         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01589         0  0.00%\\nStrontium_cation[e]          StrontiumEX 4.093E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.0271         0  0.00%\\n             Zn2[e]               ZincEX 6.549E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX      -1.62         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.03562         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.24%\\n                                    CO[e]                                COEX  -3.03E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002149         2  0.05%\\n                               Guanine[e]                           GuanineEX    -0.8001         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003963         0  0.00%\\n                                   H2O[e]                               H2OEX     -9.958         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001659         1  0.02%\\n                              Pyruvate[e]                          PyruvateEX     -5.633         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001659        15  0.30%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001622         6  1.17%\\n                                  Urea[e]                              UreaEX  -0.001622         1  0.19%\\n                                  tRNA[e]                              tRNAEX -1.858E-05        15  0.03%\\n\\n'\n",
      "stderr: None\n",
      "L_AlanineEX GuanosineEX\n",
      "L_AlanineEX PutrescineEX\n",
      "L_AlanineEX SpermidineEX\n",
      "L_AlanineEX EthanolEX\n",
      "L_AlanineEX S_MalateEX\n",
      "L_AlanineEX H2O2EX\n",
      "L_LeucineEX B12EX\n",
      "L_LeucineEX BiotinEX\n",
      "L_LeucineEX MethanolEX\n",
      "L_LeucineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_LeucineEX SMethyl5thioDriboseEX\n",
      "L_LeucineEX GlycolateEX\n",
      "L_LeucineEX AmmoniaEX\n",
      "L_LeucineEX GlycogenEX\n",
      "L_LeucineEX H2SEX\n",
      "L_LeucineEX SpermineEX\n",
      "L_LeucineEX ThiosulfateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_LeucineEX ThiosulfateEX Objective\\n=========\\n1.0 BIOMASS = 0.10399644076728215\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 5.019E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.002783         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.02474         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001614         0  0.00%\\n          Copper[e]             CopperEX 9.083E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.03778         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.378         1  0.11%\\n               H[e]                  HEX     106.6         0  0.00%\\n             Fe2[e]               IronEX 0.0003361         0  0.00%\\n       L_Leucine[e]          L_LeucineEX     813.7         6 99.89%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01652         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 7.171E-07         0  0.00%\\n          Oxygen[e]                 O2EX     123.3         0  0.00%\\n               K[e]          PotassiumEX  0.009897         0  0.00%\\n        Selenate[e]           SelenateEX  0.007941         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.04639         0  0.00%\\nStrontium_cation[e]          StrontiumEX 0.0001195         0  0.00%\\n         Sulfate[e]            SulfateEX     277.7         0  0.00%\\n             Zn2[e]               ZincEX 1.912E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX     -813.6         0  0.00%\\n                               Ammonia[e]                           AmmoniaEX     -812.7         0  0.00%\\n                               biomass[e]                           BiomassEX     -0.104         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.3489         1 89.03%\\n                                    CO[e]                                COEX -8.848E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0006273         2  0.32%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.01157         0  0.00%\\n                                   H2O[e]                               H2OEX     -143.3         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0004843         1  0.12%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0004843        15  1.85%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.004736         6  7.25%\\n                           Thiosulfate[e]                       ThiosulfateEX     -138.8         0  0.00%\\n                                  Urea[e]                              UreaEX  -0.004736         1  1.21%\\n                                  tRNA[e]                              tRNAEX -5.424E-05        15  0.21%\\n\\n'\n",
      "stderr: None\n",
      "L_LeucineEX UreaEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_LeucineEX L_AlanineEX\n",
      "L_LeucineEX L_ArginineEX\n",
      "L_LeucineEX L_AsparagineEX\n",
      "L_LeucineEX L_AspartateEX\n",
      "L_LeucineEX L_CystineEX\n",
      "L_LeucineEX L_GlutamateEX\n",
      "L_LeucineEX L_GlutamineEX\n",
      "L_LeucineEX GlycineEX\n",
      "L_LeucineEX L_HistidineEX\n",
      "L_LeucineEX L_IsoleucineEX\n",
      "L_LeucineEX L_LysineEX\n",
      "L_LeucineEX L_MethionineEX\n",
      "L_LeucineEX L_PhenylalanineEX\n",
      "L_LeucineEX L_ProlineEX\n",
      "L_LeucineEX L_SerineEX\n",
      "L_LeucineEX L_ThreonineEX\n",
      "L_LeucineEX L_TryptophanEX\n",
      "L_LeucineEX L_TyrosineEX\n",
      "L_LeucineEX L_ValineEX\n",
      "L_LeucineEX PhospholipidEX\n",
      "L_LeucineEX LPSEX\n",
      "L_LeucineEX AcetateEX\n",
      "L_LeucineEX CitrateEX\n",
      "L_LeucineEX FormateEX\n",
      "L_LeucineEX PyruvateEX\n",
      "L_LeucineEX FumarateEX\n",
      "L_LeucineEX GlutathioneEX\n",
      "L_LeucineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_LeucineEX PantothenateEX\n",
      "L_LeucineEX SuccinateEX\n",
      "L_LeucineEX ThymidineEX\n",
      "L_LeucineEX XanthosineEX\n",
      "L_LeucineEX 4_AminobenzoateEX\n",
      "L_LeucineEX 5_MethylthioadenosineEX\n",
      "L_LeucineEX AdenineEX\n",
      "L_LeucineEX DethiobiotinEX\n",
      "L_LeucineEX GuanineEX\n",
      "L_LeucineEX GuanosineEX\n",
      "L_LeucineEX PutrescineEX\n",
      "L_LeucineEX SpermidineEX\n",
      "L_LeucineEX EthanolEX\n",
      "L_LeucineEX S_MalateEX\n",
      "L_LeucineEX H2O2EX\n",
      "GlycineEX B12EX\n",
      "GlycineEX BiotinEX\n",
      "GlycineEX MethanolEX\n",
      "GlycineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "GlycineEX SMethyl5thioDriboseEX\n",
      "GlycineEX GlycolateEX\n",
      "GlycineEX AmmoniaEX\n",
      "GlycineEX GlycogenEX\n",
      "GlycineEX H2SEX\n",
      "GlycineEX SpermineEX\n",
      "GlycineEX ThiosulfateEX\n",
      "GlycineEX UreaEX\n",
      "GlycineEX L_AlanineEX\n",
      "GlycineEX L_ArginineEX\n",
      "GlycineEX L_AsparagineEX\n",
      "GlycineEX L_AspartateEX\n",
      "GlycineEX L_CystineEX\n",
      "GlycineEX L_GlutamateEX\n",
      "GlycineEX L_GlutamineEX\n",
      "GlycineEX L_HistidineEX\n",
      "GlycineEX L_IsoleucineEX\n",
      "GlycineEX L_LeucineEX\n",
      "GlycineEX L_LysineEX\n",
      "GlycineEX L_MethionineEX\n",
      "GlycineEX L_PhenylalanineEX\n",
      "GlycineEX L_ProlineEX\n",
      "GlycineEX L_SerineEX\n",
      "GlycineEX L_ThreonineEX\n",
      "GlycineEX L_TryptophanEX\n",
      "GlycineEX L_TyrosineEX\n",
      "GlycineEX L_ValineEX\n",
      "GlycineEX PhospholipidEX\n",
      "GlycineEX LPSEX\n",
      "GlycineEX AcetateEX\n",
      "GlycineEX CitrateEX\n",
      "GlycineEX FormateEX\n",
      "GlycineEX PyruvateEX\n",
      "GlycineEX FumarateEX\n",
      "GlycineEX GlutathioneEX\n",
      "GlycineEX 4_Methyl_2_oxopentanoateEX\n",
      "GlycineEX PantothenateEX\n",
      "GlycineEX SuccinateEX\n",
      "GlycineEX ThymidineEX\n",
      "GlycineEX XanthosineEX\n",
      "GlycineEX 4_AminobenzoateEX\n",
      "GlycineEX 5_MethylthioadenosineEX\n",
      "GlycineEX AdenineEX\n",
      "GlycineEX DethiobiotinEX\n",
      "GlycineEX GuanineEX\n",
      "GlycineEX GuanosineEX\n",
      "GlycineEX PutrescineEX\n",
      "GlycineEX SpermidineEX\n",
      "GlycineEX EthanolEX\n",
      "GlycineEX S_MalateEX\n",
      "GlycineEX H2O2EX\n",
      "UreaEX B12EX\n",
      "UreaEX BiotinEX\n",
      "UreaEX MethanolEX\n",
      "UreaEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "UreaEX SMethyl5thioDriboseEX\n",
      "UreaEX GlycolateEX\n",
      "UreaEX AmmoniaEX\n",
      "UreaEX GlycogenEX\n",
      "UreaEX H2SEX\n",
      "UreaEX SpermineEX\n",
      "UreaEX ThiosulfateEX\n",
      "UreaEX L_AlanineEX\n",
      "UreaEX L_ArginineEX\n",
      "UreaEX L_AsparagineEX\n",
      "UreaEX L_AspartateEX\n",
      "UreaEX L_CystineEX\n",
      "UreaEX L_GlutamateEX\n",
      "UreaEX L_GlutamineEX\n",
      "UreaEX GlycineEX\n",
      "UreaEX L_HistidineEX\n",
      "UreaEX L_IsoleucineEX\n",
      "UreaEX L_LeucineEX\n",
      "UreaEX L_LysineEX\n",
      "UreaEX L_MethionineEX\n",
      "UreaEX L_PhenylalanineEX\n",
      "UreaEX L_ProlineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX L_ProlineEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047533\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 91.41%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX     56.98         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n            Urea[e]               UreaEX    0.4584         1  8.59%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.5499         1 13.83%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.082         0  0.00%\\n                                     H[e]                                 HEX     -5.873         0  0.00%\\n                             L_Proline[e]                         L_ProlineEX    -0.6833         5 85.92%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.357         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.18%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "UreaEX L_SerineEX\n",
      "UreaEX L_ThreonineEX\n",
      "UreaEX L_TryptophanEX\n",
      "UreaEX L_TyrosineEX\n",
      "UreaEX L_ValineEX\n",
      "UreaEX PhospholipidEX\n",
      "UreaEX LPSEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX LPSEX Objective\\n=========\\n1.0 BIOMASS = 0.027525131016940436\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.329E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007367         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006548         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.271E-05         0  0.00%\\n          Copper[e]             CopperEX 2.404E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.04918         0  0.00%\\n            HCO3[e]            HCO3EXcar      5.82         1 97.35%\\n             Fe2[e]               IronEX 8.896E-05         0  0.00%\\n          Photon[e]              LightEX     62.31         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004374         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.898E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002619         0  0.00%\\n        Selenate[e]           SelenateEX  0.002102         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01228         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.163E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02094         0  0.00%\\n            Urea[e]               UreaEX    0.1587         1  2.65%\\n             Zn2[e]               ZincEX 5.061E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02753         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 23.47%\\n                                    CO[e]                                COEX -2.342E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000166         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003062         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.289         0  0.00%\\n                                     H[e]                                 HEX      -9.76         0  0.00%\\n                  Lipid_A_disaccharide[e]                               LPSEX   -0.03918        68 76.24%\\n                              Methanol[e]                          MethanolEX -0.0001282         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.074         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3649         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001282        15  0.06%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001254         6  0.22%\\n                                  tRNA[e]                              tRNAEX -1.435E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "UreaEX AcetateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UreaEX CitrateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX CitrateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600475733\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 89.17%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX     52.55         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n            Urea[e]               UreaEX    0.5918         1 10.83%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX    -0.9501         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                               Citrate[e]                           CitrateEX    -0.6833         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.266         0  0.00%\\n                                     H[e]                                 HEX     -4.506         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -4.673         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n                 ids  growth      status  uptake  secretion\\n155  {Trans_Citrate}     NaN  infeasible  UreaEX  CitrateEX\\n'\n",
      "stderr: None\n",
      "UreaEX FormateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX FormateEX Objective\\n=========\\n1.0 BIOMASS = 0.027257177933608327\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.316E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007295         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006485         0  0.00%\\n      Cobalt_ion[e]             CobaltEX  4.23E-05         0  0.00%\\n          Copper[e]             CopperEX 2.381E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009903         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.878         1 97.63%\\n             Fe2[e]               IronEX 8.809E-05         0  0.00%\\n          Photon[e]              LightEX     42.93         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004331         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.879E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002594         0  0.00%\\n        Selenate[e]           SelenateEX  0.002081         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01216         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.132E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02074         0  0.00%\\n            Urea[e]               UreaEX    0.1184         1  2.37%\\n             Zn2[e]               ZincEX 5.012E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02726         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.79%\\n                                    CO[e]                                COEX -2.319E-07         1  0.00%\\n                               Formate[e]                           FormateEX     -1.847         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001644         2  0.04%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003032         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.995         0  0.00%\\n                                     H[e]                                 HEX     -4.008         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001269         1  0.02%\\n                                Oxygen[e]                                O2EX      -3.33         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3144         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001269        15  0.23%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001241         6  0.90%\\n                                  tRNA[e]                              tRNAEX -1.421E-05        15  0.03%\\n\\n                 ids  growth      status  uptake  secretion\\n100         {R04173}     NaN  infeasible  UreaEX  FormateEX\\n423         {R00582}     NaN  infeasible  UreaEX  FormateEX\\n461  {Trans_Formate}     NaN  infeasible  UreaEX  FormateEX\\n488         {R01513}     NaN  infeasible  UreaEX  FormateEX\\n'\n",
      "stderr: None\n",
      "UreaEX PyruvateEX\n",
      "UreaEX FumarateEX\n",
      "UreaEX GlutathioneEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX GlutathioneEX Objective\\n=========\\n1.0 BIOMASS = 0.027092502243591318\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.308E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007251         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006446         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.204E-05         0  0.00%\\n          Copper[e]             CopperEX 2.366E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009843         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.877         1 88.55%\\n             Fe2[e]               IronEX 8.756E-05         0  0.00%\\n          Photon[e]              LightEX     57.54         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004305         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.868E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002578         0  0.00%\\n        Selenate[e]           SelenateEX  0.002069         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01209         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.113E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.3626         0  0.00%\\n            Urea[e]               UreaEX    0.6306         1 11.45%\\n             Zn2[e]               ZincEX 4.981E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02709         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.3764         1 97.42%\\n                                    CO[e]                                COEX -2.305E-07         1  0.00%\\n                               Formate[e]                           FormateEX    -0.3333         0  0.00%\\n                           Glutathione[e]                       GlutathioneEX     -0.342         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001634         2  0.08%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003014         0  0.00%\\n                                   H2O[e]                               H2OEX       -2.6         0  0.00%\\n                                     H[e]                                 HEX     -8.494         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001262         1  0.03%\\n                                Oxygen[e]                                O2EX     -5.368         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001262        15  0.49%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001234         6  1.92%\\n                                  tRNA[e]                              tRNAEX -1.413E-05        15  0.05%\\n\\nEmpty DataFrame\\nColumns: [ids, growth, status, uptake, secretion]\\nIndex: []\\n'\n",
      "stderr: None\n",
      "UreaEX 4_Methyl_2_oxopentanoateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UreaEX PantothenateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX PantothenateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600475487\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 93.88%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX     58.59         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n            Urea[e]               UreaEX    0.3177         1  6.12%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.2082         1 95.47%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.15%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.459         0  0.00%\\n                                     H[e]                                 HEX     -6.395         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.06%\\n                                Oxygen[e]                                O2EX     -5.618         0  0.00%\\n                          Pantothenate[e]                      PantothenateEX    -0.4019         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.86%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  3.37%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.10%\\n\\n                      ids  growth      status  uptake       secretion\\n465  {Trans_Pantothenate}     NaN  infeasible  UreaEX  PantothenateEX\\n'\n",
      "stderr: None\n",
      "UreaEX SuccinateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX SuccinateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600476586\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 82.32%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX     56.63         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n            Urea[e]               UreaEX     1.047         1 17.68%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -1.861         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.1267         0  0.00%\\n                                     H[e]                                 HEX     -9.283         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.584         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                             Succinate[e]                         SuccinateEX     -1.139         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n                   ids  growth      status  uptake    secretion\\n86   {Trans_Succinate}     NaN  infeasible  UreaEX  SuccinateEX\\n337           {R00342}     NaN  infeasible  UreaEX  SuccinateEX\\n379           {R01257}     NaN  infeasible  UreaEX  SuccinateEX\\n414           {R00408}     NaN  infeasible  UreaEX  SuccinateEX\\n459           {R10068}     NaN  infeasible  UreaEX  SuccinateEX\\n'\n",
      "stderr: None\n",
      "UreaEX ThymidineEX\n",
      "UreaEX XanthosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX XanthosineEX Objective\\n=========\\n1.0 BIOMASS = 0.027257177933608205\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.316E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007295         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006485         0  0.00%\\n      Cobalt_ion[e]             CobaltEX  4.23E-05         0  0.00%\\n          Copper[e]             CopperEX 2.381E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009903         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.196         1 87.31%\\n             Fe2[e]               IronEX 8.809E-05         0  0.00%\\n          Photon[e]              LightEX     54.88         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004331         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.879E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002594         0  0.00%\\n        Selenate[e]           SelenateEX  0.002081         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01216         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.132E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02074         0  0.00%\\n            Urea[e]               UreaEX    0.7552         1 12.69%\\n             Zn2[e]               ZincEX 5.012E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02726         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.79%\\n                                    CO[e]                                COEX -2.319E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001644         2  0.04%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003032         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.778         0  0.00%\\n                                     H[e]                                 HEX     -7.888         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001269         1  0.02%\\n                                Oxygen[e]                                O2EX     -4.317         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -0.187         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001269        15  0.23%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001241         6  0.90%\\n                            Xanthosine[e]                        XanthosineEX    -0.3184         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.421E-05        15  0.03%\\n\\n'\n",
      "stderr: None\n",
      "UreaEX 4_AminobenzoateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX 4_AminobenzoateEX Objective\\n=========\\n1.0 BIOMASS = 0.037635641600998256\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.816E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001007         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008954         0  0.00%\\n      Cobalt_ion[e]             CobaltEX  5.84E-05         0  0.00%\\n          Copper[e]             CopperEX 3.287E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01367         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.945         1 93.51%\\n             Fe2[e]               IronEX 0.0001216         0  0.00%\\n          Photon[e]              LightEX     56.08         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00598         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.595E-07         0  0.00%\\n               K[e]          PotassiumEX  0.003582         0  0.00%\\n        Selenate[e]           SelenateEX  0.002874         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01679         0  0.00%\\nStrontium_cation[e]          StrontiumEX 4.325E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02863         0  0.00%\\n            Urea[e]               UreaEX    0.3435         1  6.49%\\n             Zn2[e]               ZincEX  6.92E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                       4_Aminobenzoate[e]                   4_AminobenzoateEX    -0.3601         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.03764         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.34%\\n                                    CO[e]                                COEX -3.202E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000227         2  0.05%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.004187         0  0.00%\\n                                   H2O[e]                               H2OEX      -3.46         0  0.00%\\n                                     H[e]                                 HEX     -9.077         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001753         1  0.02%\\n                                Oxygen[e]                                O2EX     -4.787         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.0115         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001753        15  0.32%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001714         6  1.23%\\n                                  tRNA[e]                              tRNAEX -1.963E-05        15  0.04%\\n\\n                         ids  growth      status  uptake          secretion\\n157  {Trans_4_Aminobenzoate}     NaN  infeasible  UreaEX  4_AminobenzoateEX\\n'\n",
      "stderr: None\n",
      "UreaEX 5_MethylthioadenosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX 5_MethylthioadenosineEX Objective\\n=========\\n1.0 BIOMASS = 0.02746152044026794\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.325E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.000735         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006533         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.261E-05         0  0.00%\\n          Copper[e]             CopperEX 2.398E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009977         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.161         1 86.90%\\n             Fe2[e]               IronEX 8.875E-05         0  0.00%\\n          Photon[e]              LightEX     61.89         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004363         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.894E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002613         0  0.00%\\n        Selenate[e]           SelenateEX  0.002097         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01225         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.156E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.2083         0  0.00%\\n            Urea[e]               UreaEX    0.7782         1 13.10%\\n             Zn2[e]               ZincEX 5.049E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                 5_Methylthioadenosine[e]             5_MethylthioadenosineEX    -0.1886         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02746         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 46.60%\\n                                    CO[e]                                COEX -2.336E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001657         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003055         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.151         0  0.00%\\n                                     H[e]                                 HEX     -9.654         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001279         1  0.01%\\n                                Oxygen[e]                                O2EX     -5.561         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.2396         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001279        15  0.11%\\n                              Spermine[e]                          SpermineEX   -0.09369        10 53.25%\\n                                  tRNA[e]                              tRNAEX -1.432E-05        15  0.01%\\n\\n                               ids  growth  ...  uptake                secretion\\n499  {Trans_5_Methylthioadenosine}     NaN  ...  UreaEX  5_MethylthioadenosineEX\\n\\n[1 rows x 5 columns]\\n'\n",
      "stderr: None\n",
      "UreaEX AdenineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX AdenineEX Objective\\n=========\\n1.0 BIOMASS = 0.02725717793360826\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.316E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007295         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006485         0  0.00%\\n      Cobalt_ion[e]             CobaltEX  4.23E-05         0  0.00%\\n          Copper[e]             CopperEX 2.381E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009903         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.317         1 81.37%\\n             Fe2[e]               IronEX 8.809E-05         0  0.00%\\n          Photon[e]              LightEX     56.42         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004331         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.879E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002594         0  0.00%\\n        Selenate[e]           SelenateEX  0.002081         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01216         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.132E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02074         0  0.00%\\n            Urea[e]               UreaEX     1.218         1 18.63%\\n             Zn2[e]               ZincEX 5.012E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Adenine[e]                           AdenineEX    -0.4397         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02726         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.79%\\n                                    CO[e]                                COEX -2.319E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001644         2  0.04%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003032         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.589         0  0.00%\\n                                     H[e]                                 HEX      -7.91         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001269         1  0.02%\\n                                Oxygen[e]                                O2EX     -4.495         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.7101         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001269        15  0.23%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001241         6  0.90%\\n                                  tRNA[e]                              tRNAEX -1.421E-05        15  0.03%\\n\\n                 ids  growth      status  uptake  secretion\\n129  {Trans_Adenine}     NaN  infeasible  UreaEX  AdenineEX\\n'\n",
      "stderr: None\n",
      "UreaEX DethiobiotinEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX DethiobiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.027111713419450317\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.309E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007256         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.00645         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.207E-05         0  0.00%\\n          Copper[e]             CopperEX 2.368E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00985         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.482         1 89.44%\\n             Fe2[e]               IronEX 8.762E-05         0  0.00%\\n          Photon[e]              LightEX      62.9         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004308         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.869E-07         0  0.00%\\n               K[e]          PotassiumEX   0.00258         0  0.00%\\n        Selenate[e]           SelenateEX   0.00207         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01209         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.116E-05         0  0.00%\\n         Sulfate[e]            SulfateEX     0.172         0  0.00%\\n            Urea[e]               UreaEX    0.6475         1 10.56%\\n             Zn2[e]               ZincEX 4.985E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02711         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.8154         1 25.11%\\n                                    CO[e]                                COEX -2.307E-07         1  0.00%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1514         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001635         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003016         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.185         0  0.00%\\n                                     H[e]                                 HEX     -9.487         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1515         1  4.67%\\n                                Oxygen[e]                                O2EX     -5.775         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1515        15 69.98%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001235         6  0.23%\\n                                  tRNA[e]                              tRNAEX -1.414E-05        15  0.01%\\n\\n                      ids  growth      status  uptake       secretion\\n173              {R05605}     NaN  infeasible  UreaEX  DethiobiotinEX\\n488              {R10221}     NaN  infeasible  UreaEX  DethiobiotinEX\\n493  {Trans_Dethiobiotin}     NaN  infeasible  UreaEX  DethiobiotinEX\\n499              {R02036}     NaN  infeasible  UreaEX  DethiobiotinEX\\n'\n",
      "stderr: None\n",
      "UreaEX GuanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX GuanineEX Objective\\n=========\\n1.0 BIOMASS = 0.0272571779336075\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.316E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007295         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006485         0  0.00%\\n      Cobalt_ion[e]             CobaltEX  4.23E-05         0  0.00%\\n          Copper[e]             CopperEX 2.381E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009903         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.317         1 81.37%\\n             Fe2[e]               IronEX 8.809E-05         0  0.00%\\n          Photon[e]              LightEX     55.88         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004331         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.879E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002594         0  0.00%\\n        Selenate[e]           SelenateEX  0.002081         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01216         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.132E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02074         0  0.00%\\n            Urea[e]               UreaEX     1.218         1 18.63%\\n             Zn2[e]               ZincEX 5.012E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02726         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.79%\\n                                    CO[e]                                COEX -2.319E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001644         2  0.04%\\n                               Guanine[e]                           GuanineEX    -0.4397         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003032         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.589         0  0.00%\\n                                     H[e]                                 HEX     -2.389         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001269         1  0.02%\\n                                Oxygen[e]                                O2EX     -4.275         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.7101         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001269        15  0.23%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001241         6  0.90%\\n                                  tRNA[e]                              tRNAEX -1.421E-05        15  0.03%\\n\\nEmpty DataFrame\\nColumns: [ids, growth, status, uptake, secretion]\\nIndex: []\\n'\n",
      "stderr: None\n",
      "UreaEX GuanosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX GuanosineEX Objective\\n=========\\n1.0 BIOMASS = 0.027257177933608094\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.316E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007295         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006485         0  0.00%\\n      Cobalt_ion[e]             CobaltEX  4.23E-05         0  0.00%\\n          Copper[e]             CopperEX 2.381E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009903         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.176         1 85.71%\\n             Fe2[e]               IronEX 8.809E-05         0  0.00%\\n          Photon[e]              LightEX     55.98         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004331         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.879E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002594         0  0.00%\\n        Selenate[e]           SelenateEX  0.002081         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01216         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.132E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02074         0  0.00%\\n            Urea[e]               UreaEX     0.863         1 14.29%\\n             Zn2[e]               ZincEX 5.012E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02726         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.79%\\n                                    CO[e]                                COEX -2.319E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001644         2  0.04%\\n                             Guanosine[e]                         GuanosineEX    -0.2979         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003032         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.752         0  0.00%\\n                                     H[e]                                 HEX     -8.225         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001269         1  0.02%\\n                                Oxygen[e]                                O2EX     -4.417         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.2846         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001269        15  0.23%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001241         6  0.90%\\n                                  tRNA[e]                              tRNAEX -1.421E-05        15  0.03%\\n\\nEmpty DataFrame\\nColumns: [ids, growth, status, uptake, secretion]\\nIndex: []\\n'\n",
      "stderr: None\n",
      "UreaEX PutrescineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "UreaEX SpermidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX SpermidineEX Objective\\n=========\\n1.0 BIOMASS = 0.027525131016940704\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.329E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007367         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006548         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.271E-05         0  0.00%\\n          Copper[e]             CopperEX 2.404E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX      0.01         0  0.00%\\n            HCO3[e]            HCO3EXcar       5.1         1 91.90%\\n             Fe2[e]               IronEX 8.896E-05         0  0.00%\\n          Photon[e]              LightEX     63.29         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004374         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.898E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002619         0  0.00%\\n        Selenate[e]           SelenateEX  0.002102         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01228         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.163E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.2411         0  0.00%\\n            Urea[e]               UreaEX    0.4497         1  8.10%\\n             Zn2[e]               ZincEX 5.061E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02753         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 38.12%\\n                                    CO[e]                                COEX -2.342E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000166         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003062         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.065         0  0.00%\\n                                     H[e]                                 HEX      -10.5         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001282         1  0.01%\\n                                Oxygen[e]                                O2EX     -5.989         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1559         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001282        15  0.09%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.2214         6 61.76%\\n                            Spermidine[e]                        SpermidineEX    -0.2201         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.435E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "UreaEX EthanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "UreaEX S_MalateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nUreaEX S_MalateEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047495\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 82.32%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX      53.7         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n            Urea[e]               UreaEX     1.047         1 17.68%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -1.861         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.1267         0  0.00%\\n                                     H[e]                                 HEX     -8.551         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.015         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                              S_Malate[e]                          S_MalateEX     -1.139         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n                  ids  growth      status  uptake   secretion\\n98   {Trans_S_Malate}     NaN  infeasible  UreaEX  S_MalateEX\\n473          {R00342}     NaN  infeasible  UreaEX  S_MalateEX\\n'\n",
      "stderr: None\n",
      "UreaEX H2O2EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX B12EX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600475407\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.969         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX    0.1085         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX    0.1182         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.092         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     63.79         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                     Cobamide_coenzyme[e]                               B12EX    -0.1085        72 96.73%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.2541         1  3.15%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.645         0  0.00%\\n                                     H[e]                                 HEX     -10.86         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.096         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.02%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.09%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.00%\\n\\n          ids  growth      status     uptake secretion\\n175  {B12OUT}     NaN  infeasible  AmmoniaEX     B12EX\\n'\n",
      "stderr: None\n",
      "AmmoniaEX BiotinEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX BiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047477\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.296         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.483         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     64.83         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX    0.3241         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                Biotin[e]                            BiotinEX    -0.1518        10 36.81%\\n                                   CO2[e]                               CO2EX    -0.1674         1  4.06%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.831         0  0.00%\\n                                     H[e]                                 HEX     -10.27         0  0.00%\\n                              Methanol[e]                          MethanolEX     -0.152         1  3.68%\\n                                Oxygen[e]                                O2EX     -6.078         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX     -0.152        15 55.26%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.18%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX MethanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX MethanolEX Objective\\n=========\\n1.0 BIOMASS = 0.026935819530001175\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.296         0   0.00%\\n         Cadmium[e]            CadmiumEX   1.3E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007209         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006408         0   0.00%\\n      Cobalt_ion[e]             CobaltEX  4.18E-05         0   0.00%\\n          Copper[e]             CopperEX 2.353E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009786         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.483         1 100.00%\\n             Fe2[e]               IronEX 8.705E-05         0   0.00%\\n          Photon[e]              LightEX     62.87         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00428         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.857E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002563         0   0.00%\\n        Selenate[e]           SelenateEX  0.002057         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01202         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.095E-05         0   0.00%\\n         Sulfate[e]            SulfateEX    0.1722         0   0.00%\\n             Zn2[e]               ZincEX 4.953E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02694         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1675         1  6.43%\\n                                    CO[e]                                COEX -2.292E-07         1  0.00%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1517         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001625         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002997         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.831         0  0.00%\\n                                     H[e]                                 HEX     -9.479         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1519         1  5.83%\\n                                Oxygen[e]                                O2EX     -5.774         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1519        15 87.44%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001227         6  0.28%\\n                                  tRNA[e]                              tRNAEX -1.405E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX SAdenosyl4methylthio2oxobutanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.026935819530002632\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.296         0   0.00%\\n         Cadmium[e]            CadmiumEX   1.3E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007209         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006408         0   0.00%\\n      Cobalt_ion[e]             CobaltEX  4.18E-05         0   0.00%\\n          Copper[e]             CopperEX 2.353E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009786         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.483         1 100.00%\\n             Fe2[e]               IronEX 8.705E-05         0   0.00%\\n          Photon[e]              LightEX     62.87         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.00428         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.857E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002563         0   0.00%\\n        Selenate[e]           SelenateEX  0.002057         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01202         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.095E-05         0   0.00%\\n         Sulfate[e]            SulfateEX    0.1722         0   0.00%\\n             Zn2[e]               ZincEX 4.953E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02694         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1675         1  6.43%\\n                                    CO[e]                                COEX -2.292E-07         1  0.00%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1517         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001625         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002997         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.831         0  0.00%\\n                                     H[e]                                 HEX     -9.479         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1519         1  5.83%\\n                                Oxygen[e]                                O2EX     -5.774         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1519        15 87.44%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001227         6  0.28%\\n                                  tRNA[e]                              tRNAEX -1.405E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX SMethyl5thioDriboseEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX SMethyl5thioDriboseEX Objective\\n=========\\n1.0 BIOMASS = 0.027228818969082663\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.8291         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.314E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007288         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006478         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.225E-05         0   0.00%\\n          Copper[e]             CopperEX 2.378E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009892         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.026         1 100.00%\\n             Fe2[e]               IronEX   8.8E-05         0   0.00%\\n          Photon[e]              LightEX     64.23         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004327         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.877E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002591         0   0.00%\\n        Selenate[e]           SelenateEX  0.002079         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01215         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.129E-05         0   0.00%\\n         Sulfate[e]            SulfateEX     0.317         0   0.00%\\n             Zn2[e]               ZincEX 5.007E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02723         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.3814         1 10.45%\\n                                    CO[e]                                COEX -2.317E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001642         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003029         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.515         0  0.00%\\n                                     H[e]                                 HEX     -10.59         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001268         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.138         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001268        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.2976         6 48.90%\\n                              Spermine[e]                          SpermineEX    -0.1482        10 40.58%\\n                                  tRNA[e]                              tRNAEX  -1.42E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX GlycolateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX GlycolateEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047438\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     64.17         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX   -0.09144         1  2.60%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                             Glycolate[e]                         GlycolateEX     -1.708         2 97.12%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.174         0  0.00%\\n                                     H[e]                                 HEX     -6.898         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.161         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.21%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.01%\\n\\n                  ids  growth      status     uptake    secretion\\n385  {GlycolateTRANS}     NaN  infeasible  AmmoniaEX  GlycolateEX\\n451          {R03140}     NaN  infeasible  AmmoniaEX  GlycolateEX\\n457          {R01334}     NaN  infeasible  AmmoniaEX  GlycolateEX\\n'\n",
      "stderr: None\n",
      "AmmoniaEX GlycogenEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX GlycogenEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600474748\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     55.98         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX   -0.09144         1  2.60%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                              Glycogen[e]                          GlycogenEX    -0.5694         6 97.12%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.743         0  0.00%\\n                                     H[e]                                 HEX      -9.12         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.015         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.21%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX H2SEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX H2SEX Objective\\n=========\\n1.0 BIOMASS = 0.06962402098160866\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.6047         0   0.00%\\n         Cadmium[e]            CadmiumEX  3.36E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.001863         0   0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01656         0   0.00%\\n      Cobalt_ion[e]             CobaltEX  0.000108         0   0.00%\\n          Copper[e]             CopperEX 6.081E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02529         0   0.00%\\n            HCO3[e]            HCO3EXcar     3.777         1 100.00%\\n             Fe2[e]               IronEX  0.000225         0   0.00%\\n          Photon[e]              LightEX       150         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01106         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 4.801E-07         0   0.00%\\n               K[e]          PotassiumEX  0.006626         0   0.00%\\n        Selenate[e]           SelenateEX  0.005316         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.03106         0   0.00%\\nStrontium_cation[e]          StrontiumEX 8.001E-05         0   0.00%\\n         Sulfate[e]            SulfateEX     7.337         0   0.00%\\n             Zn2[e]               ZincEX  1.28E-05         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.06962         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.2367         1 90.24%\\n                                    CO[e]                                COEX -5.924E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX   -0.00042         2  0.32%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.007746         0  0.00%\\n                                   H2O[e]                               H2OEX      -3.04         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -7.284         0  0.00%\\n                                     H[e]                                 HEX     -19.13         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0003242         1  0.12%\\n                                Oxygen[e]                                O2EX     -18.71         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0003242        15  1.85%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.003171         6  7.25%\\n                                  tRNA[e]                              tRNAEX -3.631E-05        15  0.21%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX SpermineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX SpermineEX Objective\\n=========\\n1.0 BIOMASS = 0.027007779818586528\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.8291         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.304E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007229         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006425         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.191E-05         0   0.00%\\n          Copper[e]             CopperEX 2.359E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009812         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.025         1 100.00%\\n             Fe2[e]               IronEX 8.729E-05         0   0.00%\\n          Photon[e]              LightEX      64.2         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004291         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.862E-07         0   0.00%\\n               K[e]          PotassiumEX   0.00257         0   0.00%\\n        Selenate[e]           SelenateEX  0.002062         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01205         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.104E-05         0   0.00%\\n         Sulfate[e]            SulfateEX    0.3178         0   0.00%\\n             Zn2[e]               ZincEX 4.966E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02701         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.3816         1 10.42%\\n                                    CO[e]                                COEX -2.298E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001629         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003005         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.509         0  0.00%\\n                                     H[e]                                 HEX     -10.58         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001258         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.139         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001258        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.2985         6 48.91%\\n                              Spermine[e]                          SpermineEX    -0.1486        10 40.59%\\n                                  tRNA[e]                              tRNAEX -1.408E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX ThiosulfateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX ThiosulfateEX Objective\\n=========\\n1.0 BIOMASS = 0.04928843649764042\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.4281         0   0.00%\\n         Cadmium[e]            CadmiumEX 2.379E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.001319         0   0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01173         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 7.648E-05         0   0.00%\\n          Copper[e]             CopperEX 4.305E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01791         0   0.00%\\n            HCO3[e]            HCO3EXcar     2.674         1 100.00%\\n             Fe2[e]               IronEX 0.0001593         0   0.00%\\n          Photon[e]              LightEX       150         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.007832         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 3.399E-07         0   0.00%\\n               K[e]          PotassiumEX  0.004691         0   0.00%\\n        Selenate[e]           SelenateEX  0.003763         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.02199         0   0.00%\\nStrontium_cation[e]          StrontiumEX 5.664E-05         0   0.00%\\n         Sulfate[e]            SulfateEX      14.2         0   0.00%\\n             Zn2[e]               ZincEX 9.063E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.04929         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1676         1 90.24%\\n                                    CO[e]                                COEX -4.193E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002973         2  0.32%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.005483         0  0.00%\\n                                   H2O[e]                               H2OEX     -9.236         0  0.00%\\n                                     H[e]                                 HEX     -27.74         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002295         1  0.12%\\n                                Oxygen[e]                                O2EX      -17.1         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002295        15  1.85%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.002245         6  7.25%\\n                           Thiosulfate[e]                       ThiosulfateEX     -7.084         0  0.00%\\n                                  tRNA[e]                              tRNAEX  -2.57E-05        15  0.21%\\n\\n                    ids  growth      status     uptake      secretion\\n370  {ThiosulfateTRANS}     NaN  infeasible  AmmoniaEX  ThiosulfateEX\\n420            {R04859}     NaN  infeasible  AmmoniaEX  ThiosulfateEX\\n466           {O2TRANS}     NaN  infeasible  AmmoniaEX  ThiosulfateEX\\n'\n",
      "stderr: None\n",
      "AmmoniaEX UreaEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX UreaEX Objective\\n=========\\n1.0 BIOMASS = 0.028533045292480158\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     2.921         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.377E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007637         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006788         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.428E-05         0   0.00%\\n          Copper[e]             CopperEX 2.492E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01037         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.554         1 100.00%\\n             Fe2[e]               IronEX 9.221E-05         0   0.00%\\n          Photon[e]              LightEX     60.91         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004534         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.967E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002715         0   0.00%\\n        Selenate[e]           SelenateEX  0.002179         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01273         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.279E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02171         0   0.00%\\n             Zn2[e]               ZincEX 5.246E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02853         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.7633         1 52.91%\\n                                    CO[e]                                COEX -2.428E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001721         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003174         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.917         0  0.00%\\n                                     H[e]                                 HEX     -10.34         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001329         1  0.01%\\n                                Oxygen[e]                                O2EX     -5.368         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.6676         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001329        15  0.14%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001299         6  0.54%\\n                                  Urea[e]                              UreaEX    -0.6689         1 46.36%\\n                                  tRNA[e]                              tRNAEX -1.488E-05        15  0.02%\\n\\n                    ids  growth      status     uptake secretion\\n340            {R02036}     NaN  infeasible  AmmoniaEX    UreaEX\\n363            {R01082}     NaN  infeasible  AmmoniaEX    UreaEX\\n445  {Trans_Putrescine}     NaN  infeasible  AmmoniaEX    UreaEX\\n469            {R10221}     NaN  infeasible  AmmoniaEX    UreaEX\\n470            {R05605}     NaN  infeasible  AmmoniaEX    UreaEX\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_AlanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX L_AlanineEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600475247\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.942         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     6.492         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX        64         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.791         0  0.00%\\n                                     H[e]                                 HEX      -9.51         0  0.00%\\n                             L_Alanine[e]                         L_AlanineEX     -1.708         3 99.81%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.723         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.14%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_ArginineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_AsparagineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_AspartateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX L_AspartateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600474644\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.372         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.923         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX      53.7         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX      -3.36         0  0.00%\\n                                     H[e]                                 HEX     -7.504         0  0.00%\\n                           L_Aspartate[e]                       L_AspartateEX     -1.139         4 99.78%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.015         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.16%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.00%\\n\\nEmpty DataFrame\\nColumns: [ids, growth, status, uptake, secretion]\\nIndex: []\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_CystineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX L_CystineEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600474748\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.942         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     6.492         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     79.54         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX     1.729         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.353         0  0.00%\\n                                     H[e]                                 HEX     -9.978         0  0.00%\\n                             L_Cystine[e]                         L_CystineEX    -0.8541         6 99.81%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -8.858         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.14%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_GlutamateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_GlutamineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX GlycineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX GlycineEX Objective\\n=========\\n1.0 BIOMASS = 0.027092502243588962\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.375         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.308E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007251         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006446         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.204E-05         0   0.00%\\n          Copper[e]             CopperEX 2.366E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009843         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.877         1 100.00%\\n             Fe2[e]               IronEX 8.756E-05         0   0.00%\\n          Photon[e]              LightEX     46.43         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004305         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.868E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002578         0   0.00%\\n        Selenate[e]           SelenateEX  0.002069         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01209         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.113E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02061         0   0.00%\\n             Zn2[e]               ZincEX 4.981E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02709         0  0.00%\\n                                   CO2[e]                               CO2EX   -0.08778         1  3.69%\\n                                    CO[e]                                COEX -2.305E-07         1  0.00%\\n                               Formate[e]                           FormateEX     -1.131         0  0.00%\\n                               Glycine[e]                           GlycineEX      -1.14         2 95.89%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001634         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003014         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.318         0  0.00%\\n                                     H[e]                                 HEX       -1.8         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001262         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.886         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001262        15  0.08%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001234         6  0.31%\\n                                  tRNA[e]                              tRNAEX -1.413E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_HistidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_IsoleucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_LeucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_LysineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_MethionineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX L_MethionineEX Objective\\n=========\\n1.0 BIOMASS = 0.027007779818586684\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.089         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.304E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007229         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006425         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.191E-05         0   0.00%\\n          Copper[e]             CopperEX 2.359E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009812         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.646         1 100.00%\\n             Fe2[e]               IronEX 8.729E-05         0   0.00%\\n          Photon[e]              LightEX     72.96         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004291         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.862E-07         0   0.00%\\n               K[e]          PotassiumEX   0.00257         0   0.00%\\n        Selenate[e]           SelenateEX  0.002062         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01205         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.104E-05         0   0.00%\\n         Sulfate[e]            SulfateEX    0.8751         0   0.00%\\n             Zn2[e]               ZincEX 4.966E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02701         0  0.00%\\n                                    CO[e]                                COEX -2.298E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001629         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003005         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.796         0  0.00%\\n                                     H[e]                                 HEX     -10.89         0  0.00%\\n                          L_Methionine[e]                      L_MethionineEX    -0.8546         5 99.77%\\n                              Methanol[e]                          MethanolEX -0.0001258         1  0.00%\\n                                Oxygen[e]                                O2EX     -8.015         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001258        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00123         6  0.17%\\n                                  tRNA[e]                              tRNAEX -1.408E-05        15  0.00%\\n\\n                   ids  growth      status     uptake       secretion\\n240  {MethionineTRANS}     NaN  infeasible  AmmoniaEX  L_MethionineEX\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_PhenylalanineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_ProlineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX L_ProlineEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600475223\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.9168         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     56.98         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX   -0.09144         1  2.60%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.541         0  0.00%\\n                                     H[e]                                 HEX      -9.37         0  0.00%\\n                             L_Proline[e]                         L_ProlineEX    -0.6833         5 97.12%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.357         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.21%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_SerineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX L_SerineEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600475268\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.372         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     50.77         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX   -0.09144         1  2.60%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.313         0  0.00%\\n                                     H[e]                                 HEX     -7.819         0  0.00%\\n                              L_Serine[e]                          L_SerineEX     -1.139         3 97.12%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.446         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.21%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_ThreonineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_TryptophanEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_TyrosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX L_ValineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX L_ValineEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600473672\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.088         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.638         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX        64         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.937         0  0.00%\\n                                     H[e]                                 HEX     -10.36         0  0.00%\\n                              L_Valine[e]                          L_ValineEX    -0.8541         5 99.77%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.723         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.17%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX PhospholipidEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX PhospholipidEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047463\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX     1.149         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     55.11         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX   -0.09144         1  2.60%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.174         0  0.00%\\n                                     H[e]                                 HEX     -8.903         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.584         0  0.00%\\n               sn_Glycerol_3_phosphate[e]                      PhospholipidEX     -1.139         3 97.12%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.21%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX LPSEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX LPSEX Objective\\n=========\\n1.0 BIOMASS = 0.030994437879146717\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX      0.43         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.496E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0008296         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.007374         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.809E-05         0   0.00%\\n          Copper[e]             CopperEX 2.707E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.05168         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.872         1 100.00%\\n             Fe2[e]               IronEX 0.0001002         0   0.00%\\n          Photon[e]              LightEX      62.8         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004925         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 2.137E-07         0   0.00%\\n               K[e]          PotassiumEX   0.00295         0   0.00%\\n        Selenate[e]           SelenateEX  0.002367         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01383         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.562E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02358         0   0.00%\\n             Zn2[e]               ZincEX 5.699E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.03099         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 21.95%\\n                                    CO[e]                                COEX -2.637E-07         1  0.00%\\n                               Formate[e]                           FormateEX   -0.06726         0  0.00%\\n                               Glycine[e]                           GlycineEX   -0.07718         2  4.13%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000187         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003448         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.639         0  0.00%\\n                                     H[e]                                 HEX     -9.763         0  0.00%\\n                  Lipid_A_disaccharide[e]                               LPSEX   -0.04042        68 73.57%\\n                              Methanol[e]                          MethanolEX -0.0001443         1  0.00%\\n                                Oxygen[e]                                O2EX      -6.05         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1682         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001443        15  0.06%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001412         6  0.23%\\n                                  Urea[e]                              UreaEX  -0.001412         1  0.04%\\n                                  tRNA[e]                              tRNAEX -1.616E-05        15  0.01%\\n\\n            ids  growth      status     uptake secretion\\n129    {R05605}     NaN  infeasible  AmmoniaEX     LPSEX\\n254  {LPSTRANS}     NaN  infeasible  AmmoniaEX     LPSEX\\n447    {R00586}     NaN  infeasible  AmmoniaEX     LPSEX\\n481    {R02036}     NaN  infeasible  AmmoniaEX     LPSEX\\n498    {R10221}     NaN  infeasible  AmmoniaEX     LPSEX\\n'\n",
      "stderr: None\n",
      "AmmoniaEX AcetateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX CitrateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX CitrateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600474657\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.467         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     52.55         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                               Citrate[e]                           CitrateEX    -0.6833         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.449         0  0.00%\\n                                     H[e]                                 HEX     -7.672         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -4.673         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX FormateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX PyruvateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX FumarateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX FumarateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600474182\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.923         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX      53.7         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                              Fumarate[e]                          FumarateEX     -1.139         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX      -3.36         0  0.00%\\n                                     H[e]                                 HEX     -7.504         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.015         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX GlutathioneEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX 4_Methyl_2_oxopentanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600473488\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     58.15         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.5694         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX   -0.09144         1 90.24%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.32%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.743         0  0.00%\\n                                     H[e]                                 HEX     -9.663         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.12%\\n                                Oxygen[e]                                O2EX     -5.584         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  1.85%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  7.25%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.21%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX PantothenateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX SuccinateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX SuccinateEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047377\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.923         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     56.63         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.221         0  0.00%\\n                                     H[e]                                 HEX     -8.236         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.584         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                             Succinate[e]                         SuccinateEX     -1.139         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX ThymidineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX XanthosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX XanthosineEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047292\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.672         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.235         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     54.19         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.2712         1 96.48%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.12%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX      -4.95         0  0.00%\\n                                     H[e]                                 HEX     -7.595         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.04%\\n                                Oxygen[e]                                O2EX     -4.116         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.67%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  2.61%\\n                            Xanthosine[e]                        XanthosineEX    -0.3596         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.07%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX 4_AminobenzoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX 4_AminobenzoateEX Objective\\n=========\\n1.0 BIOMASS = 0.038117324175565466\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.6911         0   0.00%\\n         Cadmium[e]            CadmiumEX  1.84E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX   0.00102         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.009068         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 5.915E-05         0   0.00%\\n          Copper[e]             CopperEX 3.329E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01385         0   0.00%\\n            HCO3[e]            HCO3EXcar     4.948         1 100.00%\\n             Fe2[e]               IronEX 0.0001232         0   0.00%\\n          Photon[e]              LightEX     56.17         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.006057         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 2.628E-07         0   0.00%\\n               K[e]          PotassiumEX  0.003627         0   0.00%\\n        Selenate[e]           SelenateEX   0.00291         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.017         0   0.00%\\nStrontium_cation[e]          StrontiumEX  4.38E-05         0   0.00%\\n         Sulfate[e]            SulfateEX     0.029         0   0.00%\\n             Zn2[e]               ZincEX 7.009E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                       4_Aminobenzoate[e]                   4_AminobenzoateEX    -0.3601         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.03812         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.4897         1 97.22%\\n                                    CO[e]                                COEX -3.243E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002299         2  0.09%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.004241         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.825         0  0.00%\\n                                     H[e]                                 HEX     -9.097         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001775         1  0.04%\\n                                Oxygen[e]                                O2EX     -4.787         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001775        15  0.53%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001736         6  2.07%\\n                                  tRNA[e]                              tRNAEX -1.988E-05        15  0.06%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX 5_MethylthioadenosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX AdenineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX DethiobiotinEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX DethiobiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047498\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     1.296         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.483         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     62.87         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX    0.1723         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.1674         1  6.42%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1518         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.831         0  0.00%\\n                                     H[e]                                 HEX     -9.476         0  0.00%\\n                              Methanol[e]                          MethanolEX     -0.152         1  5.83%\\n                                Oxygen[e]                                O2EX     -5.774         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX     -0.152        15 87.45%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.28%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX GuanineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX GuanineEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047413\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     4.029         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.634         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     53.05         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX     -0.471         1 97.94%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.07%\\n                               Guanine[e]                           GuanineEX    -0.7592         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -9.145         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.03%\\n                                Oxygen[e]                                O2EX     -3.117         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.39%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  1.53%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.04%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX GuanosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX GuanosineEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047523\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     2.032         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.235         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX     55.15         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.2712         1 96.48%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.12%\\n                             Guanosine[e]                         GuanosineEX    -0.3596         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX      -5.31         0  0.00%\\n                                     H[e]                                 HEX     -7.835         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.04%\\n                                Oxygen[e]                                O2EX     -4.116         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.67%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  2.61%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.07%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX PutrescineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX PutrescineEX Objective\\n=========\\n1.0 BIOMASS = 0.02843127901646289\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX     2.812         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.372E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX  0.000761         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006764         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.412E-05         0   0.00%\\n          Copper[e]             CopperEX 2.483E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01033         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.554         1 100.00%\\n             Fe2[e]               IronEX 9.189E-05         0   0.00%\\n          Photon[e]              LightEX     60.89         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004518         0   0.00%\\n      Molybdenum[e]         MolybdenumEX  1.96E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002706         0   0.00%\\n        Selenate[e]           SelenateEX  0.002171         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01268         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.267E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02163         0   0.00%\\n             Zn2[e]               ZincEX 5.228E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02843         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 56.78%\\n                                    CO[e]                                COEX -2.419E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001715         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003163         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.861         0  0.00%\\n                                     H[e]                                 HEX     -10.34         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001324         1  0.01%\\n                                Oxygen[e]                                O2EX     -5.367         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.6686         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001324        15  0.14%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001295         6  0.54%\\n                                  Urea[e]                              UreaEX    -0.6138         1 42.50%\\n                                  tRNA[e]                              tRNAEX -1.483E-05        15  0.02%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX SpermidineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX SpermidineEX Objective\\n=========\\n1.0 BIOMASS = 0.02700777981858604\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.9671         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.304E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007229         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006425         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.191E-05         0   0.00%\\n          Copper[e]             CopperEX 2.359E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009812         0   0.00%\\n            HCO3[e]            HCO3EXcar      5.12         1 100.00%\\n             Fe2[e]               IronEX 8.729E-05         0   0.00%\\n          Photon[e]              LightEX     63.57         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004291         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.862E-07         0   0.00%\\n               K[e]          PotassiumEX   0.00257         0   0.00%\\n        Selenate[e]           SelenateEX  0.002062         0   0.00%\\n   Sodium_cation[e]             SodiumEX   0.01205         0   0.00%\\nStrontium_cation[e]          StrontiumEX 3.104E-05         0   0.00%\\n         Sulfate[e]            SulfateEX    0.2647         0   0.00%\\n             Zn2[e]               ZincEX 4.966E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02701         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.5727         1 27.97%\\n                                    CO[e]                                COEX -2.298E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001629         2  0.02%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003005         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.637         0  0.00%\\n                                     H[e]                                 HEX     -10.53         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001258         1  0.01%\\n                                Oxygen[e]                                O2EX     -6.001         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001258        15  0.09%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX    -0.2454         6 71.91%\\n                            Spermidine[e]                        SpermidineEX    -0.2442         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.408E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX EthanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "AmmoniaEX S_MalateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nAmmoniaEX S_MalateEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600474154\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Ammonia[e]            AmmoniaEX    0.2336         0   0.00%\\n         Cadmium[e]            CadmiumEX 1.298E-06         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0   0.00%\\n          Copper[e]             CopperEX 2.349E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.923         1 100.00%\\n             Fe2[e]               IronEX 8.691E-05         0   0.00%\\n          Photon[e]              LightEX      53.7         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0   0.00%\\n               K[e]          PotassiumEX  0.002559         0   0.00%\\n        Selenate[e]           SelenateEX  0.002053         0   0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0   0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0   0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.221         0  0.00%\\n                                     H[e]                                 HEX     -7.504         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.015         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                              S_Malate[e]                          S_MalateEX     -1.139         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "AmmoniaEX H2O2EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX B12EX Objective\\n=========\\n1.0 BIOMASS = 0.035413967652882905\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.709E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0009479         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008425         0  0.00%\\n      Cobalt_ion[e]             CobaltEX   0.05394         0  0.00%\\n          Copper[e]             CopperEX 3.093E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.06675         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.039         1 30.83%\\n             Fe2[e]               IronEX 0.0001145         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX     2.261         5 69.17%\\n          Photon[e]              LightEX     78.73         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005627         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.442E-07         0  0.00%\\n               K[e]          PotassiumEX   0.00337         0  0.00%\\n        Selenate[e]           SelenateEX  0.002704         0  0.00%\\n   Sodium_cation[e]             SodiumEX    0.0158         0  0.00%\\nStrontium_cation[e]          StrontiumEX  4.07E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02694         0  0.00%\\n             Zn2[e]               ZincEX 6.511E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                     Cobamide_coenzyme[e]                               B12EX   -0.05388        72 82.47%\\n                               biomass[e]                           BiomassEX   -0.03541         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.8117         1 17.25%\\n                                    CO[e]                                COEX -3.013E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002136         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.00394         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.635         0  0.00%\\n                                     H[e]                                 HEX     -14.65         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001649         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.358         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001649        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001613         6  0.21%\\n                                  tRNA[e]                              tRNAEX -1.847E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX BiotinEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX BiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.026652270472146536\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.286E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007133         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006341         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.136E-05         0  0.00%\\n          Copper[e]             CopperEX 2.328E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009683         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.314         1 36.09%\\n             Fe2[e]               IronEX 8.614E-05         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX     1.882         5 63.91%\\n          Photon[e]              LightEX     75.15         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004235         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.838E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002536         0  0.00%\\n        Selenate[e]           SelenateEX  0.002035         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01189         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.063E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.2402         0  0.00%\\n             Zn2[e]               ZincEX 4.901E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02665         0  0.00%\\n                                Biotin[e]                            BiotinEX      -0.11        10 29.81%\\n                                   CO2[e]                               CO2EX      -0.82         1 22.23%\\n                                    CO[e]                                COEX -2.268E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001608         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002965         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.063         0  0.00%\\n                                     H[e]                                 HEX     -13.15         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1101         1  2.98%\\n                                Oxygen[e]                                O2EX       -6.6         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1101        15 44.77%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001214         6  0.20%\\n                                  tRNA[e]                              tRNAEX  -1.39E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX MethanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX MethanolEX Objective\\n=========\\n1.0 BIOMASS = 0.026710599941993787\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.289E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007149         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006355         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.145E-05         0  0.00%\\n          Copper[e]             CopperEX 2.333E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009704         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.314         1 36.09%\\n             Fe2[e]               IronEX 8.632E-05         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX     1.882         5 63.91%\\n          Photon[e]              LightEX     73.73         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004244         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.842E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002542         0  0.00%\\n        Selenate[e]           SelenateEX  0.002039         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01192         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.069E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.1302         0  0.00%\\n             Zn2[e]               ZincEX 4.911E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02671         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 31.68%\\n                                    CO[e]                                COEX -2.273E-07         1  0.00%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1099         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001611         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002972         0  0.00%\\n                                   H2O[e]                               H2OEX     -5.064         0  0.00%\\n                                     H[e]                                 HEX     -12.57         0  0.00%\\n                              Methanol[e]                          MethanolEX      -0.11         1  4.25%\\n                                Oxygen[e]                                O2EX      -6.38         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX      -0.11        15 63.76%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001216         6  0.28%\\n                                  tRNA[e]                              tRNAEX -1.393E-05        15  0.01%\\n\\n          ids  growth      status         uptake   secretion\\n388  {R04173}     NaN  infeasible  L_GlutamateEX  MethanolEX\\n439  {R00582}     NaN  infeasible  L_GlutamateEX  MethanolEX\\n475  {R01513}     NaN  infeasible  L_GlutamateEX  MethanolEX\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX SMethyl5thioDriboseEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX GlycolateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX AmmoniaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX GlycogenEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX H2SEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX H2SEX Objective\\n=========\\n1.0 BIOMASS = 0.06170317222082281\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.978E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001651         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01468         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 9.575E-05         0  0.00%\\n          Copper[e]             CopperEX 5.389E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02242         0  0.00%\\n            HCO3[e]            HCO3EXcar     3.381         1 41.94%\\n             Fe2[e]               IronEX 0.0001994         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX    0.9359         5 58.06%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.009804         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 4.255E-07         0  0.00%\\n               K[e]          PotassiumEX  0.005872         0  0.00%\\n        Selenate[e]           SelenateEX  0.004711         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.02752         0  0.00%\\nStrontium_cation[e]          StrontiumEX 7.091E-05         0  0.00%\\n         Sulfate[e]            SulfateEX     7.299         0  0.00%\\n             Zn2[e]               ZincEX 1.135E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX    -0.0617         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.5224         1 95.84%\\n                                    CO[e]                                COEX  -5.25E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0003722         2  0.14%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.006865         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.435         0  0.00%\\n                      Hydrogen_sulfide[e]                               H2SEX     -7.252         0  0.00%\\n                                     H[e]                                 HEX     -19.59         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002873         1  0.05%\\n                                Oxygen[e]                                O2EX     -18.61         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002873        15  0.79%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00281         6  3.09%\\n                                  tRNA[e]                              tRNAEX -3.218E-05        15  0.09%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX SpermineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX ThiosulfateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX ThiosulfateEX Objective\\n=========\\n1.0 BIOMASS = 0.04368835020810173\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 2.109E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001169         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01039         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 6.779E-05         0  0.00%\\n          Copper[e]             CopperEX 3.816E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01587         0  0.00%\\n            HCO3[e]            HCO3EXcar     2.394         1 41.94%\\n             Fe2[e]               IronEX 0.0001412         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX    0.6627         5 58.06%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.006942         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 3.012E-07         0  0.00%\\n               K[e]          PotassiumEX  0.004158         0  0.00%\\n        Selenate[e]           SelenateEX  0.003336         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01949         0  0.00%\\nStrontium_cation[e]          StrontiumEX 5.021E-05         0  0.00%\\n         Sulfate[e]            SulfateEX      14.1         0  0.00%\\n             Zn2[e]               ZincEX 8.033E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.04369         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.3699         1 95.84%\\n                                    CO[e]                                COEX -3.717E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002635         2  0.14%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.00486         0  0.00%\\n                                   H2O[e]                               H2OEX     -9.467         0  0.00%\\n                                     H[e]                                 HEX     -28.07         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0002035         1  0.05%\\n                                Oxygen[e]                                O2EX     -16.98         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0002035        15  0.79%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00199         6  3.09%\\n                           Thiosulfate[e]                       ThiosulfateEX     -7.035         0  0.00%\\n                                  tRNA[e]                              tRNAEX -2.278E-05        15  0.09%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX UreaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_AlanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_ArginineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX L_ArginineEX Objective\\n=========\\n1.0 BIOMASS = 0.035310212165587926\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.704E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0009451         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008401         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 5.479E-05         0  0.00%\\n          Copper[e]             CopperEX 3.084E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01283         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.417         1 21.46%\\n             Fe2[e]               IronEX 0.0001141         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX     3.965         5 78.54%\\n          Photon[e]              LightEX     89.72         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005611         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.435E-07         0  0.00%\\n               K[e]          PotassiumEX   0.00336         0  0.00%\\n        Selenate[e]           SelenateEX  0.002696         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01575         0  0.00%\\nStrontium_cation[e]          StrontiumEX 4.058E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02686         0  0.00%\\n             Zn2[e]               ZincEX 6.492E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.03531         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 21.84%\\n                                    CO[e]                                COEX -3.004E-07         1  0.00%\\n                               Citrate[e]                           CitrateEX    -0.1394         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX  -0.000213         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003928         0  0.00%\\n                                   H2O[e]                               H2OEX     -6.879         0  0.00%\\n                                     H[e]                                 HEX      -17.5         0  0.00%\\n                            L_Arginine[e]                        L_ArginineEX    -0.4869         6 77.81%\\n                              Methanol[e]                          MethanolEX -0.0001644         1  0.00%\\n                                Oxygen[e]                                O2EX     -7.492         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX   -0.01625         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001644        15  0.07%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001608         6  0.26%\\n                                  tRNA[e]                              tRNAEX -1.841E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_AsparagineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_AspartateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_CystineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_GlutamineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX GlycineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_HistidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_IsoleucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_LeucineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_LysineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_MethionineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_PhenylalanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_ProlineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_SerineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_ThreonineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_TryptophanEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX L_TryptophanEX Objective\\n=========\\n1.0 BIOMASS = 0.030722273795552573\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.483E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0008223         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.007309         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.767E-05         0  0.00%\\n          Copper[e]             CopperEX 2.683E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01116         0  0.00%\\n            HCO3[e]            HCO3EXcar       4.9         1 43.52%\\n             Fe2[e]               IronEX 9.929E-05         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX     1.272         5 56.48%\\n          Photon[e]              LightEX     64.36         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004882         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.118E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002924         0  0.00%\\n        Selenate[e]           SelenateEX  0.002346         0  0.00%\\n   Sodium_cation[e]             SodiumEX    0.0137         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.53E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02337         0  0.00%\\n             Zn2[e]               ZincEX 5.649E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.03072         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 27.24%\\n                                    CO[e]                                COEX -2.614E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001853         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003418         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.599         0  0.00%\\n                                     H[e]                                 HEX     -11.19         0  0.00%\\n                          L_Tryptophan[e]                      L_TryptophanEX    -0.1981        11 72.38%\\n                              Methanol[e]                          MethanolEX -0.0001431         1  0.00%\\n                                Oxygen[e]                                O2EX      -5.46         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001431        15  0.07%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001399         6  0.28%\\n                                  tRNA[e]                              tRNAEX -1.602E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_TyrosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX L_ValineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX PhospholipidEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX LPSEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX AcetateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX CitrateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX FormateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX PyruvateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX FumarateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX GlutathioneEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX PantothenateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX SuccinateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX ThymidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX XanthosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX 4_AminobenzoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_GlutamateEX 4_AminobenzoateEX Objective\\n=========\\n1.0 BIOMASS = 0.0363544455990328\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.755E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.000973         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.008649         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 5.641E-05         0  0.00%\\n          Copper[e]             CopperEX 3.175E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01321         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.937         1 45.74%\\n             Fe2[e]               IronEX 0.0001175         0  0.00%\\n     L_Glutamate[e]        L_GlutamateEX     1.171         5 54.26%\\n          Photon[e]              LightEX     64.52         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.005777         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 2.507E-07         0  0.00%\\n               K[e]          PotassiumEX   0.00346         0  0.00%\\n        Selenate[e]           SelenateEX  0.002776         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01622         0  0.00%\\nStrontium_cation[e]          StrontiumEX 4.178E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02766         0  0.00%\\n             Zn2[e]               ZincEX 6.684E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                       4_Aminobenzoate[e]                   4_AminobenzoateEX      -0.31         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.03635         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.7728         1 98.30%\\n                                    CO[e]                                COEX -3.093E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0002193         2  0.06%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.004045         0  0.00%\\n                                   H2O[e]                               H2OEX     -4.581         0  0.00%\\n                                     H[e]                                 HEX     -11.19         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001693         1  0.02%\\n                                Oxygen[e]                                O2EX     -5.404         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001693        15  0.32%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001656         6  1.26%\\n                                  tRNA[e]                              tRNAEX -1.896E-05        15  0.04%\\n\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX 5_MethylthioadenosineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX AdenineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX DethiobiotinEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX GuanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX GuanosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX PutrescineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX SpermidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX EthanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX S_MalateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_GlutamateEX H2O2EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX BiotinEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX MethanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX SMethyl5thioDriboseEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX GlycolateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX AmmoniaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX GlycogenEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX H2SEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX SpermineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX ThiosulfateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX UreaEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_AlanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_AsparagineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_AspartateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_CystineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_GlutamateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_GlutamineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX L_GlutamineEX Objective\\n=========\\n1.0 BIOMASS = 0.026158344331537436\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.263E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007001         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006223         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.059E-05         0  0.00%\\n          Copper[e]             CopperEX 2.285E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009503         0  0.00%\\n            HCO3[e]            HCO3EXcar      4.86         1 56.90%\\n             Fe2[e]               IronEX 8.454E-05         0  0.00%\\n      L_Arginine[e]         L_ArginineEX    0.6135         6 43.10%\\n          Photon[e]              LightEX     54.87         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004156         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.804E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002489         0  0.00%\\n        Selenate[e]           SelenateEX  0.001997         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01167         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.006E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.0199         0  0.00%\\n             Zn2[e]               ZincEX  4.81E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02616         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 24.26%\\n                                    CO[e]                                COEX -2.226E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001578         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.00291         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.415         0  0.00%\\n                                     H[e]                                 HEX     -5.132         0  0.00%\\n                           L_Glutamine[e]                       L_GlutamineEX      -0.51         5 75.45%\\n                              Methanol[e]                          MethanolEX -0.0001218         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.984         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.6034         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.4758         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001218        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001191         6  0.21%\\n                                  tRNA[e]                              tRNAEX -1.364E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_ArginineEX GlycineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_HistidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_IsoleucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_LeucineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_LysineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_MethionineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_PhenylalanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_ProlineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_SerineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_ThreonineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_TryptophanEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_TyrosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX L_ValineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX PhospholipidEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX LPSEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX AcetateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX CitrateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX FormateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX PyruvateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX FumarateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX GlutathioneEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX GlutathioneEX Objective\\n=========\\n1.0 BIOMASS = 0.026387818186909107\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.274E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007063         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006278         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.095E-05         0  0.00%\\n          Copper[e]             CopperEX 2.305E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009587         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.862         1 57.58%\\n             Fe2[e]               IronEX 8.528E-05         0  0.00%\\n      L_Arginine[e]         L_ArginineEX    0.5968         6 42.42%\\n          Photon[e]              LightEX     57.48         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004193         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.819E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002511         0  0.00%\\n        Selenate[e]           SelenateEX  0.002015         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01177         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.032E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.3484         0  0.00%\\n             Zn2[e]               ZincEX 4.852E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02639         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 98.83%\\n                                    CO[e]                                COEX -2.245E-07         1  0.00%\\n                               Formate[e]                           FormateEX    -0.3199         0  0.00%\\n                           Glutathione[e]                       GlutathioneEX    -0.3283         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001592         2  0.04%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002936         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.893         0  0.00%\\n                                     H[e]                                 HEX     -8.523         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001229         1  0.01%\\n                                Oxygen[e]                                O2EX     -5.396         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.5867         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.1105         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001229        15  0.22%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001202         6  0.87%\\n                                  tRNA[e]                              tRNAEX -1.376E-05        15  0.02%\\n\\n'\n",
      "stderr: None\n",
      "L_ArginineEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX PantothenateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX SuccinateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX ThymidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX XanthosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX 4_AminobenzoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX 5_MethylthioadenosineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX AdenineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX DethiobiotinEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX DethiobiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.02615834433153795\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.263E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007001         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006223         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.059E-05         0  0.00%\\n          Copper[e]             CopperEX 2.285E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009503         0  0.00%\\n            HCO3[e]            HCO3EXcar      5.35         1 62.63%\\n             Fe2[e]               IronEX 8.454E-05         0  0.00%\\n      L_Arginine[e]         L_ArginineEX    0.5319         6 37.37%\\n          Photon[e]              LightEX     62.08         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004156         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.804E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002489         0  0.00%\\n        Selenate[e]           SelenateEX  0.001997         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01167         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.006E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.1423         0  0.00%\\n             Zn2[e]               ZincEX  4.81E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02616         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 29.41%\\n                                    CO[e]                                COEX -2.226E-07         1  0.00%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1224         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001578         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.00291         0  0.00%\\n                                   H2O[e]                               H2OEX      -2.19         0  0.00%\\n                                     H[e]                                 HEX      -9.55         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1225         1  4.39%\\n                                Oxygen[e]                                O2EX       -5.8         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.5218         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3738         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1225        15 65.92%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001191         6  0.26%\\n                                  tRNA[e]                              tRNAEX -1.364E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_ArginineEX GuanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX GuanosineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX PutrescineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX SpermidineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\n'\n",
      "stderr: None\n",
      "L_ArginineEX EthanolEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX EthanolEX Objective\\n=========\\n1.0 BIOMASS = 0.026158344331538345\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.263E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007001         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006223         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.059E-05         0  0.00%\\n          Copper[e]             CopperEX 2.285E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009503         0  0.00%\\n            HCO3[e]            HCO3EXcar      5.06         1 58.33%\\n             Fe2[e]               IronEX 8.454E-05         0  0.00%\\n      L_Arginine[e]         L_ArginineEX    0.6025         6 41.67%\\n          Photon[e]              LightEX     61.42         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004156         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.804E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002489         0  0.00%\\n        Selenate[e]           SelenateEX  0.001997         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01167         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.006E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.0199         0  0.00%\\n             Zn2[e]               ZincEX  4.81E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02616         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 21.45%\\n                                    CO[e]                                COEX -2.226E-07         1  0.00%\\n                               Ethanol[e]                           EthanolEX    -0.1996         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001578         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX   -0.00291         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.102         0  0.00%\\n                                     H[e]                                 HEX     -10.29         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001218         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.328         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.5923         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.4536         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001218        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001191         6  0.19%\\n                                  tRNA[e]                              tRNAEX    -0.1996        15 78.31%\\n\\n                    ids  growth      status        uptake  secretion\\n146     {Trans_ethanol}     NaN  infeasible  L_ArginineEX  EthanolEX\\n371            {R00190}     NaN  infeasible  L_ArginineEX  EthanolEX\\n406  {Trans_Putrescine}     NaN  infeasible  L_ArginineEX  EthanolEX\\n488            {R00754}     NaN  infeasible  L_ArginineEX  EthanolEX\\n'\n",
      "stderr: None\n",
      "L_ArginineEX S_MalateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX S_MalateEX Objective\\n=========\\n1.0 BIOMASS = 0.027722746830814774\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.338E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.000742         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006596         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.302E-05         0  0.00%\\n          Copper[e]             CopperEX 2.421E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.01007         0  0.00%\\n            HCO3[e]            HCO3EXcar      4.87         1 60.26%\\n             Fe2[e]               IronEX  8.96E-05         0  0.00%\\n      L_Arginine[e]         L_ArginineEX    0.5353         6 39.74%\\n          Photon[e]              LightEX     53.82         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004405         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.912E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002638         0  0.00%\\n        Selenate[e]           SelenateEX  0.002117         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01237         0  0.00%\\nStrontium_cation[e]          StrontiumEX 3.186E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02109         0  0.00%\\n             Zn2[e]               ZincEX 5.097E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX    -0.8512         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02772         0  0.00%\\n                                    CO[e]                                COEX -2.359E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001672         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.003084         0  0.00%\\n                                   H2O[e]                               H2OEX    -0.1076         0  0.00%\\n                                     H[e]                                 HEX     -8.575         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001291         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.019         0  0.00%\\n                            Putrescine[e]                        PutrescineEX    -0.5246         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001291        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001263         6 74.32%\\n                              S_Malate[e]                          S_MalateEX     -1.143         0  0.00%\\n                                  tRNA[e]                              tRNAEX -1.446E-05        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "L_ArginineEX H2O2EX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ArginineEX H2O2EX Objective\\n=========\\n1.0 BIOMASS = 0.07197442975615394\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 3.474E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX  0.001926         0  0.00%\\n    Chloride_ion[e]           ChlorideEX   0.01712         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 0.0001117         0  0.00%\\n          Copper[e]             CopperEX 6.286E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.02615         0  0.00%\\n             H2O[e]                H2OEX     26.66         0  0.00%\\n            HCO3[e]            HCO3EXcar     3.743         1 68.66%\\n             Fe2[e]               IronEX 0.0002326         0  0.00%\\n      L_Arginine[e]         L_ArginineEX    0.2848         6 31.34%\\n          Photon[e]              LightEX       150         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX   0.01144         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 4.963E-07         0  0.00%\\n          Oxygen[e]                 O2EX     10.45         0  0.00%\\n               K[e]          PotassiumEX  0.006849         0  0.00%\\n        Selenate[e]           SelenateEX  0.005496         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.03211         0  0.00%\\nStrontium_cation[e]          StrontiumEX 8.271E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.05475         0  0.00%\\n             Zn2[e]               ZincEX 1.323E-05         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.07197         0  0.00%\\n                                   CO2[e]                               CO2EX    -0.7641         1 96.65%\\n                                    CO[e]                                COEX -6.124E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0004342         2  0.11%\\n                     Hydrogen_peroxide[e]                              H2O2EX     -29.17         0  0.00%\\n                                     H[e]                                 HEX     -33.72         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0003352         1  0.04%\\n                            Putrescine[e]                        PutrescineEX     -0.257         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0003352        15  0.64%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.003278         6  2.49%\\n                                  tRNA[e]                              tRNAEX -3.754E-05        15  0.07%\\n\\n              ids  growth      status        uptake secretion\\n346  {Trans_H2O2}     NaN  infeasible  L_ArginineEX    H2O2EX\\n386    {H2OTRANS}     NaN  infeasible  L_ArginineEX    H2O2EX\\n485     {O2TRANS}     NaN  infeasible  L_ArginineEX    H2O2EX\\n'\n",
      "stderr: None\n",
      "CyanateEX B12EX\n",
      "CyanateEX BiotinEX\n",
      "CyanateEX MethanolEX\n",
      "CyanateEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX SAdenosyl4methylthio2oxobutanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.02487748779586613\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.201E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0006658         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.005919         0  0.00%\\n      Cobalt_ion[e]             CobaltEX  3.86E-05         0  0.00%\\n          Copper[e]             CopperEX 2.173E-06         0  0.00%\\n         Cyanate[e]            CyanateEX     1.066         1 16.62%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009038         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.348         1 83.38%\\n             Fe2[e]               IronEX  8.04E-05         0  0.00%\\n          Photon[e]              LightEX     61.83         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.003953         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.715E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002367         0  0.00%\\n        Selenate[e]           SelenateEX  0.001899         0  0.00%\\n   Sodium_cation[e]             SodiumEX    0.0111         0  0.00%\\nStrontium_cation[e]          StrontiumEX 2.859E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.1403         0  0.00%\\n             Zn2[e]               ZincEX 4.574E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02488         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 29.59%\\n                                    CO[e]                                COEX -2.117E-07         1  0.00%\\n                          Dethiobiotin[e]                      DethiobiotinEX    -0.1214         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001501         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002768         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.144         0  0.00%\\n                                     H[e]                                 HEX      -8.44         0  0.00%\\n                              Methanol[e]                          MethanolEX    -0.1215         1  4.38%\\n                                Oxygen[e]                                O2EX     -5.794         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3909         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX    -0.1215        15 65.76%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001133         6  0.25%\\n                                  tRNA[e]                              tRNAEX -1.297E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX SMethyl5thioDriboseEX\n",
      "CyanateEX GlycolateEX\n",
      "CyanateEX AmmoniaEX\n",
      "CyanateEX GlycogenEX\n",
      "CyanateEX H2SEX\n",
      "CyanateEX SpermineEX\n",
      "CyanateEX ThiosulfateEX\n",
      "CyanateEX UreaEX\n",
      "CyanateEX L_AlanineEX\n",
      "CyanateEX L_ArginineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX L_ArginineEX Objective\\n=========\\n1.0 BIOMASS = 0.024834554501557967\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.199E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0006647         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.005908         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 3.854E-05         0  0.00%\\n          Copper[e]             CopperEX 2.169E-06         0  0.00%\\n         Cyanate[e]            CyanateEX     1.617         1 23.68%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009022         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.212         1 76.32%\\n             Fe2[e]               IronEX 8.026E-05         0  0.00%\\n          Photon[e]              LightEX     59.74         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.003946         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.712E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002363         0  0.00%\\n        Selenate[e]           SelenateEX  0.001896         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01108         0  0.00%\\nStrontium_cation[e]          StrontiumEX 2.854E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.01889         0  0.00%\\n             Zn2[e]               ZincEX 4.566E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02483         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 27.98%\\n                                    CO[e]                                COEX -2.113E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001498         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002763         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.219         0  0.00%\\n                                     H[e]                                 HEX     -8.457         0  0.00%\\n                            L_Arginine[e]                        L_ArginineEX    -0.3503         6 71.71%\\n                              Methanol[e]                          MethanolEX -0.0001157         1  0.00%\\n                                Oxygen[e]                                O2EX     -5.607         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.8814         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001157        15  0.06%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001131         6  0.23%\\n                                  tRNA[e]                              tRNAEX -1.295E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX L_AsparagineEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CyanateEX L_AspartateEX\n",
      "CyanateEX L_CystineEX\n",
      "CyanateEX L_GlutamateEX\n",
      "CyanateEX L_GlutamineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX L_GlutamineEX Objective\\n=========\\n1.0 BIOMASS = 0.02483455450155641\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.199E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0006647         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.005908         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 3.854E-05         0  0.00%\\n          Copper[e]             CopperEX 2.169E-06         0  0.00%\\n         Cyanate[e]            CyanateEX     1.228         1 20.16%\\n  Orthophosphate[e] FAKEOrthophosphateEX  0.009022         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.862         1 79.84%\\n             Fe2[e]               IronEX 8.026E-05         0  0.00%\\n          Photon[e]              LightEX     54.67         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.003946         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.712E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002363         0  0.00%\\n        Selenate[e]           SelenateEX  0.001896         0  0.00%\\n   Sodium_cation[e]             SodiumEX   0.01108         0  0.00%\\nStrontium_cation[e]          StrontiumEX 2.854E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.01889         0  0.00%\\n             Zn2[e]               ZincEX 4.566E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02483         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 24.41%\\n                                    CO[e]                                COEX -2.113E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001498         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002763         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.375         0  0.00%\\n                                     H[e]                                 HEX     -7.579         0  0.00%\\n                           L_Glutamine[e]                       L_GlutamineEX     -0.506         5 75.32%\\n                              Methanol[e]                          MethanolEX -0.0001157         1  0.00%\\n                                Oxygen[e]                                O2EX     -4.984         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.4922         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001157        15  0.05%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001131         6  0.20%\\n                                  tRNA[e]                              tRNAEX -1.295E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX GlycineEX\n",
      "CyanateEX L_HistidineEX\n",
      "CyanateEX L_IsoleucineEX\n",
      "CyanateEX L_LeucineEX\n",
      "CyanateEX L_LysineEX\n",
      "CyanateEX L_MethionineEX\n",
      "CyanateEX L_PhenylalanineEX\n",
      "CyanateEX L_ProlineEX\n",
      "CyanateEX L_SerineEX\n",
      "CyanateEX L_ThreonineEX\n",
      "CyanateEX L_TryptophanEX\n",
      "CyanateEX L_TyrosineEX\n",
      "CyanateEX L_ValineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX L_ValineEX Objective\\n=========\\n1.0 BIOMASS = 0.026892497600475216\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n         Cyanate[e]            CyanateEX     1.088         1 18.24%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 81.76%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX        64         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX     -0.325         1  7.06%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX   -0.08643         0  0.00%\\n                                     H[e]                                 HEX     -10.04         0  0.00%\\n                              L_Valine[e]                          L_ValineEX    -0.8541         5 92.73%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.00%\\n                                Oxygen[e]                                O2EX     -6.723         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  0.16%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX PhospholipidEX\n",
      "CyanateEX LPSEX\n",
      "CyanateEX AcetateEX\n",
      "CyanateEX CitrateEX\n",
      "CyanateEX FormateEX\n",
      "CyanateEX PyruvateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX PyruvateEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047485\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n         Cyanate[e]            CyanateEX     1.395         1 21.49%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.097         1 78.51%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX     59.61         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               Ammonia[e]                           AmmoniaEX     -1.162         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                     H[e]                                 HEX     -8.412         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  1.27%\\n                                Oxygen[e]                                O2EX     -5.869         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -1.708         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6 74.32%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX FumarateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CyanateEX GlutathioneEX\n",
      "CyanateEX 4_Methyl_2_oxopentanoateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nCyanateEX 4_Methyl_2_oxopentanoateEX Objective\\n=========\\n1.0 BIOMASS = 0.02689249760047467\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.298E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0007198         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.006398         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 4.173E-05         0  0.00%\\n          Copper[e]             CopperEX 2.349E-06         0  0.00%\\n         Cyanate[e]            CyanateEX    0.2336         1  4.57%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00977         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.875         1 95.43%\\n             Fe2[e]               IronEX 8.691E-05         0  0.00%\\n          Photon[e]              LightEX     58.15         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.004273         0  0.00%\\n      Molybdenum[e]         MolybdenumEX 1.854E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002559         0  0.00%\\n        Selenate[e]           SelenateEX  0.002053         0  0.00%\\n   Sodium_cation[e]             SodiumEX     0.012         0  0.00%\\nStrontium_cation[e]          StrontiumEX  3.09E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.02046         0  0.00%\\n             Zn2[e]               ZincEX 4.945E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.5694         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02689         0  0.00%\\n                                   CO2[e]                               CO2EX     -0.325         1 97.05%\\n                                    CO[e]                                COEX -2.288E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001622         2  0.10%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002992         0  0.00%\\n                                   H2O[e]                               H2OEX      -1.51         0  0.00%\\n                                     H[e]                                 HEX     -9.429         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001252         1  0.04%\\n                                Oxygen[e]                                O2EX     -5.584         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001252        15  0.56%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX  -0.001225         6  2.19%\\n                                  tRNA[e]                              tRNAEX -1.402E-05        15  0.06%\\n\\n'\n",
      "stderr: None\n",
      "CyanateEX PantothenateEX\n",
      "CyanateEX SuccinateEX\n",
      "CyanateEX ThymidineEX\n",
      "CyanateEX XanthosineEX\n",
      "CyanateEX 4_AminobenzoateEX\n",
      "CyanateEX 5_MethylthioadenosineEX\n",
      "CyanateEX AdenineEX\n",
      "CyanateEX DethiobiotinEX\n",
      "CyanateEX GuanineEX\n",
      "CyanateEX GuanosineEX\n",
      "CyanateEX PutrescineEX\n",
      "CyanateEX SpermidineEX\n",
      "CyanateEX EthanolEX\n",
      "CyanateEX S_MalateEX\n",
      "CyanateEX H2O2EX\n",
      "L_ValineEX B12EX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX B12EX Objective\\n=========\\n1.0 BIOMASS = 0.022186835416095837\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.071E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0005938         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.005278         0  0.00%\\n      Cobalt_ion[e]             CobaltEX   0.04534         0  0.00%\\n          Copper[e]             CopperEX 1.938E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.05337         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.935         1 51.82%\\n             Fe2[e]               IronEX  7.17E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.9176         5 48.18%\\n          Photon[e]              LightEX     53.43         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.003525         0  0.00%\\n      Molybdenum[e]         MolybdenumEX  1.53E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002111         0  0.00%\\n        Selenate[e]           SelenateEX  0.001694         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.009897         0  0.00%\\nStrontium_cation[e]          StrontiumEX  2.55E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.01688         0  0.00%\\n             Zn2[e]               ZincEX 4.079E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.9004         0  0.00%\\n                     Cobamide_coenzyme[e]                               B12EX   -0.04531        72 79.75%\\n                               biomass[e]                           BiomassEX   -0.02219         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 20.05%\\n                                    CO[e]                                COEX -1.888E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001338         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002468         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.319         0  0.00%\\n                                     H[e]                                 HEX     -3.549         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001033         1  0.00%\\n                                Oxygen[e]                                O2EX      -4.18         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX     -0.241         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001033        15  0.04%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00101         6  0.15%\\n                                  tRNA[e]                              tRNAEX -1.157E-05        15  0.00%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX BiotinEX\n",
      "L_ValineEX MethanolEX\n",
      "L_ValineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "L_ValineEX SMethyl5thioDriboseEX\n",
      "L_ValineEX GlycolateEX\n",
      "L_ValineEX AmmoniaEX\n",
      "L_ValineEX GlycogenEX\n",
      "L_ValineEX H2SEX\n",
      "L_ValineEX SpermineEX\n",
      "L_ValineEX ThiosulfateEX\n",
      "L_ValineEX UreaEX\n",
      "L_ValineEX L_AlanineEX\n",
      "L_ValineEX L_ArginineEX\n",
      "L_ValineEX L_AsparagineEX\n",
      "L_ValineEX L_AspartateEX\n",
      "L_ValineEX L_CystineEX\n",
      "L_ValineEX L_GlutamateEX\n",
      "L_ValineEX L_GlutamineEX\n",
      "L_ValineEX GlycineEX\n",
      "L_ValineEX L_HistidineEX\n",
      "L_ValineEX L_IsoleucineEX\n",
      "L_ValineEX L_LeucineEX\n",
      "L_ValineEX L_LysineEX\n",
      "L_ValineEX L_MethionineEX\n",
      "L_ValineEX L_PhenylalanineEX\n",
      "L_ValineEX L_ProlineEX\n",
      "L_ValineEX L_SerineEX\n",
      "L_ValineEX L_ThreonineEX\n",
      "L_ValineEX L_TryptophanEX\n",
      "L_ValineEX L_TyrosineEX\n",
      "L_ValineEX PhospholipidEX\n",
      "L_ValineEX LPSEX\n",
      "L_ValineEX AcetateEX\n",
      "L_ValineEX CitrateEX\n",
      "L_ValineEX FormateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX FormateEX Objective\\n=========\\n1.0 BIOMASS = 0.02218683541609062\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.071E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0005938         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.005278         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 3.443E-05         0  0.00%\\n          Copper[e]             CopperEX 1.938E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00806         0  0.00%\\n            HCO3[e]            HCO3EXcar     4.845         1 83.41%\\n             Fe2[e]               IronEX  7.17E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.1927         5 16.59%\\n          Photon[e]              LightEX     41.09         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.003525         0  0.00%\\n      Molybdenum[e]         MolybdenumEX  1.53E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002111         0  0.00%\\n        Selenate[e]           SelenateEX  0.001694         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.009897         0  0.00%\\nStrontium_cation[e]          StrontiumEX  2.55E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.01688         0  0.00%\\n             Zn2[e]               ZincEX 4.079E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX    -0.1754         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02219         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.02%\\n                                    CO[e]                                COEX -1.888E-07         1  0.00%\\n                               Formate[e]                           FormateEX     -1.785         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001338         2  0.03%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002468         0  0.00%\\n                                   H2O[e]                               H2OEX     -2.037         0  0.00%\\n                                     H[e]                                 HEX     -3.643         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001033         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.135         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.3407         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001033        15  0.19%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00101         6  0.73%\\n                                  tRNA[e]                              tRNAEX -1.157E-05        15  0.02%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX PyruvateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_ValineEX FumarateEX\n",
      "L_ValineEX GlutathioneEX\n",
      "L_ValineEX 4_Methyl_2_oxopentanoateEX\n",
      "L_ValineEX PantothenateEX\n",
      "L_ValineEX SuccinateEX\n",
      "L_ValineEX ThymidineEX\n",
      "L_ValineEX XanthosineEX\n",
      "L_ValineEX 4_AminobenzoateEX\n",
      "L_ValineEX 5_MethylthioadenosineEX\n",
      "L_ValineEX AdenineEX\n",
      "L_ValineEX DethiobiotinEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX DethiobiotinEX Objective\\n=========\\n1.0 BIOMASS = 0.02218688003788319\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.071E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0005938         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.005278         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 3.443E-05         0  0.00%\\n          Copper[e]             CopperEX 1.938E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00806         0  0.00%\\n            HCO3[e]            HCO3EXcar     5.191         1 56.50%\\n             Fe2[e]               IronEX  7.17E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX    0.7993         5 43.50%\\n          Photon[e]              LightEX     54.46         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.003525         0  0.00%\\n      Molybdenum[e]         MolybdenumEX  1.53E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002111         0  0.00%\\n        Selenate[e]           SelenateEX  0.001694         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.009897         0  0.00%\\nStrontium_cation[e]          StrontiumEX  2.55E-05         0  0.00%\\n         Sulfate[e]            SulfateEX    0.1035         0  0.00%\\n             Zn2[e]               ZincEX 4.079E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX     -0.782         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02219         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 37.03%\\n                                    CO[e]                                COEX -1.888E-07         1  0.00%\\n                          Dethiobiotin[e]                      DethiobiotinEX   -0.08666         0  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001338         2  0.01%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002468         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.267         0  0.00%\\n                                     H[e]                                 HEX     -7.991         0  0.00%\\n                              Methanol[e]                          MethanolEX   -0.08676         1  3.92%\\n                                Oxygen[e]                                O2EX     -4.626         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX   -0.09806         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX   -0.08676        15 58.76%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00101         6  0.27%\\n                                  tRNA[e]                              tRNAEX -1.157E-05        15  0.01%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX GuanineEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nL_ValineEX GuanineEX Objective\\n=========\\n1.0 BIOMASS = 0.022186836231099585\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number C-Flux\\n         Cadmium[e]            CadmiumEX 1.071E-06         0  0.00%\\n  Calcium_cation[e]            CalciumEX 0.0005938         0  0.00%\\n    Chloride_ion[e]           ChlorideEX  0.005278         0  0.00%\\n      Cobalt_ion[e]             CobaltEX 3.443E-05         0  0.00%\\n          Copper[e]             CopperEX 1.938E-06         0  0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00806         0  0.00%\\n            HCO3[e]            HCO3EXcar      5.02         1 48.46%\\n             Fe2[e]               IronEX  7.17E-05         0  0.00%\\n        L_Valine[e]           L_ValineEX     1.068         5 51.54%\\n          Photon[e]              LightEX     49.02         0  0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.003525         0  0.00%\\n      Molybdenum[e]         MolybdenumEX  1.53E-07         0  0.00%\\n               K[e]          PotassiumEX  0.002111         0  0.00%\\n        Selenate[e]           SelenateEX  0.001694         0  0.00%\\n   Sodium_cation[e]             SodiumEX  0.009897         0  0.00%\\nStrontium_cation[e]          StrontiumEX  2.55E-05         0  0.00%\\n         Sulfate[e]            SulfateEX   0.01688         0  0.00%\\n             Zn2[e]               ZincEX 4.079E-06         0  0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n              4_Methyl_2_oxopentanoate[e]          4_Methyl_2_oxopentanoateEX     -1.051         0  0.00%\\n                               biomass[e]                           BiomassEX   -0.02219         0  0.00%\\n                                   CO2[e]                               CO2EX      -0.82         1 99.02%\\n                                    CO[e]                                COEX -1.888E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001338         2  0.03%\\n                               Guanine[e]                           GuanineEX     -0.175         0  0.00%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.002468         0  0.00%\\n                                   H2O[e]                               H2OEX     -3.857         0  0.00%\\n                                     H[e]                                 HEX     -6.886         0  0.00%\\n                              Methanol[e]                          MethanolEX -0.0001033         1  0.01%\\n                                Oxygen[e]                                O2EX     -3.643         0  0.00%\\n                              Pyruvate[e]                          PyruvateEX    -0.4107         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -0.0001033        15  0.19%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX   -0.00101         6  0.73%\\n                                  tRNA[e]                              tRNAEX -1.157E-05        15  0.02%\\n\\n'\n",
      "stderr: None\n",
      "L_ValineEX GuanosineEX\n",
      "L_ValineEX PutrescineEX\n",
      "L_ValineEX SpermidineEX\n",
      "L_ValineEX EthanolEX\n",
      "CalledProcessError 1\n",
      "stdout: Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n",
      "\n",
      "stderr: Traceback (most recent call last):\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 138, in <module>\n",
      "    resdf = _gene_deletion_feeding(model, args.uptake, args.secretion, args.bound, args.deltype, del_list)\n",
      "  File \"/fast_data/Osnat/CC1A3/COMETS/Prochlorococcus_Model/N_Starvation/./prochlorococcus_feed_knockout.py\", line 96, in _gene_deletion_feeding\n",
      "    print(uptake1, sec1, model.summary())\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1209, in summary\n",
      "    return ModelSummary(model=self, solution=solution, fva=fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 85, in __init__\n",
      "    self._generate(model, solution, fva)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/summary/model_summary.py\", line 116, in _generate\n",
      "    solution = pfba(model)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/flux_analysis/parsimonious.py\", line 95, in pfba\n",
      "    m.slim_optimize(error_value=None)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/core/model.py\", line 1063, in slim_optimize\n",
      "    assert_optimal(self, message)\n",
      "  File \"/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/cobra/util/solver.py\", line 590, in assert_optimal\n",
      "    raise exception_cls(f\"{message} ({status}).\")\n",
      "cobra.exceptions.Infeasible: None (infeasible).\n",
      "\n",
      "L_ValineEX S_MalateEX\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_ValineEX H2O2EX\n",
      "GuanineEX B12EX\n",
      "GuanineEX BiotinEX\n",
      "GuanineEX MethanolEX\n",
      "GuanineEX SAdenosyl4methylthio2oxobutanoateEX\n",
      "GuanineEX SMethyl5thioDriboseEX\n",
      "GuanineEX GlycolateEX\n",
      "GuanineEX AmmoniaEX\n",
      "GuanineEX GlycogenEX\n",
      "GuanineEX H2SEX\n",
      "GuanineEX SpermineEX\n",
      "GuanineEX ThiosulfateEX\n",
      "GuanineEX UreaEX\n",
      "GuanineEX L_AlanineEX\n",
      "GuanineEX L_ArginineEX\n",
      "GuanineEX L_AsparagineEX\n",
      "GuanineEX L_AspartateEX\n",
      "GuanineEX L_CystineEX\n",
      "GuanineEX L_GlutamateEX\n",
      "GuanineEX L_GlutamineEX\n",
      "GuanineEX GlycineEX\n",
      "GuanineEX L_HistidineEX\n",
      "GuanineEX L_IsoleucineEX\n",
      "GuanineEX L_LeucineEX\n",
      "GuanineEX L_LysineEX\n",
      "GuanineEX L_MethionineEX\n",
      "GuanineEX L_PhenylalanineEX\n",
      "GuanineEX L_ProlineEX\n",
      "GuanineEX L_SerineEX\n",
      "GuanineEX L_ThreonineEX\n",
      "GuanineEX L_TryptophanEX\n",
      "GuanineEX L_TyrosineEX\n",
      "GuanineEX L_ValineEX\n",
      "GuanineEX PhospholipidEX\n",
      "GuanineEX LPSEX\n",
      "GuanineEX AcetateEX\n",
      "GuanineEX CitrateEX\n",
      "GuanineEX FormateEX\n",
      "GuanineEX PyruvateEX\n",
      "GuanineEX FumarateEX\n",
      "GuanineEX GlutathioneEX\n",
      "GuanineEX 4_Methyl_2_oxopentanoateEX\n",
      "GuanineEX PantothenateEX\n",
      "GuanineEX SuccinateEX\n",
      "TimeoutExpired 10\n",
      "stdout: b'Scaling...\\n A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\\nProblem data seem to be well scaled\\nGuanineEX SuccinateEX Objective\\n=========\\n1.0 BIOMASS = 0.01772709460258706\\n\\nUptake\\n------\\n         Metabolite             Reaction      Flux  C-Number  C-Flux\\n         Cadmium[e]            CadmiumEX 8.556E-07         0   0.00%\\n  Calcium_cation[e]            CalciumEX 0.0004745         0   0.00%\\n    Chloride_ion[e]           ChlorideEX  0.004217         0   0.00%\\n      Cobalt_ion[e]             CobaltEX 2.751E-05         0   0.00%\\n          Copper[e]             CopperEX 1.548E-06         0   0.00%\\n  Orthophosphate[e] FAKEOrthophosphateEX   0.00644         0   0.00%\\n         Guanine[e]            GuanineEX    0.1495         0   0.00%\\n            HCO3[e]            HCO3EXcar     5.794         1 100.00%\\n             Fe2[e]               IronEX 5.729E-05         0   0.00%\\n          Photon[e]              LightEX     54.22         0   0.00%\\nMagnesium_cation[e]          MagnesiumEX  0.002817         0   0.00%\\n      Molybdenum[e]         MolybdenumEX 1.222E-07         0   0.00%\\n               K[e]          PotassiumEX  0.001687         0   0.00%\\n        Selenate[e]           SelenateEX  0.001354         0   0.00%\\n   Sodium_cation[e]             SodiumEX  0.007908         0   0.00%\\nStrontium_cation[e]          StrontiumEX 2.037E-05         0   0.00%\\n         Sulfate[e]            SulfateEX   0.01349         0   0.00%\\n             Zn2[e]               ZincEX 3.259E-06         0   0.00%\\n\\nSecretion\\n---------\\n                               Metabolite                            Reaction       Flux  C-Number C-Flux\\n                               biomass[e]                           BiomassEX   -0.01773         0  0.00%\\n                                    CO[e]                                COEX -1.508E-07         1  0.00%\\n                        Glycolaldehyde[e]                    GlycolaldehydeEX -0.0001069         2  3.28%\\n                     Hydrogen_peroxide[e]                              H2O2EX  -0.001972         0  0.00%\\n                                   H2O[e]                               H2OEX     -1.741         0  0.00%\\n                                     H[e]                                 HEX     -7.763         0  0.00%\\n                              Methanol[e]                          MethanolEX -8.255E-05         1  1.27%\\n                                Oxygen[e]                                O2EX      -5.43         0  0.00%\\nS_Adenosyl_4_methylthio_2_oxobutanoate[e] SAdenosyl4methylthio2oxobutanoateEX -8.255E-05        15 19.00%\\n              S_Methyl_5_thio_D_ribose[e]               SMethyl5thioDriboseEX -0.0008073         6 74.32%\\n                             Succinate[e]                         SuccinateEX     -1.039         0  0.00%\\n                            Xanthosine[e]                        XanthosineEX    -0.1484         0  0.00%\\n                                  tRNA[e]                              tRNAEX -9.245E-06        15  2.13%\\n\\n'\n",
      "stderr: None\n",
      "GuanineEX ThymidineEX\n",
      "GuanineEX XanthosineEX\n",
      "GuanineEX 4_AminobenzoateEX\n",
      "GuanineEX 5_MethylthioadenosineEX\n",
      "GuanineEX AdenineEX\n",
      "GuanineEX DethiobiotinEX\n",
      "GuanineEX GuanosineEX\n",
      "GuanineEX PutrescineEX\n",
      "GuanineEX SpermidineEX\n",
      "GuanineEX EthanolEX\n",
      "GuanineEX S_MalateEX\n",
      "GuanineEX H2O2EX\n"
     ]
    }
   ],
   "source": [
    "tdf = n_exchange_df\n",
    "deltype = 'reaction'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_secretion_30082022')\n",
    "knockout_success = [\n",
    "    _secretion_deletion_feeding_with_timeout(dpath, u, s, m, deltype, timeout=10)\n",
    "    for (u, s, m) in zip(tdf['uptake'], tdf['secretion'], tdf['maximum']*0.8)\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 818,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'error',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'timeout',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success',\n",
       " 'success']"
      ]
     },
     "execution_count": 818,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knockout_success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 819,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "success    635\n",
       "timeout    243\n",
       "error        3\n",
       "Name: rknockout_success1, dtype: int64"
      ]
     },
     "execution_count": 819,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_exchange_df['rknockout_success1'] = knockout_success\n",
    "n_exchange_df['rknockout_success1'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 838,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_ValineEX EthanolEX\n"
     ]
    }
   ],
   "source": [
    "tdf = n_exchange_df.loc[n_exchange_df['rknockout_success6'].isin(['timeout', 'error'])]\n",
    "deltype = 'reaction'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_secretion_30082022')\n",
    "knockout_success = [\n",
    "    _secretion_deletion_feeding_with_timeout(dpath, u, s, m, deltype, timeout=3000)\n",
    "    for (u, s, m) in zip(tdf['uptake'], tdf['secretion'], tdf['maximum']*0.5)\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 836,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['success', 'success', 'success', 'success', 'success', 'error']"
      ]
     },
     "execution_count": 836,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "knockout_success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 837,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "success    5\n",
       "error      1\n",
       "Name: rknockout_success6, dtype: int64"
      ]
     },
     "execution_count": 837,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_exchange_df.loc[n_exchange_df.rknockout_success5.isin(['error', 'timeout']), 'rknockout_success6'] = knockout_success\n",
    "n_exchange_df['rknockout_success6'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Double gene/reaction deletion - for now not running"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "essential_genes_per_uptake = gene_uptake_knockout_df.groupby('uptake').gid.apply(list)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "essential_reactions_per_uptake = reaction_uptake_knockout_df.groupby('uptake').rid.apply(list)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "def _uptake_double_deletion_feeding_with_timeout(dpath, uptake1, deltype, timeout=10):\n",
    "    try:\n",
    "        print (uptake1)\n",
    "        if deltype == '2gene':\n",
    "            c = non_essential_genes.copy()\n",
    "            c.difference_update(essential_genes_per_uptake[uptake1])\n",
    "        else: \n",
    "            c = non_essential_reactions.copy()\n",
    "            c.difference_update(essential_reactions_per_uptake[uptake1])\n",
    "        list_file = os.path.join('data', f'MED4_del_{deltype}_{uptake1}_list.txt')\n",
    "        with open(list_file, 'w') as fh:\n",
    "            fh.writelines(f'{i}\\n' for i in c)\n",
    "\n",
    "            \n",
    "        result = subprocess.run([\n",
    "            './prochlorococcus_feed_knockout.py', \n",
    "            '--uptake', uptake1, \n",
    "            '--out_dpath', dpath,\n",
    "            '--deltype', deltype,\n",
    "            '--list_file', list_file,\n",
    "            ],\n",
    "            capture_output=True, text=True, check=True, timeout=timeout,\n",
    "        )\n",
    "        #print(\"stdout:\", result.stdout)\n",
    "        #print(\"stderr:\", result.stderr)\n",
    "        return 'success'\n",
    "        \n",
    "    except subprocess.CalledProcessError as err:\n",
    "        print('CalledProcessError', err.returncode)\n",
    "        print(\"stdout:\", err.stdout)\n",
    "        print(\"stderr:\", err.stderr)\n",
    "        return 'error'\n",
    "                \n",
    "    except subprocess.TimeoutExpired as err:\n",
    "        print('TimeoutExpired', err.timeout)\n",
    "        print(\"stdout:\", err.stdout)\n",
    "        print(\"stderr:\", err.stderr)\n",
    "        return 'timeout'\n",
    "\n"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "deltype = '2gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_uptake_22082022')\n",
    "knockout_success = [\n",
    "    _uptake_double_deletion_feeding_with_timeout(dpath, u, deltype, timeout=10000)\n",
    "    for u in df_growing.uptake\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Results of knockdown"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### load the knockout results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "deltype = 'gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_uptake_29082022')\n",
    "gene_uptake_knockout_df = pd.concat([pd.read_csv(os.path.join(dpath, i)) for i in os.listdir(dpath)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "deltype = 'gene'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_secretion_30082022')\n",
    "gene_secretion_knockout_df = pd.concat([pd.read_csv(os.path.join(dpath, i)) for i in os.listdir(dpath)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "deltype = 'reaction'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_uptake_29082022')\n",
    "reaction_uptake_knockout_df = pd.concat([pd.read_csv(os.path.join(dpath, i)) for i in os.listdir(dpath)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "deltype = 'reaction'\n",
    "dpath = os.path.join('data',f'{deltype}_knockout_secretion_30082022')\n",
    "reaction_secretion_knockout_df = pd.concat([pd.read_csv(os.path.join(dpath, i)) for i in os.listdir(dpath)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "reaction_uptake_knockout_df['rid']  = reaction_uptake_knockout_df.ids.str.extract(r\"([^\\{\\}\\']+)\")\n",
    "gene_uptake_knockout_df['gid']  = gene_uptake_knockout_df.ids.str.extract(r\"([^\\{\\}\\']+)\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "reaction_secretion_knockout_df['rid']  = reaction_secretion_knockout_df.ids.str.extract(r\"([^\\{\\}\\']+)\")\n",
    "gene_secretion_knockout_df['gid']  = gene_secretion_knockout_df.ids.str.extract(r\"([^\\{\\}\\']+)\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "reaction_uptake_knockout_df['reaction'] = reaction_uptake_knockout_df.rid.map(lambda x: model.reactions.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "reaction_secretion_knockout_df['reaction'] = reaction_secretion_knockout_df.rid.map(lambda x: model.reactions.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _get_gene_ids(rid):\n",
    "    gids = model.reactions.get_by_id(rid).genes\n",
    "    return pd.Series({'gid' :[g.id for g in gids]})\n",
    "\n",
    "reaction_uptake_knockout_df['gid'] = reaction_uptake_knockout_df.rid.apply(_get_gene_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _get_gene_ids(rid):\n",
    "    gids = model.reactions.get_by_id(rid).genes\n",
    "    return pd.Series({'gid' :[g.id for g in gids]})\n",
    "\n",
    "reaction_secretion_knockout_df['gid'] = reaction_secretion_knockout_df.rid.apply(_get_gene_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ids</th>\n",
       "      <th>growth</th>\n",
       "      <th>status</th>\n",
       "      <th>uptake</th>\n",
       "      <th>secretion</th>\n",
       "      <th>gid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>{'PMM0951'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>{'PMM0796'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>{'PMM0483'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>{'PMM0001'}</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>optimal</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>{'PMM1181'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM1181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>589</td>\n",
       "      <td>{'PMM0938'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_SerineEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>590</td>\n",
       "      <td>{'PMM1472'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_SerineEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM1472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>591</td>\n",
       "      <td>{'PMM1273'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_SerineEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM1273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>592</td>\n",
       "      <td>{'PMM0723'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_SerineEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>594</td>\n",
       "      <td>{'PMM0258'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_SerineEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0258</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5648 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0          ids   growth      status         uptake  secretion  \\\n",
       "0             0  {'PMM0951'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "1             1  {'PMM0796'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "2             3  {'PMM0483'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "3             5  {'PMM0001'}  0.00001     optimal  L_GlutamateEX        NaN   \n",
       "4             7  {'PMM1181'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "..          ...          ...      ...         ...            ...        ...   \n",
       "320         589  {'PMM0938'}      NaN  infeasible     L_SerineEX        NaN   \n",
       "321         590  {'PMM1472'}      NaN  infeasible     L_SerineEX        NaN   \n",
       "322         591  {'PMM1273'}      NaN  infeasible     L_SerineEX        NaN   \n",
       "323         592  {'PMM0723'}      NaN  infeasible     L_SerineEX        NaN   \n",
       "324         594  {'PMM0258'}      NaN  infeasible     L_SerineEX        NaN   \n",
       "\n",
       "         gid  \n",
       "0    PMM0951  \n",
       "1    PMM0796  \n",
       "2    PMM0483  \n",
       "3    PMM0001  \n",
       "4    PMM1181  \n",
       "..       ...  \n",
       "320  PMM0938  \n",
       "321  PMM1472  \n",
       "322  PMM1273  \n",
       "323  PMM0723  \n",
       "324  PMM0258  \n",
       "\n",
       "[5648 rows x 7 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "gids_of_reactions_df = reaction_uptake_knockout_df.explode('gid')\n",
    "gids_of_reactions_df = gids_of_reactions_df.dropna(subset=['gid'], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "gids_of_reactions_df = gids_of_reactions_df.drop_duplicates(subset=['uptake', 'gid'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = gids_of_reactions_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove genes that are already knocked out\n",
    "gids_of_reactions_df_m = pd.merge(gids_of_reactions_df, gene_uptake_knockout_df, how='left',\n",
    "                                  suffixes=['', '_test'], on=['uptake', 'gid'])\n",
    "gids_of_reactions_df = gids_of_reactions_df_m.loc[gids_of_reactions_df_m.ids_test.isna(), cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_uptake_knockout_df['knockout'] = 'gene'\n",
    "gids_of_reactions_df['knockout'] = 'reaction'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_uptake_knockout_df = pd.concat([gene_uptake_knockout_df, gids_of_reactions_df])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ids</th>\n",
       "      <th>growth</th>\n",
       "      <th>status</th>\n",
       "      <th>uptake</th>\n",
       "      <th>secretion</th>\n",
       "      <th>gid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>167</td>\n",
       "      <td>{'PMM1354'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>AmmoniaEX</td>\n",
       "      <td>XanthosineEX</td>\n",
       "      <td>PMM1354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>157</td>\n",
       "      <td>{'PMM1354'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>GuanineEX</td>\n",
       "      <td>AdenineEX</td>\n",
       "      <td>PMM1354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>156</td>\n",
       "      <td>{'PMM1354'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>AmmoniaEX</td>\n",
       "      <td>L_TryptophanEX</td>\n",
       "      <td>PMM1354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>170</td>\n",
       "      <td>{'PMM1354'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>CyanateEX</td>\n",
       "      <td>XanthosineEX</td>\n",
       "      <td>PMM1354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>132</td>\n",
       "      <td>{'PMM1702'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_CystineEX</td>\n",
       "      <td>EthanolEX</td>\n",
       "      <td>PMM1702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>67</td>\n",
       "      <td>{'PMM1638'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_CystineEX</td>\n",
       "      <td>5_MethylthioadenosineEX</td>\n",
       "      <td>PMM1638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>144</td>\n",
       "      <td>{'PMM0642'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_CystineEX</td>\n",
       "      <td>5_MethylthioadenosineEX</td>\n",
       "      <td>PMM0642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>163</td>\n",
       "      <td>{'PMM0046'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_CystineEX</td>\n",
       "      <td>5_MethylthioadenosineEX</td>\n",
       "      <td>PMM0046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>144</td>\n",
       "      <td>{'PMM0225'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_AlanineEX</td>\n",
       "      <td>PutrescineEX</td>\n",
       "      <td>PMM0225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>168</td>\n",
       "      <td>{'PMM0770'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_AlanineEX</td>\n",
       "      <td>PutrescineEX</td>\n",
       "      <td>PMM0770</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1963 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0          ids growth      status       uptake  \\\n",
       "0         167  {'PMM1354'}    NaN  infeasible    AmmoniaEX   \n",
       "0         157  {'PMM1354'}    NaN  infeasible    GuanineEX   \n",
       "0         156  {'PMM1354'}    NaN  infeasible    AmmoniaEX   \n",
       "0         170  {'PMM1354'}    NaN  infeasible    CyanateEX   \n",
       "0         132  {'PMM1702'}    NaN  infeasible  L_CystineEX   \n",
       "..        ...          ...    ...         ...          ...   \n",
       "0          67  {'PMM1638'}    NaN  infeasible  L_CystineEX   \n",
       "1         144  {'PMM0642'}    NaN  infeasible  L_CystineEX   \n",
       "2         163  {'PMM0046'}    NaN  infeasible  L_CystineEX   \n",
       "0         144  {'PMM0225'}    NaN  infeasible  L_AlanineEX   \n",
       "1         168  {'PMM0770'}    NaN  infeasible  L_AlanineEX   \n",
       "\n",
       "                  secretion      gid  \n",
       "0              XanthosineEX  PMM1354  \n",
       "0                 AdenineEX  PMM1354  \n",
       "0            L_TryptophanEX  PMM1354  \n",
       "0              XanthosineEX  PMM1354  \n",
       "0                 EthanolEX  PMM1702  \n",
       "..                      ...      ...  \n",
       "0   5_MethylthioadenosineEX  PMM1638  \n",
       "1   5_MethylthioadenosineEX  PMM0642  \n",
       "2   5_MethylthioadenosineEX  PMM0046  \n",
       "0              PutrescineEX  PMM0225  \n",
       "1              PutrescineEX  PMM0770  \n",
       "\n",
       "[1963 rows x 7 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_secretion_knockout_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_secretion_knockout_nodup_df = gene_secretion_knockout_df.drop_duplicates(subset=['gid','secretion'], ignore_index=True)\n",
    "reaction_secretion_knockout_nodup_df = reaction_secretion_knockout_df.drop_duplicates(subset=['rid','secretion'], ignore_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "sgids_of_reactions_df = reaction_secretion_knockout_nodup_df.explode('gid')\n",
    "sgids_of_reactions_df = sgids_of_reactions_df.dropna(subset=['gid'], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "sgids_of_reactions_df = sgids_of_reactions_df.drop_duplicates(subset=['secretion', 'gid'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "sgids_of_reactions1_df = sgids_of_reactions_df.loc[\n",
    "    (~sgids_of_reactions_df.gid.isin(gene_uptake_knockout_df.gid)) & \n",
    "    (~sgids_of_reactions_df.gid.isin(essential_genes_list)) \n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((1414, 9), (518, 9))"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sgids_of_reactions_df.shape, sgids_of_reactions1_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = sgids_of_reactions1_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove genes that are already knocked out\n",
    "sgids_of_reactions_df_m = pd.merge(sgids_of_reactions1_df, gene_secretion_knockout_nodup_df, how='left',\n",
    "                                  suffixes=['', '_test'], on=['secretion', 'gid'])\n",
    "sgids_of_reactions_df = sgids_of_reactions_df_m.loc[sgids_of_reactions_df_m.ids_test.isna(), cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_202575/4100974182.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gene_secretion_knockout_nodup_df['knockout'] = 'gene'\n"
     ]
    }
   ],
   "source": [
    "gene_secretion_knockout_nodup_df['knockout'] = 'gene'\n",
    "sgids_of_reactions_df['knockout'] = 'reaction'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_secretion_knockout_nodup_df = pd.concat([gene_secretion_knockout_nodup_df, sgids_of_reactions_df])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene_id', 'gene', 'product', 'status', 'main', 'sub', 'path', 'ecpath',\n",
       "       'baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj',\n",
       "       'contrast', 'Length', 'organism', 'protein_id'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "deseq_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_uptake_knockout_df = pd.merge(gene_uptake_knockout_df, pmm_df, left_on='gid', right_on='pmm', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_secretion_knockout_nodup_df = pd.merge(gene_secretion_knockout_nodup_df, pmm_df, left_on='gid', right_on='pmm', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = ['gene_id', 'gene', 'product', 'status', 'sub', 'path', 'ecpath',\n",
    "       'baseMean', 'log2FoldChange',  'padj',\n",
    "       ]\n",
    "gene_uptake_knockout_df = pd.merge(gene_uptake_knockout_df, deseq_df[cols], on='gene_id', \n",
    "         how='left', suffixes=['', '_de'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = ['gene_id', 'gene', 'product', 'status', 'sub', 'path', 'ecpath',\n",
    "       'baseMean', 'log2FoldChange',  'padj',\n",
    "       ]\n",
    "gene_secretion_knockout_nodup_df = pd.merge(gene_secretion_knockout_nodup_df, deseq_df[cols], on='gene_id', \n",
    "         how='left', suffixes=['', '_de'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ids</th>\n",
       "      <th>growth</th>\n",
       "      <th>status</th>\n",
       "      <th>uptake</th>\n",
       "      <th>secretion</th>\n",
       "      <th>gid</th>\n",
       "      <th>knockout</th>\n",
       "      <th>rid</th>\n",
       "      <th>reaction</th>\n",
       "      <th>...</th>\n",
       "      <th>pmm</th>\n",
       "      <th>gene</th>\n",
       "      <th>product</th>\n",
       "      <th>status_de</th>\n",
       "      <th>sub</th>\n",
       "      <th>path</th>\n",
       "      <th>ecpath</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>{'PMM0951'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0951</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>PMM0951</td>\n",
       "      <td>carA</td>\n",
       "      <td>glutamine-hydrolyzing carbamoyl-phosphate synt...</td>\n",
       "      <td>down</td>\n",
       "      <td>09104 Nucleotide metabolism // 09105 Amino aci...</td>\n",
       "      <td>00240 Pyrimidine metabolism [PATH:ko00240] // ...</td>\n",
       "      <td>K01956  carA, CPA1; carbamoyl-phosphate syntha...</td>\n",
       "      <td>178.513724</td>\n",
       "      <td>-1.504660</td>\n",
       "      <td>1.544491e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>{'PMM0796'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0796</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>PMM0796</td>\n",
       "      <td>hisA</td>\n",
       "      <td>1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino...</td>\n",
       "      <td>no_significant</td>\n",
       "      <td>09105 Amino acid metabolism</td>\n",
       "      <td>00340 Histidine metabolism [PATH:ko00340]</td>\n",
       "      <td>K01814  hisA; phosphoribosylformimino-5-aminoi...</td>\n",
       "      <td>284.786091</td>\n",
       "      <td>0.285050</td>\n",
       "      <td>2.226182e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>{'PMM0483'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0483</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>PMM0483</td>\n",
       "      <td>hemL</td>\n",
       "      <td>glutamate-1-semialdehyde 2,1-aminomutase</td>\n",
       "      <td>no_significant</td>\n",
       "      <td>09108 Metabolism of cofactors and vitamins // ...</td>\n",
       "      <td>00860 Porphyrin metabolism [PATH:ko00860] // 0...</td>\n",
       "      <td>K01845  hemL; glutamate-1-semialdehyde 2,1-ami...</td>\n",
       "      <td>2468.751490</td>\n",
       "      <td>0.459796</td>\n",
       "      <td>1.200598e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>{'PMM0001'}</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>optimal</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0001</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>PMM0001</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>DNA polymerase III subunit beta</td>\n",
       "      <td>no_significant</td>\n",
       "      <td>09124 Replication and repair // 09182 Protein ...</td>\n",
       "      <td>03030 DNA replication [PATH:ko03030] // 03430 ...</td>\n",
       "      <td>K02338  dnaN; DNA polymerase III subunit beta ...</td>\n",
       "      <td>1548.906501</td>\n",
       "      <td>0.228545</td>\n",
       "      <td>4.619685e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>{'PMM1181'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM1181</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>PMM1181</td>\n",
       "      <td>aroH</td>\n",
       "      <td>chorismate mutase</td>\n",
       "      <td>down</td>\n",
       "      <td>09105 Amino acid metabolism</td>\n",
       "      <td>00400 Phenylalanine, tyrosine and tryptophan b...</td>\n",
       "      <td>K06208  aroH; chorismate mutase [EC:5.4.99.5]</td>\n",
       "      <td>195.838847</td>\n",
       "      <td>-1.247608</td>\n",
       "      <td>8.508785e-09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0          ids   growth      status         uptake  secretion  \\\n",
       "0           0  {'PMM0951'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "1           1  {'PMM0796'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "2           3  {'PMM0483'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "3           5  {'PMM0001'}  0.00001     optimal  L_GlutamateEX        NaN   \n",
       "4           7  {'PMM1181'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "\n",
       "       gid knockout  rid reaction  ...      pmm  gene  \\\n",
       "0  PMM0951     gene  NaN      NaN  ...  PMM0951  carA   \n",
       "1  PMM0796     gene  NaN      NaN  ...  PMM0796  hisA   \n",
       "2  PMM0483     gene  NaN      NaN  ...  PMM0483  hemL   \n",
       "3  PMM0001     gene  NaN      NaN  ...  PMM0001  dnaN   \n",
       "4  PMM1181     gene  NaN      NaN  ...  PMM1181  aroH   \n",
       "\n",
       "                                             product       status_de  \\\n",
       "0  glutamine-hydrolyzing carbamoyl-phosphate synt...            down   \n",
       "1  1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino...  no_significant   \n",
       "2           glutamate-1-semialdehyde 2,1-aminomutase  no_significant   \n",
       "3                    DNA polymerase III subunit beta  no_significant   \n",
       "4                                  chorismate mutase            down   \n",
       "\n",
       "                                                 sub  \\\n",
       "0  09104 Nucleotide metabolism // 09105 Amino aci...   \n",
       "1                        09105 Amino acid metabolism   \n",
       "2  09108 Metabolism of cofactors and vitamins // ...   \n",
       "3  09124 Replication and repair // 09182 Protein ...   \n",
       "4                        09105 Amino acid metabolism   \n",
       "\n",
       "                                                path  \\\n",
       "0  00240 Pyrimidine metabolism [PATH:ko00240] // ...   \n",
       "1          00340 Histidine metabolism [PATH:ko00340]   \n",
       "2  00860 Porphyrin metabolism [PATH:ko00860] // 0...   \n",
       "3  03030 DNA replication [PATH:ko03030] // 03430 ...   \n",
       "4  00400 Phenylalanine, tyrosine and tryptophan b...   \n",
       "\n",
       "                                              ecpath     baseMean  \\\n",
       "0  K01956  carA, CPA1; carbamoyl-phosphate syntha...   178.513724   \n",
       "1  K01814  hisA; phosphoribosylformimino-5-aminoi...   284.786091   \n",
       "2  K01845  hemL; glutamate-1-semialdehyde 2,1-ami...  2468.751490   \n",
       "3  K02338  dnaN; DNA polymerase III subunit beta ...  1548.906501   \n",
       "4      K06208  aroH; chorismate mutase [EC:5.4.99.5]   195.838847   \n",
       "\n",
       "   log2FoldChange          padj  \n",
       "0       -1.504660  1.544491e-08  \n",
       "1        0.285050  2.226182e-01  \n",
       "2        0.459796  1.200598e-01  \n",
       "3        0.228545  4.619685e-01  \n",
       "4       -1.247608  8.508785e-09  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Uptake visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "gpdf = gene_uptake_knockout_df.pivot_table(index='gene',columns='uptake', values='status', aggfunc='count', fill_value=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "rpdf = reaction_uptake_knockout_df.loc[\n",
    "    ~reaction_uptake_knockout_df.rid.isin(exchange_ids)\n",
    "].pivot_table(index='reaction',columns='uptake', values='status', aggfunc='count', fill_value=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16    289\n",
       "9      37\n",
       "1      31\n",
       "14     26\n",
       "15     17\n",
       "2       5\n",
       "13      3\n",
       "10      1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df.groupby('ids').size().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(4.944444444444445, 0.5, 'number of gene knockouts')"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeoAAAH9CAYAAADVvj8ZAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAA9hAAAPYQGoP6dpAABB/ElEQVR4nO3de3zP9f//8fvbZrOxLRt2YA5ljIxEOWYr54qiPkQ5RKWcmkN0NoqJTyiilJxKqm9UShgxaeUwKXLWML5baw6bsTbbXr8//Ly/3jbynvfb+7V2u14ur0u9nq/X+/l6vN7Dfa/j02IYhiEAAGBKZVxdAAAAuDKCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYJakmEYyszMFC9pAwCYjUuDes6cOWrYsKF8fX3l6+urFi1a6LvvvrMuNwxDMTExCgkJkZeXl6KiovT777/b9JGTk6Nhw4apUqVKKl++vLp27apjx47ZVceZM2fk5+enM2fOOGS/AABwFJcGdbVq1TR58mRt27ZN27Zt0z333KMHHnjAGsZTpkzRtGnTNGvWLG3dulVBQUFq3769TaBGR0dr+fLlWrp0qTZt2qSsrCzdf//9ys/Pd9VuAQDgMBazDcrh7++vqVOnasCAAQoJCVF0dLTGjh0r6cLRc2BgoN544w0NGjRIGRkZqly5shYvXqyePXtKkv73f/9XoaGhWrlypTp27HhN28zMzJSfn58yMjLk6+vrtH0DAMBeprlGnZ+fr6VLl+rs2bNq0aKFkpKSlJqaqg4dOljX8fT0VGRkpBISEiRJiYmJOn/+vM06ISEhatCggXWdouTk5CgzM9NmAgDAjFwe1Dt37lSFChXk6empp59+WsuXL1f9+vWVmpoqSQoMDLRZPzAw0LosNTVVHh4eqlix4hXXKUpsbKz8/PysU2hoqIP3CgAAx3B5UNetW1c7duzQzz//rGeeeUb9+vXT7t27rcstFovN+oZhFGq73D+t88ILLygjI8M6JScnX99OAADgJC4Pag8PD9WuXVtNmzZVbGysGjVqpLfeektBQUGSVOjIOC0tzXqUHRQUpNzcXJ06deqK6xTF09PTeqf5xQkAADNyeVBfzjAM5eTkqFatWgoKClJcXJx1WW5uruLj49WyZUtJUpMmTVS2bFmbdVJSUrRr1y7rOgAAlGTurtz4iy++qM6dOys0NFRnzpzR0qVLtWHDBq1atUoWi0XR0dGaNGmSwsLCFBYWpkmTJsnb21u9e/eWJPn5+WngwIEaNWqUAgIC5O/vr9GjRysiIkLt2rVz5a4BAOAQLg3qP//8U3369FFKSor8/PzUsGFDrVq1Su3bt5ckjRkzRtnZ2Ro8eLBOnTqlZs2aac2aNfLx8bH2MX36dLm7u6tHjx7Kzs5W27ZttWDBArm5ublqtwAAcBjTPUftCjxHDQAwK9NdowYAAP+HoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxFz6rm8AABzt6NGjSk9Pd0rflSpVUvXq1Z3S95UQ1ACAf42jR48qPLyesrPPOaV/Ly9v7d2754aGNUENAPjXSE9PV3b2OTUbME6+wTUd2ndmymFt/nC80tPTCWoAAK6Hb3BN+Vev6+oyHIKbyQAAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABNzaVDHxsbqjjvukI+Pj6pUqaIHH3xQ+/bts1mnf//+slgsNlPz5s1t1snJydGwYcNUqVIllS9fXl27dtWxY8du5K4AAOAULg3q+Ph4DRkyRD///LPi4uKUl5enDh066OzZszbrderUSSkpKdZp5cqVNsujo6O1fPlyLV26VJs2bVJWVpbuv/9+5efn38jdAQDA4dxdufFVq1bZzM+fP19VqlRRYmKi2rRpY2339PRUUFBQkX1kZGRo3rx5Wrx4sdq1aydJ+uijjxQaGqq1a9eqY8eOztsBAACczFTXqDMyMiRJ/v7+Nu0bNmxQlSpVVKdOHT355JNKS0uzLktMTNT58+fVoUMHa1tISIgaNGighISEIreTk5OjzMxMmwkAADMyTVAbhqGRI0eqdevWatCggbW9c+fO+vjjj/X999/rzTff1NatW3XPPfcoJydHkpSamioPDw9VrFjRpr/AwEClpqYWua3Y2Fj5+flZp9DQUOftGAAA18Glp74vNXToUP3222/atGmTTXvPnj2t/9+gQQM1bdpUNWrU0Lfffqvu3btfsT/DMGSxWIpc9sILL2jkyJHW+czMTMIaAGBKpjiiHjZsmL7++mutX79e1apVu+q6wcHBqlGjhg4cOCBJCgoKUm5urk6dOmWzXlpamgIDA4vsw9PTU76+vjYTAABm5NKgNgxDQ4cO1bJly/T999+rVq1a//iZEydOKDk5WcHBwZKkJk2aqGzZsoqLi7Ouk5KSol27dqlly5ZOqx0AgBvBpae+hwwZoiVLluirr76Sj4+P9Zqyn5+fvLy8lJWVpZiYGD300EMKDg7W4cOH9eKLL6pSpUrq1q2bdd2BAwdq1KhRCggIkL+/v0aPHq2IiAjrXeAAAJRULg3qOXPmSJKioqJs2ufPn6/+/fvLzc1NO3fu1KJFi3T69GkFBwfr7rvv1qeffiofHx/r+tOnT5e7u7t69Oih7OxstW3bVgsWLJCbm9uN3B0AABzOpUFtGMZVl3t5eWn16tX/2E+5cuU0c+ZMzZw501GlAQBgCqa4mQwAABSNoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABM7LqDOj8/Xzt27NCpU6ccUQ8AALiE3UEdHR2tefPmSboQ0pGRkbr99tsVGhqqDRs2OLo+AABKNbuD+n/+53/UqFEjSdKKFSuUlJSkvXv3Kjo6Wi+99JLDCwQAoDSzO6jT09MVFBQkSVq5cqX+85//qE6dOho4cKB27tzp8AIBACjN7A7qwMBA7d69W/n5+Vq1apXatWsnSTp37pzc3NwcXiAAAKWZu70fePzxx9WjRw8FBwfLYrGoffv2kqTNmzcrPDzc4QUCAFCa2R3UMTExatCggZKTk/Wf//xHnp6ekiQ3Nzc9//zzDi8QAIDSzO6gXrRokXr27GkN6It69eqlpUuXOqwwAABQjGvUjz/+uDIyMgq1nzlzRo8//rhDigIAABfYHdSGYchisRRqP3bsmPz8/BxSFAAAuOCaT303btxYFotFFotFbdu2lbv7/300Pz9fSUlJ6tSpk1OKBACgtLrmoH7wwQclSTt27FDHjh1VoUIF6zIPDw/VrFlTDz30kMMLBACgNLvmoB43bpwkqWbNmurZs6fKlSvntKIAAMAFdt/13a9fP2fUAQAAimB3UJcpU6bIm8kuys/Pv66CAADA/7E7qJctW2YT1OfPn9cvv/yihQsXavz48Q4tDgCA0s7ux7MefPBBPfDAA9bp4Ycf1sSJEzVlyhR9/fXXdvUVGxurO+64Qz4+PqpSpYoefPBB7du3z2YdwzAUExOjkJAQeXl5KSoqSr///rvNOjk5ORo2bJgqVaqk8uXLq2vXrjp27Ji9uwYAgOnYHdRX0qxZM61du9auz8THx2vIkCH6+eefFRcXp7y8PHXo0EFnz561rjNlyhRNmzZNs2bN0tatWxUUFKT27dvrzJkz1nWio6O1fPlyLV26VJs2bVJWVpbuv/9+TsMDAEo8u099FyU7O1szZ85UtWrV7PrcqlWrbObnz5+vKlWqKDExUW3atJFhGJoxY4Zeeuklde/eXZK0cOFCBQYGasmSJRo0aJAyMjI0b948LV682DqS10cffaTQ0FCtXbtWHTt2dMQuAgDgEnYHdcWKFW2uURuGoTNnzsjb21sfffTRdRVz8dWk/v7+kqSkpCSlpqaqQ4cO1nU8PT0VGRmphIQEDRo0SImJiTp//rzNOiEhIWrQoIESEhKKDOqcnBzl5ORY5zMzM6+rbgAAnMXuoJ4xY4bNfJkyZVS5cmU1a9ZMFStWLHYhhmFo5MiRat26tRo0aCBJSk1NlXRhDOxLBQYG6siRI9Z1PDw8Cm07MDDQ+vnLxcbGcuMbAKBEMM1z1EOHDtVvv/2mTZs2FVp2+eNgV3rf+LWu88ILL2jkyJHW+czMTIWGhhajagAAnKtY16hPnz6tefPmac+ePbJYLKpfv74GDBhQ7EE5hg0bpq+//lobN260uc4dFBQk6cJRc3BwsLU9LS3NepQdFBSk3NxcnTp1yuaoOi0tTS1btixye56enoWG6QQAwIzsvut727ZtuuWWWzR9+nSdPHlS6enpmjZtmm655RZt377drr4Mw9DQoUO1bNkyff/996pVq5bN8lq1aikoKEhxcXHWttzcXMXHx1tDuEmTJipbtqzNOikpKdq1a9cVgxoAgJLC7iPqESNGqGvXrnr//fetI2jl5eXpiSeeUHR0tDZu3HjNfQ0ZMkRLlizRV199JR8fH+s1ZT8/P3l5eclisSg6OlqTJk1SWFiYwsLCNGnSJHl7e6t3797WdQcOHKhRo0YpICBA/v7+Gj16tCIiIqx3gQMAUFLZHdTbtm2zCWlJcnd315gxY9S0aVO7+pozZ44kKSoqyqZ9/vz56t+/vyRpzJgxys7O1uDBg3Xq1Ck1a9ZMa9askY+Pj3X96dOny93dXT169FB2drbatm2rBQsWyM3Nzd7dAwDAVOwOal9fXx09elTh4eE27cnJyTbheS0Mw/jHdSwWi2JiYhQTE3PFdcqVK6eZM2dq5syZdm0fAACzs/sadc+ePTVw4EB9+umnSk5O1rFjx7R06VI98cQT6tWrlzNqBACg1LL7iPq///2vLBaL+vbtq7y8PElS2bJl9cwzz2jy5MkOLxAAgNKsWI9nvfXWW4qNjdWhQ4dkGIZq164tb29vpaen89gTAAAOZPep7x49eqigoEDe3t6KiIhQw4YN5e3trT///LPQTWEAAOD62B3UKSkpGjhwYKG2qKioQjeYAQCA62N3UK9cuVJbtmzRiBEjJEnHjx9XVFSUIiIi9Nlnnzm8QAAASjO7r1EHBARo9erVat26tSTp22+/1e23366PP/5YZco4bHhrAACgYt5MVq1aNcXFxal169Zq3769Fi9e/I+DZAAAAPtdU1BfPgb1RefOndOKFSsUEBBgbTt58qTjqgMAoJS7pqC+fAxqAABwY1xTUDtrDGoAAHB1xbrre/Xq1YXa16xZo++++84hRQEAgAvsDurnn39e+fn5hdoLCgr0/PPPO6QoAABwgd1BfeDAAdWvX79Qe3h4uA4ePOiQogAAwAV2B7Wfn5/++OOPQu0HDx5U+fLlHVIUAAC4wO6g7tq1q6Kjo3Xo0CFr28GDBzVq1Ch17drVocUBAFDa2R3UU6dOVfny5RUeHq5atWqpVq1aqlevngICAvTf//7XGTUCAFBq2f1mMj8/PyUkJCguLk6//vqrvLy81LBhQ7Vp08YZ9QEAUKoV6xWiFotFHTp0UIcOHRxdDwAAuESxRtGIj49Xly5dVLt2bYWFhalr16764YcfHF0bAAClnt1B/dFHH6ldu3by9vbW8OHDNXToUHl5ealt27ZasmSJM2oEAKDUsvvU98SJEzVlyhTreNSS9Oyzz2ratGl67bXX1Lt3b4cWCABAaWb3EfUff/yhLl26FGrv2rWrkpKSHFIUAAC4wO6gDg0N1bp16wq1r1u3TqGhoQ4pCgAAXGD3qe9Ro0Zp+PDh2rFjh1q2bCmLxaJNmzZpwYIFeuutt5xRIwAApZbdQf3MM88oKChIb775pj777DNJUr169fTpp5/qgQcecHiBAACUZsV6jrpbt27q1q2bo2sBAACXKVZQS1Jubq7S0tJUUFBg0169evXrLgoAAFxgd1AfOHBAAwYMUEJCgk27YRiyWCxFjlUNAACKx+6g7t+/v9zd3fXNN98oODhYFovFGXUBAAAVI6h37NihxMREhYeHO6MeAABwCbufo65fv77S09OdUQsAALiM3UH9xhtvaMyYMdqwYYNOnDihzMxMmwkAADiO3ae+27VrJ0lq27atTTs3kwEA4Hh2B/X69eudUQcAACiC3UHdokULeXh4FLmMa9cAADiW3deoe/ToUeglJ5L0559/KioqyhE1AQCA/8/uoE5JSdHAgQMLtUVFRfHIFgAADmZ3UK9cuVJbtmzRiBEjJEnHjx9XVFSUIiIirIN0AAAAx7D7GnVAQIBWr16t1q1bS5K+/fZb3X777fr4449VpozduQ8AAK6iWINyVKtWTXFxcWrdurXat2+vxYsX8ypRAACc4JqCumLFikUG8blz57RixQoFBARY206ePOm46gAAKOWuKahnzJjh5DIAAEBRrimo+/Xr5+w6AABAEbj7CwAAEyOoAQAwMYIaAAATI6gBADCxYgf1wYMHtXr1amVnZ0u6MMwlAABwLLuD+sSJE2rXrp3q1Kmje++9VykpKZKkJ554QqNGjXJ4gQAAlGZ2B/WIESPk7u6uo0ePytvb29res2dPrVq1yqHFAQBQ2tn9CtE1a9Zo9erVqlatmk17WFiYjhw54rDCAABAMY6oz549a3MkfVF6ero8PT0dUhQAALjA7qBu06aNFi1aZJ23WCwqKCjQ1KlTdffddzu0OAAASju7T31PnTpVUVFR2rZtm3JzczVmzBj9/vvvOnnypH788Udn1AgAQKll9xF1/fr19dtvv+nOO+9U+/btdfbsWXXv3l2//PKLbrnlFmfUCABAqVWs8aiDgoI0fvx4R9cCAAAuU6ygPn36tLZs2aK0tDQVFBTYLOvbt69DCgMAAMUI6hUrVujRRx/V2bNn5ePjI4vFYl1msVgIagAAHMjua9SjRo3SgAEDdObMGZ0+fVqnTp2yTidPnnRGjQAAlFp2B/Xx48c1fPjwIp+lBgAAjmV3UHfs2FHbtm1zRi0AAOAydgf1fffdp+eee04xMTH64osv9PXXX9tM9ti4caO6dOmikJAQWSwWffnllzbL+/fvL4vFYjM1b97cZp2cnBwNGzZMlSpVUvny5dW1a1cdO3bM3t0CAMCU7L6Z7Mknn5QkTZgwodAyi8Wi/Pz8a+7r7NmzatSokR5//HE99NBDRa7TqVMnzZ8/3zrv4eFhszw6OlorVqzQ0qVLFRAQoFGjRun+++9XYmKi3NzcrrkWAADMyO6gvvxxrOvRuXNnde7c+arreHp6KigoqMhlGRkZmjdvnhYvXqx27dpJkj766COFhoZq7dq16tixo8NqBQDAFew+9X2pv//+21F1XNGGDRtUpUoV1alTR08++aTS0tKsyxITE3X+/Hl16NDB2hYSEqIGDRooISHhin3m5OQoMzPTZgIAwIzsDur8/Hy99tprqlq1qipUqKA//vhDkvTKK69o3rx5Di2uc+fO+vjjj/X999/rzTff1NatW3XPPfcoJydHkpSamioPDw9VrFjR5nOBgYFKTU29Yr+xsbHy8/OzTqGhoQ6tGwAAR7E7qCdOnKgFCxZoypQpNteLIyIi9MEHHzi0uJ49e+q+++5TgwYN1KVLF3333Xfav3+/vv3226t+zjAMmxexXO6FF15QRkaGdUpOTnZo3QAAOIrdQb1o0SLNnTtXjz76qM3NWg0bNtTevXsdWtzlgoODVaNGDR04cEDShXeO5+bm6tSpUzbrpaWlKTAw8Ir9eHp6ytfX12YCAMCMivXCk9q1axdqLygo0Pnz5x1S1JWcOHFCycnJCg4OliQ1adJEZcuWVVxcnHWdlJQU7dq1Sy1btnRqLQAA3Ah23/V966236ocfflCNGjVs2j///HM1btzYrr6ysrJ08OBB63xSUpJ27Nghf39/+fv7KyYmRg899JCCg4N1+PBhvfjii6pUqZK6desmSfLz89PAgQM1atQoBQQEyN/fX6NHj1ZERIT1LnAAAEoyu4N63Lhx6tOnj44fP66CggItW7ZM+/bt06JFi/TNN9/Y1de2bdt09913W+dHjhwpSerXr5/mzJmjnTt3atGiRTp9+rSCg4N1991369NPP5WPj4/1M9OnT5e7u7t69Oih7OxstW3bVgsWLOAZagDAv4LdQd2lSxd9+umnmjRpkiwWi1599VXdfvvtWrFihdq3b29XX1FRUTIM44rLV69e/Y99lCtXTjNnztTMmTPt2jYAACVBscaj7tixIy8TAQDgBriuF54AAADnsvuIumLFikU+o2yxWFSuXDnVrl1b/fv31+OPP+6QAgEAKM3sDupXX31VEydOVOfOnXXnnXfKMAxt3bpVq1at0pAhQ5SUlKRnnnlGeXl51gE8AABA8dgd1Js2bdLrr7+up59+2qb9vffe05o1a/TFF1+oYcOGevvttwlqAACuk93XqFevXl3kM8pt27a13qV97733Wt8BDgAAis/uoPb399eKFSsKta9YsUL+/v6SLowzfemzzgAAoHjsPvX9yiuv6JlnntH69et15513ymKxaMuWLVq5cqXeffddSVJcXJwiIyMdXiwAAKWN3UH95JNPqn79+po1a5aWLVsmwzAUHh6u+Ph46/u1R40a5fBCAQAojYr1wpNWrVqpVatWjq4FAABchheeAABgYgQ1AAAmRlADAGBi1xTUv/32mwoKCpxdCwAAuMw1BXXjxo2Vnp4uSbr55pt14sQJpxYFAAAuuKagvummm5SUlCRJOnz4MEfXAADcINf0eNZDDz2kyMhIBQcHy2KxqGnTpnJzcytyXV4dCgCA41xTUM+dO1fdu3fXwYMHNXz4cD355JO8IhQAgBvgml940qlTJ0lSYmKinn32WYIaAIAbwO43k82fP9/6/8eOHZPFYlHVqlUdWhQAALjA7ueoCwoKNGHCBPn5+alGjRqqXr26brrpJr322mvcZAYAgIPZfUT90ksvad68eZo8ebJatWolwzD0448/KiYmRn///bcmTpzojDoBACiV7A7qhQsX6oMPPlDXrl2tbY0aNVLVqlU1ePBgghoAAAey+9T3yZMnFR4eXqg9PDxcJ0+edEhRAADgAruDulGjRpo1a1ah9lmzZqlRo0YOKQoAAFxg96nvKVOm6L777tPatWvVokULWSwWJSQkKDk5WStXrnRGjQAAlFp2H1FHRkZq//796tatm06fPq2TJ0+qe/fu2rdvn+666y5n1AgAQKll9xG1JIWEhHDTGAAANwDjUQMAYGIENQAAJkZQAwBgYnYFtWEYOnLkiLKzs51VDwAAuITdQR0WFqZjx445qx4AAHAJu4K6TJkyCgsL04kTJ5xVDwAAuITd16inTJmi5557Trt27XJGPQAA4BJ2P0f92GOP6dy5c2rUqJE8PDzk5eVls5z3fQMA4Dh2B/WMGTOcUAYAACiK3UHdr18/Z9QBAACKUKznqA8dOqSXX35ZvXr1UlpamiRp1apV+v333x1aHAAApZ3dQR0fH6+IiAht3rxZy5YtU1ZWliTpt99+07hx4xxeIAAApZndQf3888/r9ddfV1xcnDw8PKztd999t3766SeHFgcAQGlnd1Dv3LlT3bp1K9ReuXJlnq8GAMDB7A7qm266SSkpKYXaf/nlF1WtWtUhRQEAgAvsDurevXtr7NixSk1NlcViUUFBgX788UeNHj1affv2dUaNAACUWnYH9cSJE1W9enVVrVpVWVlZql+/vtq0aaOWLVvq5ZdfdkaNAACUWnY/R122bFl9/PHHmjBhgn755RcVFBSocePGCgsLc0Z9AACUanYH9UW33HKLbr75ZkmSxWJxWEEAAOD/FOuFJ/PmzVODBg1Urlw5lStXTg0aNNAHH3zg6NoAACj17D6ifuWVVzR9+nQNGzZMLVq0kCT99NNPGjFihA4fPqzXX3/d4UUCAFBa2R3Uc+bM0fvvv69evXpZ27p27aqGDRtq2LBhBDUAAA5k96nv/Px8NW3atFB7kyZNlJeX55CiAADABXYH9WOPPaY5c+YUap87d64effRRhxQFAAAuuKZT3yNHjrT+v8Vi0QcffKA1a9aoefPmkqSff/5ZycnJvPAEAAAHu6ag/uWXX2zmmzRpIunCcJfShfd8V65cmWEuAQBwsGsK6vXr1zu7DgAAUIRiPUcNAABuDLsfz/r77781c+ZMrV+/XmlpaSooKLBZvn37docVBwBAaWd3UA8YMEBxcXF6+OGHdeedd/L6UAAAnMjuoP7222+1cuVKtWrVyhn1AACAS9h9jbpq1ary8fFxRi0AAOAydgf1m2++qbFjx+rIkSPOqAcAAFzC7lPfTZs21d9//62bb75Z3t7eKlu2rM3ykydPOqw4AABKO7uDulevXjp+/LgmTZqkwMBAbiYDAMCJ7A7qhIQE/fTTT2rUqNF1b3zjxo2aOnWqEhMTlZKSouXLl+vBBx+0LjcMQ+PHj9fcuXN16tQpNWvWTO+8845uvfVW6zo5OTkaPXq0PvnkE2VnZ6tt27aaPXu2qlWrdt31AQDganZfow4PD1d2drZDNn727Fk1atRIs2bNKnL5lClTNG3aNM2aNUtbt25VUFCQ2rdvrzNnzljXiY6O1vLly7V06VJt2rRJWVlZuv/++5Wfn++QGgEAcCW7j6gnT56sUaNGaeLEiYqIiCh0jdrX1/ea++rcubM6d+5c5DLDMDRjxgy99NJL6t69uyRp4cKFCgwM1JIlSzRo0CBlZGRo3rx5Wrx4sdq1aydJ+uijjxQaGqq1a9eqY8eO9u4eAACmYndQd+rUSZLUtm1bm3bDMGSxWBx2JJuUlKTU1FR16NDB2ubp6anIyEglJCRo0KBBSkxM1Pnz523WCQkJUYMGDZSQkHDFoM7JyVFOTo51PjMz0yE1AwDgaHYH9Y0aoCM1NVWSFBgYaNMeGBhofTQsNTVVHh4eqlixYqF1Ln6+KLGxsRo/fryDKwYAwPHsDurIyEhn1HFFl99VfvHI/Wr+aZ0XXnjBZoztzMxMhYaGXl+hAAA4gd1BvXHjxqsub9OmTbGLuVRQUJCkC0fNwcHB1va0tDTrUXZQUJByc3N16tQpm6PqtLQ0tWzZ8op9e3p6ytPT0yF1AgDgTHYHdVRUVKG2S49eHXWNulatWgoKClJcXJwaN24sScrNzVV8fLzeeOMNSVKTJk1UtmxZxcXFqUePHpKklJQU7dq1S1OmTHFIHQAAuJLdQX3q1Cmb+fPnz+uXX37RK6+8ookTJ9rVV1ZWlg4ePGidT0pK0o4dO+Tv76/q1asrOjpakyZNUlhYmMLCwjRp0iR5e3urd+/ekiQ/Pz8NHDhQo0aNUkBAgPz9/TV69GhFRERY7wIHAKAkszuo/fz8CrW1b99enp6eGjFihBITE6+5r23btunuu++2zl+8btyvXz8tWLBAY8aMUXZ2tgYPHmx94cmaNWtsBgWZPn263N3d1aNHD+sLTxYsWCA3Nzd7dw0AANOxO6ivpHLlytq3b59dn4mKipJhGFdcbrFYFBMTo5iYmCuuU65cOc2cOVMzZ860a9sAAJQEdgf1b7/9ZjNvGIZSUlI0efJkh7xWFAAA/B+7g/q2226TxWIpdCTcvHlzffjhhw4rDAAAFCOok5KSbObLlCmjypUrq1y5cg4rCgAAXGB3UNeoUcMZdQAAgCIU62aydevWad26dUpLS1NBQYHNMk5/AwDgOHYH9fjx4zVhwgQ1bdpUwcHB//g6TwAAUHx2B/W7776rBQsWqE+fPs6oBwAAXKKMvR/Izc296nu0AQCA49gd1E888YSWLFnijFoAAMBl7D71/ffff2vu3Llau3atGjZsqLJly9osnzZtmsOKAwCgtCvWm8luu+02SdKuXbtslnFjGQAAjmV3UK9fv94ZdQAAgCLYfY0aAADcOAQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmZuqgjomJkcVisZmCgoKsyw3DUExMjEJCQuTl5aWoqCj9/vvvLqwYAADHMnVQS9Ktt96qlJQU67Rz507rsilTpmjatGmaNWuWtm7dqqCgILVv315nzpxxYcUAADiO6YPa3d1dQUFB1qly5cqSLhxNz5gxQy+99JK6d++uBg0aaOHChTp37pyWLFni4qoBAHAM0wf1gQMHFBISolq1aumRRx7RH3/8IUlKSkpSamqqOnToYF3X09NTkZGRSkhIuGqfOTk5yszMtJkAADAjUwd1s2bNtGjRIq1evVrvv/++UlNT1bJlS504cUKpqamSpMDAQJvPBAYGWpddSWxsrPz8/KxTaGio0/YBAIDrYeqg7ty5sx566CFFRESoXbt2+vbbbyVJCxcutK5jsVhsPmMYRqG2y73wwgvKyMiwTsnJyY4vHgAABzB1UF+ufPnyioiI0IEDB6x3f19+9JyWllboKPtynp6e8vX1tZkAADCjEhXUOTk52rNnj4KDg1WrVi0FBQUpLi7Oujw3N1fx8fFq2bKlC6sEAMBx3F1dwNWMHj1aXbp0UfXq1ZWWlqbXX39dmZmZ6tevnywWi6KjozVp0iSFhYUpLCxMkyZNkre3t3r37u3Suo8ePar09HSn9F2pUiVVr17dKX0DAMzH1EF97Ngx9erVS+np6apcubKaN2+un3/+WTVq1JAkjRkzRtnZ2Ro8eLBOnTqlZs2aac2aNfLx8XFZzUePHlV4eD1lZ59zSv9eXt7au3cPYQ0ApYSpg3rp0qVXXW6xWBQTE6OYmJgbU9A1SE9PV3b2OTUbME6+wTUd2ndmymFt/nC80tPTCWoAKCVMHdQlmW9wTflXr+vqMgAAJVyJupkMAIDShqAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMR4hSiA68JocYBzEdQAio3R4gDnI6gBFBujxQHOR1ADuG6MFgc4DzeTAQBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYgQ1AAAmRlADAGBiBDUAACZGUAMAYGIENQAAJkZQAwBgYoyeBQAo5OjRo0pPT3dK35UqVWLoUjsQ1AAAG0ePHlV4eD1lZ59zSv9eXt7au3cPYX2NCGoAgI309HRlZ59TswHj5Btc06F9Z6Yc1uYPxys9PZ2gvkYENQCgSL7BNeVfva6ryyj1uJkMAAATI6gBADAxghoAABMjqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABPjzWQAgBtuz549JapfVyKoAQA3THbGCUkWPfbYY07dzvmcXKf2fyMR1ACAG+b8uTOSDN3We6wq1wp3eP8pO3/Srq/nKi8vz+F9uwpBDQC44SpUqe6UAT8yUw47vE9X42YyAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjKAGAMDECGoAAEyMoAYAwMQIagAATIygBgDAxAhqAABMjNGzSiBnDoxeqVIlVa9e3Wn9AwDsQ1CXIDdiwHUvL2/t3buHsAYAkyCoSxBnD7iemXJYmz8cr/T0dIIaAEyCoC6BnDXgOoCS4+jRo0pPT3dK3868vAb7EdQAUMIcPXpU4eH1lJ19zqnbOZ+T69T+cW3+NUE9e/ZsTZ06VSkpKbr11ls1Y8YM3XXXXa4uCwAcLj09XdnZ59RswDj5Btd0eP8pO3/Srq/nKi8vz+F9w37/iqD+9NNPFR0drdmzZ6tVq1Z677331LlzZ+3evZtrrUAJx1MOV+YbXNMpl8EyUw47vE8U378iqKdNm6aBAwfqiSeekCTNmDFDq1ev1pw5cxQbG+vi6gAUB085ABeU+KDOzc1VYmKinn/+eZv2Dh06KCEhocjP5OTkKCcnxzqfkZEhScrMzLzuerKysiRJJ4/sU15O9nX3d6nMlCOSpIzjB1TW3eLQviUpM/WoJCkxMdG6H45UpkwZFRQUOLzfG9E/tRdt3759kpzz5/3EoV2SDN0c9R/5BVZzaN+SdO7kn9oXt0SrV69W3brOuTnTWd+9M793ybn/1jj93zFn1v7//43MyspySF5Iko+PjyyWf6jTKOGOHz9uSDJ+/PFHm/aJEycaderUKfIz48aNMyQxMTExMTG5dMrIyPjHnCvxR9QXXf4biWEYV/wt5YUXXtDIkSOt8wUFBTp58qQCAgL++TcbE8jMzFRoaKiSk5Pl6+vr6nKKjf0wl3/Lfkj/nn1hP8zFGfvh4+Pzj+uU+KCuVKmS3NzclJqaatOelpamwMDAIj/j6ekpT09Pm7abbrrJWSU6ja+vb4n+Q38R+2Eu/5b9kP49+8J+mMuN3o8SPyiHh4eHmjRpori4OJv2uLg4tWzZ0kVVAQDgGCX+iFqSRo4cqT59+qhp06Zq0aKF5s6dq6NHj+rpp592dWkAAFyXf0VQ9+zZUydOnNCECROUkpKiBg0aaOXKlapRo4arS3MKT09PjRs3rtDp+5KG/TCXf8t+SP+efWE/zMVV+2ExDMO4oVsEAADXrMRfowYA4N+MoAYAwMQIagAATIygBgDAxAjqEiI2NlZ33HGHfHx8VKVKFT344IPW9/2WZLGxsbJYLIqOjnZ1KcVy/PhxPfbYYwoICJC3t7duu+02JSYmurosu+Tl5enll19WrVq15OXlpZtvvlkTJkxw6rvNHWHjxo3q0qWLQkJCZLFY9OWXX9osNwxDMTExCgkJkZeXl6KiovT777+7ptiruNp+nD9/XmPHjlVERITKly+vkJAQ9e3bV//7v//ruoKv4p9+JpcaNGiQLBaLZsyYccPqu1bXsh979uxR165d5efnJx8fHzVv3lxHjx51Sj0EdQkRHx+vIUOG6Oeff1ZcXJzy8vLUoUMHnT171tWlFdvWrVs1d+5cNWzY0NWlFMupU6fUqlUrlS1bVt999512796tN998s8S95e6NN97Qu+++q1mzZmnPnj2aMmWKpk6dqpkzZ7q6tKs6e/asGjVqpFmzZhW5fMqUKZo2bZpmzZqlrVu3KigoSO3bt9eZM2ducKVXd7X9OHfunLZv365XXnlF27dv17Jly7R//3517drVBZX+s3/6mVz05ZdfavPmzQoJCblBldnnn/bj0KFDat26tcLDw7Vhwwb9+uuveuWVV1SuXDnnFHTdo2LAJdLS0gxJRnx8vKtLKZYzZ84YYWFhRlxcnBEZGWk8++yzri7JbmPHjjVat27t6jKu23333WcMGDDApq179+7GY4895qKK7CfJWL58uXW+oKDACAoKMiZPnmxt+/vvvw0/Pz/j3XffdUGF1+by/SjKli1bDEnGkSNHbkxRxXSlfTl27JhRtWpVY9euXUaNGjWM6dOn3/Da7FHUfvTs2fOG/v3giLqEujg0p7+/v4srKZ4hQ4bovvvuU7t27VxdSrF9/fXXatq0qf7zn/+oSpUqaty4sd5//31Xl2W31q1ba926ddq/f78k6ddff9WmTZt07733uriy4ktKSlJqaqo6dOhgbfP09FRkZOQVh78tKTIyMmSxWErcmRvpwgBIffr00XPPPadbb73V1eUUS0FBgb799lvVqVNHHTt2VJUqVdSsWbOrnua/XgR1CWQYhkaOHKnWrVurQYMGri7HbkuXLtX27dsVGxvr6lKuyx9//KE5c+YoLCxMq1ev1tNPP63hw4dr0aJFri7NLmPHjlWvXr0UHh6usmXLqnHjxoqOjlavXr1cXVqxXRyk5/KBeQIDAwsN4FOS/P3333r++efVu3fvEjm4xRtvvCF3d3cNHz7c1aUUW1pamrKysjR58mR16tRJa9asUbdu3dS9e3fFx8c7ZZv/ileIljZDhw7Vb7/9pk2bNrm6FLslJyfr2Wef1Zo1a5x3PecGKSgoUNOmTTVp0iRJUuPGjfX7779rzpw56tu3r4uru3affvqpPvroIy1ZskS33nqrduzYoejoaIWEhKhfv36uLu+62DP8rdmdP39ejzzyiAoKCjR79mxXl2O3xMREvfXWW9q+fXuJ/RlIst5k+cADD2jEiBGSpNtuu00JCQl69913FRkZ6fBtckRdwgwbNkxff/211q9fr2rVqrm6HLslJiYqLS1NTZo0kbu7u9zd3RUfH6+3335b7u7uys/Pd3WJ1yw4OFj169e3aatXr57T7vx0lueee07PP/+8HnnkEUVERKhPnz4aMWJEiT7jERQUJEl2DX9rZufPn1ePHj2UlJSkuLi4Enk0/cMPPygtLU3Vq1e3/t0/cuSIRo0apZo1a7q6vGtWqVIlubu739C/+xxRlxCGYWjYsGFavny5NmzYoFq1arm6pGJp27atdu7cadP2+OOPKzw8XGPHjpWbm5uLKrNfq1atCj0it3///hI3GMy5c+dUpozt7+xubm6mfzzramrVqqWgoCDFxcWpcePGkqTc3FzFx8frjTfecHF19rkY0gcOHND69esVEBDg6pKKpU+fPoXuSenYsaP69Omjxx9/3EVV2c/Dw0N33HHHDf27T1CXEEOGDNGSJUv01VdfycfHx3qk4OfnJy8vLxdXd+18fHwKXVcvX768AgICStz19hEjRqhly5aaNGmSevTooS1btmju3LmaO3euq0uzS5cuXTRx4kRVr15dt956q3755RdNmzZNAwYMcHVpV5WVlaWDBw9a55OSkrRjxw75+/urevXqio6O1qRJkxQWFqawsDBNmjRJ3t7e6t27twurLuxq+xESEqKHH35Y27dv1zfffKP8/Hzr331/f395eHi4quwi/dPP5PJfMsqWLaugoCDVrVv3Rpd6Vf+0H88995x69uypNm3a6O6779aqVau0YsUKbdiwwTkF3bD7y3FdJBU5zZ8/39WlXbeS+niWYRjGihUrjAYNGhienp5GeHi4MXfuXFeXZLfMzEzj2WefNapXr26UK1fOuPnmm42XXnrJyMnJcXVpV7V+/foi/07069fPMIwLj2iNGzfOCAoKMjw9PY02bdoYO3fudG3RRbjafiQlJV3x7/769etdXXoh//QzuZxZH8+6lv2YN2+eUbt2baNcuXJGo0aNjC+//NJp9TDMJQAAJsbNZAAAmBhBDQCAiRHUAACYGEENAICJEdQAAJgYQQ0AgIkR1AAAmBhBDQCAiRHUwBVERUUpOjra1WVYGYahp556Sv7+/rJYLNqxY4erSyqkf//+evDBB11dhlNZLJbrHnv48u/JbH/WYC4ENVBCrFq1SgsWLNA333yjlJQUp74b/d8eHIcPHzbVLzvLli3Ta6+9Zp2vWbOmZsyY4bqCYCoMygHcQPn5+bJYLIVGq7oWhw4dUnBwsFq2bOmEyuBK/v7+ri4BJsYRNUwtKipKw4cP15gxY+Tv76+goCDFxMRYlxd1ZHT69GlZLBbrSDYbNmyQxWLR6tWr1bhxY3l5eemee+5RWlqavvvuO9WrV0++vr7q1auXzp07Z7P9vLw8DR06VDfddJMCAgL08ssv69LX4+fm5mrMmDGqWrWqypcvr2bNmtmMoLNgwQLddNNN+uabb1S/fn15enrqyJEjRe5rfHy87rzzTnl6eio4OFjPP/+88vLyJF04VTps2DAdPXpUFovliuP3xsTE6LbbbrNpmzFjhs36F0+7jh8/XlWqVJGvr68GDRqk3Nxc6/L4+Hi99dZbslgsslgsOnz4sPLz8zVw4EDVqlVLXl5eqlu3rt56660i67goMTFRVapU0cSJEyVJGRkZeuqpp6zbveeee/Trr79e8fMXf76fffaZ7rrrLnl5eemOO+7Q/v37tXXrVjVt2lQVKlRQp06d9Ndff9l8dv78+apXr57KlSun8PBwzZ4927rs4jCxjRs3lsViUVRUlCRp69atat++vSpVqiQ/Pz9FRkZq+/bthepKSUlR586d5eXlpVq1aunzzz+3Wb5z507dc8898vLyUkBAgJ566illZWVdcT8vPYMRFRWlI0eOaMSIEdbv/6KEhAS1adNGXl5eCg0N1fDhw3X27Fnr8tmzZyssLEzlypVTYGCgHn744StuEyWI04b7ABwgMjLS8PX1NWJiYoz9+/cbCxcuNCwWi7FmzRrDMAzr6EK//PKL9TOnTp2yGV3o4kg4zZs3NzZt2mRs377dqF27thEZGWl06NDB2L59u7Fx40YjICDAmDx5ss22K1SoYDz77LPG3r17jY8++sjw9va2GSGrd+/eRsuWLY2NGzcaBw8eNKZOnWp4enoa+/fvNwzDMObPn2+ULVvWaNmypfHjjz8ae/fuNbKysgrt57Fjxwxvb29j8ODBxp49e4zly5cblSpVMsaNG2cYhmGcPn3amDBhglGtWjUjJSXFSEtLK/L7GjdunNGoUSObtunTpxs1atSwzvfr18+oUKGC0bNnT2PXrl3GN998Y1SuXNl48cUXrdtq0aKF8eSTTxopKSlGSkqKkZeXZ+Tm5hqvvvqqsWXLFuOPP/6wfh+ffvqpTd8PPPCA9Xv38/MzZs+ebRjGhdGsWrVqZXTp0sXYunWrsX//fmPUqFFGQECAceLEiSL35+LPNzw83Fi1apWxe/duo3nz5sbtt99uREVF2fw8n376aevn5s6dawQHBxtffPGF8ccffxhffPGF4e/vbyxYsMAwDMPYsmWLIclYu3atkZKSYt3+unXrjMWLFxu7d+82du/ebQwcONAIDAw0MjMzrX1LMgICAoz333/f2Ldvn/Hyyy8bbm5uxu7duw3DMIyzZ88aISEhRvfu3Y2dO3ca69atM2rVqmUz8tKl35Nh2I4gd+LECaNatWrGhAkTrN+/YRjGb7/9ZlSoUMGYPn26sX//fuPHH380GjdubPTv398wDMPYunWr4ebmZixZssQ4fPiwsX37duOtt94q8ntFyUJQw9QiIyON1q1b27TdcccdxtixYw3DsC+o165da10nNjbWkGQcOnTI2jZo0CCjY8eONtuuV6+eUVBQYG0bO3asUa9ePcMwDOPgwYOGxWIxjh8/blNf27ZtjRdeeMEwjAtBLcnYsWPHVffzxRdfNOrWrWuzrXfeeceoUKGCkZ+fbxhG4cAtyrUGtb+/v3H27Flr25w5c2y2da1Djw4ePNh46KGHbPp+4IEHjC+//NLw8fExlixZYl22bt06w9fX1/j7779t+rjllluM9957r8j+L/58P/jgA2vbJ598Ykgy1q1bZ22LjY016tata50PDQ212bZhGMZrr71mtGjRwqbfS//cFCUvL8/w8fExVqxYYW2TZPNLgWEYRrNmzYxnnnnGMIwLvyRUrFjR5heyb7/91ihTpoyRmppqGMbVg9owih7+sU+fPsZTTz1l0/bDDz8YZcqUMbKzs40vvvjC8PX1tfmlAv8OXKOG6TVs2NBmPjg4WGlpadfVT2BgoLy9vXXzzTfbtG3ZssXmM82bN7c59diiRQu9+eabys/P1/bt22UYhurUqWPzmZycHAUEBFjnPTw8Cu3D5fbs2aMWLVrYbKtVq1bKysrSsWPHVL16dft29h80atRI3t7e1vkWLVooKytLycnJqlGjxhU/9+677+qDDz7QkSNHlJ2drdzc3EKn2jdv3qxvvvlGn3/+ubp162ZtT0xMVFZWls13I0nZ2dk6dOjQVeu9/GcnSRERETZtF/9M/PXXX0pOTtbAgQP15JNPWtfJy8uTn5/fVbeTlpamV199Vd9//73+/PNP5efn69y5czp69KjNei1atCg0f/Hyy549e9SoUSOVL1/eurxVq1YqKCjQvn37rPXbKzExUQcPHtTHH39sbTMMQwUFBUpKSlL79u1Vo0YN3XzzzerUqZM6deqkbt262fycUTIR1DC9smXL2sxbLBYVFBRIkvWmLOOS68bnz5//x34sFstV+70WBQUFcnNzU2Jiotzc3GyWVahQwfr/Xl5eNgFcFMMwCq1zcZ/+6bOXKlOmjM13IV35+yjK1bb12WefacSIEXrzzTfVokUL+fj4aOrUqdq8ebPNerfccosCAgL04Ycf6r777pOHh4ekC99XcHCwzTX8i2666aar1nX5z66otos/u4v/ff/999WsWTObfi7/OV2uf//++uuvvzRjxgzVqFFDnp6eatGihfX6/dVcrKuon+Xl6xRHQUGBBg0apOHDhxdaVr16dXl4eGj79u3asGGD1qxZo1dffVUxMTHaunXrP36/MDeCGiVa5cqVJV24uadx48aS5NBHbn7++edC82FhYXJzc1Pjxo2Vn5+vtLQ03XXXXde1nfr16+uLL76w+Uc+ISFBPj4+qlq16jX3U7lyZaWmptr0U9T38euvvyo7O1teXl7W/apQoYKqVasm6cJZgPz8fJvP/PDDD2rZsqUGDx5sbSvqSLhSpUpatmyZoqKi1LNnT3322WcqW7asbr/9dqWmpsrd3f2KN8M5QmBgoKpWrao//vhDjz76aJHrXPzloah9nD17tu69915JUnJystLT0wt9/ueff1bfvn1t5i/++atfv74WLlyos2fPWo+qf/zxR5UpU6bQ2ZcrKer7v/322/X777+rdu3aV/ycu7u72rVrp3bt2mncuHG66aab9P3336t79+7XtF2YE3d9o0Tz8vJS8+bNNXnyZO3evVsbN27Uyy+/7LD+k5OTNXLkSO3bt0+ffPKJZs6cqWeffVaSVKdOHT366KPq27evli1bpqSkJG3dulVvvPGGVq5cadd2Bg8erOTkZA0bNkx79+7VV199pXHjxmnkyJF2PcoVFRWlv/76S1OmTNGhQ4f0zjvv6Lvvviu0Xm5urgYOHKjdu3fru+++07hx4zR06FDrtmrWrKnNmzfr8OHDSk9PV0FBgWrXrq1t27Zp9erV2r9/v1555RVt3bq1yDqqVKmi77//Xnv37lWvXr2Ul5endu3aqUWLFnrwwQe1evVqHT58WAkJCXr55Ze1bds2u76vfxITE6PY2Fi99dZb2r9/v3bu3Kn58+dr2rRp1vq8vLy0atUq/fnnn8rIyJAk1a5dW4sXL9aePXu0efNmPfroo9ZfZi71+eef68MPP9T+/fs1btw4bdmyRUOHDpUkPfrooypXrpz69eunXbt2af369Ro2bJj69Olzzae9a9asqY0bN+r48ePWXxTGjh2rn376SUOGDNGOHTt04MABff311xo2bJgk6ZtvvtHbb7+tHTt26MiRI1q0aJEKCgpUt27d6/4+4VoENUq8Dz/8UOfPn1fTpk317LPP6vXXX3dY33379lV2drbuvPNODRkyRMOGDdNTTz1lXT5//nz17dtXo0aNUt26ddW1a1dt3rxZoaGhdm2natWqWrlypbZs2aJGjRrp6aef1sCBA+3+paNevXqaPXu23nnnHTVq1EhbtmzR6NGjC63Xtm1bhYWFqU2bNurRo4e6dOli89jb6NGj5ebmpvr166ty5co6evSonn76aXXv3l09e/ZUs2bNdOLECZuj68sFBQXp+++/186dO/Xoo4+qoKBAK1euVJs2bTRgwADVqVNHjzzyiA4fPlzs67ZX8sQTT+iDDz7QggULFBERocjISC1YsMD6WJa7u7vefvttvffeewoJCdEDDzwg6cKfpVOnTqlx48bq06ePhg8fripVqhTqf/z48Vq6dKkaNmyohQsX6uOPP1b9+vUlSd7e3lq9erVOnjypO+64Qw8//LDatm2rWbNmXXP9EyZM0OHDh3XLLbdYzxo1bNhQ8fHxOnDggO666y41btxYr7zyioKDgyVduHywbNky3XPPPapXr57effddffLJJ7r11luv67uE61mMyy9oAfhX69+/v06fPn3dr8EEcGNwRA0AgIkR1AAAmBinvgEAMDGOqAEAMDGCGgAAEyOoAQAwMYIaAAATI6gBADAxghoAABMjqAEAMDGCGgAAE/t/isMwisWPADIAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 500x500 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.displot(gene_uptake_knockout_df.groupby('gid').size(), bins=16)\n",
    "plt.xlabel('number of uptake metabolites')\n",
    "plt.ylabel('number of gene knockouts')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(13.819444444444445, 0.5, 'number of reaction knockouts')"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sns.displot(reaction_uptake_knockout_df.groupby('rid').size(), bins=16)\n",
    "plt.xlabel('number of uptake metabolites')\n",
    "plt.ylabel('number of reaction knockouts')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'ids', 'growth', 'status', 'uptake', 'secretion', 'gid',\n",
       "       'knockout', 'rid', 'reaction', 'gene_id', 'pmm', 'gene', 'product',\n",
       "       'status_de', 'sub', 'path', 'ecpath', 'baseMean', 'log2FoldChange',\n",
       "       'padj'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gene        5648\n",
       "reaction      18\n",
       "Name: knockout, dtype: int64"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df['knockout'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "gpdf = gene_uptake_knockout_df.pivot_table(index='gene',columns='uptake', values='log2FoldChange', aggfunc='mean',\n",
    "                                           fill_value=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_uptake_knockout_df['stars'] = ''\n",
    "gene_uptake_knockout_df.loc[gene_uptake_knockout_df.padj < 0.1, 'stars'] = '*'\n",
    "gene_uptake_knockout_df.loc[gene_uptake_knockout_df.padj < 0.05, 'stars'] = '**'\n",
    "gene_uptake_knockout_df.loc[gene_uptake_knockout_df.padj < 0.01, 'stars'] = '***'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>uptake</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3339</th>\n",
       "      <td>cysK</td>\n",
       "      <td>AmmoniaEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3340</th>\n",
       "      <td>cysK</td>\n",
       "      <td>CyanateEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3341</th>\n",
       "      <td>cysK</td>\n",
       "      <td>GlutathioneEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3342</th>\n",
       "      <td>cysK</td>\n",
       "      <td>GlycineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3343</th>\n",
       "      <td>cysK</td>\n",
       "      <td>GuanineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3344</th>\n",
       "      <td>cysK</td>\n",
       "      <td>GuanosineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3345</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_AlanineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3346</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_ArginineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3347</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_AspartateEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3348</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_CystineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3349</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3350</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_GlutamineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3351</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_LeucineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3352</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_SerineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3353</th>\n",
       "      <td>cysK</td>\n",
       "      <td>L_ValineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3354</th>\n",
       "      <td>cysK</td>\n",
       "      <td>UreaEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TX50_RS00135</td>\n",
       "      <td>AmmoniaEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TX50_RS00135</td>\n",
       "      <td>CyanateEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TX50_RS00135</td>\n",
       "      <td>GlutathioneEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TX50_RS00135</td>\n",
       "      <td>GlycineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene         uptake  0\n",
       "3339          cysK      AmmoniaEX  2\n",
       "3340          cysK      CyanateEX  2\n",
       "3341          cysK  GlutathioneEX  2\n",
       "3342          cysK      GlycineEX  2\n",
       "3343          cysK      GuanineEX  2\n",
       "3344          cysK    GuanosineEX  2\n",
       "3345          cysK    L_AlanineEX  2\n",
       "3346          cysK   L_ArginineEX  2\n",
       "3347          cysK  L_AspartateEX  2\n",
       "3348          cysK    L_CystineEX  2\n",
       "3349          cysK  L_GlutamateEX  2\n",
       "3350          cysK  L_GlutamineEX  2\n",
       "3351          cysK    L_LeucineEX  2\n",
       "3352          cysK     L_SerineEX  2\n",
       "3353          cysK     L_ValineEX  2\n",
       "3354          cysK         UreaEX  2\n",
       "0     TX50_RS00135      AmmoniaEX  1\n",
       "1     TX50_RS00135      CyanateEX  1\n",
       "2     TX50_RS00135  GlutathioneEX  1\n",
       "3     TX50_RS00135      GlycineEX  1"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df.groupby( ['gene','uptake'] ).size().reset_index().nlargest(columns=0, n=20)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "gpdf_padj = gene_uptake_knockout_df.pivot_table(index='gene',columns='uptake', values='stars', aggfunc='first',\n",
    "                                           fill_value='')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ids</th>\n",
       "      <th>growth</th>\n",
       "      <th>status</th>\n",
       "      <th>uptake</th>\n",
       "      <th>secretion</th>\n",
       "      <th>gid</th>\n",
       "      <th>knockout</th>\n",
       "      <th>rid</th>\n",
       "      <th>reaction</th>\n",
       "      <th>...</th>\n",
       "      <th>gene</th>\n",
       "      <th>product</th>\n",
       "      <th>status_de</th>\n",
       "      <th>sub</th>\n",
       "      <th>path</th>\n",
       "      <th>ecpath</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>padj</th>\n",
       "      <th>stars</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>{'PMM0951'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0951</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>carA</td>\n",
       "      <td>glutamine-hydrolyzing carbamoyl-phosphate synt...</td>\n",
       "      <td>down</td>\n",
       "      <td>09104 Nucleotide metabolism // 09105 Amino aci...</td>\n",
       "      <td>00240 Pyrimidine metabolism [PATH:ko00240] // ...</td>\n",
       "      <td>K01956  carA, CPA1; carbamoyl-phosphate syntha...</td>\n",
       "      <td>178.513724</td>\n",
       "      <td>-1.504660</td>\n",
       "      <td>1.544491e-08</td>\n",
       "      <td>***</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>{'PMM0796'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0796</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>hisA</td>\n",
       "      <td>1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino...</td>\n",
       "      <td>no_significant</td>\n",
       "      <td>09105 Amino acid metabolism</td>\n",
       "      <td>00340 Histidine metabolism [PATH:ko00340]</td>\n",
       "      <td>K01814  hisA; phosphoribosylformimino-5-aminoi...</td>\n",
       "      <td>284.786091</td>\n",
       "      <td>0.285050</td>\n",
       "      <td>2.226182e-01</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>{'PMM0483'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0483</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>hemL</td>\n",
       "      <td>glutamate-1-semialdehyde 2,1-aminomutase</td>\n",
       "      <td>no_significant</td>\n",
       "      <td>09108 Metabolism of cofactors and vitamins // ...</td>\n",
       "      <td>00860 Porphyrin metabolism [PATH:ko00860] // 0...</td>\n",
       "      <td>K01845  hemL; glutamate-1-semialdehyde 2,1-ami...</td>\n",
       "      <td>2468.751490</td>\n",
       "      <td>0.459796</td>\n",
       "      <td>1.200598e-01</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>{'PMM0001'}</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>optimal</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM0001</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>DNA polymerase III subunit beta</td>\n",
       "      <td>no_significant</td>\n",
       "      <td>09124 Replication and repair // 09182 Protein ...</td>\n",
       "      <td>03030 DNA replication [PATH:ko03030] // 03430 ...</td>\n",
       "      <td>K02338  dnaN; DNA polymerase III subunit beta ...</td>\n",
       "      <td>1548.906501</td>\n",
       "      <td>0.228545</td>\n",
       "      <td>4.619685e-01</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>{'PMM1181'}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infeasible</td>\n",
       "      <td>L_GlutamateEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMM1181</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>aroH</td>\n",
       "      <td>chorismate mutase</td>\n",
       "      <td>down</td>\n",
       "      <td>09105 Amino acid metabolism</td>\n",
       "      <td>00400 Phenylalanine, tyrosine and tryptophan b...</td>\n",
       "      <td>K06208  aroH; chorismate mutase [EC:5.4.99.5]</td>\n",
       "      <td>195.838847</td>\n",
       "      <td>-1.247608</td>\n",
       "      <td>8.508785e-09</td>\n",
       "      <td>***</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0          ids   growth      status         uptake  secretion  \\\n",
       "0           0  {'PMM0951'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "1           1  {'PMM0796'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "2           3  {'PMM0483'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "3           5  {'PMM0001'}  0.00001     optimal  L_GlutamateEX        NaN   \n",
       "4           7  {'PMM1181'}      NaN  infeasible  L_GlutamateEX        NaN   \n",
       "\n",
       "       gid knockout  rid reaction  ...  gene  \\\n",
       "0  PMM0951     gene  NaN      NaN  ...  carA   \n",
       "1  PMM0796     gene  NaN      NaN  ...  hisA   \n",
       "2  PMM0483     gene  NaN      NaN  ...  hemL   \n",
       "3  PMM0001     gene  NaN      NaN  ...  dnaN   \n",
       "4  PMM1181     gene  NaN      NaN  ...  aroH   \n",
       "\n",
       "                                             product       status_de  \\\n",
       "0  glutamine-hydrolyzing carbamoyl-phosphate synt...            down   \n",
       "1  1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino...  no_significant   \n",
       "2           glutamate-1-semialdehyde 2,1-aminomutase  no_significant   \n",
       "3                    DNA polymerase III subunit beta  no_significant   \n",
       "4                                  chorismate mutase            down   \n",
       "\n",
       "                                                 sub  \\\n",
       "0  09104 Nucleotide metabolism // 09105 Amino aci...   \n",
       "1                        09105 Amino acid metabolism   \n",
       "2  09108 Metabolism of cofactors and vitamins // ...   \n",
       "3  09124 Replication and repair // 09182 Protein ...   \n",
       "4                        09105 Amino acid metabolism   \n",
       "\n",
       "                                                path  \\\n",
       "0  00240 Pyrimidine metabolism [PATH:ko00240] // ...   \n",
       "1          00340 Histidine metabolism [PATH:ko00340]   \n",
       "2  00860 Porphyrin metabolism [PATH:ko00860] // 0...   \n",
       "3  03030 DNA replication [PATH:ko03030] // 03430 ...   \n",
       "4  00400 Phenylalanine, tyrosine and tryptophan b...   \n",
       "\n",
       "                                              ecpath     baseMean  \\\n",
       "0  K01956  carA, CPA1; carbamoyl-phosphate syntha...   178.513724   \n",
       "1  K01814  hisA; phosphoribosylformimino-5-aminoi...   284.786091   \n",
       "2  K01845  hemL; glutamate-1-semialdehyde 2,1-ami...  2468.751490   \n",
       "3  K02338  dnaN; DNA polymerase III subunit beta ...  1548.906501   \n",
       "4      K06208  aroH; chorismate mutase [EC:5.4.99.5]   195.838847   \n",
       "\n",
       "  log2FoldChange          padj  stars  \n",
       "0      -1.504660  1.544491e-08    ***  \n",
       "1       0.285050  2.226182e-01         \n",
       "2       0.459796  1.200598e-01         \n",
       "3       0.228545  4.619685e-01         \n",
       "4      -1.247608  8.508785e-09    ***  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "09105 Amino acid metabolism                                 33\n",
       "09181 Protein families: metabolism                          32\n",
       "09101 Carbohydrate metabolism                               27\n",
       "09102 Energy metabolism                                     22\n",
       "09183 Protein families: signaling and cellular processes    21\n",
       "09108 Metabolism of cofactors and vitamins                  20\n",
       "09104 Nucleotide metabolism                                 18\n",
       "09106 Metabolism of other amino acids                       15\n",
       "09103 Lipid metabolism                                      12\n",
       "09107 Glycan biosynthesis and metabolism                    10\n",
       "09110 Biosynthesis of other secondary metabolites           10\n",
       "09132 Signal transduction                                   10\n",
       "09182 Protein families: genetic information processing       9\n",
       "09175 Drug resistance: antimicrobial                         6\n",
       "09111 Xenobiotics biodegradation and metabolism              6\n",
       "09109 Metabolism of terpenoids and polyketides               6\n",
       "09131 Membrane transport                                     4\n",
       "09161 Cancer: overview                                       4\n",
       "09166 Cardiovascular disease                                 3\n",
       "09145 Cellular community - prokaryotes                       3\n",
       "09143 Cell growth and death                                  3\n",
       "09167 Endocrine and metabolic disease                        3\n",
       "09171 Infectious disease: bacterial                          2\n",
       "09124 Replication and repair                                 2\n",
       "09122 Translation                                            2\n",
       "09141 Transport and catabolism                               2\n",
       "09176 Drug resistance: antineoplastic                        2\n",
       "09162 Cancer: specific types                                 1\n",
       "09152 Endocrine system                                       1\n",
       "09191 Unclassified: metabolism                               1\n",
       "09158 Development and regeneration                           1\n",
       "09123 Folding, sorting and degradation                       1\n",
       "09194 Poorly characterized                                   1\n",
       "09193 Unclassified: signaling and cellular processes         1\n",
       "09121 Transcription                                          1\n",
       "09156 Nervous system                                         1\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df['sub'].drop_duplicates().str.split(' // ', \n",
    "                                         expand=True).melt().dropna()['value'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "09108 Metabolism of cofactors and vitamins                  1509\n",
       "09105 Amino acid metabolism                                 1279\n",
       "09181 Protein families: metabolism                          1140\n",
       "09102 Energy metabolism                                      964\n",
       "09101 Carbohydrate metabolism                                646\n",
       "09104 Nucleotide metabolism                                  642\n",
       "09183 Protein families: signaling and cellular processes     627\n",
       "09182 Protein families: genetic information processing       415\n",
       "09103 Lipid metabolism                                       336\n",
       "09106 Metabolism of other amino acids                        300\n",
       "09107 Glycan biosynthesis and metabolism                     289\n",
       "09110 Biosynthesis of other secondary metabolites            286\n",
       "09109 Metabolism of terpenoids and polyketides               282\n",
       "09131 Membrane transport                                     232\n",
       "09124 Replication and repair                                 192\n",
       "09132 Signal transduction                                    175\n",
       "09175 Drug resistance: antimicrobial                         128\n",
       "09191 Unclassified: metabolism                                77\n",
       "09111 Xenobiotics biodegradation and metabolism               67\n",
       "09145 Cellular community - prokaryotes                        64\n",
       "09143 Cell growth and death                                   47\n",
       "09166 Cardiovascular disease                                  33\n",
       "09167 Endocrine and metabolic disease                         33\n",
       "09161 Cancer: overview                                        32\n",
       "09122 Translation                                             32\n",
       "09176 Drug resistance: antineoplastic                         30\n",
       "09171 Infectious disease: bacterial                           18\n",
       "09193 Unclassified: signaling and cellular processes          16\n",
       "09194 Poorly characterized                                    16\n",
       "09158 Development and regeneration                            16\n",
       "09156 Nervous system                                          15\n",
       "09123 Folding, sorting and degradation                        15\n",
       "09141 Transport and catabolism                                14\n",
       "09152 Endocrine system                                         2\n",
       "09162 Cancer: specific types                                   1\n",
       "09121 Transcription                                            1\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_uptake_knockout_df['sub'].str.split(' // ', \n",
    "                                         expand=True).melt().dropna()['value'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "kegg_categories_map = {\n",
    "    '09183 Protein families: signaling and cellular processes' : 'Signal/cellular',\n",
    "    '09132 Signal transduction' : 'Signal/cellular',                                \n",
    "    '09193 Unclassified: signaling and cellular processes' : 'Signal/cellular',\n",
    "    '09143 Cell growth and death' : 'Signal/cellular',                              \n",
    "    #'09132 Signal transduction': 'Signal transduction',\n",
    "    '09109 Metabolism of terpenoids and polyketides' : 'Secondary',\n",
    "    '09182 Protein families: genetic information processing' : 'genetic info',\n",
    "    '09110 Biosynthesis of other secondary metabolites' : 'Secondary', \n",
    "    '09108 Metabolism of cofactors and vitamins': 'Secondary',\n",
    "    '09102 Energy metabolism' : 'Energy',\n",
    "    '09101 Carbohydrate metabolism': 'Carbohydrate',\n",
    "    '09104 Nucleotide metabolism' : 'Nucleotide',\n",
    "    '09105 Amino acid metabolism' : 'Amino acid',\n",
    "    '09106 Metabolism of other amino acids' : 'Amino acid',\n",
    "    #'09143 Cell growth and death', 'Cell growth and death',\n",
    "    #'09123 Folding, sorting and degradation' : 'Folding'\n",
    "    #'09141 Transport and catabolism' : ''\n",
    "#09124 Replication and repair                                 12\n",
    "    #'09131 Membrane transport' : 'Membrane transport',\n",
    "    '09103 Lipid metabolism' : 'Lipid',\n",
    "    '09107 Glycan biosynthesis and metabolism' : 'Glycan',\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene_id', 'gene', 'product', 'status', 'main', 'sub', 'path', 'ecpath',\n",
       "       'baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj',\n",
       "       'contrast', 'Length', 'organism', 'protein_id'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "deseq_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "kegg_rowanno = deseq_df[['sub', 'log2FoldChange']]\n",
    "kegg_rowanno.index= deseq_df['gene']\n",
    "kegg_rowanno = kegg_rowanno[~kegg_rowanno.index.duplicated(keep='first')]\n",
    "kegg_rowanno = kegg_rowanno.loc[gpdf.index,:]\n",
    "#kegg_rowanno = kegg_rowanno.to_frame(name='cat')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['sub', 'log2FoldChange'], dtype='object')"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kegg_rowanno.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "for i, j in kegg_categories_map.items():\n",
    "    kegg_rowanno[j] = kegg_rowanno.cat.str.contains(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "kegg_rowanno['sub'] = kegg_rowanno['sub'].fillna('')\n",
    "for i, j in kegg_categories_map.items():\n",
    "    kegg_rowanno.loc[kegg_rowanno['sub'].str.contains(i), 'Category']  = j\n",
    "kegg_rowanno.loc[kegg_rowanno['Category'].isna(), 'Category']  = 'Other'\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Amino acid         90\n",
       "Secondary          88\n",
       "Energy             67\n",
       "Signal/cellular    33\n",
       "Other              32\n",
       "Nucleotide         31\n",
       "Carbohydrate       23\n",
       "Lipid              21\n",
       "genetic info       19\n",
       "Glycan             19\n",
       "Name: Category, dtype: int64"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kegg_rowanno.Category.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                                      24\n",
       "09191 Unclassified: metabolism         6\n",
       "09194 Poorly characterized             1\n",
       "09181 Protein families: metabolism     1\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kegg_rowanno.loc[kegg_rowanno.Category.isin(['Other']), 'sub'].str.split(' // ', \n",
    "                                         expand=True).melt().dropna()['value'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "kegg_rowanno_sorted = kegg_rowanno.sort_values(by=['Category', 'log2FoldChange' ], ascending=[True, False])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "gpdf_sorted = gpdf.reindex(kegg_rowanno_sorted.index)\n",
    "gpdf_padj_sorted = gpdf_padj.reindex(kegg_rowanno_sorted.index)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "cl_labels = kegg_rowanno.Category.unique()\n",
    "cl_labels.sort()\n",
    "\n",
    "lut = dict(zip(cl_labels, sns.color_palette(palette='Set3', n_colors=len(cl_labels))))\n",
    "row_anno = kegg_rowanno_sorted.Category.map(lut)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 uptake gene knockouts')"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "#sel = ((gpdf_sorted != 0).sum(axis=1) >8) & ((gpdf != 0).sum(axis=1) < gpdf.shape[1])\n",
    "sel = ((gpdf_sorted != 0).sum(axis=1) < gpdf_sorted.shape[1])\n",
    "sns.clustermap(data=gpdf_sorted.loc[sel], \n",
    "               mask=(gpdf_sorted.loc[sel] == 0),\n",
    "               #annot=gpdf_padj_sorted.loc[sel], fmt='',\n",
    "               row_colors=row_anno.loc[sel],\n",
    "               figsize=(5,10), lw=0.001, cmap='coolwarm', center=0,\n",
    "              colors_ratio=0.04, \n",
    "               dendrogram_ratio=0.01,\n",
    "               linecolor='lightgrey',\n",
    "               row_cluster=False,\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "               yticklabels=False,\n",
    "              )\n",
    "\n",
    "handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "plt.legend(handles, lut, title='Category',\n",
    "           bbox_to_anchor=(1.3, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "plt.suptitle('med4 uptake gene knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 uptake gene knockouts')"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "sel = (gpdf_sorted != 0).sum(axis=1) <=8\n",
    "col_sel = (gpdf_sorted.loc[sel] != 0).sum(axis=0) != 0\n",
    "sns.clustermap(data=gpdf_sorted.loc[sel, col_sel], \n",
    "               mask=(gpdf_sorted.loc[sel, col_sel] == 0),\n",
    "               annot=gpdf_padj_sorted.loc[sel, col_sel], fmt='',\n",
    "               row_colors=row_anno.loc[sel],\n",
    "               figsize=(5,10), lw=0.01, cmap='coolwarm', center=0,\n",
    "              colors_ratio=0.04, \n",
    "               dendrogram_ratio=0.01,\n",
    "               linecolor='lightgrey',\n",
    "               row_cluster=False,\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "              )\n",
    "\n",
    "handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "plt.legend(handles, lut, title='Category',\n",
    "           bbox_to_anchor=(1.3, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "plt.suptitle('med4 uptake gene knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 uptake gene knockouts')"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "sel = (gpdf_sorted != 0).sum(axis=1) <=8\n",
    "col_sel = (gpdf_sorted.loc[sel] != 0).sum(axis=0) != 0\n",
    "sns.clustermap(data=gpdf_sorted.loc[sel, col_sel], \n",
    "               mask=(gpdf_sorted.loc[sel, col_sel] == 0),\n",
    "               annot=gpdf_padj_sorted.loc[sel, col_sel], fmt='',\n",
    "               row_colors=row_anno.loc[sel],\n",
    "               figsize=(5,10), lw=0.01, cmap='coolwarm', center=0,\n",
    "              colors_ratio=0.04, \n",
    "               dendrogram_ratio=0.01,\n",
    "               linecolor='lightgrey',\n",
    "               #row_cluster=False,\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "              )\n",
    "\n",
    "handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "plt.legend(handles, lut, title='Category',\n",
    "           bbox_to_anchor=(1.3, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "plt.suptitle('med4 uptake gene knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 uptake reaction knockouts')"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sel = rpdf.sum(axis=1)< 14\n",
    "sns.set_context('paper')\n",
    "sns.clustermap(data=rpdf.loc[sel], \n",
    "               yticklabels= rpdf.loc[sel].index.str.slice(0, 20),\n",
    "               figsize=(5,10), lw=0.01,\n",
    "               dendrogram_ratio=0.01,\n",
    "               linecolor='lightgrey',\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "#                 #cmap='coolwarm',\n",
    "#                row_cluster=False,\n",
    "              )\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "plt.suptitle('med4 uptake reaction knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 uptake reaction knockouts')"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sel = rpdf.sum(axis=1)< 16\n",
    "sns.set_context('paper')\n",
    "sns.clustermap(data=rpdf.loc[sel], \n",
    "               yticklabels= rpdf.loc[sel].index.str.slice(0, 35),\n",
    "               figsize=(7,12), lw=0.01,\n",
    "               dendrogram_ratio=0.01,\n",
    "               linecolor='lightgrey',\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "#                 #cmap='coolwarm',\n",
    "#                row_cluster=False,\n",
    "              )\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "plt.suptitle('med4 uptake reaction knockouts', y=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Secretion visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'ids', 'growth', 'status', 'uptake', 'secretion', 'gid',\n",
       "       'knockout', 'rid', 'reaction', 'gene_id', 'pmm', 'gene', 'product',\n",
       "       'status_de', 'sub', 'path', 'ecpath', 'baseMean', 'log2FoldChange',\n",
       "       'padj'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_secretion_knockout_nodup_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "sgpdf = gene_secretion_knockout_nodup_df.pivot_table(index='gene',columns='secretion', values='log2FoldChange', aggfunc='mean',\n",
    "                                           fill_value=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "srpdf = reaction_secretion_knockout_nodup_df.loc[\n",
    "    ~reaction_secretion_knockout_nodup_df.rid.isin(exchange_ids)\n",
    "].pivot_table(index='reaction',columns='secretion', values='status', aggfunc='count', fill_value=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(4.96944444444445, 0.5, 'number of gene knockouts')"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sns.displot(gene_secretion_knockout_nodup_df.groupby('gid').size(), bins=16)\n",
    "plt.xlabel('number of secretion metabolites')\n",
    "plt.ylabel('number of gene knockouts')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(4.96944444444445, 0.5, 'number of reaction knockouts')"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sns.displot(reaction_secretion_knockout_nodup_df.groupby('rid').size(), bins=16)\n",
    "plt.xlabel('number of secretion metabolites')\n",
    "plt.ylabel('number of reaction knockouts')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gene        641\n",
       "reaction    248\n",
       "Name: knockout, dtype: int64"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_secretion_knockout_nodup_df['knockout'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_secretion_knockout_nodup_df['stars'] = ''\n",
    "gene_secretion_knockout_nodup_df.loc[gene_secretion_knockout_nodup_df.padj < 0.1, 'stars'] = '*'\n",
    "gene_secretion_knockout_nodup_df.loc[gene_secretion_knockout_nodup_df.padj < 0.05, 'stars'] = '**'\n",
    "gene_secretion_knockout_nodup_df.loc[gene_secretion_knockout_nodup_df.padj < 0.01, 'stars'] = '***'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>secretion</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>668</th>\n",
       "      <td>lipA</td>\n",
       "      <td>L_IsoleucineEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TX50_RS00370</td>\n",
       "      <td>4_Methyl_2_oxopentanoateEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TX50_RS00370</td>\n",
       "      <td>5_MethylthioadenosineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TX50_RS00370</td>\n",
       "      <td>ThiosulfateEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TX50_RS00510</td>\n",
       "      <td>5_MethylthioadenosineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TX50_RS00655</td>\n",
       "      <td>BiotinEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TX50_RS00655</td>\n",
       "      <td>EthanolEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TX50_RS00655</td>\n",
       "      <td>L_CystineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TX50_RS00740</td>\n",
       "      <td>5_MethylthioadenosineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TX50_RS00740</td>\n",
       "      <td>L_LeucineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TX50_RS00795</td>\n",
       "      <td>L_LeucineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TX50_RS00795</td>\n",
       "      <td>L_ValineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TX50_RS00885</td>\n",
       "      <td>4_Methyl_2_oxopentanoateEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TX50_RS00885</td>\n",
       "      <td>5_MethylthioadenosineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TX50_RS00885</td>\n",
       "      <td>AcetateEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TX50_RS00900</td>\n",
       "      <td>MethanolEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TX50_RS00905</td>\n",
       "      <td>5_MethylthioadenosineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TX50_RS00905</td>\n",
       "      <td>SpermidineEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TX50_RS00905</td>\n",
       "      <td>ThiosulfateEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TX50_RS00910</td>\n",
       "      <td>4_Methyl_2_oxopentanoateEX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene                   secretion  0\n",
       "668          lipA              L_IsoleucineEX  2\n",
       "0    TX50_RS00370  4_Methyl_2_oxopentanoateEX  1\n",
       "1    TX50_RS00370     5_MethylthioadenosineEX  1\n",
       "2    TX50_RS00370               ThiosulfateEX  1\n",
       "3    TX50_RS00510     5_MethylthioadenosineEX  1\n",
       "4    TX50_RS00655                    BiotinEX  1\n",
       "5    TX50_RS00655                   EthanolEX  1\n",
       "6    TX50_RS00655                 L_CystineEX  1\n",
       "7    TX50_RS00740     5_MethylthioadenosineEX  1\n",
       "8    TX50_RS00740                 L_LeucineEX  1\n",
       "9    TX50_RS00795                 L_LeucineEX  1\n",
       "10   TX50_RS00795                  L_ValineEX  1\n",
       "11   TX50_RS00885  4_Methyl_2_oxopentanoateEX  1\n",
       "12   TX50_RS00885     5_MethylthioadenosineEX  1\n",
       "13   TX50_RS00885                   AcetateEX  1\n",
       "14   TX50_RS00900                  MethanolEX  1\n",
       "15   TX50_RS00905     5_MethylthioadenosineEX  1\n",
       "16   TX50_RS00905                SpermidineEX  1\n",
       "17   TX50_RS00905               ThiosulfateEX  1\n",
       "18   TX50_RS00910  4_Methyl_2_oxopentanoateEX  1"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_secretion_knockout_nodup_df.groupby( ['gene','secretion'] ).size().reset_index().nlargest(columns=0, n=20)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "sgpdf_padj = gene_secretion_knockout_nodup_df.pivot_table(index='gene',columns='secretion', values='stars', aggfunc='first',\n",
    "                                           fill_value='')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "09101 Carbohydrate metabolism                               16\n",
       "09182 Protein families: genetic information processing      15\n",
       "09102 Energy metabolism                                     12\n",
       "09181 Protein families: metabolism                          10\n",
       "09183 Protein families: signaling and cellular processes     9\n",
       "09108 Metabolism of cofactors and vitamins                   9\n",
       "09105 Amino acid metabolism                                  8\n",
       "09106 Metabolism of other amino acids                        7\n",
       "09145 Cellular community - prokaryotes                       5\n",
       "09152 Endocrine system                                       5\n",
       "09104 Nucleotide metabolism                                  5\n",
       "09122 Translation                                            4\n",
       "09109 Metabolism of terpenoids and polyketides               4\n",
       "09159 Environmental adaptation                               3\n",
       "09132 Signal transduction                                    3\n",
       "09110 Biosynthesis of other secondary metabolites            3\n",
       "09167 Endocrine and metabolic disease                        3\n",
       "09111 Xenobiotics biodegradation and metabolism              3\n",
       "09143 Cell growth and death                                  3\n",
       "09103 Lipid metabolism                                       3\n",
       "09123 Folding, sorting and degradation                       2\n",
       "09161 Cancer: overview                                       2\n",
       "09166 Cardiovascular disease                                 2\n",
       "09191 Unclassified: metabolism                               1\n",
       "09193 Unclassified: signaling and cellular processes         1\n",
       "09107 Glycan biosynthesis and metabolism                     1\n",
       "09141 Transport and catabolism                               1\n",
       "09121 Transcription                                          1\n",
       "09172 Infectious disease: viral                              1\n",
       "09164 Neurodegenerative disease                              1\n",
       "09131 Membrane transport                                     1\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_secretion_knockout_nodup_df['sub'].drop_duplicates().str.split(' // ', \n",
    "                                         expand=True).melt().dropna()['value'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "09101 Carbohydrate metabolism                               282\n",
       "09102 Energy metabolism                                     186\n",
       "09105 Amino acid metabolism                                 137\n",
       "09183 Protein families: signaling and cellular processes    128\n",
       "09182 Protein families: genetic information processing      106\n",
       "09145 Cellular community - prokaryotes                      104\n",
       "09181 Protein families: metabolism                          104\n",
       "09106 Metabolism of other amino acids                        94\n",
       "09108 Metabolism of cofactors and vitamins                   89\n",
       "09191 Unclassified: metabolism                               48\n",
       "09104 Nucleotide metabolism                                  40\n",
       "09122 Translation                                            37\n",
       "09131 Membrane transport                                     31\n",
       "09161 Cancer: overview                                       29\n",
       "09166 Cardiovascular disease                                 27\n",
       "09132 Signal transduction                                    26\n",
       "09159 Environmental adaptation                               24\n",
       "09152 Endocrine system                                       23\n",
       "09111 Xenobiotics biodegradation and metabolism              20\n",
       "09109 Metabolism of terpenoids and polyketides               14\n",
       "09167 Endocrine and metabolic disease                        13\n",
       "09110 Biosynthesis of other secondary metabolites            10\n",
       "09123 Folding, sorting and degradation                       10\n",
       "09107 Glycan biosynthesis and metabolism                      9\n",
       "09143 Cell growth and death                                   9\n",
       "09172 Infectious disease: viral                               8\n",
       "09103 Lipid metabolism                                        6\n",
       "09121 Transcription                                           5\n",
       "09193 Unclassified: signaling and cellular processes          3\n",
       "09164 Neurodegenerative disease                               3\n",
       "09141 Transport and catabolism                                2\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_secretion_knockout_nodup_df['sub'].str.split(' // ', \n",
    "                                         expand=True).melt().dropna()['value'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "kegg_categories_map = {\n",
    "    '09183 Protein families: signaling and cellular processes' : 'Signal/cellular',\n",
    "    '09132 Signal transduction' : 'Signal/cellular',                                \n",
    "    '09193 Unclassified: signaling and cellular processes' : 'Signal/cellular',\n",
    "    '09143 Cell growth and death' : 'Signal/cellular',                              \n",
    "    #'09132 Signal transduction': 'Signal transduction',\n",
    "    '09109 Metabolism of terpenoids and polyketides' : 'Secondary',\n",
    "    '09182 Protein families: genetic information processing' : 'genetic info',\n",
    "    '09110 Biosynthesis of other secondary metabolites' : 'Secondary', \n",
    "    '09108 Metabolism of cofactors and vitamins': 'Secondary',\n",
    "    '09102 Energy metabolism' : 'Energy',\n",
    "    '09101 Carbohydrate metabolism': 'Carbohydrate',\n",
    "    '09104 Nucleotide metabolism' : 'Nucleotide',\n",
    "    '09105 Amino acid metabolism' : 'Amino acid',\n",
    "    '09106 Metabolism of other amino acids' : 'Amino acid',\n",
    "    #'09143 Cell growth and death', 'Cell growth and death',\n",
    "    #'09123 Folding, sorting and degradation' : 'Folding'\n",
    "    #'09141 Transport and catabolism' : ''\n",
    "#09124 Replication and repair                                 12\n",
    "    #'09131 Membrane transport' : 'Membrane transport',\n",
    "    '09103 Lipid metabolism' : 'Lipid',\n",
    "    '09107 Glycan biosynthesis and metabolism' : 'Glycan',\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene_id', 'gene', 'product', 'status', 'main', 'sub', 'path', 'ecpath',\n",
       "       'baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj',\n",
       "       'contrast', 'Length', 'organism', 'protein_id'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "deseq_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "skegg_rowanno = deseq_df[['sub', 'log2FoldChange']]\n",
    "skegg_rowanno.index= deseq_df['gene']\n",
    "skegg_rowanno = skegg_rowanno[~skegg_rowanno.index.duplicated(keep='first')]\n",
    "skegg_rowanno = skegg_rowanno.loc[sgpdf.index,:]\n",
    "#kegg_rowanno = kegg_rowanno.to_frame(name='cat')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['sub', 'log2FoldChange'], dtype='object')"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "skegg_rowanno.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "for i, j in kegg_categories_map.items():\n",
    "    kegg_rowanno[j] = kegg_rowanno.cat.str.contains(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "skegg_rowanno['sub'] = skegg_rowanno['sub'].fillna('')\n",
    "for i, j in kegg_categories_map.items():\n",
    "    skegg_rowanno.loc[skegg_rowanno['sub'].str.contains(i), 'Category']  = j\n",
    "skegg_rowanno.loc[skegg_rowanno['Category'].isna(), 'Category']  = 'Other'\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Other              28\n",
       "Secondary          24\n",
       "Amino acid         22\n",
       "genetic info       22\n",
       "Carbohydrate       21\n",
       "Signal/cellular    11\n",
       "Energy             10\n",
       "Nucleotide          8\n",
       "Glycan              5\n",
       "Lipid               3\n",
       "Name: Category, dtype: int64"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "skegg_rowanno.Category.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                                          18\n",
       "09191 Unclassified: metabolism             9\n",
       "09123 Folding, sorting and degradation     1\n",
       "Name: value, dtype: int64"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "skegg_rowanno.loc[skegg_rowanno.Category.isin(['Other']), 'sub'].str.split(' // ', \n",
    "                                         expand=True).melt().dropna()['value'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "skegg_rowanno_sorted = skegg_rowanno.sort_values(by=['Category', 'log2FoldChange' ], ascending=[True, False])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "sgpdf_sorted = sgpdf.reindex(skegg_rowanno_sorted.index)\n",
    "sgpdf_padj_sorted = sgpdf_padj.reindex(skegg_rowanno_sorted.index)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "cl_labels = skegg_rowanno.Category.unique()\n",
    "cl_labels.sort()\n",
    "\n",
    "lut = dict(zip(cl_labels, sns.color_palette(palette='Set3', n_colors=len(cl_labels))))\n",
    "row_anno = skegg_rowanno_sorted.Category.map(lut)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 secretion gene knockouts')"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "#sel = ((sgpdf_sorted != 0).sum(axis=1) >8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "sel = ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "g = sns.clustermap(data=sgpdf_sorted.loc[sel], \n",
    "               mask=(sgpdf_sorted.loc[sel] == 0),\n",
    "               #annot=gpdf_padj_sorted.loc[sel], fmt='',\n",
    "               row_colors=row_anno.loc[sel],\n",
    "               figsize=(15,10), lw=0.001, cmap='coolwarm', center=0,\n",
    "              colors_ratio=0.04, \n",
    "               #dendrogram_ratio=0.01,\n",
    "               linecolor='lightgrey',\n",
    "               row_cluster=False,\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "               yticklabels=False,\n",
    "               xticklabels= sgpdf_sorted.loc[sel].columns.str.slice(0, 20),\n",
    "               \n",
    "              )\n",
    "\n",
    "handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "plt.legend(handles, lut, title='Category',\n",
    "           bbox_to_anchor=(1.1, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "plt.suptitle('med4 secretion gene knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1])"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.cluster import AgglomerativeClustering\n",
    "sel = ((sgpdf_sorted != 0).sum(axis=1) <=8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "\n",
    "clustering_model = AgglomerativeClustering(n_clusters=2, affinity='euclidean', linkage='ward')\n",
    "clustering_model.fit(sgpdf_sorted.loc[sel].T == 0)\n",
    "clustering_model.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "from matplotlib.patches import Patch\n",
    "for i in range(2):\n",
    "    \n",
    "    col_sel = clustering_model.labels_ == i\n",
    "    sel = ((sgpdf_sorted != 0).sum(axis=1) <=8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "    #sel = ((sgpdf_sorted != 0).sum(axis=1) >8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "    #sel = ((sgpdf_sorted.loc[:,col_sel] != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "    #sel = ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "    g = sns.clustermap(data=sgpdf_sorted.loc[sel,col_sel], \n",
    "                   mask=(sgpdf_sorted.loc[sel,col_sel] == 0),\n",
    "                   #annot=gpdf_padj_sorted.loc[sel], fmt='',\n",
    "                   row_colors=row_anno.loc[sel],\n",
    "                   figsize=(10,10), lw=0.001, cmap='coolwarm', center=0,\n",
    "                  colors_ratio=0.04, \n",
    "                   #dendrogram_ratio=0.01,\n",
    "                   linecolor='lightgrey',\n",
    "                   row_cluster=False,\n",
    "                   cbar_pos=(1, .2, .01, .4),\n",
    "                   yticklabels=False,\n",
    "                   xticklabels= sgpdf_sorted.loc[sel].columns.str.slice(0, 20),\n",
    "\n",
    "                  )\n",
    "\n",
    "    handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "    plt.legend(handles, lut, title='Category',\n",
    "               bbox_to_anchor=(1.1, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "    plt.suptitle('med4 secretion gene knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sub</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>Category</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>mtnP</th>\n",
       "      <td>09105 Amino acid metabolism</td>\n",
       "      <td>0.839025</td>\n",
       "      <td>Amino acid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>argJ</th>\n",
       "      <td>09105 Amino acid metabolism</td>\n",
       "      <td>0.790923</td>\n",
       "      <td>Amino acid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cysE</th>\n",
       "      <td>09102 Energy metabolism // 09105 Amino acid me...</td>\n",
       "      <td>0.687886</td>\n",
       "      <td>Amino acid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TX50_RS00370</th>\n",
       "      <td>09106 Metabolism of other amino acids</td>\n",
       "      <td>0.294328</td>\n",
       "      <td>Amino acid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TX50_RS03145</th>\n",
       "      <td>09105 Amino acid metabolism // 09110 Biosynthe...</td>\n",
       "      <td>0.258452</td>\n",
       "      <td>Amino acid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>serS</th>\n",
       "      <td>09122 Translation // 09181 Protein families: m...</td>\n",
       "      <td>-0.524039</td>\n",
       "      <td>genetic info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TX50_RS08230</th>\n",
       "      <td>09121 Transcription // 09182 Protein families:...</td>\n",
       "      <td>-0.565906</td>\n",
       "      <td>genetic info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trpS</th>\n",
       "      <td>09122 Translation // 09181 Protein families: m...</td>\n",
       "      <td>-0.592245</td>\n",
       "      <td>genetic info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>leuS</th>\n",
       "      <td>09122 Translation // 09181 Protein families: m...</td>\n",
       "      <td>-0.627136</td>\n",
       "      <td>genetic info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>glyS</th>\n",
       "      <td>09122 Translation // 09181 Protein families: m...</td>\n",
       "      <td>-0.837062</td>\n",
       "      <td>genetic info</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>154 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            sub  \\\n",
       "gene                                                              \n",
       "mtnP                                09105 Amino acid metabolism   \n",
       "argJ                                09105 Amino acid metabolism   \n",
       "cysE          09102 Energy metabolism // 09105 Amino acid me...   \n",
       "TX50_RS00370              09106 Metabolism of other amino acids   \n",
       "TX50_RS03145  09105 Amino acid metabolism // 09110 Biosynthe...   \n",
       "...                                                         ...   \n",
       "serS          09122 Translation // 09181 Protein families: m...   \n",
       "TX50_RS08230  09121 Transcription // 09182 Protein families:...   \n",
       "trpS          09122 Translation // 09181 Protein families: m...   \n",
       "leuS          09122 Translation // 09181 Protein families: m...   \n",
       "glyS          09122 Translation // 09181 Protein families: m...   \n",
       "\n",
       "              log2FoldChange      Category  \n",
       "gene                                        \n",
       "mtnP                0.839025    Amino acid  \n",
       "argJ                0.790923    Amino acid  \n",
       "cysE                0.687886    Amino acid  \n",
       "TX50_RS00370        0.294328    Amino acid  \n",
       "TX50_RS03145        0.258452    Amino acid  \n",
       "...                      ...           ...  \n",
       "serS               -0.524039  genetic info  \n",
       "TX50_RS08230       -0.565906  genetic info  \n",
       "trpS               -0.592245  genetic info  \n",
       "leuS               -0.627136  genetic info  \n",
       "glyS               -0.837062  genetic info  \n",
       "\n",
       "[154 rows x 3 columns]"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "skegg_rowanno_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(gene\n",
       " mtnP            True\n",
       " argJ            True\n",
       " cysE            True\n",
       " TX50_RS00370    True\n",
       " TX50_RS03145    True\n",
       "                 ... \n",
       " serS            True\n",
       " TX50_RS08230    True\n",
       " trpS            True\n",
       " leuS            True\n",
       " glyS            True\n",
       " Length: 154, dtype: bool,\n",
       " gene\n",
       " mtnP            False\n",
       " argJ            False\n",
       " cysE            False\n",
       " TX50_RS00370    False\n",
       " TX50_RS03145    False\n",
       "                 ...  \n",
       " serS             True\n",
       " TX50_RS08230     True\n",
       " trpS             True\n",
       " leuS             True\n",
       " glyS             True\n",
       " Name: Category, Length: 154, dtype: bool)"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "#sel = ((sgpdf_sorted != 0).sum(axis=1) >8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "for cat in cl_labels:\n",
    "    sel = ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "    row_sel = skegg_rowanno_sorted.Category.isin([cat])\n",
    "sel, row_sel\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(-3.0108353196405595, 1.8419680121935724)"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vmin = sgpdf_sorted.min().min()\n",
    "vmax = sgpdf_sorted.max().max()\n",
    "vmin, vmax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Amino acid 0.48214285714285715 0.11038961038961038 0.22895622895622894\n",
      "Carbohydrate 0.375 0.09740259740259741 0.25974025974025977\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/seaborn/matrix.py:1203: UserWarning: Tight layout not applied. The bottom and top margins cannot be made large enough to accommodate all axes decorations.\n",
      "  self._figure.tight_layout(**tight_params)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Energy 0.19642857142857142 0.039285714285714285 0.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/seaborn/matrix.py:1203: UserWarning: Tight layout not applied. The bottom and top margins cannot be made large enough to accommodate all axes decorations.\n",
      "  self._figure.tight_layout(**tight_params)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Glycan 0.14285714285714285 0.032467532467532464 0.22727272727272727\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/seaborn/matrix.py:1203: UserWarning: Tight layout not applied. The bottom and top margins cannot be made large enough to accommodate all axes decorations.\n",
      "  self._figure.tight_layout(**tight_params)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lipid 0.10714285714285714 0.02142857142857143 0.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/seaborn/matrix.py:1203: UserWarning: Tight layout not applied. The bottom and top margins cannot be made large enough to accommodate all axes decorations.\n",
      "  self._figure.tight_layout(**tight_params)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nucleotide 0.14285714285714285 0.03896103896103896 0.27272727272727276\n",
      "Other 0.39285714285714285 0.14285714285714285 0.36363636363636365\n",
      "Secondary 0.42857142857142855 0.15584415584415584 0.36363636363636365\n",
      "Signal/cellular 0.375 0.07500000000000001 0.20000000000000004\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/seaborn/matrix.py:805: RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).\n",
      "  self._figure = plt.figure(figsize=figsize)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genetic info 0.26785714285714285 0.14285714285714285 0.5333333333333333\n"
     ]
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "#sel = ((sgpdf_sorted != 0).sum(axis=1) >8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "vmin = sgpdf_sorted.min().min()\n",
    "vmax = sgpdf_sorted.max().max()\n",
    "\n",
    "for cat in cl_labels:\n",
    "    sel = ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1]) &((sgpdf_sorted != 0).sum(axis=1) <=8) & skegg_rowanno_sorted.Category.isin([cat])\n",
    "    col_sel = (sgpdf_sorted.loc[sel] != 0).sum(axis=0) > 0\n",
    "    number_of_cols = col_sel.sum()\n",
    "    total_number_of_cols = sgpdf_sorted.shape[1]\n",
    "    number_of_rows = sel.sum()\n",
    "    total_number_of_rows = sgpdf_sorted.shape[0]\n",
    "    width = number_of_cols / total_number_of_cols\n",
    "    height = number_of_rows / total_number_of_rows\n",
    "    if (height/width < 0.2):\n",
    "        height = width*0.2\n",
    "    g = sns.clustermap(data=sgpdf_sorted.loc[sel, col_sel], \n",
    "                   mask=(sgpdf_sorted.loc[sel, col_sel] == 0),\n",
    "                   annot=sgpdf_padj_sorted.loc[sel, col_sel], fmt='',\n",
    "                   row_colors=row_anno.loc[sel],\n",
    "                   figsize=(15*width,40*height), lw=0.01, cmap='coolwarm', center=0,\n",
    "                   vmin=vmin, vmax=vmax,\n",
    "                   \n",
    "                   colors_ratio=0.04, \n",
    "                   dendrogram_ratio=0.01,\n",
    "                   linecolor='lightgrey',\n",
    "                   row_cluster=False,\n",
    "                   cbar_pos=(1, .2, .01, .4),\n",
    "                   #yticklabels=False,\n",
    "                   xticklabels= sgpdf_sorted.loc[sel, col_sel].columns.str.slice(0, 20),\n",
    "                       \n",
    "                  )\n",
    "    g.cax.set_visible(False)\n",
    "    print(cat, width, height, height/width)\n",
    "\n",
    "# handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "# plt.legend(handles, lut, title='Category',\n",
    "#            bbox_to_anchor=(1.3, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "    plt.suptitle(f'med4 secretion gene knockouts - {cat}', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Amino acid 1.0 0.2 0.2\n",
      "Carbohydrate 0.9642857142857143 0.19285714285714287 0.2\n",
      "Energy 0.4107142857142857 0.08214285714285714 0.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/seaborn/matrix.py:1203: UserWarning: Tight layout not applied. The bottom and top margins cannot be made large enough to accommodate all axes decorations.\n",
      "  self._figure.tight_layout(**tight_params)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Glycan 0.14285714285714285 0.032467532467532464 0.22727272727272727\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/seaborn/matrix.py:1203: UserWarning: Tight layout not applied. The bottom and top margins cannot be made large enough to accommodate all axes decorations.\n",
      "  self._figure.tight_layout(**tight_params)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lipid 0.10714285714285714 0.02142857142857143 0.2\n",
      "Nucleotide 0.4107142857142857 0.08214285714285714 0.2\n",
      "Other 1.0 0.2 0.2\n",
      "Secondary 0.42857142857142855 0.15584415584415584 0.36363636363636365\n",
      "Signal/cellular 0.375 0.07500000000000001 0.20000000000000004\n",
      "genetic info 0.26785714285714285 0.14285714285714285 0.5333333333333333\n"
     ]
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "#sel = ((sgpdf_sorted != 0).sum(axis=1) >8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "vmin = sgpdf_sorted.min().min()\n",
    "vmax = sgpdf_sorted.max().max()\n",
    "\n",
    "for cat in cl_labels:\n",
    "    sel = ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])  & skegg_rowanno_sorted.Category.isin([cat])\n",
    "    col_sel = (sgpdf_sorted.loc[sel] != 0).sum(axis=0) > 0\n",
    "    number_of_cols = col_sel.sum()\n",
    "    total_number_of_cols = sgpdf_sorted.shape[1]\n",
    "    number_of_rows = sel.sum()\n",
    "    total_number_of_rows = sgpdf_sorted.shape[0]\n",
    "    width = number_of_cols / total_number_of_cols\n",
    "    height = number_of_rows / total_number_of_rows\n",
    "    if (height/width < 0.2):\n",
    "        height = width*0.2\n",
    "    g = sns.clustermap(data=sgpdf_sorted.loc[sel, col_sel], \n",
    "                   mask=(sgpdf_sorted.loc[sel, col_sel] == 0),\n",
    "                   annot=sgpdf_padj_sorted.loc[sel, col_sel], fmt='',\n",
    "                   row_colors=row_anno.loc[sel],\n",
    "                   figsize=(15*width,40*height), lw=0.01, cmap='coolwarm', center=0,\n",
    "                   vmin=vmin, vmax=vmax,\n",
    "                   \n",
    "                   colors_ratio=0.04, \n",
    "                   dendrogram_ratio=0.01,\n",
    "                   linecolor='lightgrey',\n",
    "                   row_cluster=False,\n",
    "                   cbar_pos=(1, .2, .01, .4),\n",
    "                   #yticklabels=False,\n",
    "                   xticklabels= sgpdf_sorted.loc[sel, col_sel].columns.str.slice(0, 20),\n",
    "                       \n",
    "                  )\n",
    "    g.cax.set_visible(False)\n",
    "    print(cat, width, height, height/width)\n",
    "\n",
    "# handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "# plt.legend(handles, lut, title='Category',\n",
    "#            bbox_to_anchor=(1.3, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "    plt.suptitle(f'med4 secretion gene knockouts - {cat}', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 secretion gene knockouts')"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "#sel = ((sgpdf_sorted != 0).sum(axis=1) >8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "sel = ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "sns.clustermap(data=sgpdf_sorted.loc[sel], \n",
    "               mask=(sgpdf_sorted.loc[sel] == 0),\n",
    "               #annot=gpdf_padj_sorted.loc[sel], fmt='',\n",
    "               row_colors=row_anno.loc[sel],\n",
    "               figsize=(15,10), lw=0.001, cmap='coolwarm', center=0,\n",
    "              colors_ratio=0.04, \n",
    "               dendrogram_ratio=0.01,\n",
    "               linecolor='lightgrey',\n",
    "               #row_cluster=False,\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "               yticklabels=False,\n",
    "              )\n",
    "\n",
    "handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "plt.legend(handles, lut, title='Category',\n",
    "           bbox_to_anchor=(1.3, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "plt.suptitle('med4 secretion gene knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 secretion gene knockouts')"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "#sel = ((sgpdf_sorted != 0).sum(axis=1) >8) & ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "sel = ((sgpdf_sorted != 0).sum(axis=1) < sgpdf_sorted.shape[1])\n",
    "sns.clustermap(data=sgpdf_sorted.loc[sel], \n",
    "               mask=(sgpdf_sorted.loc[sel] == 0),\n",
    "               #annot=gpdf_padj_sorted.loc[sel], fmt='',\n",
    "               row_colors=row_anno.loc[sel],\n",
    "               figsize=(15,10), lw=0.001, cmap='coolwarm', center=0,\n",
    "              colors_ratio=0.04, \n",
    "               dendrogram_ratio=0.01,\n",
    "               linecolor='lightgrey',\n",
    "               row_cluster=False,\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "               yticklabels=False,\n",
    "              )\n",
    "\n",
    "handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "plt.legend(handles, lut, title='Category',\n",
    "           bbox_to_anchor=(1.3, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "plt.suptitle('med4 secretion gene knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 secretion gene knockouts')"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from matplotlib.patches import Patch\n",
    "sel = (sgpdf_sorted != 0).sum(axis=1) <=5\n",
    "col_sel = (sgpdf_sorted.loc[sel] != 0).sum(axis=0) != 0\n",
    "sns.clustermap(data=sgpdf_sorted.loc[sel, col_sel], \n",
    "               mask=(sgpdf_sorted.loc[sel, col_sel] == 0),\n",
    "               annot=sgpdf_padj_sorted.loc[sel, col_sel], fmt='',\n",
    "               row_colors=row_anno.loc[sel],\n",
    "               figsize=(15,10), #lw=0.01,\n",
    "               cmap='coolwarm', center=0,\n",
    "              colors_ratio=0.04, \n",
    "               dendrogram_ratio=0.01,\n",
    "               #linecolor='lightgrey',\n",
    "               row_cluster=False,\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "              )\n",
    "\n",
    "handles = [Patch(facecolor=lut[name]) for name in lut]\n",
    "plt.legend(handles, lut, title='Category',\n",
    "           bbox_to_anchor=(1.3, 1), bbox_transform=plt.gcf().transFigure, loc='upper right')\n",
    "\n",
    "plt.suptitle('med4 secretion gene knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [],
   "source": [
    "g_uptake = set(gene_uptake_knockout_df.gid.unique())\n",
    "g_sec = set(gene_secretion_knockout_df.gid.unique())\n",
    "g2_sec = set(gene_secretion_knockout_nodup_df.gid.unique())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g2_sec & g_uptake"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(424, 106, 155, (154, 56))"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(g_uptake) , len(g_sec), len(g2_sec), sgpdf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/oweissberg/work/tools/anaconda3/lib/python3.9/site-packages/seaborn/matrix.py:654: UserWarning: Clustering large matrix with scipy. Installing `fastcluster` may give better performance.\n",
      "  warnings.warn(msg)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Text(0.5, 1, 'med4 uptake reaction knockouts')"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sel = srpdf.sum(axis=1)>0\n",
    "sns.set_context('paper')\n",
    "sns.clustermap(data=srpdf.loc[sel], \n",
    "               yticklabels= srpdf.loc[sel].index.str.slice(0, 20),\n",
    "               #yticklabels= None,\n",
    "               figsize=(15,10), #lw=0.01,\n",
    "               dendrogram_ratio=0.01,\n",
    "               #linecolor='lightgrey',\n",
    "               cbar_pos=(1, .2, .01, .4),\n",
    "#                 #cmap='coolwarm',\n",
    "#                row_cluster=False,\n",
    "              )\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "plt.suptitle('med4 uptake reaction knockouts', y=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "from openpyxl import load_workbook\n",
    "\n",
    "fpath = 'data/PRO_N_sources.xlsx'\n",
    "\n",
    "book = load_workbook(fpath)\n",
    "writer = pd.ExcelWriter(fpath, engine = 'openpyxl')\n",
    "writer.book = book\n",
    "writer.sheets = dict((ws.title, ws) for ws in book.worksheets)\n",
    "\n",
    "gene_uptake_knockout_df.to_excel(writer, sheet_name = 'gene_knockout_uptake')\n",
    "reaction_uptake_knockout_df.to_excel(writer, sheet_name = 'reactin_knockout_uptake')\n",
    "#writer.save()\n",
    "writer.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>secretion</th>\n",
       "      <th>minimum</th>\n",
       "      <th>maximum</th>\n",
       "      <th>uptake</th>\n",
       "      <th>metabolite_uptake</th>\n",
       "      <th>mid_uptake</th>\n",
       "      <th>N_uptake</th>\n",
       "      <th>C_uptake</th>\n",
       "      <th>metabolite_secretion</th>\n",
       "      <th>mid_secretion</th>\n",
       "      <th>N_secretion</th>\n",
       "      <th>C_secretion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>B12EX</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.941910</td>\n",
       "      <td>GuanosineEX</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Cobamide coenzyme</td>\n",
       "      <td>Cobamide_coenzyme[e]</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>BiotinEX</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.985477</td>\n",
       "      <td>GuanosineEX</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Biotin</td>\n",
       "      <td>Biotin[e]</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>MethanolEX</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>1.985519</td>\n",
       "      <td>GuanosineEX</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Methanol</td>\n",
       "      <td>Methanol[e]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>SAdenosyl4methylthio2oxobutanoateEX</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>1.985519</td>\n",
       "      <td>GuanosineEX</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>S-Adenosyl-4-methylthio-2-oxobutanoate</td>\n",
       "      <td>S_Adenosyl_4_methylthio_2_oxobutanoate[e]</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>SMethyl5thioDriboseEX</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>15.884229</td>\n",
       "      <td>GuanosineEX</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>Guanosine[e]</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>S-Methyl-5-thio-D-ribose</td>\n",
       "      <td>S_Methyl_5_thio_D_ribose[e]</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0                            secretion   minimum    maximum  \\\n",
       "0           0                                B12EX  0.000000   7.941910   \n",
       "1           1                             BiotinEX  0.000000   1.985477   \n",
       "2           2                           MethanolEX  0.000042   1.985519   \n",
       "3           3  SAdenosyl4methylthio2oxobutanoateEX  0.000042   1.985519   \n",
       "4           4                SMethyl5thioDriboseEX  0.000000  15.884229   \n",
       "\n",
       "        uptake metabolite_uptake    mid_uptake  N_uptake  C_uptake  \\\n",
       "0  GuanosineEX      Guanosine[e]  Guanosine[e]      True      True   \n",
       "1  GuanosineEX      Guanosine[e]  Guanosine[e]      True      True   \n",
       "2  GuanosineEX      Guanosine[e]  Guanosine[e]      True      True   \n",
       "3  GuanosineEX      Guanosine[e]  Guanosine[e]      True      True   \n",
       "4  GuanosineEX      Guanosine[e]  Guanosine[e]      True      True   \n",
       "\n",
       "                     metabolite_secretion  \\\n",
       "0                       Cobamide coenzyme   \n",
       "1                                  Biotin   \n",
       "2                                Methanol   \n",
       "3  S-Adenosyl-4-methylthio-2-oxobutanoate   \n",
       "4                S-Methyl-5-thio-D-ribose   \n",
       "\n",
       "                               mid_secretion  N_secretion  C_secretion  \n",
       "0                       Cobamide_coenzyme[e]         True         True  \n",
       "1                                  Biotin[e]         True         True  \n",
       "2                                Methanol[e]        False         True  \n",
       "3  S_Adenosyl_4_methylthio_2_oxobutanoate[e]         True         True  \n",
       "4                S_Methyl_5_thio_D_ribose[e]        False         True  "
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_exchange_df.head()                          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(881, 13)"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_exchange_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
